{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "a75890eb",
   "metadata": {
    "papermill": {
     "duration": 0.021199,
     "end_time": "2023-08-23T17:38:05.792521",
     "exception": false,
     "start_time": "2023-08-23T17:38:05.771322",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2fa2e50f",
   "metadata": {
    "papermill": {
     "duration": 0.011382,
     "end_time": "2023-08-23T17:38:05.845845",
     "exception": false,
     "start_time": "2023-08-23T17:38:05.834463",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d707ce2c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-10-14T14:33:37.863523Z",
     "start_time": "2022-10-14T14:33:14.145712Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T17:38:05.895775Z",
     "iopub.status.busy": "2023-08-23T17:38:05.895399Z",
     "iopub.status.idle": "2023-08-23T17:38:12.042694Z",
     "shell.execute_reply": "2023-08-23T17:38:12.041739Z"
    },
    "papermill": {
     "duration": 6.188339,
     "end_time": "2023-08-23T17:38:12.045937",
     "exception": false,
     "start_time": "2023-08-23T17:38:05.857598",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization, Metadata\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired, trim_single\n",
    "from qiime2.plugins.demux.methods import filter_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cfbb760c",
   "metadata": {
    "papermill": {
     "duration": 0.010661,
     "end_time": "2023-08-23T17:38:12.074144",
     "exception": false,
     "start_time": "2023-08-23T17:38:12.063483",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "615bd466",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.594341Z",
     "start_time": "2022-06-28T18:50:41.591544Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T17:38:12.096851Z",
     "iopub.status.busy": "2023-08-23T17:38:12.096336Z",
     "iopub.status.idle": "2023-08-23T17:38:12.101972Z",
     "shell.execute_reply": "2023-08-23T17:38:12.101290Z"
    },
    "papermill": {
     "duration": 0.018736,
     "end_time": "2023-08-23T17:38:12.103533",
     "exception": false,
     "start_time": "2023-08-23T17:38:12.084797",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'parameter-set-01.yaml')\n",
    "experiment_name = 'exp-01'\n",
    "base_dir = os.path.join('/', 'home', 'username', 'pipeline-dir')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'manifest.csv')\n",
    "replace_files = False\n",
    "trim = None\n",
    "metadata_file = None\n",
    "threads = 1\n",
    "demux_file = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5001c7a6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T17:38:12.127111Z",
     "iopub.status.busy": "2023-08-23T17:38:12.126880Z",
     "iopub.status.idle": "2023-08-23T17:38:12.131202Z",
     "shell.execute_reply": "2023-08-23T17:38:12.130679Z"
    },
    "papermill": {
     "duration": 0.017221,
     "end_time": "2023-08-23T17:38:12.132663",
     "exception": false,
     "start_time": "2023-08-23T17:38:12.115442",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "experiment_name = \"ana-flavia-NCxHSD-NC-trim\"\n",
    "base_dir = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri\"\n",
    "manifest_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-NCxHSD-NC.csv\"\n",
    "metadata_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-NCxHSD-NC.tsv\"\n",
    "class_col = \"group-id\"\n",
    "classifier_file = \"/home/lauro/nupeb/rede-micro/models/silva-138-99-nb-classifier.qza\"\n",
    "top_n = 20\n",
    "replace_files = False\n",
    "phred = 20\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n",
    "overlap = 12\n",
    "threads = 6\n",
    "trim = {\n",
    "    \"overlap\": 8,\n",
    "    \"forward_primer\": \"CCTACGGGRSGCAGCAG\",\n",
    "    \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\",\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8802bcfa",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-18T15:40:31.070885Z",
     "start_time": "2023-08-18T15:40:30.611671Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T17:38:12.154935Z",
     "iopub.status.busy": "2023-08-23T17:38:12.154710Z",
     "iopub.status.idle": "2023-08-23T17:38:12.159080Z",
     "shell.execute_reply": "2023-08-23T17:38:12.158354Z"
    },
    "papermill": {
     "duration": 0.017165,
     "end_time": "2023-08-23T17:38:12.160606",
     "exception": false,
     "start_time": "2023-08-23T17:38:12.143441",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "data_dir =  os.path.abspath(os.path.join(base_dir, 'data'))\n",
    "raw_data_dir =  os.path.abspath(os.path.join(data_dir, 'raw'))\n",
    "interim_data_dir =  os.path.abspath(os.path.join(data_dir, 'interim'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d1e0d91e",
   "metadata": {
    "papermill": {
     "duration": 0.013435,
     "end_time": "2023-08-23T17:38:12.185262",
     "exception": false,
     "start_time": "2023-08-23T17:38:12.171827",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "87359ef1",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.632125Z",
     "start_time": "2022-06-28T18:50:41.619710Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T17:38:12.207728Z",
     "iopub.status.busy": "2023-08-23T17:38:12.207502Z",
     "iopub.status.idle": "2023-08-23T17:38:12.213250Z",
     "shell.execute_reply": "2023-08-23T17:38:12.212580Z"
    },
    "papermill": {
     "duration": 0.018864,
     "end_time": "2023-08-23T17:38:12.214897",
     "exception": false,
     "start_time": "2023-08-23T17:38:12.196033",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New qiime-artifacts folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-NCxHSD-NC-trim/qiime-artifacts\n",
      "New img folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-NCxHSD-NC-trim/imgs\n"
     ]
    }
   ],
   "source": [
    "# Create path if it not exists\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exists\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')\n",
    "\n",
    "# Define the output artifact full path\n",
    "if demux_file is None:\n",
    "    demux_file = os.path.join(out_dir, 'demux-paired.qza')\n",
    "demux_view = os.path.join(out_dir, 'demux-paired.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c55f526b",
   "metadata": {
    "papermill": {
     "duration": 0.011563,
     "end_time": "2023-08-23T17:38:12.238262",
     "exception": false,
     "start_time": "2023-08-23T17:38:12.226699",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file of `SampleData[PairedEndSequencesWithQuality]`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8f47f564",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T17:38:12.261639Z",
     "iopub.status.busy": "2023-08-23T17:38:12.261413Z",
     "iopub.status.idle": "2023-08-23T17:38:12.330690Z",
     "shell.execute_reply": "2023-08-23T17:38:12.329996Z"
    },
    "papermill": {
     "duration": 0.08325,
     "end_time": "2023-08-23T17:38:12.332808",
     "exception": false,
     "start_time": "2023-08-23T17:38:12.249558",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "manifest_df = pd.read_csv(manifest_file)\n",
    "n_directions = len(manifest_df['direction'].unique())\n",
    "if n_directions == 1:\n",
    "    d_type = 'SampleData[SequencesWithQuality]'\n",
    "    v_type = 'SingleEndFastqManifestPhred33'\n",
    "elif n_directions == 2:\n",
    "    d_type = 'SampleData[PairedEndSequencesWithQuality]'\n",
    "    v_type = 'PairedEndFastqManifestPhred33'\n",
    "else:\n",
    "    print(f'ERROR: invalid number of directions {n_directions}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "229eef19",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T17:38:12.363262Z",
     "iopub.status.busy": "2023-08-23T17:38:12.362890Z",
     "iopub.status.idle": "2023-08-23T17:38:12.385001Z",
     "shell.execute_reply": "2023-08-23T17:38:12.384073Z"
    },
    "papermill": {
     "duration": 0.03772,
     "end_time": "2023-08-23T17:38:12.387304",
     "exception": false,
     "start_time": "2023-08-23T17:38:12.349584",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the metadata file is not defined, use the default metadata file\n",
    "metadata_qa = Metadata.load(metadata_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2878c8d4",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:44:58.640338Z",
     "start_time": "2022-06-28T18:50:41.633657Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T17:38:12.416419Z",
     "iopub.status.busy": "2023-08-23T17:38:12.416060Z",
     "iopub.status.idle": "2023-08-23T17:39:12.692183Z",
     "shell.execute_reply": "2023-08-23T17:39:12.691540Z"
    },
    "papermill": {
     "duration": 60.2924,
     "end_time": "2023-08-23T17:39:12.695189",
     "exception": false,
     "start_time": "2023-08-23T17:38:12.402789",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    interim_demux_path = os.path.join(interim_data_dir, \"demux-paired.qza\")\n",
    "    # Load Artifact from interim folder\n",
    "    if os.path.isfile(interim_demux_path):\n",
    "        artifact = Artifact.load(interim_demux_path)\n",
    "    # Create new Artifact using Manifest\n",
    "    else:\n",
    "        artifact = Artifact.import_data(d_type, manifest_file, view_type=v_type)\n",
    "        if not os.path.isfile(interim_demux_path):\n",
    "            artifact.save(interim_demux_path)\n",
    "    # Filter and Save Artifact\n",
    "    artifact = filter_samples(demux=artifact, metadata=metadata_qa).filtered_demux\n",
    "    artifact.save(demux_file)\n",
    "    # Visualization\n",
    "    if os.path.isfile(demux_view):\n",
    "        demux_view_obj = Visualization.load(demux_view)\n",
    "    else:\n",
    "        demux_view_obj = demux.visualizers.summarize(artifact).visualization\n",
    "        Visualization.save(demux_view_obj, filepath=demux_view)\n",
    "elif os.path.isfile(demux_file):\n",
    "    # Load Artifact\n",
    "    artifact = Artifact.load(demux_file)\n",
    "    artifact = filter_samples(demux=artifact, metadata=metadata_qa).filtered_demux\n",
    "    # Visualization\n",
    "    if os.path.isfile(demux_view):\n",
    "        demux_view_obj = Visualization.load(demux_view)\n",
    "    else:\n",
    "        demux_view_obj = demux.visualizers.summarize(artifact).visualization\n",
    "        Visualization.save(demux_view_obj, filepath=demux_view)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b40a82f3",
   "metadata": {
    "papermill": {
     "duration": 0.099463,
     "end_time": "2023-08-23T17:39:12.813534",
     "exception": false,
     "start_time": "2023-08-23T17:39:12.714071",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "fd408bad",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.609737Z",
     "start_time": "2022-06-28T19:47:56.604820Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T17:39:12.846895Z",
     "iopub.status.busy": "2023-08-23T17:39:12.846595Z",
     "iopub.status.idle": "2023-08-23T17:39:12.851496Z",
     "shell.execute_reply": "2023-08-23T17:39:12.850643Z"
    },
    "papermill": {
     "duration": 0.018858,
     "end_time": "2023-08-23T17:39:12.853185",
     "exception": false,
     "start_time": "2023-08-23T17:39:12.834327",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<visualization: Visualization uuid: 903634ee-a0f2-47af-9fce-9b4eec0b2d9c>\n"
     ]
    }
   ],
   "source": [
    "print(demux_view_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b38b9506",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.704841Z",
     "start_time": "2022-06-28T19:47:56.611846Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T17:39:12.879134Z",
     "iopub.status.busy": "2023-08-23T17:39:12.878782Z",
     "iopub.status.idle": "2023-08-23T17:39:12.889795Z",
     "shell.execute_reply": "2023-08-23T17:39:12.888902Z"
    },
    "papermill": {
     "duration": 0.026315,
     "end_time": "2023-08-23T17:39:12.891695",
     "exception": false,
     "start_time": "2023-08-23T17:39:12.865380",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-o6bopdzg')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 903634ee-a0f2-47af-9fce-9b4eec0b2d9c>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed300ee6",
   "metadata": {
    "papermill": {
     "duration": 0.013261,
     "end_time": "2023-08-23T17:39:12.920370",
     "exception": false,
     "start_time": "2023-08-23T17:39:12.907109",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "bc00f6ae",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T17:39:12.973653Z",
     "iopub.status.busy": "2023-08-23T17:39:12.973276Z",
     "iopub.status.idle": "2023-08-23T17:40:28.286200Z",
     "shell.execute_reply": "2023-08-23T17:40:28.284998Z"
    },
    "papermill": {
     "duration": 75.340546,
     "end_time": "2023-08-23T17:40:28.289473",
     "exception": false,
     "start_time": "2023-08-23T17:39:12.948927",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121673_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121673_1_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121673_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121673_1_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121673_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121673_1_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121673_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121673_1_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.71 s (10 µs/read; 5.99 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            370,042\n",
      "  Read 1 with adapter:                 359,997 (97.3%)\n",
      "  Read 2 with adapter:                 343,009 (92.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           36,262 (9.8%)\n",
      "Pairs written (passing filters):       333,780 (90.2%)\n",
      "\n",
      "Total basepairs processed:   188,721,420 bp\n",
      "  Read 1:   112,862,810 bp\n",
      "  Read 2:    75,858,610 bp\n",
      "Total written (filtered):    155,609,062 bp (82.5%)\n",
      "  Read 1:    94,978,565 bp\n",
      "  Read 2:    60,630,497 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 359997 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t5.6\t0\t3\n",
      "9\t4\t1.4\t0\t4\n",
      "10\t3\t0.4\t0\t3\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t4\t0.0\t0\t4\n",
      "14\t12\t0.0\t0\t12\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t19\t0.0\t0\t19\n",
      "17\t109\t0.0\t0\t109\n",
      "18\t56158\t0.0\t0\t56158\n",
      "19\t64949\t0.0\t0\t64949\n",
      "20\t66248\t0.0\t0\t66248\n",
      "21\t59860\t0.0\t0\t59860\n",
      "22\t60699\t0.0\t0\t60699\n",
      "23\t51867\t0.0\t0\t51867\n",
      "24\t42\t0.0\t0\t42\n",
      "25\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 343009 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.4\t0\t1\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t14\t0.0\t0\t14\n",
      "20\t137\t0.0\t0\t137\n",
      "21\t64389\t0.0\t0\t64389\n",
      "22\t61501\t0.0\t0\t61501\n",
      "23\t56595\t0.0\t0\t56595\n",
      "24\t58584\t0.0\t0\t58584\n",
      "25\t52061\t0.0\t0\t52061\n",
      "26\t49654\t0.0\t0\t49654\n",
      "27\t42\t0.0\t0\t42\n",
      "29\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121674_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121674_3_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121674_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121674_3_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121674_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121674_3_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121674_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121674_3_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 5.10 s (9 µs/read; 6.58 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            559,807\n",
      "  Read 1 with adapter:                 544,656 (97.3%)\n",
      "  Read 2 with adapter:                 519,894 (92.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           53,686 (9.6%)\n",
      "Pairs written (passing filters):       506,121 (90.4%)\n",
      "\n",
      "Total basepairs processed:   285,501,570 bp\n",
      "  Read 1:   170,741,135 bp\n",
      "  Read 2:   114,760,435 bp\n",
      "Total written (filtered):    235,957,214 bp (82.6%)\n",
      "  Read 1:   144,019,667 bp\n",
      "  Read 2:    91,937,547 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 544656 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t8.5\t0\t5\n",
      "9\t4\t2.1\t0\t4\n",
      "10\t3\t0.5\t0\t3\n",
      "11\t7\t0.1\t0\t7\n",
      "12\t7\t0.0\t0\t7\n",
      "13\t6\t0.0\t0\t6\n",
      "14\t17\t0.0\t0\t17\n",
      "15\t16\t0.0\t0\t16\n",
      "16\t22\t0.0\t0\t22\n",
      "17\t155\t0.0\t0\t155\n",
      "18\t85410\t0.0\t0\t85410\n",
      "19\t98095\t0.0\t0\t98095\n",
      "20\t100478\t0.0\t0\t100478\n",
      "21\t90125\t0.0\t0\t90125\n",
      "22\t91840\t0.0\t0\t91840\n",
      "23\t78402\t0.0\t0\t78402\n",
      "24\t55\t0.0\t0\t55\n",
      "26\t3\t0.0\t0\t3\n",
      "29\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "35\t1\t0.0\t0\t1\n",
      "42\t1\t0.0\t0\t1\n",
      "43\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 519894 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t5\t0.0\t0\t5\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t15\t0.0\t0\t15\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t9\t0.0\t0\t9\n",
      "18\t6\t0.0\t0\t6\n",
      "19\t17\t0.0\t0\t17\n",
      "20\t180\t0.0\t0\t180\n",
      "21\t98232\t0.0\t0\t98232\n",
      "22\t93652\t0.0\t0\t93652\n",
      "23\t85816\t0.0\t0\t85816\n",
      "24\t87651\t0.0\t0\t87651\n",
      "25\t78341\t0.0\t0\t78341\n",
      "26\t75895\t0.0\t0\t75895\n",
      "27\t62\t0.0\t0\t62\n",
      "36\t1\t0.0\t0\t1\n",
      "57\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121675_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121675_5_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121675_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121675_5_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121675_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121675_5_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121675_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121675_5_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.30 s (10 µs/read; 6.30 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            345,762\n",
      "  Read 1 with adapter:                 336,819 (97.4%)\n",
      "  Read 2 with adapter:                 321,022 (92.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           32,941 (9.5%)\n",
      "Pairs written (passing filters):       312,821 (90.5%)\n",
      "\n",
      "Total basepairs processed:   176,338,620 bp\n",
      "  Read 1:   105,457,410 bp\n",
      "  Read 2:    70,881,210 bp\n",
      "Total written (filtered):    145,841,282 bp (82.7%)\n",
      "  Read 1:    89,015,404 bp\n",
      "  Read 2:    56,825,878 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 336819 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t5.3\t0\t5\n",
      "9\t3\t1.3\t0\t3\n",
      "10\t5\t0.3\t0\t5\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t5\t0.0\t0\t5\n",
      "14\t12\t0.0\t0\t12\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t11\t0.0\t0\t11\n",
      "17\t92\t0.0\t0\t92\n",
      "18\t52679\t0.0\t0\t52679\n",
      "19\t60788\t0.0\t0\t60788\n",
      "20\t62024\t0.0\t0\t62024\n",
      "21\t55828\t0.0\t0\t55828\n",
      "22\t57192\t0.0\t0\t57192\n",
      "23\t48121\t0.0\t0\t48121\n",
      "24\t34\t0.0\t0\t34\n",
      "25\t2\t0.0\t0\t2\n",
      "28\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "36\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 321022 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t3\t0.0\t0\t3\n",
      "15\t12\t0.0\t0\t12\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t11\t0.0\t0\t11\n",
      "20\t106\t0.0\t0\t106\n",
      "21\t60450\t0.0\t0\t60450\n",
      "22\t58128\t0.0\t0\t58128\n",
      "23\t53277\t0.0\t0\t53277\n",
      "24\t54587\t0.0\t0\t54587\n",
      "25\t48050\t0.0\t0\t48050\n",
      "26\t46353\t0.0\t0\t46353\n",
      "27\t34\t0.0\t0\t34\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121676_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121676_7_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121676_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121676_7_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121676_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121676_7_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121676_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121676_7_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 6.85 s (10 µs/read; 6.24 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            712,750\n",
      "  Read 1 with adapter:                 694,205 (97.4%)\n",
      "  Read 2 with adapter:                 664,033 (93.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           65,715 (9.2%)\n",
      "Pairs written (passing filters):       647,035 (90.8%)\n",
      "\n",
      "Total basepairs processed:   363,502,500 bp\n",
      "  Read 1:   217,388,750 bp\n",
      "  Read 2:   146,113,750 bp\n",
      "Total written (filtered):    301,650,677 bp (83.0%)\n",
      "  Read 1:   184,115,305 bp\n",
      "  Read 2:   117,535,372 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 694205 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t10.9\t0\t4\n",
      "9\t9\t2.7\t0\t9\n",
      "10\t5\t0.7\t0\t5\n",
      "11\t14\t0.2\t0\t14\n",
      "12\t8\t0.0\t0\t8\n",
      "13\t6\t0.0\t0\t6\n",
      "14\t30\t0.0\t0\t30\n",
      "15\t18\t0.0\t0\t18\n",
      "16\t33\t0.0\t0\t33\n",
      "17\t183\t0.0\t0\t183\n",
      "18\t108490\t0.0\t0\t108490\n",
      "19\t125122\t0.0\t0\t125122\n",
      "20\t127377\t0.0\t0\t127377\n",
      "21\t114840\t0.0\t0\t114840\n",
      "22\t117757\t0.0\t0\t117757\n",
      "23\t100233\t0.0\t0\t100233\n",
      "24\t68\t0.0\t0\t68\n",
      "25\t1\t0.0\t0\t1\n",
      "28\t2\t0.0\t0\t2\n",
      "35\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "41\t1\t0.0\t0\t1\n",
      "49\t1\t0.0\t0\t1\n",
      "57\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 664033 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.7\t0\t2\n",
      "11\t4\t0.2\t0\t4\n",
      "12\t5\t0.0\t0\t5\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t18\t0.0\t0\t18\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t4\t0.0\t0\t4\n",
      "18\t9\t0.0\t0\t9\n",
      "19\t16\t0.0\t0\t16\n",
      "20\t229\t0.0\t0\t229\n",
      "21\t124781\t0.0\t0\t124781\n",
      "22\t120299\t0.0\t0\t120299\n",
      "23\t109785\t0.0\t0\t109785\n",
      "24\t112135\t0.0\t0\t112135\n",
      "25\t100654\t0.0\t0\t100654\n",
      "26\t95988\t0.0\t0\t95988\n",
      "27\t87\t0.0\t0\t87\n",
      "28\t1\t0.0\t0\t1\n",
      "29\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "47\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "56\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121688_30_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121688_31_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121688_30_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121688_31_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121688_30_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121688_31_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121688_30_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121688_31_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.46 s (11 µs/read; 5.63 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            136,741\n",
      "  Read 1 with adapter:                 132,822 (97.1%)\n",
      "  Read 2 with adapter:                 126,455 (92.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           13,839 (10.1%)\n",
      "Pairs written (passing filters):       122,902 (89.9%)\n",
      "\n",
      "Total basepairs processed:    69,737,910 bp\n",
      "  Read 1:    41,706,005 bp\n",
      "  Read 2:    28,031,905 bp\n",
      "Total written (filtered):     57,294,347 bp (82.2%)\n",
      "  Read 1:    34,969,730 bp\n",
      "  Read 2:    22,324,617 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 132822 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.1\t0\t1\n",
      "9\t2\t0.5\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t11\t0.0\t0\t11\n",
      "17\t32\t0.0\t0\t32\n",
      "18\t20270\t0.0\t0\t20270\n",
      "19\t23588\t0.0\t0\t23588\n",
      "20\t24502\t0.0\t0\t24502\n",
      "21\t22084\t0.0\t0\t22084\n",
      "22\t22934\t0.0\t0\t22934\n",
      "23\t19365\t0.0\t0\t19365\n",
      "24\t14\t0.0\t0\t14\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 126455 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t38\t0.0\t0\t38\n",
      "21\t23579\t0.0\t0\t23579\n",
      "22\t22708\t0.0\t0\t22708\n",
      "23\t21038\t0.0\t0\t21038\n",
      "24\t21560\t0.0\t0\t21560\n",
      "25\t19189\t0.0\t0\t19189\n",
      "26\t18299\t0.0\t0\t18299\n",
      "27\t23\t0.0\t0\t23\n",
      "37\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121689_32_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121689_33_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121689_32_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121689_33_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121689_32_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121689_33_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121689_32_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121689_33_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.84 s (10 µs/read; 5.92 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            181,624\n",
      "  Read 1 with adapter:                 176,525 (97.2%)\n",
      "  Read 2 with adapter:                 167,756 (92.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           18,504 (10.2%)\n",
      "Pairs written (passing filters):       163,120 (89.8%)\n",
      "\n",
      "Total basepairs processed:    92,628,240 bp\n",
      "  Read 1:    55,395,320 bp\n",
      "  Read 2:    37,232,920 bp\n",
      "Total written (filtered):     76,044,296 bp (82.1%)\n",
      "  Read 1:    46,413,857 bp\n",
      "  Read 2:    29,630,439 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 176525 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.8\t0\t1\n",
      "10\t1\t0.2\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t7\t0.0\t0\t7\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t42\t0.0\t0\t42\n",
      "18\t27235\t0.0\t0\t27235\n",
      "19\t31463\t0.0\t0\t31463\n",
      "20\t32408\t0.0\t0\t32408\n",
      "21\t29357\t0.0\t0\t29357\n",
      "22\t30155\t0.0\t0\t30155\n",
      "23\t25810\t0.0\t0\t25810\n",
      "24\t27\t0.0\t0\t27\n",
      "25\t2\t0.0\t0\t2\n",
      "33\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 167756 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.2\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t56\t0.0\t0\t56\n",
      "21\t31305\t0.0\t0\t31305\n",
      "22\t30487\t0.0\t0\t30487\n",
      "23\t27621\t0.0\t0\t27621\n",
      "24\t28517\t0.0\t0\t28517\n",
      "25\t25431\t0.0\t0\t25431\n",
      "26\t24303\t0.0\t0\t24303\n",
      "27\t18\t0.0\t0\t18\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121690_34_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121690_35_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121690_34_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121690_35_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121690_34_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121690_35_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121690_34_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121690_35_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.42 s (11 µs/read; 5.69 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            134,399\n",
      "  Read 1 with adapter:                 130,633 (97.2%)\n",
      "  Read 2 with adapter:                 124,093 (92.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           13,697 (10.2%)\n",
      "Pairs written (passing filters):       120,702 (89.8%)\n",
      "\n",
      "Total basepairs processed:    68,543,490 bp\n",
      "  Read 1:    40,991,695 bp\n",
      "  Read 2:    27,551,795 bp\n",
      "Total written (filtered):     56,270,832 bp (82.1%)\n",
      "  Read 1:    34,345,858 bp\n",
      "  Read 2:    21,924,974 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 130633 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.1\t0\t2\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t44\t0.0\t0\t44\n",
      "18\t20220\t0.0\t0\t20220\n",
      "19\t23666\t0.0\t0\t23666\n",
      "20\t23982\t0.0\t0\t23982\n",
      "21\t21631\t0.0\t0\t21631\n",
      "22\t22313\t0.0\t0\t22313\n",
      "23\t18746\t0.0\t0\t18746\n",
      "24\t10\t0.0\t0\t10\n",
      "26\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 124093 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t33\t0.0\t0\t33\n",
      "21\t23126\t0.0\t0\t23126\n",
      "22\t22324\t0.0\t0\t22324\n",
      "23\t20527\t0.0\t0\t20527\n",
      "24\t21205\t0.0\t0\t21205\n",
      "25\t18881\t0.0\t0\t18881\n",
      "26\t17957\t0.0\t0\t17957\n",
      "27\t22\t0.0\t0\t22\n",
      "31\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121698_50_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121698_51_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121698_50_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121698_51_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121698_50_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121698_51_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121698_50_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121698_51_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.30 s (21 µs/read; 2.87 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             14,319\n",
      "  Read 1 with adapter:                  13,866 (96.8%)\n",
      "  Read 2 with adapter:                  12,833 (89.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,875 (13.1%)\n",
      "Pairs written (passing filters):        12,444 (86.9%)\n",
      "\n",
      "Total basepairs processed:     7,302,690 bp\n",
      "  Read 1:     4,367,295 bp\n",
      "  Read 2:     2,935,395 bp\n",
      "Total written (filtered):      5,801,125 bp (79.4%)\n",
      "  Read 1:     3,540,895 bp\n",
      "  Read 2:     2,260,230 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 13866 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "17\t5\t0.0\t0\t5\n",
      "18\t2170\t0.0\t0\t2170\n",
      "19\t2437\t0.0\t0\t2437\n",
      "20\t2494\t0.0\t0\t2494\n",
      "21\t2324\t0.0\t0\t2324\n",
      "22\t2426\t0.0\t0\t2426\n",
      "23\t2006\t0.0\t0\t2006\n",
      "24\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 12833 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t3\t0.0\t0\t3\n",
      "21\t2351\t0.0\t0\t2351\n",
      "22\t2314\t0.0\t0\t2314\n",
      "23\t2129\t0.0\t0\t2129\n",
      "24\t2175\t0.0\t0\t2175\n",
      "25\t2020\t0.0\t0\t2020\n",
      "26\t1841\t0.0\t0\t1841\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121699_52_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121699_53_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121699_52_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121699_53_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121699_52_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121699_53_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121699_52_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121699_53_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.41 s (19 µs/read; 3.09 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             20,959\n",
      "  Read 1 with adapter:                  20,357 (97.1%)\n",
      "  Read 2 with adapter:                  19,037 (90.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,450 (11.7%)\n",
      "Pairs written (passing filters):        18,509 (88.3%)\n",
      "\n",
      "Total basepairs processed:    10,689,090 bp\n",
      "  Read 1:     6,392,495 bp\n",
      "  Read 2:     4,296,595 bp\n",
      "Total written (filtered):      8,628,512 bp (80.7%)\n",
      "  Read 1:     5,266,608 bp\n",
      "  Read 2:     3,361,904 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 20357 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t7\t0.0\t0\t7\n",
      "18\t3085\t0.0\t0\t3085\n",
      "19\t3612\t0.0\t0\t3612\n",
      "20\t3759\t0.0\t0\t3759\n",
      "21\t3463\t0.0\t0\t3463\n",
      "22\t3487\t0.0\t0\t3487\n",
      "23\t2937\t0.0\t0\t2937\n",
      "24\t5\t0.0\t0\t5\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 19037 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t6\t0.0\t0\t6\n",
      "21\t3547\t0.0\t0\t3547\n",
      "22\t3449\t0.0\t0\t3449\n",
      "23\t3103\t0.0\t0\t3103\n",
      "24\t3209\t0.0\t0\t3209\n",
      "25\t2928\t0.0\t0\t2928\n",
      "26\t2792\t0.0\t0\t2792\n",
      "27\t2\t0.0\t0\t2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121700_54_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121700_55_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121700_54_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121700_55_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121700_54_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121700_55_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121700_54_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121700_55_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.41 s (20 µs/read; 2.97 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             20,194\n",
      "  Read 1 with adapter:                  19,610 (97.1%)\n",
      "  Read 2 with adapter:                  18,045 (89.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,655 (13.1%)\n",
      "Pairs written (passing filters):        17,539 (86.9%)\n",
      "\n",
      "Total basepairs processed:    10,298,940 bp\n",
      "  Read 1:     6,159,170 bp\n",
      "  Read 2:     4,139,770 bp\n",
      "Total written (filtered):      8,176,314 bp (79.4%)\n",
      "  Read 1:     4,990,682 bp\n",
      "  Read 2:     3,185,632 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 19610 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t3\t0.0\t0\t3\n",
      "17\t7\t0.0\t0\t7\n",
      "18\t3003\t0.0\t0\t3003\n",
      "19\t3491\t0.0\t0\t3491\n",
      "20\t3617\t0.0\t0\t3617\n",
      "21\t3224\t0.0\t0\t3224\n",
      "22\t3416\t0.0\t0\t3416\n",
      "23\t2844\t0.0\t0\t2844\n",
      "24\t4\t0.0\t0\t4\n",
      "31\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 18045 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t10\t0.0\t0\t10\n",
      "21\t3352\t0.0\t0\t3352\n",
      "22\t3231\t0.0\t0\t3231\n",
      "23\t2977\t0.0\t0\t2977\n",
      "24\t3007\t0.0\t0\t3007\n",
      "25\t2837\t0.0\t0\t2837\n",
      "26\t2625\t0.0\t0\t2625\n",
      "27\t4\t0.0\t0\t4\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121701_56_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121701_57_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121701_56_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121701_57_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121701_56_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121701_57_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121701_56_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121701_57_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.78 s (12 µs/read; 5.18 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             67,293\n",
      "  Read 1 with adapter:                  65,501 (97.3%)\n",
      "  Read 2 with adapter:                  61,930 (92.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            6,987 (10.4%)\n",
      "Pairs written (passing filters):        60,306 (89.6%)\n",
      "\n",
      "Total basepairs processed:    34,319,430 bp\n",
      "  Read 1:    20,524,365 bp\n",
      "  Read 2:    13,795,065 bp\n",
      "Total written (filtered):     28,113,964 bp (81.9%)\n",
      "  Read 1:    17,159,382 bp\n",
      "  Read 2:    10,954,582 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 65501 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t19\t0.0\t0\t19\n",
      "18\t10169\t0.0\t0\t10169\n",
      "19\t11601\t0.0\t0\t11601\n",
      "20\t12117\t0.0\t0\t12117\n",
      "21\t10719\t0.0\t0\t10719\n",
      "22\t11255\t0.0\t0\t11255\n",
      "23\t9603\t0.0\t0\t9603\n",
      "24\t7\t0.0\t0\t7\n",
      "27\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 61930 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t2\t0.0\t0\t2\n",
      "20\t18\t0.0\t0\t18\n",
      "21\t11697\t0.0\t0\t11697\n",
      "22\t11095\t0.0\t0\t11095\n",
      "23\t10267\t0.0\t0\t10267\n",
      "24\t10352\t0.0\t0\t10352\n",
      "25\t9484\t0.0\t0\t9484\n",
      "26\t9007\t0.0\t0\t9007\n",
      "27\t6\t0.0\t0\t6\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121702_58_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121702_59_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121702_58_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121702_59_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121702_58_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-40bna5mt/S210421121702_59_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121702_58_L001_R1_001.fastq.gz /tmp/qiime2-archive-bvd8ips_/186779aa-411a-46c4-9dc0-75b543daf6bc/data/S210421121702_59_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.26 s (31 µs/read; 1.95 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              8,334\n",
      "  Read 1 with adapter:                   8,081 (97.0%)\n",
      "  Read 2 with adapter:                   7,509 (90.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,046 (12.6%)\n",
      "Pairs written (passing filters):         7,288 (87.4%)\n",
      "\n",
      "Total basepairs processed:     4,250,340 bp\n",
      "  Read 1:     2,541,870 bp\n",
      "  Read 2:     1,708,470 bp\n",
      "Total written (filtered):      3,397,413 bp (79.9%)\n",
      "  Read 1:     2,073,898 bp\n",
      "  Read 2:     1,323,515 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 8081 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t1267\t0.0\t0\t1267\n",
      "19\t1465\t0.0\t0\t1465\n",
      "20\t1495\t0.0\t0\t1495\n",
      "21\t1281\t0.0\t0\t1281\n",
      "22\t1390\t0.0\t0\t1390\n",
      "23\t1178\t0.0\t0\t1178\n",
      "24\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 7509 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t3\t0.0\t0\t3\n",
      "21\t1388\t0.0\t0\t1388\n",
      "22\t1308\t0.0\t0\t1308\n",
      "23\t1248\t0.0\t0\t1248\n",
      "24\t1240\t0.0\t0\t1240\n",
      "25\t1164\t0.0\t0\t1164\n",
      "26\t1154\t0.0\t0\t1154\n",
      "27\t2\t0.0\t0\t2\n",
      "31\t1\t0.0\t0\t1\n",
      "59\t1\t0.0\t0\t1\n"
     ]
    }
   ],
   "source": [
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')\n",
    "\n",
    "if trim and (not os.path.isfile(demux_file_trim) or replace_files):\n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = None\n",
    "    \n",
    "    if n_directions == 1:\n",
    "        res = trim_single(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front=forward_primer,\n",
    "            adapter=reverse_reverse_complement,\n",
    "            cores=threads,\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    elif n_directions == 2:\n",
    "        res = trim_paired(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front_f=forward_primer,\n",
    "            front_r=reverse_primer,\n",
    "            # adapter_f=reverse_reverse_complement,\n",
    "            # adapter_r=forward_reverse_complement,\n",
    "            cores=threads,\n",
    "            overlap=trim['overlap'],\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    else:\n",
    "        print(f'ERROR: invalid number of directions {n_directions}')\n",
    "\n",
    "    if res is not None:\n",
    "        res.save(demux_file_trim)\n",
    "        Visualization.save(demux.visualizers.summarize(res).visualization, filepath=demux_view_trim)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 145.057561,
   "end_time": "2023-08-23T17:40:29.538382",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-templates/step-prepare-data.ipynb",
   "output_path": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-NCxHSD-NC-trim/nb-executed-steps/step-prepare-data-ana-flavia-NCxHSD-NC-trim.ipynb",
   "parameters": {
    "base_dir": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri",
    "class_col": "group-id",
    "classifier_file": "/home/lauro/nupeb/rede-micro/models/silva-138-99-nb-classifier.qza",
    "experiment_name": "ana-flavia-NCxHSD-NC-trim",
    "manifest_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-NCxHSD-NC.csv",
    "metadata_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-NCxHSD-NC.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "top_n": 20,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2023-08-23T17:38:04.480821",
   "version": "2.3.4"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}