{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "ea337206",
   "metadata": {
    "papermill": {
     "duration": 0.018581,
     "end_time": "2022-10-24T18:45:17.297498",
     "exception": false,
     "start_time": "2022-10-24T18:45:17.278917",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "15ce67d0",
   "metadata": {
    "papermill": {
     "duration": 0.011173,
     "end_time": "2022-10-24T18:45:17.320438",
     "exception": false,
     "start_time": "2022-10-24T18:45:17.309265",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0786143e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-10-14T14:33:37.863523Z",
     "start_time": "2022-10-14T14:33:14.145712Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T18:45:17.344206Z",
     "iopub.status.busy": "2022-10-24T18:45:17.343768Z",
     "iopub.status.idle": "2022-10-24T18:45:37.260511Z",
     "shell.execute_reply": "2022-10-24T18:45:37.259635Z"
    },
    "papermill": {
     "duration": 19.932669,
     "end_time": "2022-10-24T18:45:37.264102",
     "exception": false,
     "start_time": "2022-10-24T18:45:17.331433",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7550098a",
   "metadata": {
    "papermill": {
     "duration": 0.011414,
     "end_time": "2022-10-24T18:45:37.294958",
     "exception": false,
     "start_time": "2022-10-24T18:45:37.283544",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "397dd6bc",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.594341Z",
     "start_time": "2022-06-28T18:50:41.591544Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T18:45:37.337255Z",
     "iopub.status.busy": "2022-10-24T18:45:37.337000Z",
     "iopub.status.idle": "2022-10-24T18:45:37.343336Z",
     "shell.execute_reply": "2022-10-24T18:45:37.342572Z"
    },
    "papermill": {
     "duration": 0.020633,
     "end_time": "2022-10-24T18:45:37.344968",
     "exception": false,
     "start_time": "2022-10-24T18:45:37.324335",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'ana-flavia-superlactacao.yaml')\n",
    "experiment_name = 'jenneffer-vs-01'\n",
    "base_dir = os.path.join('/', 'home', 'lauro', 'nupeb', 'redemicro')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'not-hist-vs-manifest.csv')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'imgs'))\n",
    "replace_files = False\n",
    "trim = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d852d08c",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-10-24T18:45:37.386173Z",
     "iopub.status.busy": "2022-10-24T18:45:37.385841Z",
     "iopub.status.idle": "2022-10-24T18:45:37.392941Z",
     "shell.execute_reply": "2022-10-24T18:45:37.392198Z"
    },
    "papermill": {
     "duration": 0.037729,
     "end_time": "2022-10-24T18:45:37.394570",
     "exception": false,
     "start_time": "2022-10-24T18:45:37.356841",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "experiment_name = \"ana-flavia-STD-NCxHSD-NC-trim\"\n",
    "base_dir = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri\"\n",
    "manifest_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-STD-NCxHSD-NC.csv\"\n",
    "metadata_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-STD-NCxHSD-NC.tsv\"\n",
    "class_col = \"group-id\"\n",
    "classifier_file = \"/home/lauro/nupeb/dados_brutos_rede_genoma/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza\"\n",
    "replace_files = False\n",
    "phred = 20\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n",
    "overlap = 12\n",
    "threads = 6\n",
    "trim = {\n",
    "    \"overlap\": 8,\n",
    "    \"forward_primer\": \"CCTACGGGRSGCAGCAG\",\n",
    "    \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\",\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0beafd64",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.610682Z",
     "start_time": "2022-06-28T18:50:41.595886Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T18:45:37.424392Z",
     "iopub.status.busy": "2022-10-24T18:45:37.424038Z",
     "iopub.status.idle": "2022-10-24T18:45:37.430990Z",
     "shell.execute_reply": "2022-10-24T18:45:37.430209Z"
    },
    "papermill": {
     "duration": 0.020642,
     "end_time": "2022-10-24T18:45:37.432595",
     "exception": false,
     "start_time": "2022-10-24T18:45:37.411953",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Setting hardcoded parameters for development and debbuging\n",
    "production = True\n",
    "if not production:\n",
    "    params_path = os.path.join('..', 'params', 'ana-flavia-hipotese-01')\n",
    "    with open(params_path, 'r') as stream:\n",
    "        params = yaml.safe_load(stream)\n",
    "        experiment_name = params['experiment_name']\n",
    "        base_dir = params['base_dir']\n",
    "        manifest_file = params['manifest_file']\n",
    "        replace_files = params['replace_files']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "099d83a0",
   "metadata": {
    "papermill": {
     "duration": 0.010865,
     "end_time": "2022-10-24T18:45:37.456779",
     "exception": false,
     "start_time": "2022-10-24T18:45:37.445914",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2ce61d42",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.618098Z",
     "start_time": "2022-06-28T18:50:41.614152Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T18:45:37.479813Z",
     "iopub.status.busy": "2022-10-24T18:45:37.479462Z",
     "iopub.status.idle": "2022-10-24T18:45:37.483839Z",
     "shell.execute_reply": "2022-10-24T18:45:37.483096Z"
    },
    "papermill": {
     "duration": 0.017882,
     "end_time": "2022-10-24T18:45:37.485412",
     "exception": false,
     "start_time": "2022-10-24T18:45:37.467530",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# new_manifest = '/home/lauro/nupeb/redemicro/data/raw/manifest/karina-manifest.csv'\n",
    "# with open(manifest_file, 'r') as oldm, open(new_manifest, 'w') as newm:\n",
    "#     header = 'sample-id,absolute-filepath,direction\\n'\n",
    "#     newm.write(header)\n",
    "#     for line in oldm.readlines()[1:]:\n",
    "#         sid, forward, reverse = line[:-1].split(',')\n",
    "#         fline = ','.join((sid, forward, 'forward')) + '\\n'\n",
    "#         rline = ','.join((sid, reverse, 'reverse')) + '\\n'\n",
    "#         newm.write(fline)\n",
    "#         newm.write(rline)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a7d966e3",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.632125Z",
     "start_time": "2022-06-28T18:50:41.619710Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T18:45:37.509470Z",
     "iopub.status.busy": "2022-10-24T18:45:37.509125Z",
     "iopub.status.idle": "2022-10-24T18:45:37.520044Z",
     "shell.execute_reply": "2022-10-24T18:45:37.519278Z"
    },
    "papermill": {
     "duration": 0.024839,
     "end_time": "2022-10-24T18:45:37.521547",
     "exception": false,
     "start_time": "2022-10-24T18:45:37.496708",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New qiime-artifacts folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-STD-NCxHSD-NC-trim/qiime-artifacts\n",
      "New img folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-STD-NCxHSD-NC-trim/imgs\n"
     ]
    }
   ],
   "source": [
    "# Define the output folder path\n",
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "\n",
    "# Create path if it not exist\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exist\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')\n",
    "\n",
    "# Define the output artifact full path\n",
    "demux_file = os.path.join(out_dir, 'demux-paired.qza')\n",
    "demux_view = os.path.join(out_dir, 'demux-paired.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a2c9cc29",
   "metadata": {
    "papermill": {
     "duration": 0.011119,
     "end_time": "2022-10-24T18:45:37.544876",
     "exception": false,
     "start_time": "2022-10-24T18:45:37.533757",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file of `SampleData[PairedEndSequencesWithQuality]`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d6f09785",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-10-24T18:45:37.567552Z",
     "iopub.status.busy": "2022-10-24T18:45:37.567396Z",
     "iopub.status.idle": "2022-10-24T18:45:37.575074Z",
     "shell.execute_reply": "2022-10-24T18:45:37.574551Z"
    },
    "papermill": {
     "duration": 0.020876,
     "end_time": "2022-10-24T18:45:37.576744",
     "exception": false,
     "start_time": "2022-10-24T18:45:37.555868",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "manifest_df = pd.read_csv(manifest_file)\n",
    "n_directions = len(manifest_df['direction'].unique())\n",
    "if n_directions == 1:\n",
    "    d_type = 'SampleData[SequencesWithQuality]'\n",
    "    v_type = 'SingleEndFastqManifestPhred33'\n",
    "elif n_directions == 2:\n",
    "    d_type = 'SampleData[PairedEndSequencesWithQuality]'\n",
    "    v_type = 'PairedEndFastqManifestPhred33'\n",
    "else:\n",
    "    print(f'ERROR: invalid number of directions {n_directions}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8a3f0e86",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:44:58.640338Z",
     "start_time": "2022-06-28T18:50:41.633657Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T18:45:37.601929Z",
     "iopub.status.busy": "2022-10-24T18:45:37.601524Z",
     "iopub.status.idle": "2022-10-24T18:52:34.027085Z",
     "shell.execute_reply": "2022-10-24T18:52:34.026339Z"
    },
    "papermill": {
     "duration": 416.440828,
     "end_time": "2022-10-24T18:52:34.030265",
     "exception": false,
     "start_time": "2022-10-24T18:45:37.589437",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    \n",
    "    # Import data and create an artifact object\n",
    "    artifact = Artifact.import_data(\n",
    "        d_type, \n",
    "        manifest_file, \n",
    "        view_type=v_type)\n",
    "    \n",
    "    # Save the artifact object to a new qza file\n",
    "    artifact.save(demux_file)\n",
    "\n",
    "else:\n",
    "    artifact = Artifact.load(demux_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "cb35260e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.602855Z",
     "start_time": "2022-06-28T19:44:58.642520Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T18:52:34.059108Z",
     "iopub.status.busy": "2022-10-24T18:52:34.058734Z",
     "iopub.status.idle": "2022-10-24T18:53:02.045078Z",
     "shell.execute_reply": "2022-10-24T18:53:02.043880Z"
    },
    "papermill": {
     "duration": 28.002435,
     "end_time": "2022-10-24T18:53:02.048704",
     "exception": false,
     "start_time": "2022-10-24T18:52:34.046269",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "if not os.path.isfile(demux_view) or replace_files: \n",
    "    # Generate e visualization of the Artifact\n",
    "    demux_summary = demux.visualizers.summarize(artifact)\n",
    "\n",
    "    # Save a new visualization file based on the qza file\n",
    "    Visualization.save(demux_summary.visualization, filepath=demux_view)\n",
    "    \n",
    "    demux_view_obj = demux_summary.visualization\n",
    "else:\n",
    "    demux_view_obj = Visualization.load(demux_view)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0faccc87",
   "metadata": {
    "papermill": {
     "duration": 0.013505,
     "end_time": "2022-10-24T18:53:02.083181",
     "exception": false,
     "start_time": "2022-10-24T18:53:02.069676",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "db75346a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.609737Z",
     "start_time": "2022-06-28T19:47:56.604820Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T18:53:02.108919Z",
     "iopub.status.busy": "2022-10-24T18:53:02.108596Z",
     "iopub.status.idle": "2022-10-24T18:53:02.114665Z",
     "shell.execute_reply": "2022-10-24T18:53:02.113701Z"
    },
    "papermill": {
     "duration": 0.021106,
     "end_time": "2022-10-24T18:53:02.116510",
     "exception": false,
     "start_time": "2022-10-24T18:53:02.095404",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<visualization: Visualization uuid: df1f840a-bfcc-4c67-a065-4110bc896010>\n"
     ]
    }
   ],
   "source": [
    "print(demux_view_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "4087664c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.704841Z",
     "start_time": "2022-06-28T19:47:56.611846Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T18:53:02.144045Z",
     "iopub.status.busy": "2022-10-24T18:53:02.143713Z",
     "iopub.status.idle": "2022-10-24T18:53:02.194705Z",
     "shell.execute_reply": "2022-10-24T18:53:02.194040Z"
    },
    "papermill": {
     "duration": 0.067169,
     "end_time": "2022-10-24T18:53:02.197690",
     "exception": false,
     "start_time": "2022-10-24T18:53:02.130521",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-fqwew1hy')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: df1f840a-bfcc-4c67-a065-4110bc896010>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "afccd63c",
   "metadata": {
    "papermill": {
     "duration": 0.013148,
     "end_time": "2022-10-24T18:53:02.227797",
     "exception": false,
     "start_time": "2022-10-24T18:53:02.214649",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ee22c540",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-10-24T18:53:02.254996Z",
     "iopub.status.busy": "2022-10-24T18:53:02.254679Z",
     "iopub.status.idle": "2022-10-24T18:54:33.267155Z",
     "shell.execute_reply": "2022-10-24T18:54:33.266518Z"
    },
    "papermill": {
     "duration": 91.029513,
     "end_time": "2022-10-24T18:54:33.270294",
     "exception": false,
     "start_time": "2022-10-24T18:53:02.240781",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121682_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121682_1_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121682_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121682_1_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121682_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121682_1_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121682_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121682_1_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.67 s (16 µs/read; 3.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            165,821\n",
      "  Read 1 with adapter:                 164,234 (99.0%)\n",
      "  Read 2 with adapter:                 162,538 (98.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   2 (0.0%)\n",
      "Pairs discarded as untrimmed:            4,837 (2.9%)\n",
      "Pairs written (passing filters):       160,982 (97.1%)\n",
      "\n",
      "Total basepairs processed:    84,568,710 bp\n",
      "  Read 1:    50,575,405 bp\n",
      "  Read 2:    33,993,305 bp\n",
      "Total written (filtered):     75,046,538 bp (88.7%)\n",
      "  Read 1:    45,805,681 bp\n",
      "  Read 2:    29,240,857 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 38 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 18.4%\n",
      "  G: 73.7%\n",
      "  T: 7.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t2.5\t1\t0 2\n",
      "9\t3\t0.6\t1\t0 3\n",
      "20\t1\t0.0\t3\t1\n",
      "21\t3\t0.0\t3\t2 1\n",
      "22\t3\t0.0\t3\t3\n",
      "23\t3\t0.0\t3\t2 1\n",
      "24\t8\t0.0\t3\t7 1\n",
      "25\t1\t0.0\t3\t1\n",
      "29\t1\t0.0\t3\t1\n",
      "177\t1\t0.0\t3\t1\n",
      "255\t1\t0.0\t3\t0 1\n",
      "258\t1\t0.0\t3\t1\n",
      "265\t1\t0.0\t3\t1\n",
      "266\t1\t0.0\t3\t1\n",
      "268\t1\t0.0\t3\t0 0 1\n",
      "277\t2\t0.0\t3\t2\n",
      "278\t2\t0.0\t3\t2\n",
      "280\t1\t0.0\t3\t0 1\n",
      "281\t1\t0.0\t3\t1\n",
      "284\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 164196 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t2.5\t1\t3\n",
      "9\t1\t0.6\t1\t1\n",
      "10\t6\t0.2\t1\t3 3\n",
      "11\t7\t0.0\t1\t4 3\n",
      "12\t2\t0.0\t1\t1 1\n",
      "13\t3\t0.0\t1\t3\n",
      "14\t13\t0.0\t2\t8 2 3\n",
      "15\t5\t0.0\t2\t1 2 2\n",
      "16\t11\t0.0\t2\t4 4 3\n",
      "17\t85\t0.0\t2\t46 27 12\n",
      "18\t25564\t0.0\t2\t25119 428 17\n",
      "19\t29498\t0.0\t2\t28946 517 35\n",
      "20\t30404\t0.0\t2\t29904 482 18\n",
      "21\t27173\t0.0\t2\t26703 442 28\n",
      "22\t28128\t0.0\t2\t27677 430 21\n",
      "23\t23272\t0.0\t2\t22942 325 5\n",
      "24\t17\t0.0\t2\t12 1 4\n",
      "26\t1\t0.0\t2\t1\n",
      "31\t1\t0.0\t2\t1\n",
      "37\t1\t0.0\t2\t1\n",
      "38\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 8 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 37.5%\n",
      "  G: 25.0%\n",
      "  T: 12.5%\n",
      "  none/other: 25.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.5\t1\t0 1\n",
      "203\t2\t0.0\t2\t2\n",
      "204\t3\t0.0\t2\t3\n",
      "205\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 162530 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.5\t1\t0 1\n",
      "11\t2\t0.0\t1\t1 1\n",
      "12\t1\t0.0\t1\t1\n",
      "14\t2\t0.0\t2\t1 1\n",
      "15\t10\t0.0\t2\t3 3 4\n",
      "16\t5\t0.0\t2\t0 4 1\n",
      "17\t4\t0.0\t2\t1 2 1\n",
      "18\t7\t0.0\t2\t4 1 2\n",
      "19\t9\t0.0\t2\t5 3 1\n",
      "20\t121\t0.0\t3\t54 32 18 17\n",
      "21\t30398\t0.0\t3\t28983 1051 239 125\n",
      "22\t29264\t0.0\t3\t27868 1039 219 138\n",
      "23\t26760\t0.0\t3\t25302 1067 262 129\n",
      "24\t27456\t0.0\t3\t25940 1130 248 138\n",
      "25\t24814\t0.0\t3\t23454 977 250 133\n",
      "26\t23641\t0.0\t3\t22373 932 245 91\n",
      "27\t28\t0.0\t3\t24 2 0 2\n",
      "36\t1\t0.0\t3\t0 0 1\n",
      "52\t1\t0.0\t3\t0 0 0 1\n",
      "54\t1\t0.0\t3\t0 0 0 1\n",
      "59\t1\t0.0\t3\t0 0 0 1\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "62\t2\t0.0\t3\t0 0 0 2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121683_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121683_3_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121683_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121683_3_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121683_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121683_3_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121683_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121683_3_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.30 s (15 µs/read; 3.91 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            149,412\n",
      "  Read 1 with adapter:                 147,947 (99.0%)\n",
      "  Read 2 with adapter:                 146,155 (97.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   9 (0.0%)\n",
      "Pairs discarded as untrimmed:            4,686 (3.1%)\n",
      "Pairs written (passing filters):       144,717 (96.9%)\n",
      "\n",
      "Total basepairs processed:    76,200,120 bp\n",
      "  Read 1:    45,570,660 bp\n",
      "  Read 2:    30,629,460 bp\n",
      "Total written (filtered):     67,462,124 bp (88.5%)\n",
      "  Read 1:    41,176,861 bp\n",
      "  Read 2:    26,285,263 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 105 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 27.6%\n",
      "  G: 71.4%\n",
      "  T: 1.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t8\t2.3\t1\t0 8\n",
      "9\t16\t0.6\t1\t0 16\n",
      "20\t5\t0.0\t3\t5\n",
      "21\t5\t0.0\t3\t5\n",
      "22\t4\t0.0\t3\t4\n",
      "23\t4\t0.0\t3\t4\n",
      "24\t3\t0.0\t3\t3\n",
      "25\t3\t0.0\t3\t3\n",
      "26\t1\t0.0\t3\t1\n",
      "29\t4\t0.0\t3\t4\n",
      "30\t11\t0.0\t3\t10 1\n",
      "31\t7\t0.0\t3\t7\n",
      "32\t12\t0.0\t3\t11 1\n",
      "33\t3\t0.0\t3\t3\n",
      "34\t6\t0.0\t3\t6\n",
      "153\t1\t0.0\t3\t1\n",
      "257\t1\t0.0\t3\t1\n",
      "258\t3\t0.0\t3\t3\n",
      "260\t2\t0.0\t3\t2\n",
      "261\t1\t0.0\t3\t1\n",
      "280\t1\t0.0\t3\t1\n",
      "281\t1\t0.0\t3\t1\n",
      "282\t2\t0.0\t3\t2\n",
      "284\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 147842 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t2.3\t1\t1 1\n",
      "10\t1\t0.1\t1\t1\n",
      "11\t4\t0.0\t1\t2 2\n",
      "12\t4\t0.0\t1\t2 2\n",
      "13\t5\t0.0\t1\t3 2\n",
      "14\t14\t0.0\t2\t8 5 1\n",
      "15\t7\t0.0\t2\t2 1 4\n",
      "16\t13\t0.0\t2\t10 2 1\n",
      "17\t75\t0.0\t2\t45 21 9\n",
      "18\t23193\t0.0\t2\t22745 428 20\n",
      "19\t26424\t0.0\t2\t25902 503 19\n",
      "20\t27036\t0.0\t2\t26577 441 18\n",
      "21\t24700\t0.0\t2\t24277 404 19\n",
      "22\t25051\t0.0\t2\t24615 412 24\n",
      "23\t21290\t0.0\t2\t20941 343 6\n",
      "24\t22\t0.0\t2\t17 3 2\n",
      "37\t1\t0.0\t2\t0 0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 24 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 12.5%\n",
      "  C: 16.7%\n",
      "  G: 20.8%\n",
      "  T: 12.5%\n",
      "  none/other: 37.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t9\t0.0\t2\t7 2\n",
      "204\t6\t0.0\t2\t6\n",
      "205\t9\t0.0\t2\t9\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 146131 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t2\t0.0\t1\t2\n",
      "14\t2\t0.0\t2\t1 0 1\n",
      "15\t3\t0.0\t2\t3\n",
      "16\t2\t0.0\t2\t2\n",
      "17\t1\t0.0\t2\t1\n",
      "18\t5\t0.0\t2\t0 1 4\n",
      "19\t11\t0.0\t2\t5 5 1\n",
      "20\t116\t0.0\t3\t53 33 14 16\n",
      "21\t27337\t0.0\t3\t25986 1000 239 112\n",
      "22\t26355\t0.0\t3\t25058 941 213 143\n",
      "23\t24213\t0.0\t3\t22833 1030 227 123\n",
      "24\t24844\t0.0\t3\t23540 980 222 102\n",
      "25\t22091\t0.0\t3\t20833 932 228 98\n",
      "26\t21122\t0.0\t3\t19937 843 226 116\n",
      "27\t22\t0.0\t3\t16 2 2 2\n",
      "33\t2\t0.0\t3\t2\n",
      "40\t1\t0.0\t3\t1\n",
      "45\t1\t0.0\t3\t1\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121684_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121684_5_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121684_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121684_5_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121684_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121684_5_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121684_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121684_5_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 4.99 s (15 µs/read; 4.13 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            343,261\n",
      "  Read 1 with adapter:                 340,079 (99.1%)\n",
      "  Read 2 with adapter:                 336,731 (98.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   1 (0.0%)\n",
      "Pairs discarded as untrimmed:            9,645 (2.8%)\n",
      "Pairs written (passing filters):       333,615 (97.2%)\n",
      "\n",
      "Total basepairs processed:   175,063,110 bp\n",
      "  Read 1:   104,694,605 bp\n",
      "  Read 2:    70,368,505 bp\n",
      "Total written (filtered):    155,517,885 bp (88.8%)\n",
      "  Read 1:    94,921,854 bp\n",
      "  Read 2:    60,596,031 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 162 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.6%\n",
      "  C: 56.2%\n",
      "  G: 22.8%\n",
      "  T: 20.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t38\t5.2\t1\t0 38\n",
      "9\t41\t1.3\t1\t0 41\n",
      "20\t3\t0.0\t3\t2 1\n",
      "21\t7\t0.0\t3\t7\n",
      "22\t4\t0.0\t3\t3 0 1\n",
      "23\t6\t0.0\t3\t5 1\n",
      "24\t4\t0.0\t3\t4\n",
      "25\t7\t0.0\t3\t7\n",
      "26\t1\t0.0\t3\t1\n",
      "29\t1\t0.0\t3\t1\n",
      "136\t1\t0.0\t3\t1\n",
      "218\t2\t0.0\t3\t0 0 2\n",
      "256\t1\t0.0\t3\t1\n",
      "277\t8\t0.0\t3\t7 1\n",
      "278\t5\t0.0\t3\t5\n",
      "279\t5\t0.0\t3\t5\n",
      "280\t8\t0.0\t3\t7 1\n",
      "281\t11\t0.0\t3\t11\n",
      "282\t6\t0.0\t3\t6\n",
      "283\t3\t0.0\t3\t2 1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 339917 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.2\t1\t1\n",
      "9\t6\t1.3\t1\t5 1\n",
      "10\t4\t0.3\t1\t2 2\n",
      "11\t6\t0.1\t1\t5 1\n",
      "12\t4\t0.0\t1\t2 2\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t17\t0.0\t2\t11 1 5\n",
      "15\t20\t0.0\t2\t9 6 5\n",
      "16\t34\t0.0\t2\t13 12 9\n",
      "17\t167\t0.0\t2\t90 51 26\n",
      "18\t53282\t0.0\t2\t52362 885 35\n",
      "19\t61544\t0.0\t2\t60504 999 41\n",
      "20\t62323\t0.0\t2\t61346 940 37\n",
      "21\t55889\t0.0\t2\t54937 902 50\n",
      "22\t57711\t0.0\t2\t56814 848 49\n",
      "23\t48851\t0.0\t2\t48142 690 19\n",
      "24\t52\t0.0\t2\t46 4 2\n",
      "33\t1\t0.0\t2\t1\n",
      "35\t1\t0.0\t2\t1\n",
      "40\t1\t0.0\t2\t1\n",
      "44\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 14 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 7.1%\n",
      "  C: 7.1%\n",
      "  G: 50.0%\n",
      "  T: 28.6%\n",
      "  none/other: 7.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t1.3\t1\t0 1\n",
      "10\t1\t0.3\t1\t0 1\n",
      "180\t1\t0.0\t2\t1\n",
      "203\t6\t0.0\t2\t3 2 1\n",
      "204\t4\t0.0\t2\t4\n",
      "205\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 336717 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t5.2\t1\t0 2\n",
      "10\t1\t0.3\t1\t0 1\n",
      "11\t1\t0.1\t1\t1\n",
      "12\t2\t0.0\t1\t2\n",
      "13\t2\t0.0\t1\t0 2\n",
      "14\t8\t0.0\t2\t2 5 1\n",
      "15\t20\t0.0\t2\t11 7 2\n",
      "16\t6\t0.0\t2\t1 3 2\n",
      "17\t12\t0.0\t2\t5 1 6\n",
      "18\t7\t0.0\t2\t5 0 2\n",
      "19\t22\t0.0\t2\t9 7 6\n",
      "20\t266\t0.0\t3\t93 92 49 32\n",
      "21\t62624\t0.0\t3\t59527 2272 574 251\n",
      "22\t60388\t0.0\t3\t57389 2182 559 258\n",
      "23\t55640\t0.0\t3\t52503 2291 570 276\n",
      "24\t57281\t0.0\t3\t54259 2205 530 287\n",
      "25\t50943\t0.0\t3\t47921 2110 603 309\n",
      "26\t49434\t0.0\t3\t46600 2071 525 238\n",
      "27\t51\t0.0\t3\t35 7 3 6\n",
      "35\t1\t0.0\t3\t1\n",
      "61\t4\t0.0\t3\t0 1 3\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "64\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121688_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121688_7_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121688_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121688_7_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121688_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121688_7_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121688_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121688_7_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.08 s (15 µs/read; 3.94 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            136,741\n",
      "  Read 1 with adapter:                 135,344 (99.0%)\n",
      "  Read 2 with adapter:                 133,661 (97.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  10 (0.0%)\n",
      "Pairs discarded as untrimmed:            4,417 (3.2%)\n",
      "Pairs written (passing filters):       132,314 (96.8%)\n",
      "\n",
      "Total basepairs processed:    69,737,910 bp\n",
      "  Read 1:    41,706,005 bp\n",
      "  Read 2:    28,031,905 bp\n",
      "Total written (filtered):     61,659,397 bp (88.4%)\n",
      "  Read 1:    37,628,045 bp\n",
      "  Read 2:    24,031,352 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 111 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 1.8%\n",
      "  C: 23.4%\n",
      "  G: 73.0%\n",
      "  T: 1.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t6\t2.1\t1\t0 6\n",
      "9\t5\t0.5\t1\t0 5\n",
      "21\t3\t0.0\t3\t2 1\n",
      "22\t2\t0.0\t3\t1 1\n",
      "24\t2\t0.0\t3\t2\n",
      "25\t1\t0.0\t3\t1\n",
      "26\t1\t0.0\t3\t1\n",
      "29\t2\t0.0\t3\t2\n",
      "30\t1\t0.0\t3\t1\n",
      "31\t2\t0.0\t3\t2\n",
      "32\t2\t0.0\t3\t2\n",
      "33\t1\t0.0\t3\t1\n",
      "34\t1\t0.0\t3\t1\n",
      "133\t1\t0.0\t3\t1\n",
      "151\t1\t0.0\t3\t0 1\n",
      "193\t1\t0.0\t3\t1\n",
      "216\t1\t0.0\t3\t1\n",
      "245\t3\t0.0\t3\t3\n",
      "246\t3\t0.0\t3\t3\n",
      "247\t3\t0.0\t3\t3\n",
      "248\t2\t0.0\t3\t1 1\n",
      "249\t1\t0.0\t3\t1\n",
      "250\t3\t0.0\t3\t2 1\n",
      "278\t1\t0.0\t3\t1\n",
      "280\t1\t0.0\t3\t0 1\n",
      "281\t2\t0.0\t3\t2\n",
      "282\t10\t0.0\t3\t9 1\n",
      "283\t14\t0.0\t3\t10 3 1\n",
      "284\t7\t0.0\t3\t7\n",
      "285\t8\t0.0\t3\t8\n",
      "286\t8\t0.0\t3\t8\n",
      "287\t10\t0.0\t3\t9 1\n",
      "288\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 135233 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.1\t1\t1\n",
      "9\t2\t0.5\t1\t2\n",
      "10\t1\t0.1\t1\t0 1\n",
      "11\t1\t0.0\t1\t0 1\n",
      "12\t5\t0.0\t1\t4 1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t14\t0.0\t2\t8 3 3\n",
      "15\t10\t0.0\t2\t5 1 4\n",
      "16\t19\t0.0\t2\t11 4 4\n",
      "17\t71\t0.0\t2\t32 27 12\n",
      "18\t20647\t0.0\t2\t20253 372 22\n",
      "19\t24041\t0.0\t2\t23576 437 28\n",
      "20\t24968\t0.0\t2\t24488 458 22\n",
      "21\t22490\t0.0\t2\t22070 392 28\n",
      "22\t23290\t0.0\t2\t22913 357 20\n",
      "23\t19653\t0.0\t2\t19351 295 7\n",
      "24\t17\t0.0\t2\t14 2 1\n",
      "45\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 26 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.8%\n",
      "  C: 11.5%\n",
      "  G: 30.8%\n",
      "  T: 15.4%\n",
      "  none/other: 38.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t2.1\t1\t0 2\n",
      "203\t2\t0.0\t2\t2\n",
      "204\t12\t0.0\t2\t12\n",
      "205\t10\t0.0\t2\t10\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 133635 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.1\t1\t0 1\n",
      "11\t1\t0.0\t1\t0 1\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t8\t0.0\t2\t7 0 1\n",
      "16\t7\t0.0\t2\t4 1 2\n",
      "17\t2\t0.0\t2\t1 0 1\n",
      "18\t5\t0.0\t2\t1 1 3\n",
      "19\t7\t0.0\t2\t4 3\n",
      "20\t93\t0.0\t3\t38 24 16 15\n",
      "21\t24828\t0.0\t3\t23579 909 223 117\n",
      "22\t23875\t0.0\t3\t22708 848 216 103\n",
      "23\t22280\t0.0\t3\t21038 929 213 100\n",
      "24\t22734\t0.0\t3\t21560 866 212 96\n",
      "25\t20325\t0.0\t3\t19189 821 199 116\n",
      "26\t19434\t0.0\t3\t18299 827 201 107\n",
      "27\t29\t0.0\t3\t23 4 0 2\n",
      "37\t1\t0.0\t3\t1\n",
      "57\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121689_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121689_9_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121689_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121689_9_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121689_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121689_9_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121689_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121689_9_L001_R2_001.fastq.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.99 s (16 µs/read; 3.65 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            181,624\n",
      "  Read 1 with adapter:                 179,835 (99.0%)\n",
      "  Read 2 with adapter:                 177,952 (98.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   7 (0.0%)\n",
      "Pairs discarded as untrimmed:            5,418 (3.0%)\n",
      "Pairs written (passing filters):       176,199 (97.0%)\n",
      "\n",
      "Total basepairs processed:    92,628,240 bp\n",
      "  Read 1:    55,395,320 bp\n",
      "  Read 2:    37,232,920 bp\n",
      "Total written (filtered):     82,133,955 bp (88.7%)\n",
      "  Read 1:    50,131,407 bp\n",
      "  Read 2:    32,002,548 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 369 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 1.6%\n",
      "  C: 13.8%\n",
      "  G: 83.7%\n",
      "  T: 0.8%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'G' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t26\t2.8\t1\t0 26\n",
      "9\t24\t0.7\t1\t0 24\n",
      "20\t3\t0.0\t3\t3\n",
      "21\t5\t0.0\t3\t5\n",
      "22\t2\t0.0\t3\t2\n",
      "23\t3\t0.0\t3\t2 1\n",
      "24\t3\t0.0\t3\t3\n",
      "25\t5\t0.0\t3\t5\n",
      "29\t37\t0.0\t3\t35 2\n",
      "30\t44\t0.0\t3\t43 1\n",
      "31\t66\t0.0\t3\t64 2\n",
      "32\t44\t0.0\t3\t41 3\n",
      "33\t42\t0.0\t3\t40 2\n",
      "34\t53\t0.0\t3\t52 1\n",
      "36\t1\t0.0\t3\t0 1\n",
      "192\t1\t0.0\t3\t1\n",
      "195\t2\t0.0\t3\t2\n",
      "196\t1\t0.0\t3\t1\n",
      "197\t2\t0.0\t3\t2\n",
      "233\t1\t0.0\t3\t1\n",
      "250\t2\t0.0\t3\t2\n",
      "253\t1\t0.0\t3\t1\n",
      "282\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 179466 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.8\t1\t1\n",
      "10\t3\t0.2\t1\t1 2\n",
      "11\t2\t0.0\t1\t2\n",
      "12\t5\t0.0\t1\t4 1\n",
      "13\t3\t0.0\t1\t2 1\n",
      "14\t11\t0.0\t2\t7 2 2\n",
      "15\t11\t0.0\t2\t3 6 2\n",
      "16\t11\t0.0\t2\t3 5 3\n",
      "17\t98\t0.0\t2\t42 35 21\n",
      "18\t27693\t0.0\t2\t27177 498 18\n",
      "19\t32014\t0.0\t2\t31417 571 26\n",
      "20\t32946\t0.0\t2\t32359 561 26\n",
      "21\t29829\t0.0\t2\t29292 506 31\n",
      "22\t30649\t0.0\t2\t30106 528 15\n",
      "23\t26157\t0.0\t2\t25767 380 10\n",
      "24\t29\t0.0\t2\t27 1 1\n",
      "25\t2\t0.0\t2\t2\n",
      "33\t1\t0.0\t2\t1\n",
      "39\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 21 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 4.8%\n",
      "  C: 19.0%\n",
      "  G: 14.3%\n",
      "  T: 28.6%\n",
      "  none/other: 33.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "179\t1\t0.0\t2\t0 0 1\n",
      "203\t4\t0.0\t2\t4\n",
      "204\t9\t0.0\t2\t9\n",
      "205\t7\t0.0\t2\t7\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 177931 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.2\t1\t1\n",
      "13\t1\t0.0\t1\t1\n",
      "14\t2\t0.0\t2\t0 2\n",
      "15\t8\t0.0\t2\t8\n",
      "16\t2\t0.0\t2\t1 1\n",
      "17\t1\t0.0\t2\t1\n",
      "18\t8\t0.0\t2\t3 4 1\n",
      "19\t8\t0.0\t2\t3 1 4\n",
      "20\t137\t0.0\t3\t56 38 22 21\n",
      "21\t33009\t0.0\t3\t31305 1276 277 151\n",
      "22\t32157\t0.0\t3\t30487 1181 344 145\n",
      "23\t29386\t0.0\t3\t27621 1320 299 146\n",
      "24\t30221\t0.0\t3\t28517 1204 328 172\n",
      "25\t27046\t0.0\t3\t25431 1158 304 153\n",
      "26\t25904\t0.0\t3\t24303 1154 306 141\n",
      "27\t27\t0.0\t3\t18 4 5\n",
      "28\t1\t0.0\t3\t0 0 1\n",
      "60\t2\t0.0\t3\t0 0 0 2\n",
      "61\t1\t0.0\t3\t0 0 1\n",
      "62\t4\t0.0\t3\t0 0 1 3\n",
      "63\t1\t0.0\t3\t0 1\n",
      "64\t2\t0.0\t3\t0 0 1 1\n",
      "65\t2\t0.0\t3\t0 0 0 2\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121690_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121690_11_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121690_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121690_11_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121690_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121690_11_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121690_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121690_11_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.09 s (16 µs/read; 3.86 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            134,399\n",
      "  Read 1 with adapter:                 133,122 (99.0%)\n",
      "  Read 2 with adapter:                 131,694 (98.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   2 (0.0%)\n",
      "Pairs discarded as untrimmed:            3,948 (2.9%)\n",
      "Pairs written (passing filters):       130,449 (97.1%)\n",
      "\n",
      "Total basepairs processed:    68,543,490 bp\n",
      "  Read 1:    40,991,695 bp\n",
      "  Read 2:    27,551,795 bp\n",
      "Total written (filtered):     60,814,970 bp (88.7%)\n",
      "  Read 1:    37,120,467 bp\n",
      "  Read 2:    23,694,503 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 78 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 46.2%\n",
      "  G: 53.8%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t23\t2.1\t1\t0 23\n",
      "9\t13\t0.5\t1\t0 13\n",
      "20\t6\t0.0\t3\t4 1 1\n",
      "21\t5\t0.0\t3\t3 1 1\n",
      "22\t6\t0.0\t3\t5 0 1\n",
      "23\t5\t0.0\t3\t5\n",
      "24\t8\t0.0\t3\t8\n",
      "25\t8\t0.0\t3\t7 1\n",
      "28\t1\t0.0\t3\t1\n",
      "31\t1\t0.0\t3\t1\n",
      "34\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 133044 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.1\t1\t0 1\n",
      "10\t2\t0.1\t1\t2\n",
      "11\t3\t0.0\t1\t2 1\n",
      "12\t3\t0.0\t1\t1 2\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t7\t0.0\t2\t5 2\n",
      "15\t2\t0.0\t2\t2\n",
      "16\t8\t0.0\t2\t6 0 2\n",
      "17\t85\t0.0\t2\t44 31 10\n",
      "18\t20575\t0.0\t2\t20211 344 20\n",
      "19\t24150\t0.0\t2\t23659 464 27\n",
      "20\t24402\t0.0\t2\t23976 398 28\n",
      "21\t22045\t0.0\t2\t21625 397 23\n",
      "22\t22706\t0.0\t2\t22309 374 23\n",
      "23\t19038\t0.0\t2\t18739 293 6\n",
      "24\t14\t0.0\t2\t10 0 4\n",
      "26\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 6 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 16.7%\n",
      "  G: 16.7%\n",
      "  T: 33.3%\n",
      "  none/other: 33.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t1\t0 1\n",
      "203\t2\t0.0\t2\t1 1\n",
      "204\t1\t0.0\t2\t1\n",
      "205\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 131688 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t5\t0.0\t2\t2 1 2\n",
      "16\t2\t0.0\t2\t1 1\n",
      "17\t6\t0.0\t2\t3 1 2\n",
      "18\t2\t0.0\t2\t2\n",
      "19\t12\t0.0\t2\t5 4 3\n",
      "20\t107\t0.0\t3\t33 38 20 16\n",
      "21\t24486\t0.0\t3\t23126 1008 235 117\n",
      "22\t23576\t0.0\t3\t22324 901 230 121\n",
      "23\t21821\t0.0\t3\t20527 954 222 118\n",
      "24\t22473\t0.0\t3\t21205 922 222 124\n",
      "25\t20054\t0.0\t3\t18881 860 208 105\n",
      "26\t19109\t0.0\t3\t17957 833 205 114\n",
      "27\t26\t0.0\t3\t22 3 0 1\n",
      "31\t1\t0.0\t3\t1\n",
      "39\t1\t0.0\t3\t1\n",
      "62\t1\t0.0\t3\t0 0 1\n",
      "63\t1\t0.0\t3\t0 0 0 1\n",
      "64\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121694_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121694_13_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121694_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121694_13_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121694_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121694_13_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121694_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121694_13_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.07 s (18 µs/read; 3.32 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             58,992\n",
      "  Read 1 with adapter:                  58,444 (99.1%)\n",
      "  Read 2 with adapter:                  57,446 (97.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  12 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,078 (3.5%)\n",
      "Pairs written (passing filters):        56,902 (96.5%)\n",
      "\n",
      "Total basepairs processed:    30,085,920 bp\n",
      "  Read 1:    17,992,560 bp\n",
      "  Read 2:    12,093,360 bp\n",
      "Total written (filtered):     26,521,205 bp (88.2%)\n",
      "  Read 1:    16,187,827 bp\n",
      "  Read 2:    10,333,378 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 88 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 2.3%\n",
      "  C: 35.2%\n",
      "  G: 59.1%\n",
      "  T: 3.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t6\t0.9\t1\t0 6\n",
      "9\t11\t0.2\t1\t0 11\n",
      "20\t1\t0.0\t3\t1\n",
      "21\t1\t0.0\t3\t1\n",
      "22\t1\t0.0\t3\t1\n",
      "24\t1\t0.0\t3\t1\n",
      "29\t4\t0.0\t3\t4\n",
      "30\t9\t0.0\t3\t8 1\n",
      "31\t8\t0.0\t3\t8\n",
      "32\t7\t0.0\t3\t6 1\n",
      "33\t13\t0.0\t3\t12 0 1\n",
      "34\t7\t0.0\t3\t7\n",
      "110\t1\t0.0\t3\t1\n",
      "246\t1\t0.0\t3\t1\n",
      "252\t4\t0.0\t3\t4\n",
      "253\t2\t0.0\t3\t2\n",
      "254\t2\t0.0\t3\t2\n",
      "255\t2\t0.0\t3\t2\n",
      "256\t5\t0.0\t3\t5\n",
      "257\t1\t0.0\t3\t1\n",
      "283\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 58356 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.2\t1\t1\n",
      "10\t1\t0.1\t1\t0 1\n",
      "11\t1\t0.0\t1\t1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t8\t0.0\t2\t6 1 1\n",
      "16\t4\t0.0\t2\t3 1\n",
      "17\t23\t0.0\t2\t10 9 4\n",
      "18\t9079\t0.0\t2\t8912 158 9\n",
      "19\t10538\t0.0\t2\t10322 207 9\n",
      "20\t10856\t0.0\t2\t10657 191 8\n",
      "21\t9542\t0.0\t2\t9365 166 11\n",
      "22\t9915\t0.0\t2\t9746 162 7\n",
      "23\t8378\t0.0\t2\t8239 136 3\n",
      "24\t6\t0.0\t2\t6\n",
      "40\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 28 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.6%\n",
      "  C: 14.3%\n",
      "  G: 17.9%\n",
      "  T: 21.4%\n",
      "  none/other: 42.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.9\t1\t0 1\n",
      "11\t1\t0.0\t1\t0 1\n",
      "150\t1\t0.0\t2\t0 1\n",
      "203\t4\t0.0\t2\t4\n",
      "204\t9\t0.0\t2\t9\n",
      "205\t12\t0.0\t2\t12\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 57418 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t1\t1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t2\t0.0\t2\t2\n",
      "16\t1\t0.0\t2\t1\n",
      "17\t2\t0.0\t2\t1 0 1\n",
      "18\t3\t0.0\t2\t2 1\n",
      "19\t5\t0.0\t2\t0 2 3\n",
      "20\t38\t0.0\t3\t18 6 7 7\n",
      "21\t10758\t0.0\t3\t10166 438 100 54\n",
      "22\t10350\t0.0\t3\t9802 401 99 48\n",
      "23\t9350\t0.0\t3\t8793 400 114 43\n",
      "24\t9766\t0.0\t3\t9258 373 80 55\n",
      "25\t8706\t0.0\t3\t8219 343 90 54\n",
      "26\t8422\t0.0\t3\t7938 358 82 44\n",
      "27\t10\t0.0\t3\t6 2 0 2\n",
      "58\t1\t0.0\t3\t0 0 0 1\n",
      "61\t2\t0.0\t3\t0 0 0 2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121695_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121695_15_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121695_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121695_15_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121695_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121695_15_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121695_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121695_15_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.68 s (16 µs/read; 3.75 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            229,809\n",
      "  Read 1 with adapter:                 227,628 (99.1%)\n",
      "  Read 2 with adapter:                 224,970 (97.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  10 (0.0%)\n",
      "Pairs discarded as untrimmed:            6,969 (3.0%)\n",
      "Pairs written (passing filters):       222,830 (97.0%)\n",
      "\n",
      "Total basepairs processed:   117,202,590 bp\n",
      "  Read 1:    70,091,745 bp\n",
      "  Read 2:    47,110,845 bp\n",
      "Total written (filtered):    103,871,094 bp (88.6%)\n",
      "  Read 1:    63,399,872 bp\n",
      "  Read 2:    40,471,222 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 372 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.2%\n",
      "  C: 12.1%\n",
      "  G: 82.0%\n",
      "  T: 2.7%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'G' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t21\t3.5\t1\t0 21\n",
      "9\t23\t0.9\t1\t0 23\n",
      "10\t1\t0.2\t1\t0 1\n",
      "20\t2\t0.0\t3\t2\n",
      "21\t3\t0.0\t3\t3\n",
      "22\t4\t0.0\t3\t4\n",
      "23\t6\t0.0\t3\t6\n",
      "24\t3\t0.0\t3\t3\n",
      "25\t2\t0.0\t3\t2\n",
      "29\t38\t0.0\t3\t35 3\n",
      "30\t52\t0.0\t3\t49 1 2\n",
      "31\t52\t0.0\t3\t51 1\n",
      "32\t52\t0.0\t3\t51 1\n",
      "33\t43\t0.0\t3\t40 3\n",
      "34\t42\t0.0\t3\t42\n",
      "129\t2\t0.0\t3\t2\n",
      "187\t1\t0.0\t3\t1\n",
      "192\t1\t0.0\t3\t1\n",
      "193\t1\t0.0\t3\t1\n",
      "195\t3\t0.0\t3\t2 1\n",
      "196\t1\t0.0\t3\t1\n",
      "197\t2\t0.0\t3\t2\n",
      "209\t1\t0.0\t3\t1\n",
      "215\t1\t0.0\t3\t1\n",
      "217\t2\t0.0\t3\t1 1\n",
      "218\t1\t0.0\t3\t1\n",
      "219\t3\t0.0\t3\t2 1\n",
      "239\t1\t0.0\t3\t1\n",
      "243\t2\t0.0\t3\t1 1\n",
      "247\t1\t0.0\t3\t1\n",
      "248\t2\t0.0\t3\t2\n",
      "249\t1\t0.0\t3\t1\n",
      "277\t1\t0.0\t3\t1\n",
      "281\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 227256 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t3.5\t1\t2\n",
      "9\t4\t0.9\t1\t4\n",
      "10\t4\t0.2\t1\t3 1\n",
      "11\t6\t0.1\t1\t6\n",
      "12\t8\t0.0\t1\t6 2\n",
      "13\t3\t0.0\t1\t2 1\n",
      "14\t19\t0.0\t2\t14 3 2\n",
      "15\t17\t0.0\t2\t8 5 4\n",
      "16\t10\t0.0\t2\t6 1 3\n",
      "17\t119\t0.0\t2\t64 43 12\n",
      "18\t35246\t0.0\t2\t34628 590 28\n",
      "19\t40769\t0.0\t2\t39998 740 31\n",
      "20\t41965\t0.0\t2\t41231 693 41\n",
      "21\t37951\t0.0\t2\t37242 666 43\n",
      "22\t38582\t0.0\t2\t37947 606 29\n",
      "23\t32515\t0.0\t2\t32044 462 9\n",
      "24\t31\t0.0\t2\t27 1 3\n",
      "26\t1\t0.0\t2\t1\n",
      "27\t1\t0.0\t2\t0 1\n",
      "31\t1\t0.0\t2\t1\n",
      "34\t1\t0.0\t2\t1\n",
      "35\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 26 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 11.5%\n",
      "  C: 7.7%\n",
      "  G: 19.2%\n",
      "  T: 23.1%\n",
      "  none/other: 38.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.9\t1\t0 1\n",
      "10\t1\t0.2\t1\t0 1\n",
      "86\t1\t0.0\t2\t1\n",
      "203\t6\t0.0\t2\t6\n",
      "204\t7\t0.0\t2\t7\n",
      "205\t10\t0.0\t2\t9 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 224944 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t1\t0 1\n",
      "12\t1\t0.0\t1\t0 1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t2\t0.0\t2\t0 1 1\n",
      "15\t5\t0.0\t2\t5\n",
      "16\t7\t0.0\t2\t1 4 2\n",
      "17\t6\t0.0\t2\t4 2\n",
      "18\t4\t0.0\t2\t0 3 1\n",
      "19\t14\t0.0\t2\t5 6 3\n",
      "20\t160\t0.0\t3\t63 42 25 30\n",
      "21\t41811\t0.0\t3\t39586 1635 404 186\n",
      "22\t40375\t0.0\t3\t38305 1501 372 197\n",
      "23\t36809\t0.0\t3\t34643 1598 382 186\n",
      "24\t38209\t0.0\t3\t36017 1623 373 196\n",
      "25\t34398\t0.0\t3\t32350 1484 373 191\n",
      "26\t33094\t0.0\t3\t31075 1494 350 175\n",
      "27\t33\t0.0\t3\t26 5 1 1\n",
      "54\t1\t0.0\t3\t0 0 0 1\n",
      "58\t1\t0.0\t3\t0 0 0 1\n",
      "59\t2\t0.0\t3\t0 0 1 1\n",
      "60\t2\t0.0\t3\t0 0 0 2\n",
      "61\t4\t0.0\t3\t0 1 1 2\n",
      "63\t1\t0.0\t3\t0 0 1\n",
      "64\t1\t0.0\t3\t0 0 0 1\n",
      "65\t1\t0.0\t3\t0 1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121696_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121696_17_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121696_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121696_17_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121696_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121696_17_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121696_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121696_17_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.86 s (16 µs/read; 3.70 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            114,634\n",
      "  Read 1 with adapter:                 113,440 (99.0%)\n",
      "  Read 2 with adapter:                 111,771 (97.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  14 (0.0%)\n",
      "Pairs discarded as untrimmed:            4,005 (3.5%)\n",
      "Pairs written (passing filters):       110,615 (96.5%)\n",
      "\n",
      "Total basepairs processed:    58,463,340 bp\n",
      "  Read 1:    34,963,370 bp\n",
      "  Read 2:    23,499,970 bp\n",
      "Total written (filtered):     51,562,559 bp (88.2%)\n",
      "  Read 1:    31,472,632 bp\n",
      "  Read 2:    20,089,927 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 47 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 34.0%\n",
      "  G: 63.8%\n",
      "  T: 2.1%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t7\t1.7\t1\t0 7\n",
      "21\t1\t0.0\t3\t1\n",
      "22\t4\t0.0\t3\t3 0 1\n",
      "23\t2\t0.0\t3\t2\n",
      "24\t6\t0.0\t3\t6\n",
      "25\t2\t0.0\t3\t2\n",
      "26\t4\t0.0\t3\t4\n",
      "30\t2\t0.0\t3\t2\n",
      "31\t2\t0.0\t3\t2\n",
      "32\t3\t0.0\t3\t3\n",
      "34\t3\t0.0\t3\t3\n",
      "244\t1\t0.0\t3\t1\n",
      "279\t1\t0.0\t3\t1\n",
      "280\t2\t0.0\t3\t2\n",
      "281\t2\t0.0\t3\t2\n",
      "282\t2\t0.0\t3\t2\n",
      "283\t2\t0.0\t3\t2\n",
      "284\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 113393 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.7\t1\t1\n",
      "9\t1\t0.4\t1\t0 1\n",
      "10\t4\t0.1\t1\t4\n",
      "11\t6\t0.0\t1\t5 1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t6\t0.0\t2\t5 0 1\n",
      "15\t10\t0.0\t2\t5 1 4\n",
      "16\t5\t0.0\t2\t3 1 1\n",
      "17\t60\t0.0\t2\t33 19 8\n",
      "18\t17704\t0.0\t2\t17399 295 10\n",
      "19\t20264\t0.0\t2\t19910 336 18\n",
      "20\t20686\t0.0\t2\t20306 361 19\n",
      "21\t18830\t0.0\t2\t18485 323 22\n",
      "22\t19209\t0.0\t2\t18874 322 13\n",
      "23\t16580\t0.0\t2\t16322 254 4\n",
      "24\t22\t0.0\t2\t19 1 2\n",
      "28\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 25 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 4.0%\n",
      "  C: 16.0%\n",
      "  G: 0.0%\n",
      "  T: 24.0%\n",
      "  none/other: 56.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.7\t1\t0 1\n",
      "203\t6\t0.0\t2\t6\n",
      "204\t4\t0.0\t2\t4\n",
      "205\t14\t0.0\t2\t14\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 111746 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t1.7\t1\t1 3\n",
      "13\t1\t0.0\t1\t1\n",
      "15\t4\t0.0\t2\t2 2\n",
      "16\t3\t0.0\t2\t2 0 1\n",
      "17\t4\t0.0\t2\t1 2 1\n",
      "18\t6\t0.0\t2\t2 2 2\n",
      "19\t8\t0.0\t2\t3 2 3\n",
      "20\t93\t0.0\t3\t38 28 17 10\n",
      "21\t20831\t0.0\t3\t19666 863 188 114\n",
      "22\t19961\t0.0\t3\t18723 875 244 119\n",
      "23\t18302\t0.0\t3\t17184 798 210 110\n",
      "24\t19164\t0.0\t3\t18017 823 198 126\n",
      "25\t16910\t0.0\t3\t15803 792 202 113\n",
      "26\t16422\t0.0\t3\t15339 774 200 109\n",
      "27\t22\t0.0\t3\t17 3 1 1\n",
      "49\t1\t0.0\t3\t1\n",
      "55\t1\t0.0\t3\t0 0 0 1\n",
      "59\t1\t0.0\t3\t0 0 1\n",
      "60\t2\t0.0\t3\t0 2\n",
      "61\t3\t0.0\t3\t0 0 1 2\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "63\t1\t0.0\t3\t0 0 0 1\n",
      "64\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121697_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121697_19_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121697_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121697_19_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121697_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121697_19_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121697_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121697_19_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.96 s (20 µs/read; 3.07 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             49,335\n",
      "  Read 1 with adapter:                  48,895 (99.1%)\n",
      "  Read 2 with adapter:                  48,247 (97.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   5 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,514 (3.1%)\n",
      "Pairs written (passing filters):        47,816 (96.9%)\n",
      "\n",
      "Total basepairs processed:    25,160,850 bp\n",
      "  Read 1:    15,047,175 bp\n",
      "  Read 2:    10,113,675 bp\n",
      "Total written (filtered):     22,289,755 bp (88.6%)\n",
      "  Read 1:    13,605,875 bp\n",
      "  Read 2:     8,683,880 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 31 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.2%\n",
      "  C: 74.2%\n",
      "  G: 19.4%\n",
      "  T: 3.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t14\t0.8\t1\t0 14\n",
      "9\t10\t0.2\t1\t0 10\n",
      "20\t4\t0.0\t3\t3 0 1\n",
      "21\t1\t0.0\t3\t1\n",
      "25\t1\t0.0\t3\t1\n",
      "262\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 48864 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.2\t1\t1\n",
      "10\t1\t0.0\t1\t0 1\n",
      "11\t2\t0.0\t1\t2\n",
      "12\t1\t0.0\t1\t1\n",
      "14\t2\t0.0\t2\t1 0 1\n",
      "15\t5\t0.0\t2\t2 1 2\n",
      "16\t7\t0.0\t2\t3 3 1\n",
      "17\t31\t0.0\t2\t16 10 5\n",
      "18\t7412\t0.0\t2\t7268 138 6\n",
      "19\t8842\t0.0\t2\t8668 166 8\n",
      "20\t9116\t0.0\t2\t8944 161 11\n",
      "21\t7915\t0.0\t2\t7771 135 9\n",
      "22\t8401\t0.0\t2\t8228 165 8\n",
      "23\t7119\t0.0\t2\t7011 108\n",
      "24\t9\t0.0\t2\t7 2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 11 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 9.1%\n",
      "  G: 18.2%\n",
      "  T: 27.3%\n",
      "  none/other: 45.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t5\t0.0\t2\t4 1\n",
      "204\t1\t0.0\t2\t1\n",
      "205\t5\t0.0\t2\t4 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 48236 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t1\t1\n",
      "15\t2\t0.0\t2\t1 0 1\n",
      "16\t1\t0.0\t2\t1\n",
      "17\t1\t0.0\t2\t1\n",
      "18\t2\t0.0\t2\t1 1\n",
      "19\t1\t0.0\t2\t0 1\n",
      "20\t34\t0.0\t3\t14 11 6 3\n",
      "21\t8963\t0.0\t3\t8529 330 67 37\n",
      "22\t8658\t0.0\t3\t8218 313 77 50\n",
      "23\t7982\t0.0\t3\t7506 350 89 37\n",
      "24\t8054\t0.0\t3\t7601 339 67 47\n",
      "25\t7425\t0.0\t3\t6998 304 89 34\n",
      "26\t7102\t0.0\t3\t6678 304 77 43\n",
      "27\t3\t0.0\t3\t3\n",
      "32\t1\t0.0\t3\t1\n",
      "57\t1\t0.0\t3\t0 0 0 1\n",
      "58\t1\t0.0\t3\t0 0 0 1\n",
      "63\t2\t0.0\t3\t0 0 2\n",
      "64\t2\t0.0\t3\t0 0 0 2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121698_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121698_21_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121698_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121698_21_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121698_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121698_21_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121698_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121698_21_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.43 s (30 µs/read; 1.99 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             14,319\n",
      "  Read 1 with adapter:                  14,165 (98.9%)\n",
      "  Read 2 with adapter:                  13,639 (95.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  11 (0.1%)\n",
      "Pairs discarded as untrimmed:              822 (5.7%)\n",
      "Pairs written (passing filters):        13,486 (94.2%)\n",
      "\n",
      "Total basepairs processed:     7,302,690 bp\n",
      "  Read 1:     4,367,295 bp\n",
      "  Read 2:     2,935,395 bp\n",
      "Total written (filtered):      6,284,106 bp (86.1%)\n",
      "  Read 1:     3,837,039 bp\n",
      "  Read 2:     2,447,067 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 13 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 7.7%\n",
      "  C: 38.5%\n",
      "  G: 38.5%\n",
      "  T: 7.7%\n",
      "  none/other: 7.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.2\t1\t0 1\n",
      "9\t4\t0.1\t1\t0 4\n",
      "20\t1\t0.0\t3\t1\n",
      "21\t1\t0.0\t3\t1\n",
      "23\t1\t0.0\t3\t1\n",
      "24\t1\t0.0\t3\t1\n",
      "25\t1\t0.0\t3\t1\n",
      "257\t1\t0.0\t3\t1\n",
      "282\t1\t0.0\t3\t0 1\n",
      "305\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 14152 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t1\t1\n",
      "15\t1\t0.0\t2\t1\n",
      "17\t12\t0.0\t2\t5 6 1\n",
      "18\t2212\t0.0\t2\t2169 43\n",
      "19\t2485\t0.0\t2\t2436 48 1\n",
      "20\t2558\t0.0\t2\t2493 64 1\n",
      "21\t2375\t0.0\t2\t2324 50 1\n",
      "22\t2475\t0.0\t2\t2424 49 2\n",
      "23\t2031\t0.0\t2\t2005 25 1\n",
      "24\t1\t0.0\t2\t1\n",
      "45\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 23 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 4.3%\n",
      "  C: 21.7%\n",
      "  G: 21.7%\n",
      "  T: 8.7%\n",
      "  none/other: 43.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t4\t0.0\t2\t4\n",
      "204\t9\t0.0\t2\t6 3\n",
      "205\t10\t0.0\t2\t10\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 13616 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t1\t0.0\t2\t0 1\n",
      "19\t1\t0.0\t2\t0 1\n",
      "20\t10\t0.0\t3\t3 3 1 3\n",
      "21\t2476\t0.0\t3\t2351 86 24 15\n",
      "22\t2460\t0.0\t3\t2314 101 32 13\n",
      "23\t2282\t0.0\t3\t2129 112 26 15\n",
      "24\t2295\t0.0\t3\t2175 89 20 11\n",
      "25\t2135\t0.0\t3\t2020 78 28 9\n",
      "26\t1952\t0.0\t3\t1841 87 12 12\n",
      "27\t1\t0.0\t3\t0 0 1\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "64\t2\t0.0\t3\t0 1 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121699_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121699_23_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121699_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121699_23_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121699_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121699_23_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121699_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121699_23_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.57 s (27 µs/read; 2.20 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             20,959\n",
      "  Read 1 with adapter:                  20,764 (99.1%)\n",
      "  Read 2 with adapter:                  20,210 (96.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   6 (0.0%)\n",
      "Pairs discarded as untrimmed:              930 (4.4%)\n",
      "Pairs written (passing filters):        20,023 (95.5%)\n",
      "\n",
      "Total basepairs processed:    10,689,090 bp\n",
      "  Read 1:     6,392,495 bp\n",
      "  Read 2:     4,296,595 bp\n",
      "Total written (filtered):      9,331,198 bp (87.3%)\n",
      "  Read 1:     5,696,317 bp\n",
      "  Read 2:     3,634,881 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 14 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 21.4%\n",
      "  G: 64.3%\n",
      "  T: 14.3%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.3\t1\t0 1\n",
      "9\t2\t0.1\t1\t0 2\n",
      "21\t1\t0.0\t3\t1\n",
      "22\t1\t0.0\t3\t1\n",
      "26\t1\t0.0\t3\t1\n",
      "30\t1\t0.0\t3\t1\n",
      "32\t1\t0.0\t3\t1\n",
      "33\t1\t0.0\t3\t1\n",
      "215\t1\t0.0\t3\t1\n",
      "248\t2\t0.0\t3\t2\n",
      "277\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 20750 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t3\t0.0\t2\t1 1 1\n",
      "16\t2\t0.0\t2\t1 0 1\n",
      "17\t9\t0.0\t2\t7 1 1\n",
      "18\t3158\t0.0\t2\t3084 70 4\n",
      "19\t3678\t0.0\t2\t3609 63 6\n",
      "20\t3818\t0.0\t2\t3758 57 3\n",
      "21\t3532\t0.0\t2\t3462 65 5\n",
      "22\t3549\t0.0\t2\t3485 59 5\n",
      "23\t2996\t0.0\t2\t2934 61 1\n",
      "24\t5\t0.0\t2\t5\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 16 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 12.5%\n",
      "  G: 25.0%\n",
      "  T: 25.0%\n",
      "  none/other: 37.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t3\t0.0\t2\t3\n",
      "204\t7\t0.0\t2\t6 1\n",
      "205\t6\t0.0\t2\t6\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 20194 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t2\t0.0\t2\t0 1 1\n",
      "19\t1\t0.0\t2\t1\n",
      "20\t15\t0.0\t3\t6 5 1 3\n",
      "21\t3733\t0.0\t3\t3547 126 39 21\n",
      "22\t3631\t0.0\t3\t3449 129 39 14\n",
      "23\t3299\t0.0\t3\t3103 146 28 22\n",
      "24\t3404\t0.0\t3\t3209 139 34 22\n",
      "25\t3123\t0.0\t3\t2928 142 30 23\n",
      "26\t2979\t0.0\t3\t2792 136 36 15\n",
      "27\t5\t0.0\t3\t2 1 1 1\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "66\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121700_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121700_25_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121700_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121700_25_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121700_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121700_25_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121700_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121700_25_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.52 s (26 µs/read; 2.34 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             20,194\n",
      "  Read 1 with adapter:                  19,984 (99.0%)\n",
      "  Read 2 with adapter:                  19,270 (95.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   8 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,124 (5.6%)\n",
      "Pairs written (passing filters):        19,062 (94.4%)\n",
      "\n",
      "Total basepairs processed:    10,298,940 bp\n",
      "  Read 1:     6,159,170 bp\n",
      "  Read 2:     4,139,770 bp\n",
      "Total written (filtered):      8,882,443 bp (86.2%)\n",
      "  Read 1:     5,423,098 bp\n",
      "  Read 2:     3,459,345 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 9 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 22.2%\n",
      "  C: 22.2%\n",
      "  G: 55.6%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.3\t1\t0 1\n",
      "9\t1\t0.1\t1\t0 1\n",
      "24\t1\t0.0\t3\t1\n",
      "26\t1\t0.0\t3\t1\n",
      "34\t1\t0.0\t3\t1\n",
      "277\t3\t0.0\t3\t3\n",
      "278\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 19975 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t1\t0 1\n",
      "15\t3\t0.0\t2\t3\n",
      "17\t12\t0.0\t2\t7 2 3\n",
      "18\t3069\t0.0\t2\t3002 64 3\n",
      "19\t3549\t0.0\t2\t3490 56 3\n",
      "20\t3683\t0.0\t2\t3617 61 5\n",
      "21\t3281\t0.0\t2\t3224 53 4\n",
      "22\t3480\t0.0\t2\t3413 62 5\n",
      "23\t2892\t0.0\t2\t2843 48 1\n",
      "24\t4\t0.0\t2\t4\n",
      "31\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 24 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 16.7%\n",
      "  G: 29.2%\n",
      "  T: 20.8%\n",
      "  none/other: 33.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t6\t0.0\t2\t6\n",
      "204\t10\t0.0\t2\t9 1\n",
      "205\t8\t0.0\t2\t6 2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 19246 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t2\t1\n",
      "18\t1\t0.0\t2\t0 1\n",
      "19\t2\t0.0\t2\t1 1\n",
      "20\t21\t0.0\t3\t10 6 3 2\n",
      "21\t3560\t0.0\t3\t3352 152 41 15\n",
      "22\t3422\t0.0\t3\t3231 139 32 20\n",
      "23\t3185\t0.0\t3\t2977 150 42 16\n",
      "24\t3237\t0.0\t3\t3007 169 46 15\n",
      "25\t3020\t0.0\t3\t2837 133 34 16\n",
      "26\t2791\t0.0\t3\t2625 129 20 17\n",
      "27\t4\t0.0\t3\t4\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121701_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121701_27_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121701_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121701_27_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121701_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121701_27_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121701_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121701_27_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.22 s (18 µs/read; 3.32 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             67,293\n",
      "  Read 1 with adapter:                  66,636 (99.0%)\n",
      "  Read 2 with adapter:                  65,550 (97.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  14 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,368 (3.5%)\n",
      "Pairs written (passing filters):        64,911 (96.5%)\n",
      "\n",
      "Total basepairs processed:    34,319,430 bp\n",
      "  Read 1:    20,524,365 bp\n",
      "  Read 2:    13,795,065 bp\n",
      "Total written (filtered):     30,254,663 bp (88.2%)\n",
      "  Read 1:    18,466,767 bp\n",
      "  Read 2:    11,787,896 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 44 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 2.3%\n",
      "  C: 47.7%\n",
      "  G: 50.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t10\t1.0\t1\t0 10\n",
      "9\t11\t0.3\t1\t0 11\n",
      "22\t1\t0.0\t3\t1\n",
      "30\t4\t0.0\t3\t4\n",
      "32\t1\t0.0\t3\t1\n",
      "33\t2\t0.0\t3\t2\n",
      "34\t1\t0.0\t3\t1\n",
      "275\t1\t0.0\t3\t1\n",
      "277\t3\t0.0\t3\t3\n",
      "278\t1\t0.0\t3\t1\n",
      "279\t4\t0.0\t3\t4\n",
      "280\t2\t0.0\t3\t2\n",
      "281\t3\t0.0\t3\t3\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 66592 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t1\t1\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t1\t0.0\t1\t0 1\n",
      "14\t6\t0.0\t2\t4 2\n",
      "15\t5\t0.0\t2\t2 2 1\n",
      "16\t4\t0.0\t2\t1 1 2\n",
      "17\t36\t0.0\t2\t19 10 7\n",
      "18\t10342\t0.0\t2\t10168 164 10\n",
      "19\t11793\t0.0\t2\t11595 184 14\n",
      "20\t12312\t0.0\t2\t12115 189 8\n",
      "21\t10918\t0.0\t2\t10714 191 13\n",
      "22\t11422\t0.0\t2\t11250 161 11\n",
      "23\t9740\t0.0\t2\t9600 138 2\n",
      "24\t9\t0.0\t2\t7 2\n",
      "27\t1\t0.0\t2\t1\n",
      "31\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 28 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 14.3%\n",
      "  G: 17.9%\n",
      "  T: 17.9%\n",
      "  none/other: 50.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t5\t0.0\t2\t4 1\n",
      "204\t9\t0.0\t2\t9\n",
      "205\t14\t0.0\t2\t14\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 65522 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t1\t0.0\t2\t0 0 1\n",
      "15\t2\t0.0\t2\t1 0 1\n",
      "18\t1\t0.0\t2\t1\n",
      "19\t4\t0.0\t2\t2 1 1\n",
      "20\t39\t0.0\t3\t18 9 4 8\n",
      "21\t12322\t0.0\t3\t11697 449 118 58\n",
      "22\t11671\t0.0\t3\t11095 425 97 54\n",
      "23\t10860\t0.0\t3\t10267 443 101 49\n",
      "24\t10970\t0.0\t3\t10352 471 85 62\n",
      "25\t10039\t0.0\t3\t9484 406 103 46\n",
      "26\t9598\t0.0\t3\t9007 414 113 64\n",
      "27\t6\t0.0\t3\t6\n",
      "50\t1\t0.0\t3\t0 0 1\n",
      "57\t1\t0.0\t3\t0 0 0 1\n",
      "59\t1\t0.0\t3\t0 0 0 1\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "62\t1\t0.0\t3\t0 0 1\n",
      "63\t2\t0.0\t3\t0 1 1\n",
      "64\t2\t0.0\t3\t0 0 1 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121702_28_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121702_29_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121702_28_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121702_29_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121702_28_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-08o7obtr/210421121702_29_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121702_28_L001_R1_001.fastq.gz /tmp/qiime2-archive-0tiamqlh/c1918fcd-fca3-4f28-b410-5c40896f2153/data/210421121702_29_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.42 s (50 µs/read; 1.19 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              8,334\n",
      "  Read 1 with adapter:                   8,233 (98.8%)\n",
      "  Read 2 with adapter:                   7,983 (95.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   7 (0.1%)\n",
      "Pairs discarded as untrimmed:              443 (5.3%)\n",
      "Pairs written (passing filters):         7,884 (94.6%)\n",
      "\n",
      "Total basepairs processed:     4,250,340 bp\n",
      "  Read 1:     2,541,870 bp\n",
      "  Read 2:     1,708,470 bp\n",
      "Total written (filtered):      3,674,455 bp (86.5%)\n",
      "  Read 1:     2,243,538 bp\n",
      "  Read 2:     1,430,917 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 1 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 100.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "24\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 8232 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t2\t0 0 1\n",
      "16\t1\t0.0\t2\t1\n",
      "17\t3\t0.0\t2\t3\n",
      "18\t1297\t0.0\t2\t1267 30\n",
      "19\t1484\t0.0\t2\t1464 18 2\n",
      "20\t1518\t0.0\t2\t1495 20 3\n",
      "21\t1305\t0.0\t2\t1281 23 1\n",
      "22\t1420\t0.0\t2\t1390 28 2\n",
      "23\t1202\t0.0\t2\t1178 23 1\n",
      "24\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 11 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 18.2%\n",
      "  T: 18.2%\n",
      "  none/other: 63.6%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "204\t4\t0.0\t2\t4\n",
      "205\t7\t0.0\t2\t7\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 7972 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t11\t0.0\t3\t3 5 1 2\n",
      "21\t1453\t0.0\t3\t1388 47 12 6\n",
      "22\t1398\t0.0\t3\t1308 56 24 10\n",
      "23\t1322\t0.0\t3\t1248 58 12 4\n",
      "24\t1306\t0.0\t3\t1240 45 18 3\n",
      "25\t1239\t0.0\t3\t1164 57 13 5\n",
      "26\t1237\t0.0\t3\t1154 67 8 8\n",
      "27\t2\t0.0\t3\t2\n",
      "31\t1\t0.0\t3\t1\n",
      "59\t1\t0.0\t3\t1\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "62\t1\t0.0\t3\t0 0 0 1\n"
     ]
    }
   ],
   "source": [
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')\n",
    "\n",
    "if trim and (not os.path.isfile(demux_file_trim) or replace_files):\n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = trim_paired(\n",
    "        demultiplexed_sequences=artifact,\n",
    "        front_f=forward_primer,\n",
    "        front_r=reverse_primer,\n",
    "        adapter_f=reverse_reverse_complement,\n",
    "        adapter_r=forward_reverse_complement,\n",
    "        cores=threads,\n",
    "        overlap=trim['overlap'],\n",
    "        indels=False,\n",
    "        match_read_wildcards=True,\n",
    "        match_adapter_wildcards=True,\n",
    "        error_rate=0.15,\n",
    "        discard_untrimmed=True,\n",
    "    ).trimmed_sequences\n",
    "\n",
    "    demux.visualizers.summarize(res).visualization\n",
    "    res.save(demux_file_trim)\n",
    "    Visualization.save(demux.visualizers.summarize(res).visualization, filepath=demux_view_trim)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 558.773802,
   "end_time": "2022-10-24T18:54:34.419690",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-templates/step-prepare-data.ipynb",
   "output_path": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-STD-NCxHSD-NC-trim/nb-executed-steps/step-prepare-data-ana-flavia-STD-NCxHSD-NC-trim.ipynb",
   "parameters": {
    "base_dir": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri",
    "class_col": "group-id",
    "classifier_file": "/home/lauro/nupeb/dados_brutos_rede_genoma/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza",
    "experiment_name": "ana-flavia-STD-NCxHSD-NC-trim",
    "manifest_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-STD-NCxHSD-NC.csv",
    "metadata_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-STD-NCxHSD-NC.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2022-10-24T18:45:15.645888",
   "version": "2.3.4"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}