{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "6ef54b0f",
   "metadata": {
    "tags": [
     "papermill-error-cell-tag"
    ]
   },
   "source": [
    "<span style=\"color:red; font-family:Helvetica Neue, Helvetica, Arial, sans-serif; font-size:2em;\">An Exception was encountered at '<a href=\"#papermill-error-cell\">In [7]</a>'.</span>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "092e3812",
   "metadata": {
    "papermill": {
     "duration": 0.019379,
     "end_time": "2023-08-07T20:15:32.510605",
     "exception": false,
     "start_time": "2023-08-07T20:15:32.491226",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e3cb560e",
   "metadata": {
    "papermill": {
     "duration": 0.031276,
     "end_time": "2023-08-07T20:15:32.553764",
     "exception": false,
     "start_time": "2023-08-07T20:15:32.522488",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "8d51de90",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-23T10:42:21.279135Z",
     "start_time": "2023-08-23T10:42:01.596638Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-07T20:15:32.577620Z",
     "iopub.status.busy": "2023-08-07T20:15:32.577024Z",
     "iopub.status.idle": "2023-08-07T20:15:39.177681Z",
     "shell.execute_reply": "2023-08-07T20:15:39.176734Z"
    },
    "papermill": {
     "duration": 6.616013,
     "end_time": "2023-08-07T20:15:39.180840",
     "exception": false,
     "start_time": "2023-08-07T20:15:32.564827",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization, Metadata\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired, trim_single\n",
    "from qiime2.plugins.demux.methods import filter_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7378b679",
   "metadata": {
    "papermill": {
     "duration": 0.037478,
     "end_time": "2023-08-07T20:15:39.235978",
     "exception": false,
     "start_time": "2023-08-07T20:15:39.198500",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b6be8d57",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-23T10:42:21.286174Z",
     "start_time": "2023-08-23T10:42:21.282183Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-07T20:15:39.291514Z",
     "iopub.status.busy": "2023-08-07T20:15:39.290973Z",
     "iopub.status.idle": "2023-08-07T20:15:39.296140Z",
     "shell.execute_reply": "2023-08-07T20:15:39.295443Z"
    },
    "papermill": {
     "duration": 0.01803,
     "end_time": "2023-08-07T20:15:39.297692",
     "exception": false,
     "start_time": "2023-08-07T20:15:39.279662",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'parameter-set-01.yaml')\n",
    "experiment_name = 'exp-01'\n",
    "base_dir = os.path.join('/', 'home', 'username', 'pipeline-dir')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'manifest.csv')\n",
    "replace_files = False\n",
    "trim = None\n",
    "metadata_file = None\n",
    "threads = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6d29ba48",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-23T10:42:21.330783Z",
     "start_time": "2023-08-23T10:42:21.287900Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-07T20:15:39.320756Z",
     "iopub.status.busy": "2023-08-07T20:15:39.320415Z",
     "iopub.status.idle": "2023-08-07T20:15:39.325765Z",
     "shell.execute_reply": "2023-08-07T20:15:39.325035Z"
    },
    "papermill": {
     "duration": 0.01886,
     "end_time": "2023-08-07T20:15:39.327468",
     "exception": false,
     "start_time": "2023-08-07T20:15:39.308608",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "experiment_name = \"ana-flavia-all\"\n",
    "base_dir = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri\"\n",
    "manifest_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest-all-ana.csv\"\n",
    "metadata_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata-all-ana-types.tsv\"\n",
    "class_col = \"group-id\"\n",
    "classifier_file = \"/home/lauro/nupeb/rede-micro/models/silva-138-99-nb-classifier.qza\"\n",
    "top_n = 20\n",
    "replace_files = False\n",
    "phred = 20\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n",
    "overlap = 12\n",
    "threads = 6\n",
    "trim = {\n",
    "    \"overlap\": 8,\n",
    "    \"forward_primer\": \"CCTACGGGRSGCAGCAG\",\n",
    "    \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\",\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4b8cdec9",
   "metadata": {
    "papermill": {
     "duration": 0.010513,
     "end_time": "2023-08-07T20:15:39.349721",
     "exception": false,
     "start_time": "2023-08-07T20:15:39.339208",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e8af500d",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-23T10:42:21.361818Z",
     "start_time": "2023-08-23T10:42:21.334022Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-07T20:15:39.371621Z",
     "iopub.status.busy": "2023-08-07T20:15:39.371266Z",
     "iopub.status.idle": "2023-08-07T20:15:39.378258Z",
     "shell.execute_reply": "2023-08-07T20:15:39.377274Z"
    },
    "papermill": {
     "duration": 0.019991,
     "end_time": "2023-08-07T20:15:39.380156",
     "exception": false,
     "start_time": "2023-08-07T20:15:39.360165",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Define the output folder path\n",
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "\n",
    "# Create path if it not exists\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exists\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')\n",
    "\n",
    "# Define the output artifact full path\n",
    "demux_file = os.path.join(out_dir, 'demux-paired.qza')\n",
    "demux_view = os.path.join(out_dir, 'demux-paired.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6103aef9",
   "metadata": {
    "papermill": {
     "duration": 0.012455,
     "end_time": "2023-08-07T20:15:39.404789",
     "exception": false,
     "start_time": "2023-08-07T20:15:39.392334",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file of `SampleData[PairedEndSequencesWithQuality]`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "bf4e0f3a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-23T10:42:21.581031Z",
     "start_time": "2023-08-23T10:42:21.364399Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-07T20:15:39.427097Z",
     "iopub.status.busy": "2023-08-07T20:15:39.426758Z",
     "iopub.status.idle": "2023-08-07T20:15:39.436835Z",
     "shell.execute_reply": "2023-08-07T20:15:39.436207Z"
    },
    "papermill": {
     "duration": 0.023032,
     "end_time": "2023-08-07T20:15:39.438427",
     "exception": false,
     "start_time": "2023-08-07T20:15:39.415395",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "manifest_df = pd.read_csv(manifest_file)\n",
    "n_directions = len(manifest_df['direction'].unique())\n",
    "if n_directions == 1:\n",
    "    d_type = 'SampleData[SequencesWithQuality]'\n",
    "    v_type = 'SingleEndFastqManifestPhred33'\n",
    "elif n_directions == 2:\n",
    "    d_type = 'SampleData[PairedEndSequencesWithQuality]'\n",
    "    v_type = 'PairedEndFastqManifestPhred33'\n",
    "else:\n",
    "    print(f'ERROR: invalid number of directions {n_directions}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ecf2da39",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-23T10:42:21.661712Z",
     "start_time": "2023-08-23T10:42:21.582950Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-07T20:15:39.464260Z",
     "iopub.status.busy": "2023-08-07T20:15:39.463904Z",
     "iopub.status.idle": "2023-08-07T20:15:39.478820Z",
     "shell.execute_reply": "2023-08-07T20:15:39.478197Z"
    },
    "papermill": {
     "duration": 0.028277,
     "end_time": "2023-08-07T20:15:39.480957",
     "exception": false,
     "start_time": "2023-08-07T20:15:39.452680",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample-desc</th>\n",
       "      <th>group-id</th>\n",
       "      <th>group-desc</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample-id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S210421121673</th>\n",
       "      <td>N9.01</td>\n",
       "      <td>NC</td>\n",
       "      <td>Ninhada-controle</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121674</th>\n",
       "      <td>N11.01</td>\n",
       "      <td>NC</td>\n",
       "      <td>Ninhada-controle</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121675</th>\n",
       "      <td>N9.05</td>\n",
       "      <td>NC</td>\n",
       "      <td>Ninhada-controle</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121676</th>\n",
       "      <td>N11.02</td>\n",
       "      <td>NC</td>\n",
       "      <td>Ninhada-controle</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121677</th>\n",
       "      <td>N9.03</td>\n",
       "      <td>NR</td>\n",
       "      <td>Ninhada-reduzida</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121678</th>\n",
       "      <td>N10.03</td>\n",
       "      <td>NR</td>\n",
       "      <td>Ninhada-reduzida</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121679</th>\n",
       "      <td>N10.04</td>\n",
       "      <td>NR</td>\n",
       "      <td>Ninhada-reduzida</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121680</th>\n",
       "      <td>N10.07</td>\n",
       "      <td>NR</td>\n",
       "      <td>Ninhada-reduzida</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121681</th>\n",
       "      <td>N10.08</td>\n",
       "      <td>NR</td>\n",
       "      <td>Ninhada-reduzida</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121682</th>\n",
       "      <td>N7.01</td>\n",
       "      <td>STD-NC</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121683</th>\n",
       "      <td>N7.02</td>\n",
       "      <td>STD-NC</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121684</th>\n",
       "      <td>N8.01</td>\n",
       "      <td>STD-NC</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121685</th>\n",
       "      <td>N7.03</td>\n",
       "      <td>STD-NR</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121686</th>\n",
       "      <td>N7.04</td>\n",
       "      <td>STD-NR</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121687</th>\n",
       "      <td>N8.03</td>\n",
       "      <td>STD-NR</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121688</th>\n",
       "      <td>N7.05</td>\n",
       "      <td>HSD-NC</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121689</th>\n",
       "      <td>N8.05</td>\n",
       "      <td>HSD-NC</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121690</th>\n",
       "      <td>N8.06</td>\n",
       "      <td>HSD-NC</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121691</th>\n",
       "      <td>N7.07</td>\n",
       "      <td>HSD-NR</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121692</th>\n",
       "      <td>N7.07</td>\n",
       "      <td>HSD-NR</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121693</th>\n",
       "      <td>N8.08</td>\n",
       "      <td>HSD-NR</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121694</th>\n",
       "      <td>N1.01</td>\n",
       "      <td>STD-NC</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121695</th>\n",
       "      <td>N3.01</td>\n",
       "      <td>STD-NC</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121696</th>\n",
       "      <td>N4.01</td>\n",
       "      <td>STD-NC</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121697</th>\n",
       "      <td>N4.02</td>\n",
       "      <td>STD-NC</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121699</th>\n",
       "      <td>N2.04</td>\n",
       "      <td>HSD-NC</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121700</th>\n",
       "      <td>N3.04</td>\n",
       "      <td>HSD-NC</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121701</th>\n",
       "      <td>N4.03</td>\n",
       "      <td>HSD-NC</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121702</th>\n",
       "      <td>N4.04</td>\n",
       "      <td>HSD-NC</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121703</th>\n",
       "      <td>N1.02</td>\n",
       "      <td>STD-NR</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121704</th>\n",
       "      <td>N1.03</td>\n",
       "      <td>STD-NR</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121713</th>\n",
       "      <td>N1.03</td>\n",
       "      <td>STD-NR</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121705</th>\n",
       "      <td>N2.02</td>\n",
       "      <td>STD-NR</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121714</th>\n",
       "      <td>N2.02</td>\n",
       "      <td>STD-NR</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121706</th>\n",
       "      <td>N2.03</td>\n",
       "      <td>STD-NR</td>\n",
       "      <td>Grupo-dieta-padrao</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121707</th>\n",
       "      <td>N1.05</td>\n",
       "      <td>HSD-NR</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121708</th>\n",
       "      <td>N2.05</td>\n",
       "      <td>HSD-NR</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121709</th>\n",
       "      <td>N3.05</td>\n",
       "      <td>HSD-NR</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121710</th>\n",
       "      <td>N3.06</td>\n",
       "      <td>HSD-NR</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S210421121711</th>\n",
       "      <td>N1.04</td>\n",
       "      <td>HSD-NC</td>\n",
       "      <td>Grupo-dieta-high-sugar</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              sample-desc group-id              group-desc\n",
       "sample-id                                                 \n",
       "S210421121673       N9.01       NC        Ninhada-controle\n",
       "S210421121674      N11.01       NC        Ninhada-controle\n",
       "S210421121675       N9.05       NC        Ninhada-controle\n",
       "S210421121676      N11.02       NC        Ninhada-controle\n",
       "S210421121677       N9.03       NR        Ninhada-reduzida\n",
       "S210421121678      N10.03       NR        Ninhada-reduzida\n",
       "S210421121679      N10.04       NR        Ninhada-reduzida\n",
       "S210421121680      N10.07       NR        Ninhada-reduzida\n",
       "S210421121681      N10.08       NR        Ninhada-reduzida\n",
       "S210421121682       N7.01   STD-NC      Grupo-dieta-padrao\n",
       "S210421121683       N7.02   STD-NC      Grupo-dieta-padrao\n",
       "S210421121684       N8.01   STD-NC      Grupo-dieta-padrao\n",
       "S210421121685       N7.03   STD-NR      Grupo-dieta-padrao\n",
       "S210421121686       N7.04   STD-NR      Grupo-dieta-padrao\n",
       "S210421121687       N8.03   STD-NR      Grupo-dieta-padrao\n",
       "S210421121688       N7.05   HSD-NC  Grupo-dieta-high-sugar\n",
       "S210421121689       N8.05   HSD-NC  Grupo-dieta-high-sugar\n",
       "S210421121690       N8.06   HSD-NC  Grupo-dieta-high-sugar\n",
       "S210421121691       N7.07   HSD-NR  Grupo-dieta-high-sugar\n",
       "S210421121692       N7.07   HSD-NR  Grupo-dieta-high-sugar\n",
       "S210421121693       N8.08   HSD-NR  Grupo-dieta-high-sugar\n",
       "S210421121694       N1.01   STD-NC      Grupo-dieta-padrao\n",
       "S210421121695       N3.01   STD-NC      Grupo-dieta-padrao\n",
       "S210421121696       N4.01   STD-NC      Grupo-dieta-padrao\n",
       "S210421121697       N4.02   STD-NC      Grupo-dieta-padrao\n",
       "S210421121699       N2.04   HSD-NC  Grupo-dieta-high-sugar\n",
       "S210421121700       N3.04   HSD-NC  Grupo-dieta-high-sugar\n",
       "S210421121701       N4.03   HSD-NC  Grupo-dieta-high-sugar\n",
       "S210421121702       N4.04   HSD-NC  Grupo-dieta-high-sugar\n",
       "S210421121703       N1.02   STD-NR      Grupo-dieta-padrao\n",
       "S210421121704       N1.03   STD-NR      Grupo-dieta-padrao\n",
       "S210421121713       N1.03   STD-NR      Grupo-dieta-padrao\n",
       "S210421121705       N2.02   STD-NR      Grupo-dieta-padrao\n",
       "S210421121714       N2.02   STD-NR      Grupo-dieta-padrao\n",
       "S210421121706       N2.03   STD-NR      Grupo-dieta-padrao\n",
       "S210421121707       N1.05   HSD-NR  Grupo-dieta-high-sugar\n",
       "S210421121708       N2.05   HSD-NR  Grupo-dieta-high-sugar\n",
       "S210421121709       N3.05   HSD-NR  Grupo-dieta-high-sugar\n",
       "S210421121710       N3.06   HSD-NR  Grupo-dieta-high-sugar\n",
       "S210421121711       N1.04   HSD-NC  Grupo-dieta-high-sugar"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# If the metadata file is not defined, use the default metadata file\n",
    "metadata_qa = Metadata.load(metadata_file)\n",
    "metadata_qa.to_dataframe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "83cc518e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-23T10:42:21.807460Z",
     "start_time": "2023-08-23T10:42:21.663556Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest-all-ana.csv\n",
      "sample-id,absolute-filepath,direction\r\n",
      "S210421121673,/home/lauro/nupeb/dados_brutos_rede_genoma/data/16s_karina/workspace/decompressed_reads/210421121673-1-1-1_S81_L001_R1_001.fastq,forward\r\n",
      "S210421121673,/home/lauro/nupeb/dados_brutos_rede_genoma/data/16s_karina/workspace/decompressed_reads/210421121673-1-1-1_S81_L001_R2_001.fastq,reverse\r\n",
      "S210421121674,/home/lauro/nupeb/dados_brutos_rede_genoma/data/16s_karina/workspace/decompressed_reads/210421121674-1-1-1_S83_L001_R1_001.fastq,forward\r\n",
      "S210421121674,/home/lauro/nupeb/dados_brutos_rede_genoma/data/16s_karina/workspace/decompressed_reads/210421121674-1-1-1_S83_L001_R2_001.fastq,reverse\r\n",
      "S210421121675,/home/lauro/nupeb/dados_brutos_rede_genoma/data/16s_karina/workspace/decompressed_reads/210421121675-1-1-1_S85_L001_R1_001.fastq,forward\r\n",
      "S210421121675,/home/lauro/nupeb/dados_brutos_rede_genoma/data/16s_karina/workspace/decompressed_reads/210421121675-1-1-1_S85_L001_R2_001.fastq,reverse\r\n",
      "S210421121676,/home/lauro/nupeb/dados_brutos_rede_genoma/data/16s_karina/workspace/decompressed_reads/210421121676-1-1-1_S87_L001_R1_001.fastq,forward\r\n",
      "S210421121676,/home/lauro/nupeb/dados_brutos_rede_genoma/data/16s_karina/workspace/decompressed_reads/210421121676-1-1-1_S87_L001_R2_001.fastq,reverse\r\n",
      "S210421121677,/home/lauro/nupeb/dados_brutos_rede_genoma/data/16s_karina/workspace/decompressed_reads/210421121677-1-1-1_S89_L001_R1_001.fastq,forward\r\n"
     ]
    }
   ],
   "source": [
    "print(manifest_file)\n",
    "!head {manifest_file}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d624cfa9",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-23T11:38:21.466298Z",
     "start_time": "2023-08-23T10:43:58.739531Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-all/qiime-artifacts/demux-paired.qza'"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Import data and create an artifact object\n",
    "artifact = Artifact.import_data(\n",
    "    d_type, \n",
    "    manifest_file, \n",
    "    view_type=v_type)\n",
    "\n",
    "#     artifact = filter_samples(demux=artifact, metadata=metadata_qa).filtered_demux\n",
    "# Save the artifact object to a new qza file\n",
    "artifact.save(demux_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eae4a2ed",
   "metadata": {
    "ExecuteTime": {
     "start_time": "2023-08-23T00:37:50.216Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-07T20:15:39.506792Z",
     "iopub.status.busy": "2023-08-07T20:15:39.506444Z",
     "iopub.status.idle": "2023-08-07T21:08:40.334423Z",
     "shell.execute_reply": "2023-08-07T21:08:40.333519Z"
    },
    "papermill": {
     "duration": 3180.860812,
     "end_time": "2023-08-07T21:08:40.355444",
     "exception": true,
     "start_time": "2023-08-07T20:15:39.494632",
     "status": "failed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    \n",
    "    # Import data and create an artifact object\n",
    "    artifact = Artifact.import_data(\n",
    "        d_type, \n",
    "        manifest_file, \n",
    "        view_type=v_type)\n",
    "\n",
    "#     artifact = filter_samples(demux=artifact, metadata=metadata_qa).filtered_demux\n",
    "    # Save the artifact object to a new qza file\n",
    "    artifact.save(demux_file)\n",
    "\n",
    "else:\n",
    "    artifact = Artifact.load(demux_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "929a8eae",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-23T11:53:13.056476Z",
     "start_time": "2023-08-23T11:50:13.853885Z"
    },
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "if not os.path.isfile(demux_view) or replace_files: \n",
    "    # Generate e visualization of the Artifact\n",
    "    demux_summary = demux.visualizers.summarize(artifact)\n",
    "\n",
    "    # Save a new visualization file based on the qza file\n",
    "    Visualization.save(demux_summary.visualization, filepath=demux_view)\n",
    "    \n",
    "    demux_view_obj = demux_summary.visualization\n",
    "else:\n",
    "    demux_view_obj = Visualization.load(demux_view)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "bf4dfacf",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-23T11:53:13.063889Z",
     "start_time": "2023-08-23T11:53:13.059167Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-all/qiime-artifacts/demux-paired.qza'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "demux_file"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90ba6448",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "08c5b267",
   "metadata": {
    "ExecuteTime": {
     "start_time": "2023-08-23T00:37:57.760Z"
    },
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "print(demux_view_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "157d9d03",
   "metadata": {
    "ExecuteTime": {
     "start_time": "2023-08-23T00:37:58.600Z"
    },
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0134e10",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "bf280a2c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-23T14:29:39.564331Z",
     "start_time": "2023-08-23T14:23:20.424020Z"
    },
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121673_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121673_1_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121673_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121673_1_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121673_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121673_1_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121673_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121673_1_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 3.77 s (10 µs/read; 5.89 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            370,042\n",
      "  Read 1 with adapter:                 359,997 (97.3%)\n",
      "  Read 2 with adapter:                 343,009 (92.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           36,262 (9.8%)\n",
      "Pairs written (passing filters):       333,780 (90.2%)\n",
      "\n",
      "Total basepairs processed:   188,721,420 bp\n",
      "  Read 1:   112,862,810 bp\n",
      "  Read 2:    75,858,610 bp\n",
      "Total written (filtered):    155,609,062 bp (82.5%)\n",
      "  Read 1:    94,978,565 bp\n",
      "  Read 2:    60,630,497 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 359997 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t5.6\t0\t3\n",
      "9\t4\t1.4\t0\t4\n",
      "10\t3\t0.4\t0\t3\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t4\t0.0\t0\t4\n",
      "14\t12\t0.0\t0\t12\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t19\t0.0\t0\t19\n",
      "17\t109\t0.0\t0\t109\n",
      "18\t56158\t0.0\t0\t56158\n",
      "19\t64949\t0.0\t0\t64949\n",
      "20\t66248\t0.0\t0\t66248\n",
      "21\t59860\t0.0\t0\t59860\n",
      "22\t60699\t0.0\t0\t60699\n",
      "23\t51867\t0.0\t0\t51867\n",
      "24\t42\t0.0\t0\t42\n",
      "25\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 343009 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.4\t0\t1\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t14\t0.0\t0\t14\n",
      "20\t137\t0.0\t0\t137\n",
      "21\t64389\t0.0\t0\t64389\n",
      "22\t61501\t0.0\t0\t61501\n",
      "23\t56595\t0.0\t0\t56595\n",
      "24\t58584\t0.0\t0\t58584\n",
      "25\t52061\t0.0\t0\t52061\n",
      "26\t49654\t0.0\t0\t49654\n",
      "27\t42\t0.0\t0\t42\n",
      "29\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121674_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121674_3_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121674_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121674_3_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121674_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121674_3_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121674_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121674_3_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 5.01 s (9 µs/read; 6.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            559,807\n",
      "  Read 1 with adapter:                 544,656 (97.3%)\n",
      "  Read 2 with adapter:                 519,894 (92.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           53,686 (9.6%)\n",
      "Pairs written (passing filters):       506,121 (90.4%)\n",
      "\n",
      "Total basepairs processed:   285,501,570 bp\n",
      "  Read 1:   170,741,135 bp\n",
      "  Read 2:   114,760,435 bp\n",
      "Total written (filtered):    235,957,214 bp (82.6%)\n",
      "  Read 1:   144,019,667 bp\n",
      "  Read 2:    91,937,547 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 544656 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t8.5\t0\t5\n",
      "9\t4\t2.1\t0\t4\n",
      "10\t3\t0.5\t0\t3\n",
      "11\t7\t0.1\t0\t7\n",
      "12\t7\t0.0\t0\t7\n",
      "13\t6\t0.0\t0\t6\n",
      "14\t17\t0.0\t0\t17\n",
      "15\t16\t0.0\t0\t16\n",
      "16\t22\t0.0\t0\t22\n",
      "17\t155\t0.0\t0\t155\n",
      "18\t85410\t0.0\t0\t85410\n",
      "19\t98095\t0.0\t0\t98095\n",
      "20\t100478\t0.0\t0\t100478\n",
      "21\t90125\t0.0\t0\t90125\n",
      "22\t91840\t0.0\t0\t91840\n",
      "23\t78402\t0.0\t0\t78402\n",
      "24\t55\t0.0\t0\t55\n",
      "26\t3\t0.0\t0\t3\n",
      "29\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "35\t1\t0.0\t0\t1\n",
      "42\t1\t0.0\t0\t1\n",
      "43\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 519894 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t5\t0.0\t0\t5\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t15\t0.0\t0\t15\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t9\t0.0\t0\t9\n",
      "18\t6\t0.0\t0\t6\n",
      "19\t17\t0.0\t0\t17\n",
      "20\t180\t0.0\t0\t180\n",
      "21\t98232\t0.0\t0\t98232\n",
      "22\t93652\t0.0\t0\t93652\n",
      "23\t85816\t0.0\t0\t85816\n",
      "24\t87651\t0.0\t0\t87651\n",
      "25\t78341\t0.0\t0\t78341\n",
      "26\t75895\t0.0\t0\t75895\n",
      "27\t62\t0.0\t0\t62\n",
      "36\t1\t0.0\t0\t1\n",
      "57\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121675_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121675_5_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121675_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121675_5_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121675_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121675_5_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121675_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121675_5_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 3.23 s (9 µs/read; 6.43 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            345,762\n",
      "  Read 1 with adapter:                 336,819 (97.4%)\n",
      "  Read 2 with adapter:                 321,022 (92.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           32,941 (9.5%)\n",
      "Pairs written (passing filters):       312,821 (90.5%)\n",
      "\n",
      "Total basepairs processed:   176,338,620 bp\n",
      "  Read 1:   105,457,410 bp\n",
      "  Read 2:    70,881,210 bp\n",
      "Total written (filtered):    145,841,282 bp (82.7%)\n",
      "  Read 1:    89,015,404 bp\n",
      "  Read 2:    56,825,878 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 336819 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t5.3\t0\t5\n",
      "9\t3\t1.3\t0\t3\n",
      "10\t5\t0.3\t0\t5\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t5\t0.0\t0\t5\n",
      "14\t12\t0.0\t0\t12\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t11\t0.0\t0\t11\n",
      "17\t92\t0.0\t0\t92\n",
      "18\t52679\t0.0\t0\t52679\n",
      "19\t60788\t0.0\t0\t60788\n",
      "20\t62024\t0.0\t0\t62024\n",
      "21\t55828\t0.0\t0\t55828\n",
      "22\t57192\t0.0\t0\t57192\n",
      "23\t48121\t0.0\t0\t48121\n",
      "24\t34\t0.0\t0\t34\n",
      "25\t2\t0.0\t0\t2\n",
      "28\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "36\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 321022 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t3\t0.0\t0\t3\n",
      "15\t12\t0.0\t0\t12\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t11\t0.0\t0\t11\n",
      "20\t106\t0.0\t0\t106\n",
      "21\t60450\t0.0\t0\t60450\n",
      "22\t58128\t0.0\t0\t58128\n",
      "23\t53277\t0.0\t0\t53277\n",
      "24\t54587\t0.0\t0\t54587\n",
      "25\t48050\t0.0\t0\t48050\n",
      "26\t46353\t0.0\t0\t46353\n",
      "27\t34\t0.0\t0\t34\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121676_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121676_7_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121676_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121676_7_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121676_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121676_7_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121676_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121676_7_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 6.49 s (9 µs/read; 6.59 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            712,750\n",
      "  Read 1 with adapter:                 694,205 (97.4%)\n",
      "  Read 2 with adapter:                 664,033 (93.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           65,715 (9.2%)\n",
      "Pairs written (passing filters):       647,035 (90.8%)\n",
      "\n",
      "Total basepairs processed:   363,502,500 bp\n",
      "  Read 1:   217,388,750 bp\n",
      "  Read 2:   146,113,750 bp\n",
      "Total written (filtered):    301,650,677 bp (83.0%)\n",
      "  Read 1:   184,115,305 bp\n",
      "  Read 2:   117,535,372 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 694205 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t10.9\t0\t4\n",
      "9\t9\t2.7\t0\t9\n",
      "10\t5\t0.7\t0\t5\n",
      "11\t14\t0.2\t0\t14\n",
      "12\t8\t0.0\t0\t8\n",
      "13\t6\t0.0\t0\t6\n",
      "14\t30\t0.0\t0\t30\n",
      "15\t18\t0.0\t0\t18\n",
      "16\t33\t0.0\t0\t33\n",
      "17\t183\t0.0\t0\t183\n",
      "18\t108490\t0.0\t0\t108490\n",
      "19\t125122\t0.0\t0\t125122\n",
      "20\t127377\t0.0\t0\t127377\n",
      "21\t114840\t0.0\t0\t114840\n",
      "22\t117757\t0.0\t0\t117757\n",
      "23\t100233\t0.0\t0\t100233\n",
      "24\t68\t0.0\t0\t68\n",
      "25\t1\t0.0\t0\t1\n",
      "28\t2\t0.0\t0\t2\n",
      "35\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "41\t1\t0.0\t0\t1\n",
      "49\t1\t0.0\t0\t1\n",
      "57\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 664033 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.7\t0\t2\n",
      "11\t4\t0.2\t0\t4\n",
      "12\t5\t0.0\t0\t5\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t18\t0.0\t0\t18\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t4\t0.0\t0\t4\n",
      "18\t9\t0.0\t0\t9\n",
      "19\t16\t0.0\t0\t16\n",
      "20\t229\t0.0\t0\t229\n",
      "21\t124781\t0.0\t0\t124781\n",
      "22\t120299\t0.0\t0\t120299\n",
      "23\t109785\t0.0\t0\t109785\n",
      "24\t112135\t0.0\t0\t112135\n",
      "25\t100654\t0.0\t0\t100654\n",
      "26\t95988\t0.0\t0\t95988\n",
      "27\t87\t0.0\t0\t87\n",
      "28\t1\t0.0\t0\t1\n",
      "29\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "47\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "56\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121677_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121677_9_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121677_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121677_9_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121677_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121677_9_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121677_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121677_9_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 3.47 s (9 µs/read; 6.33 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            365,427\n",
      "  Read 1 with adapter:                 355,998 (97.4%)\n",
      "  Read 2 with adapter:                 339,330 (92.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           34,713 (9.5%)\n",
      "Pairs written (passing filters):       330,714 (90.5%)\n",
      "\n",
      "Total basepairs processed:   186,367,770 bp\n",
      "  Read 1:   111,455,235 bp\n",
      "  Read 2:    74,912,535 bp\n",
      "Total written (filtered):    154,181,570 bp (82.7%)\n",
      "  Read 1:    94,107,785 bp\n",
      "  Read 2:    60,073,785 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 355998 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.6\t0\t1\n",
      "9\t2\t1.4\t0\t2\n",
      "10\t4\t0.3\t0\t4\n",
      "11\t4\t0.1\t0\t4\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t19\t0.0\t0\t19\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t25\t0.0\t0\t25\n",
      "17\t99\t0.0\t0\t99\n",
      "18\t56045\t0.0\t0\t56045\n",
      "19\t64739\t0.0\t0\t64739\n",
      "20\t65079\t0.0\t0\t65079\n",
      "21\t58610\t0.0\t0\t58610\n",
      "22\t60073\t0.0\t0\t60073\n",
      "23\t51236\t0.0\t0\t51236\n",
      "24\t48\t0.0\t0\t48\n",
      "39\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 339330 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t1.4\t0\t1\n",
      "10\t1\t0.3\t0\t1\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t16\t0.0\t0\t16\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t8\t0.0\t0\t8\n",
      "19\t9\t0.0\t0\t9\n",
      "20\t112\t0.0\t0\t112\n",
      "21\t63644\t0.0\t0\t63644\n",
      "22\t61206\t0.0\t0\t61206\n",
      "23\t56423\t0.0\t0\t56423\n",
      "24\t57151\t0.0\t0\t57151\n",
      "25\t51497\t0.0\t0\t51497\n",
      "26\t49209\t0.0\t0\t49209\n",
      "27\t38\t0.0\t0\t38\n",
      "28\t2\t0.0\t0\t2\n",
      "38\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121678_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121678_11_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121678_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121678_11_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121678_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121678_11_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121678_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121678_11_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.00 s (10 µs/read; 5.77 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             95,852\n",
      "  Read 1 with adapter:                  93,250 (97.3%)\n",
      "  Read 2 with adapter:                  88,539 (92.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            9,674 (10.1%)\n",
      "Pairs written (passing filters):        86,178 (89.9%)\n",
      "\n",
      "Total basepairs processed:    48,884,520 bp\n",
      "  Read 1:    29,234,860 bp\n",
      "  Read 2:    19,649,660 bp\n",
      "Total written (filtered):     40,178,039 bp (82.2%)\n",
      "  Read 1:    24,522,719 bp\n",
      "  Read 2:    15,655,320 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 93250 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t5\t0.4\t0\t5\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t5\t0.0\t0\t5\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t24\t0.0\t0\t24\n",
      "18\t14644\t0.0\t0\t14644\n",
      "19\t16889\t0.0\t0\t16889\n",
      "20\t17011\t0.0\t0\t17011\n",
      "21\t15408\t0.0\t0\t15408\n",
      "22\t16019\t0.0\t0\t16019\n",
      "23\t13225\t0.0\t0\t13225\n",
      "24\t9\t0.0\t0\t9\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 88539 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t34\t0.0\t0\t34\n",
      "21\t16889\t0.0\t0\t16889\n",
      "22\t15980\t0.0\t0\t15980\n",
      "23\t14709\t0.0\t0\t14709\n",
      "24\t14923\t0.0\t0\t14923\n",
      "25\t13235\t0.0\t0\t13235\n",
      "26\t12749\t0.0\t0\t12749\n",
      "27\t11\t0.0\t0\t11\n",
      "60\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121679_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121679_13_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121679_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121679_13_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121679_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121679_13_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121679_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121679_13_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.95 s (9 µs/read; 6.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            316,849\n",
      "  Read 1 with adapter:                 308,351 (97.3%)\n",
      "  Read 2 with adapter:                 292,785 (92.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           31,765 (10.0%)\n",
      "Pairs written (passing filters):       285,084 (90.0%)\n",
      "\n",
      "Total basepairs processed:   161,592,990 bp\n",
      "  Read 1:    96,638,945 bp\n",
      "  Read 2:    64,954,045 bp\n",
      "Total written (filtered):    132,905,965 bp (82.2%)\n",
      "  Read 1:    81,120,487 bp\n",
      "  Read 2:    51,785,478 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 308351 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t4.8\t0\t4\n",
      "9\t1\t1.2\t0\t1\n",
      "10\t1\t0.3\t0\t1\n",
      "11\t9\t0.1\t0\t9\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t5\t0.0\t0\t5\n",
      "14\t17\t0.0\t0\t17\n",
      "15\t20\t0.0\t0\t20\n",
      "16\t21\t0.0\t0\t21\n",
      "17\t83\t0.0\t0\t83\n",
      "18\t48009\t0.0\t0\t48009\n",
      "19\t55379\t0.0\t0\t55379\n",
      "20\t56655\t0.0\t0\t56655\n",
      "21\t51178\t0.0\t0\t51178\n",
      "22\t52436\t0.0\t0\t52436\n",
      "23\t44491\t0.0\t0\t44491\n",
      "24\t30\t0.0\t0\t30\n",
      "30\t1\t0.0\t0\t1\n",
      "31\t2\t0.0\t0\t2\n",
      "32\t1\t0.0\t0\t1\n",
      "41\t1\t0.0\t0\t1\n",
      "43\t1\t0.0\t0\t1\n",
      "60\t2\t0.0\t0\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 292785 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t3\t0.1\t0\t3\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t5\t0.0\t0\t5\n",
      "19\t16\t0.0\t0\t16\n",
      "20\t87\t0.0\t0\t87\n",
      "21\t54723\t0.0\t0\t54723\n",
      "22\t52812\t0.0\t0\t52812\n",
      "23\t48903\t0.0\t0\t48903\n",
      "24\t49680\t0.0\t0\t49680\n",
      "25\t44102\t0.0\t0\t44102\n",
      "26\t42389\t0.0\t0\t42389\n",
      "27\t46\t0.0\t0\t46\n",
      "30\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121680_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121680_15_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121680_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121680_15_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121680_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121680_15_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121680_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121680_15_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 2.43 s (10 µs/read; 6.14 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            248,647\n",
      "  Read 1 with adapter:                 241,815 (97.3%)\n",
      "  Read 2 with adapter:                 229,975 (92.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           24,848 (10.0%)\n",
      "Pairs written (passing filters):       223,799 (90.0%)\n",
      "\n",
      "Total basepairs processed:   126,809,970 bp\n",
      "  Read 1:    75,837,335 bp\n",
      "  Read 2:    50,972,635 bp\n",
      "Total written (filtered):    104,332,265 bp (82.3%)\n",
      "  Read 1:    63,680,774 bp\n",
      "  Read 2:    40,651,491 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 241815 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t3.8\t0\t3\n",
      "9\t1\t0.9\t0\t1\n",
      "10\t4\t0.2\t0\t4\n",
      "12\t5\t0.0\t0\t5\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t69\t0.0\t0\t69\n",
      "18\t37474\t0.0\t0\t37474\n",
      "19\t43244\t0.0\t0\t43244\n",
      "20\t44420\t0.0\t0\t44420\n",
      "21\t40069\t0.0\t0\t40069\n",
      "22\t41569\t0.0\t0\t41569\n",
      "23\t34901\t0.0\t0\t34901\n",
      "24\t28\t0.0\t0\t28\n",
      "25\t2\t0.0\t0\t2\n",
      "27\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 229975 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t4\t0.0\t0\t4\n",
      "18\t5\t0.0\t0\t5\n",
      "19\t6\t0.0\t0\t6\n",
      "20\t74\t0.0\t0\t74\n",
      "21\t42774\t0.0\t0\t42774\n",
      "22\t41717\t0.0\t0\t41717\n",
      "23\t37843\t0.0\t0\t37843\n",
      "24\t38947\t0.0\t0\t38947\n",
      "25\t35038\t0.0\t0\t35038\n",
      "26\t33531\t0.0\t0\t33531\n",
      "27\t22\t0.0\t0\t22\n",
      "28\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121681_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121681_17_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121681_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121681_17_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121681_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121681_17_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121681_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121681_17_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 2.02 s (10 µs/read; 6.17 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            207,454\n",
      "  Read 1 with adapter:                 201,975 (97.4%)\n",
      "  Read 2 with adapter:                 192,058 (92.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           20,423 (9.8%)\n",
      "Pairs written (passing filters):       187,031 (90.2%)\n",
      "\n",
      "Total basepairs processed:   105,801,540 bp\n",
      "  Read 1:    63,273,470 bp\n",
      "  Read 2:    42,528,070 bp\n",
      "Total written (filtered):     87,193,544 bp (82.4%)\n",
      "  Read 1:    53,220,325 bp\n",
      "  Read 2:    33,973,219 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 201975 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t3.2\t0\t3\n",
      "9\t3\t0.8\t0\t3\n",
      "10\t2\t0.2\t0\t2\n",
      "11\t3\t0.0\t0\t3\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t11\t0.0\t0\t11\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t8\t0.0\t0\t8\n",
      "17\t60\t0.0\t0\t60\n",
      "18\t31587\t0.0\t0\t31587\n",
      "19\t36316\t0.0\t0\t36316\n",
      "20\t37050\t0.0\t0\t37050\n",
      "21\t33459\t0.0\t0\t33459\n",
      "22\t34304\t0.0\t0\t34304\n",
      "23\t29134\t0.0\t0\t29134\n",
      "24\t19\t0.0\t0\t19\n",
      "25\t2\t0.0\t0\t2\n",
      "31\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 192058 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t5\t0.0\t0\t5\n",
      "19\t7\t0.0\t0\t7\n",
      "20\t54\t0.0\t0\t54\n",
      "21\t35775\t0.0\t0\t35775\n",
      "22\t34503\t0.0\t0\t34503\n",
      "23\t32154\t0.0\t0\t32154\n",
      "24\t32732\t0.0\t0\t32732\n",
      "25\t28715\t0.0\t0\t28715\n",
      "26\t28079\t0.0\t0\t28079\n",
      "27\t22\t0.0\t0\t22\n",
      "32\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121682_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121682_19_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121682_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121682_19_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121682_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121682_19_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121682_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121682_19_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.71 s (10 µs/read; 5.82 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            165,821\n",
      "  Read 1 with adapter:                 161,411 (97.3%)\n",
      "  Read 2 with adapter:                 154,014 (92.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           15,834 (9.5%)\n",
      "Pairs written (passing filters):       149,987 (90.5%)\n",
      "\n",
      "Total basepairs processed:    84,568,710 bp\n",
      "  Read 1:    50,575,405 bp\n",
      "  Read 2:    33,993,305 bp\n",
      "Total written (filtered):     69,924,570 bp (82.7%)\n",
      "  Read 1:    42,679,301 bp\n",
      "  Read 2:    27,245,269 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 161411 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t2.5\t0\t3\n",
      "9\t1\t0.6\t0\t1\n",
      "10\t3\t0.2\t0\t3\n",
      "11\t4\t0.0\t0\t4\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t46\t0.0\t0\t46\n",
      "18\t25121\t0.0\t0\t25121\n",
      "19\t28953\t0.0\t0\t28953\n",
      "20\t29908\t0.0\t0\t29908\n",
      "21\t26709\t0.0\t0\t26709\n",
      "22\t27683\t0.0\t0\t27683\n",
      "23\t22947\t0.0\t0\t22947\n",
      "24\t12\t0.0\t0\t12\n",
      "26\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "38\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 154014 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t3\t0.0\t0\t3\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t4\t0.0\t0\t4\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t54\t0.0\t0\t54\n",
      "21\t28983\t0.0\t0\t28983\n",
      "22\t27868\t0.0\t0\t27868\n",
      "23\t25302\t0.0\t0\t25302\n",
      "24\t25940\t0.0\t0\t25940\n",
      "25\t23454\t0.0\t0\t23454\n",
      "26\t22373\t0.0\t0\t22373\n",
      "27\t24\t0.0\t0\t24\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121683_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121683_21_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121683_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121683_21_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121683_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121683_21_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121683_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121683_21_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.59 s (11 µs/read; 5.65 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            149,412\n",
      "  Read 1 with adapter:                 145,228 (97.2%)\n",
      "  Read 2 with adapter:                 138,274 (92.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           14,953 (10.0%)\n",
      "Pairs written (passing filters):       134,459 (90.0%)\n",
      "\n",
      "Total basepairs processed:    76,200,120 bp\n",
      "  Read 1:    45,570,660 bp\n",
      "  Read 2:    30,629,460 bp\n",
      "Total written (filtered):     62,685,460 bp (82.3%)\n",
      "  Read 1:    38,260,442 bp\n",
      "  Read 2:    24,425,018 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 145228 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.3\t0\t1\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t10\t0.0\t0\t10\n",
      "17\t45\t0.0\t0\t45\n",
      "18\t22753\t0.0\t0\t22753\n",
      "19\t25911\t0.0\t0\t25911\n",
      "20\t26596\t0.0\t0\t26596\n",
      "21\t24288\t0.0\t0\t24288\n",
      "22\t24636\t0.0\t0\t24636\n",
      "23\t20953\t0.0\t0\t20953\n",
      "24\t17\t0.0\t0\t17\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 138274 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t1\t0.0\t0\t1\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t53\t0.0\t0\t53\n",
      "21\t25986\t0.0\t0\t25986\n",
      "22\t25058\t0.0\t0\t25058\n",
      "23\t22833\t0.0\t0\t22833\n",
      "24\t23540\t0.0\t0\t23540\n",
      "25\t20833\t0.0\t0\t20833\n",
      "26\t19937\t0.0\t0\t19937\n",
      "27\t16\t0.0\t0\t16\n",
      "33\t2\t0.0\t0\t2\n",
      "40\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121684_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121684_23_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121684_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121684_23_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121684_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121684_23_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121684_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121684_23_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 3.28 s (10 µs/read; 6.28 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            343,261\n",
      "  Read 1 with adapter:                 334,370 (97.4%)\n",
      "  Read 2 with adapter:                 318,364 (92.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           32,990 (9.6%)\n",
      "Pairs written (passing filters):       310,271 (90.4%)\n",
      "\n",
      "Total basepairs processed:   175,063,110 bp\n",
      "  Read 1:   104,694,605 bp\n",
      "  Read 2:    70,368,505 bp\n",
      "Total written (filtered):    144,648,355 bp (82.6%)\n",
      "  Read 1:    88,289,164 bp\n",
      "  Read 2:    56,359,191 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 334370 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.2\t0\t1\n",
      "9\t5\t1.3\t0\t5\n",
      "10\t2\t0.3\t0\t2\n",
      "11\t5\t0.1\t0\t5\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t11\t0.0\t0\t11\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t13\t0.0\t0\t13\n",
      "17\t90\t0.0\t0\t90\n",
      "18\t52368\t0.0\t0\t52368\n",
      "19\t60523\t0.0\t0\t60523\n",
      "20\t61360\t0.0\t0\t61360\n",
      "21\t54944\t0.0\t0\t54944\n",
      "22\t56828\t0.0\t0\t56828\n",
      "23\t48158\t0.0\t0\t48158\n",
      "24\t46\t0.0\t0\t46\n",
      "33\t1\t0.0\t0\t1\n",
      "35\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "44\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 318364 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t11\t0.0\t0\t11\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t5\t0.0\t0\t5\n",
      "18\t5\t0.0\t0\t5\n",
      "19\t9\t0.0\t0\t9\n",
      "20\t93\t0.0\t0\t93\n",
      "21\t59527\t0.0\t0\t59527\n",
      "22\t57389\t0.0\t0\t57389\n",
      "23\t52503\t0.0\t0\t52503\n",
      "24\t54259\t0.0\t0\t54259\n",
      "25\t47921\t0.0\t0\t47921\n",
      "26\t46600\t0.0\t0\t46600\n",
      "27\t35\t0.0\t0\t35\n",
      "35\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121685_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121685_25_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121685_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121685_25_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121685_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121685_25_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121685_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121685_25_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.48 s (10 µs/read; 6.17 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            254,393\n",
      "  Read 1 with adapter:                 247,382 (97.2%)\n",
      "  Read 2 with adapter:                 234,645 (92.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           26,087 (10.3%)\n",
      "Pairs written (passing filters):       228,306 (89.7%)\n",
      "\n",
      "Total basepairs processed:   129,740,430 bp\n",
      "  Read 1:    77,589,865 bp\n",
      "  Read 2:    52,150,565 bp\n",
      "Total written (filtered):    106,436,280 bp (82.0%)\n",
      "  Read 1:    64,964,669 bp\n",
      "  Read 2:    41,471,611 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 247382 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t3.9\t0\t3\n",
      "9\t3\t1.0\t0\t3\n",
      "10\t2\t0.2\t0\t2\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t6\t0.0\t0\t6\n",
      "13\t5\t0.0\t0\t5\n",
      "14\t13\t0.0\t0\t13\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t82\t0.0\t0\t82\n",
      "18\t38495\t0.0\t0\t38495\n",
      "19\t44787\t0.0\t0\t44787\n",
      "20\t45421\t0.0\t0\t45421\n",
      "21\t40692\t0.0\t0\t40692\n",
      "22\t41815\t0.0\t0\t41815\n",
      "23\t36011\t0.0\t0\t36011\n",
      "24\t29\t0.0\t0\t29\n",
      "25\t2\t0.0\t0\t2\n",
      "47\t2\t0.0\t0\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 234645 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t7\t0.0\t0\t7\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t74\t0.0\t0\t74\n",
      "21\t44245\t0.0\t0\t44245\n",
      "22\t42188\t0.0\t0\t42188\n",
      "23\t38808\t0.0\t0\t38808\n",
      "24\t39553\t0.0\t0\t39553\n",
      "25\t35622\t0.0\t0\t35622\n",
      "26\t34102\t0.0\t0\t34102\n",
      "27\t20\t0.0\t0\t20\n",
      "33\t1\t0.0\t0\t1\n",
      "44\t1\t0.0\t0\t1\n",
      "64\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121686_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121686_27_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121686_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121686_27_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121686_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121686_27_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121686_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121686_27_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.16 s (11 µs/read; 5.35 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            103,905\n",
      "  Read 1 with adapter:                 101,232 (97.4%)\n",
      "  Read 2 with adapter:                  96,332 (92.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           10,027 (9.7%)\n",
      "Pairs written (passing filters):        93,878 (90.3%)\n",
      "\n",
      "Total basepairs processed:    52,991,550 bp\n",
      "  Read 1:    31,691,025 bp\n",
      "  Read 2:    21,300,525 bp\n",
      "Total written (filtered):     43,766,915 bp (82.6%)\n",
      "  Read 1:    26,714,313 bp\n",
      "  Read 2:    17,052,602 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 101232 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.6\t0\t1\n",
      "9\t2\t0.4\t0\t2\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t5\t0.0\t0\t5\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t32\t0.0\t0\t32\n",
      "18\t16068\t0.0\t0\t16068\n",
      "19\t18255\t0.0\t0\t18255\n",
      "20\t18746\t0.0\t0\t18746\n",
      "21\t16620\t0.0\t0\t16620\n",
      "22\t16935\t0.0\t0\t16935\n",
      "23\t14537\t0.0\t0\t14537\n",
      "24\t12\t0.0\t0\t12\n",
      "40\t1\t0.0\t0\t1\n",
      "44\t1\t0.0\t0\t1\n",
      "51\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 96332 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.6\t0\t1\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t6\t0.0\t0\t6\n",
      "20\t47\t0.0\t0\t47\n",
      "21\t18052\t0.0\t0\t18052\n",
      "22\t17427\t0.0\t0\t17427\n",
      "23\t15720\t0.0\t0\t15720\n",
      "24\t16399\t0.0\t0\t16399\n",
      "25\t14599\t0.0\t0\t14599\n",
      "26\t14056\t0.0\t0\t14056\n",
      "27\t15\t0.0\t0\t15\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121687_28_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121687_29_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121687_28_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121687_29_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121687_28_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121687_29_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121687_28_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121687_29_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.81 s (12 µs/read; 4.90 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             66,368\n",
      "  Read 1 with adapter:                  64,521 (97.2%)\n",
      "  Read 2 with adapter:                  61,032 (92.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            7,022 (10.6%)\n",
      "Pairs written (passing filters):        59,346 (89.4%)\n",
      "\n",
      "Total basepairs processed:    33,847,680 bp\n",
      "  Read 1:    20,242,240 bp\n",
      "  Read 2:    13,605,440 bp\n",
      "Total written (filtered):     27,666,882 bp (81.7%)\n",
      "  Read 1:    16,886,558 bp\n",
      "  Read 2:    10,780,324 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 64521 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t3\t0.0\t0\t3\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t22\t0.0\t0\t22\n",
      "18\t9869\t0.0\t0\t9869\n",
      "19\t11662\t0.0\t0\t11662\n",
      "20\t11874\t0.0\t0\t11874\n",
      "21\t10704\t0.0\t0\t10704\n",
      "22\t11032\t0.0\t0\t11032\n",
      "23\t9337\t0.0\t0\t9337\n",
      "24\t9\t0.0\t0\t9\n",
      "26\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 61032 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t16\t0.0\t0\t16\n",
      "21\t11652\t0.0\t0\t11652\n",
      "22\t10779\t0.0\t0\t10779\n",
      "23\t10214\t0.0\t0\t10214\n",
      "24\t10220\t0.0\t0\t10220\n",
      "25\t9211\t0.0\t0\t9211\n",
      "26\t8923\t0.0\t0\t8923\n",
      "27\t7\t0.0\t0\t7\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121688_30_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121688_31_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121688_30_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121688_31_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121688_30_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121688_31_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121688_30_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121688_31_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.51 s (11 µs/read; 5.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            136,741\n",
      "  Read 1 with adapter:                 132,822 (97.1%)\n",
      "  Read 2 with adapter:                 126,455 (92.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           13,839 (10.1%)\n",
      "Pairs written (passing filters):       122,902 (89.9%)\n",
      "\n",
      "Total basepairs processed:    69,737,910 bp\n",
      "  Read 1:    41,706,005 bp\n",
      "  Read 2:    28,031,905 bp\n",
      "Total written (filtered):     57,294,347 bp (82.2%)\n",
      "  Read 1:    34,969,730 bp\n",
      "  Read 2:    22,324,617 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 132822 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.1\t0\t1\n",
      "9\t2\t0.5\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t11\t0.0\t0\t11\n",
      "17\t32\t0.0\t0\t32\n",
      "18\t20270\t0.0\t0\t20270\n",
      "19\t23588\t0.0\t0\t23588\n",
      "20\t24502\t0.0\t0\t24502\n",
      "21\t22084\t0.0\t0\t22084\n",
      "22\t22934\t0.0\t0\t22934\n",
      "23\t19365\t0.0\t0\t19365\n",
      "24\t14\t0.0\t0\t14\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 126455 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t38\t0.0\t0\t38\n",
      "21\t23579\t0.0\t0\t23579\n",
      "22\t22708\t0.0\t0\t22708\n",
      "23\t21038\t0.0\t0\t21038\n",
      "24\t21560\t0.0\t0\t21560\n",
      "25\t19189\t0.0\t0\t19189\n",
      "26\t18299\t0.0\t0\t18299\n",
      "27\t23\t0.0\t0\t23\n",
      "37\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121689_32_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121689_33_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121689_32_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121689_33_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121689_32_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121689_33_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121689_32_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121689_33_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.82 s (10 µs/read; 5.98 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            181,624\n",
      "  Read 1 with adapter:                 176,525 (97.2%)\n",
      "  Read 2 with adapter:                 167,756 (92.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           18,504 (10.2%)\n",
      "Pairs written (passing filters):       163,120 (89.8%)\n",
      "\n",
      "Total basepairs processed:    92,628,240 bp\n",
      "  Read 1:    55,395,320 bp\n",
      "  Read 2:    37,232,920 bp\n",
      "Total written (filtered):     76,044,296 bp (82.1%)\n",
      "  Read 1:    46,413,857 bp\n",
      "  Read 2:    29,630,439 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 176525 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.8\t0\t1\n",
      "10\t1\t0.2\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t7\t0.0\t0\t7\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t42\t0.0\t0\t42\n",
      "18\t27235\t0.0\t0\t27235\n",
      "19\t31463\t0.0\t0\t31463\n",
      "20\t32408\t0.0\t0\t32408\n",
      "21\t29357\t0.0\t0\t29357\n",
      "22\t30155\t0.0\t0\t30155\n",
      "23\t25810\t0.0\t0\t25810\n",
      "24\t27\t0.0\t0\t27\n",
      "25\t2\t0.0\t0\t2\n",
      "33\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 167756 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.2\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t56\t0.0\t0\t56\n",
      "21\t31305\t0.0\t0\t31305\n",
      "22\t30487\t0.0\t0\t30487\n",
      "23\t27621\t0.0\t0\t27621\n",
      "24\t28517\t0.0\t0\t28517\n",
      "25\t25431\t0.0\t0\t25431\n",
      "26\t24303\t0.0\t0\t24303\n",
      "27\t18\t0.0\t0\t18\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121690_34_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121690_35_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121690_34_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121690_35_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121690_34_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121690_35_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121690_34_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121690_35_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.44 s (11 µs/read; 5.59 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            134,399\n",
      "  Read 1 with adapter:                 130,633 (97.2%)\n",
      "  Read 2 with adapter:                 124,093 (92.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           13,697 (10.2%)\n",
      "Pairs written (passing filters):       120,702 (89.8%)\n",
      "\n",
      "Total basepairs processed:    68,543,490 bp\n",
      "  Read 1:    40,991,695 bp\n",
      "  Read 2:    27,551,795 bp\n",
      "Total written (filtered):     56,270,832 bp (82.1%)\n",
      "  Read 1:    34,345,858 bp\n",
      "  Read 2:    21,924,974 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 130633 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.1\t0\t2\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t44\t0.0\t0\t44\n",
      "18\t20220\t0.0\t0\t20220\n",
      "19\t23666\t0.0\t0\t23666\n",
      "20\t23982\t0.0\t0\t23982\n",
      "21\t21631\t0.0\t0\t21631\n",
      "22\t22313\t0.0\t0\t22313\n",
      "23\t18746\t0.0\t0\t18746\n",
      "24\t10\t0.0\t0\t10\n",
      "26\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 124093 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t33\t0.0\t0\t33\n",
      "21\t23126\t0.0\t0\t23126\n",
      "22\t22324\t0.0\t0\t22324\n",
      "23\t20527\t0.0\t0\t20527\n",
      "24\t21205\t0.0\t0\t21205\n",
      "25\t18881\t0.0\t0\t18881\n",
      "26\t17957\t0.0\t0\t17957\n",
      "27\t22\t0.0\t0\t22\n",
      "31\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121691_36_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121691_37_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121691_36_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121691_37_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121691_36_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121691_37_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121691_36_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121691_37_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.35 s (10 µs/read; 5.72 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            224,175\n",
      "  Read 1 with adapter:                 217,928 (97.2%)\n",
      "  Read 2 with adapter:                 207,950 (92.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           21,909 (9.8%)\n",
      "Pairs written (passing filters):       202,266 (90.2%)\n",
      "\n",
      "Total basepairs processed:   114,329,250 bp\n",
      "  Read 1:    68,373,375 bp\n",
      "  Read 2:    45,955,875 bp\n",
      "Total written (filtered):     94,295,455 bp (82.5%)\n",
      "  Read 1:    57,554,676 bp\n",
      "  Read 2:    36,740,779 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 217928 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t3.4\t0\t1\n",
      "9\t4\t0.9\t0\t4\n",
      "10\t1\t0.2\t0\t1\n",
      "11\t5\t0.1\t0\t5\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t7\t0.0\t0\t7\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t7\t0.0\t0\t7\n",
      "17\t51\t0.0\t0\t51\n",
      "18\t33939\t0.0\t0\t33939\n",
      "19\t39206\t0.0\t0\t39206\n",
      "20\t40233\t0.0\t0\t40233\n",
      "21\t35873\t0.0\t0\t35873\n",
      "22\t37230\t0.0\t0\t37230\n",
      "23\t31336\t0.0\t0\t31336\n",
      "24\t21\t0.0\t0\t21\n",
      "27\t1\t0.0\t0\t1\n",
      "29\t2\t0.0\t0\t2\n",
      "31\t1\t0.0\t0\t1\n",
      "33\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "234\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 207950 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t2\t0.0\t0\t2\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t64\t0.0\t0\t64\n",
      "21\t38764\t0.0\t0\t38764\n",
      "22\t37770\t0.0\t0\t37770\n",
      "23\t34391\t0.0\t0\t34391\n",
      "24\t34925\t0.0\t0\t34925\n",
      "25\t31714\t0.0\t0\t31714\n",
      "26\t30279\t0.0\t0\t30279\n",
      "27\t25\t0.0\t0\t25\n",
      "28\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "53\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121692_38_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121692_39_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121692_38_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121692_39_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121692_38_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121692_39_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121692_38_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121692_39_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.87 s (11 µs/read; 5.51 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             80,216\n",
      "  Read 1 with adapter:                  77,965 (97.2%)\n",
      "  Read 2 with adapter:                  73,953 (92.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            8,302 (10.3%)\n",
      "Pairs written (passing filters):        71,914 (89.7%)\n",
      "\n",
      "Total basepairs processed:    40,910,160 bp\n",
      "  Read 1:    24,465,880 bp\n",
      "  Read 2:    16,444,280 bp\n",
      "Total written (filtered):     33,525,312 bp (81.9%)\n",
      "  Read 1:    20,462,757 bp\n",
      "  Read 2:    13,062,555 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 77965 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.2\t0\t1\n",
      "9\t1\t0.3\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t23\t0.0\t0\t23\n",
      "18\t12006\t0.0\t0\t12006\n",
      "19\t14026\t0.0\t0\t14026\n",
      "20\t14281\t0.0\t0\t14281\n",
      "21\t13017\t0.0\t0\t13017\n",
      "22\t13328\t0.0\t0\t13328\n",
      "23\t11273\t0.0\t0\t11273\n",
      "24\t4\t0.0\t0\t4\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 73953 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t2\t0.0\t0\t2\n",
      "18\t2\t0.0\t0\t2\n",
      "20\t30\t0.0\t0\t30\n",
      "21\t13728\t0.0\t0\t13728\n",
      "22\t13253\t0.0\t0\t13253\n",
      "23\t12470\t0.0\t0\t12470\n",
      "24\t12441\t0.0\t0\t12441\n",
      "25\t11187\t0.0\t0\t11187\n",
      "26\t10833\t0.0\t0\t10833\n",
      "27\t3\t0.0\t0\t3\n",
      "46\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121693_40_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121693_41_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121693_40_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121693_41_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121693_40_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121693_41_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121693_40_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121693_41_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.63 s (10 µs/read; 6.04 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            164,608\n",
      "  Read 1 with adapter:                 160,048 (97.2%)\n",
      "  Read 2 with adapter:                 152,781 (92.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           15,982 (9.7%)\n",
      "Pairs written (passing filters):       148,626 (90.3%)\n",
      "\n",
      "Total basepairs processed:    83,950,080 bp\n",
      "  Read 1:    50,205,440 bp\n",
      "  Read 2:    33,744,640 bp\n",
      "Total written (filtered):     69,288,385 bp (82.5%)\n",
      "  Read 1:    42,290,904 bp\n",
      "  Read 2:    26,997,481 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 160048 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.5\t0\t1\n",
      "9\t3\t0.6\t0\t3\n",
      "10\t2\t0.2\t0\t2\n",
      "12\t2\t0.0\t0\t2\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t40\t0.0\t0\t40\n",
      "18\t25002\t0.0\t0\t25002\n",
      "19\t28512\t0.0\t0\t28512\n",
      "20\t29241\t0.0\t0\t29241\n",
      "21\t26726\t0.0\t0\t26726\n",
      "22\t27383\t0.0\t0\t27383\n",
      "23\t23098\t0.0\t0\t23098\n",
      "24\t19\t0.0\t0\t19\n",
      "32\t2\t0.0\t0\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 152781 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t49\t0.0\t0\t49\n",
      "21\t28677\t0.0\t0\t28677\n",
      "22\t27458\t0.0\t0\t27458\n",
      "23\t25231\t0.0\t0\t25231\n",
      "24\t26023\t0.0\t0\t26023\n",
      "25\t22983\t0.0\t0\t22983\n",
      "26\t22323\t0.0\t0\t22323\n",
      "27\t20\t0.0\t0\t20\n",
      "38\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121694_42_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121694_43_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121694_42_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121694_43_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121694_42_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121694_43_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121694_42_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121694_43_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.73 s (12 µs/read; 4.84 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             58,992\n",
      "  Read 1 with adapter:                  57,341 (97.2%)\n",
      "  Read 2 with adapter:                  54,208 (91.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            6,281 (10.6%)\n",
      "Pairs written (passing filters):        52,711 (89.4%)\n",
      "\n",
      "Total basepairs processed:    30,085,920 bp\n",
      "  Read 1:    17,992,560 bp\n",
      "  Read 2:    12,093,360 bp\n",
      "Total written (filtered):     24,574,036 bp (81.7%)\n",
      "  Read 1:    14,999,414 bp\n",
      "  Read 2:     9,574,622 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 57341 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.2\t0\t1\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t6\t0.0\t0\t6\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t10\t0.0\t0\t10\n",
      "18\t8921\t0.0\t0\t8921\n",
      "19\t10341\t0.0\t0\t10341\n",
      "20\t10666\t0.0\t0\t10666\n",
      "21\t9376\t0.0\t0\t9376\n",
      "22\t9759\t0.0\t0\t9759\n",
      "23\t8248\t0.0\t0\t8248\n",
      "24\t6\t0.0\t0\t6\n",
      "40\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 54208 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "20\t18\t0.0\t0\t18\n",
      "21\t10166\t0.0\t0\t10166\n",
      "22\t9802\t0.0\t0\t9802\n",
      "23\t8793\t0.0\t0\t8793\n",
      "24\t9258\t0.0\t0\t9258\n",
      "25\t8219\t0.0\t0\t8219\n",
      "26\t7938\t0.0\t0\t7938\n",
      "27\t6\t0.0\t0\t6\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121695_44_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121695_45_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121695_44_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121695_45_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121695_44_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121695_45_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121695_44_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121695_45_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 2.42 s (11 µs/read; 5.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            229,809\n",
      "  Read 1 with adapter:                 223,555 (97.3%)\n",
      "  Read 2 with adapter:                 212,081 (92.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           23,391 (10.2%)\n",
      "Pairs written (passing filters):       206,418 (89.8%)\n",
      "\n",
      "Total basepairs processed:   117,202,590 bp\n",
      "  Read 1:    70,091,745 bp\n",
      "  Read 2:    47,110,845 bp\n",
      "Total written (filtered):     96,229,804 bp (82.1%)\n",
      "  Read 1:    58,736,074 bp\n",
      "  Read 2:    37,493,730 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 223555 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t3.5\t0\t2\n",
      "9\t4\t0.9\t0\t4\n",
      "10\t3\t0.2\t0\t3\n",
      "11\t6\t0.1\t0\t6\n",
      "12\t6\t0.0\t0\t6\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t14\t0.0\t0\t14\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t64\t0.0\t0\t64\n",
      "18\t34673\t0.0\t0\t34673\n",
      "19\t40049\t0.0\t0\t40049\n",
      "20\t41294\t0.0\t0\t41294\n",
      "21\t37301\t0.0\t0\t37301\n",
      "22\t38005\t0.0\t0\t38005\n",
      "23\t32087\t0.0\t0\t32087\n",
      "24\t27\t0.0\t0\t27\n",
      "26\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "35\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 212081 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t4\t0.0\t0\t4\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t63\t0.0\t0\t63\n",
      "21\t39586\t0.0\t0\t39586\n",
      "22\t38305\t0.0\t0\t38305\n",
      "23\t34643\t0.0\t0\t34643\n",
      "24\t36017\t0.0\t0\t36017\n",
      "25\t32350\t0.0\t0\t32350\n",
      "26\t31075\t0.0\t0\t31075\n",
      "27\t26\t0.0\t0\t26\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121696_46_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121696_47_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121696_46_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121696_47_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121696_46_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121696_47_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121696_46_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121696_47_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.25 s (11 µs/read; 5.52 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            114,634\n",
      "  Read 1 with adapter:                 111,413 (97.2%)\n",
      "  Read 2 with adapter:                 104,800 (91.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,724 (11.1%)\n",
      "Pairs written (passing filters):       101,910 (88.9%)\n",
      "\n",
      "Total basepairs processed:    58,463,340 bp\n",
      "  Read 1:    34,963,370 bp\n",
      "  Read 2:    23,499,970 bp\n",
      "Total written (filtered):     47,509,206 bp (81.3%)\n",
      "  Read 1:    28,997,978 bp\n",
      "  Read 2:    18,511,228 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 111413 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.7\t0\t1\n",
      "10\t4\t0.1\t0\t4\n",
      "11\t5\t0.0\t0\t5\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t33\t0.0\t0\t33\n",
      "18\t17408\t0.0\t0\t17408\n",
      "19\t19914\t0.0\t0\t19914\n",
      "20\t20315\t0.0\t0\t20315\n",
      "21\t18493\t0.0\t0\t18493\n",
      "22\t18881\t0.0\t0\t18881\n",
      "23\t16324\t0.0\t0\t16324\n",
      "24\t19\t0.0\t0\t19\n",
      "28\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 104800 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.7\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t38\t0.0\t0\t38\n",
      "21\t19666\t0.0\t0\t19666\n",
      "22\t18723\t0.0\t0\t18723\n",
      "23\t17184\t0.0\t0\t17184\n",
      "24\t18017\t0.0\t0\t18017\n",
      "25\t15803\t0.0\t0\t15803\n",
      "26\t15339\t0.0\t0\t15339\n",
      "27\t17\t0.0\t0\t17\n",
      "49\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121697_48_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121697_49_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121697_48_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121697_49_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121697_48_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121697_49_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121697_48_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121697_49_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.63 s (13 µs/read; 4.72 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             49,335\n",
      "  Read 1 with adapter:                  47,930 (97.2%)\n",
      "  Read 2 with adapter:                  45,553 (92.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            5,055 (10.2%)\n",
      "Pairs written (passing filters):        44,280 (89.8%)\n",
      "\n",
      "Total basepairs processed:    25,160,850 bp\n",
      "  Read 1:    15,047,175 bp\n",
      "  Read 2:    10,113,675 bp\n",
      "Total written (filtered):     20,642,581 bp (82.0%)\n",
      "  Read 1:    12,599,485 bp\n",
      "  Read 2:     8,043,096 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 47930 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.2\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t16\t0.0\t0\t16\n",
      "18\t7269\t0.0\t0\t7269\n",
      "19\t8668\t0.0\t0\t8668\n",
      "20\t8944\t0.0\t0\t8944\n",
      "21\t7771\t0.0\t0\t7771\n",
      "22\t8230\t0.0\t0\t8230\n",
      "23\t7015\t0.0\t0\t7015\n",
      "24\t7\t0.0\t0\t7\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 45553 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "20\t14\t0.0\t0\t14\n",
      "21\t8529\t0.0\t0\t8529\n",
      "22\t8218\t0.0\t0\t8218\n",
      "23\t7506\t0.0\t0\t7506\n",
      "24\t7601\t0.0\t0\t7601\n",
      "25\t6998\t0.0\t0\t6998\n",
      "26\t6678\t0.0\t0\t6678\n",
      "27\t3\t0.0\t0\t3\n",
      "32\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121698_50_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121698_51_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121698_50_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121698_51_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121698_50_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121698_51_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121698_50_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121698_51_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.32 s (22 µs/read; 2.69 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             14,319\n",
      "  Read 1 with adapter:                  13,866 (96.8%)\n",
      "  Read 2 with adapter:                  12,833 (89.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,875 (13.1%)\n",
      "Pairs written (passing filters):        12,444 (86.9%)\n",
      "\n",
      "Total basepairs processed:     7,302,690 bp\n",
      "  Read 1:     4,367,295 bp\n",
      "  Read 2:     2,935,395 bp\n",
      "Total written (filtered):      5,801,125 bp (79.4%)\n",
      "  Read 1:     3,540,895 bp\n",
      "  Read 2:     2,260,230 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 13866 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "17\t5\t0.0\t0\t5\n",
      "18\t2170\t0.0\t0\t2170\n",
      "19\t2437\t0.0\t0\t2437\n",
      "20\t2494\t0.0\t0\t2494\n",
      "21\t2324\t0.0\t0\t2324\n",
      "22\t2426\t0.0\t0\t2426\n",
      "23\t2006\t0.0\t0\t2006\n",
      "24\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 12833 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t3\t0.0\t0\t3\n",
      "21\t2351\t0.0\t0\t2351\n",
      "22\t2314\t0.0\t0\t2314\n",
      "23\t2129\t0.0\t0\t2129\n",
      "24\t2175\t0.0\t0\t2175\n",
      "25\t2020\t0.0\t0\t2020\n",
      "26\t1841\t0.0\t0\t1841\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121699_52_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121699_53_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121699_52_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121699_53_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121699_52_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121699_53_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121699_52_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121699_53_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.41 s (20 µs/read; 3.07 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             20,959\n",
      "  Read 1 with adapter:                  20,357 (97.1%)\n",
      "  Read 2 with adapter:                  19,037 (90.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,450 (11.7%)\n",
      "Pairs written (passing filters):        18,509 (88.3%)\n",
      "\n",
      "Total basepairs processed:    10,689,090 bp\n",
      "  Read 1:     6,392,495 bp\n",
      "  Read 2:     4,296,595 bp\n",
      "Total written (filtered):      8,628,512 bp (80.7%)\n",
      "  Read 1:     5,266,608 bp\n",
      "  Read 2:     3,361,904 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 20357 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t7\t0.0\t0\t7\n",
      "18\t3085\t0.0\t0\t3085\n",
      "19\t3612\t0.0\t0\t3612\n",
      "20\t3759\t0.0\t0\t3759\n",
      "21\t3463\t0.0\t0\t3463\n",
      "22\t3487\t0.0\t0\t3487\n",
      "23\t2937\t0.0\t0\t2937\n",
      "24\t5\t0.0\t0\t5\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 19037 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t6\t0.0\t0\t6\n",
      "21\t3547\t0.0\t0\t3547\n",
      "22\t3449\t0.0\t0\t3449\n",
      "23\t3103\t0.0\t0\t3103\n",
      "24\t3209\t0.0\t0\t3209\n",
      "25\t2928\t0.0\t0\t2928\n",
      "26\t2792\t0.0\t0\t2792\n",
      "27\t2\t0.0\t0\t2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121700_54_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121700_55_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121700_54_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121700_55_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121700_54_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121700_55_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121700_54_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121700_55_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.38 s (19 µs/read; 3.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             20,194\n",
      "  Read 1 with adapter:                  19,610 (97.1%)\n",
      "  Read 2 with adapter:                  18,045 (89.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,655 (13.1%)\n",
      "Pairs written (passing filters):        17,539 (86.9%)\n",
      "\n",
      "Total basepairs processed:    10,298,940 bp\n",
      "  Read 1:     6,159,170 bp\n",
      "  Read 2:     4,139,770 bp\n",
      "Total written (filtered):      8,176,314 bp (79.4%)\n",
      "  Read 1:     4,990,682 bp\n",
      "  Read 2:     3,185,632 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 19610 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t3\t0.0\t0\t3\n",
      "17\t7\t0.0\t0\t7\n",
      "18\t3003\t0.0\t0\t3003\n",
      "19\t3491\t0.0\t0\t3491\n",
      "20\t3617\t0.0\t0\t3617\n",
      "21\t3224\t0.0\t0\t3224\n",
      "22\t3416\t0.0\t0\t3416\n",
      "23\t2844\t0.0\t0\t2844\n",
      "24\t4\t0.0\t0\t4\n",
      "31\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 18045 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t10\t0.0\t0\t10\n",
      "21\t3352\t0.0\t0\t3352\n",
      "22\t3231\t0.0\t0\t3231\n",
      "23\t2977\t0.0\t0\t2977\n",
      "24\t3007\t0.0\t0\t3007\n",
      "25\t2837\t0.0\t0\t2837\n",
      "26\t2625\t0.0\t0\t2625\n",
      "27\t4\t0.0\t0\t4\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121701_56_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121701_57_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121701_56_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121701_57_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121701_56_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121701_57_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121701_56_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121701_57_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.83 s (12 µs/read; 4.86 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             67,293\n",
      "  Read 1 with adapter:                  65,501 (97.3%)\n",
      "  Read 2 with adapter:                  61,930 (92.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            6,987 (10.4%)\n",
      "Pairs written (passing filters):        60,306 (89.6%)\n",
      "\n",
      "Total basepairs processed:    34,319,430 bp\n",
      "  Read 1:    20,524,365 bp\n",
      "  Read 2:    13,795,065 bp\n",
      "Total written (filtered):     28,113,964 bp (81.9%)\n",
      "  Read 1:    17,159,382 bp\n",
      "  Read 2:    10,954,582 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 65501 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t19\t0.0\t0\t19\n",
      "18\t10169\t0.0\t0\t10169\n",
      "19\t11601\t0.0\t0\t11601\n",
      "20\t12117\t0.0\t0\t12117\n",
      "21\t10719\t0.0\t0\t10719\n",
      "22\t11255\t0.0\t0\t11255\n",
      "23\t9603\t0.0\t0\t9603\n",
      "24\t7\t0.0\t0\t7\n",
      "27\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 61930 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t2\t0.0\t0\t2\n",
      "20\t18\t0.0\t0\t18\n",
      "21\t11697\t0.0\t0\t11697\n",
      "22\t11095\t0.0\t0\t11095\n",
      "23\t10267\t0.0\t0\t10267\n",
      "24\t10352\t0.0\t0\t10352\n",
      "25\t9484\t0.0\t0\t9484\n",
      "26\t9007\t0.0\t0\t9007\n",
      "27\t6\t0.0\t0\t6\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121702_58_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121702_59_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121702_58_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121702_59_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121702_58_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121702_59_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121702_58_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121702_59_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.28 s (33 µs/read; 1.81 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              8,334\n",
      "  Read 1 with adapter:                   8,081 (97.0%)\n",
      "  Read 2 with adapter:                   7,509 (90.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,046 (12.6%)\n",
      "Pairs written (passing filters):         7,288 (87.4%)\n",
      "\n",
      "Total basepairs processed:     4,250,340 bp\n",
      "  Read 1:     2,541,870 bp\n",
      "  Read 2:     1,708,470 bp\n",
      "Total written (filtered):      3,397,413 bp (79.9%)\n",
      "  Read 1:     2,073,898 bp\n",
      "  Read 2:     1,323,515 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 8081 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t1267\t0.0\t0\t1267\n",
      "19\t1465\t0.0\t0\t1465\n",
      "20\t1495\t0.0\t0\t1495\n",
      "21\t1281\t0.0\t0\t1281\n",
      "22\t1390\t0.0\t0\t1390\n",
      "23\t1178\t0.0\t0\t1178\n",
      "24\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 7509 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t3\t0.0\t0\t3\n",
      "21\t1388\t0.0\t0\t1388\n",
      "22\t1308\t0.0\t0\t1308\n",
      "23\t1248\t0.0\t0\t1248\n",
      "24\t1240\t0.0\t0\t1240\n",
      "25\t1164\t0.0\t0\t1164\n",
      "26\t1154\t0.0\t0\t1154\n",
      "27\t2\t0.0\t0\t2\n",
      "31\t1\t0.0\t0\t1\n",
      "59\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121703_60_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121703_61_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121703_60_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121703_61_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121703_60_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121703_61_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121703_60_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121703_61_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.43 s (18 µs/read; 3.39 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             24,189\n",
      "  Read 1 with adapter:                  23,503 (97.2%)\n",
      "  Read 2 with adapter:                  21,685 (89.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            3,102 (12.8%)\n",
      "Pairs written (passing filters):        21,087 (87.2%)\n",
      "\n",
      "Total basepairs processed:    12,336,390 bp\n",
      "  Read 1:     7,377,645 bp\n",
      "  Read 2:     4,958,745 bp\n",
      "Total written (filtered):      9,830,774 bp (79.7%)\n",
      "  Read 1:     6,000,254 bp\n",
      "  Read 2:     3,830,520 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 23503 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t4\t0.0\t0\t4\n",
      "18\t3615\t0.0\t0\t3615\n",
      "19\t4126\t0.0\t0\t4126\n",
      "20\t4339\t0.0\t0\t4339\n",
      "21\t3919\t0.0\t0\t3919\n",
      "22\t4058\t0.0\t0\t4058\n",
      "23\t3438\t0.0\t0\t3438\n",
      "24\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 21685 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t10\t0.0\t0\t10\n",
      "21\t4101\t0.0\t0\t4101\n",
      "22\t3941\t0.0\t0\t3941\n",
      "23\t3556\t0.0\t0\t3556\n",
      "24\t3609\t0.0\t0\t3609\n",
      "25\t3363\t0.0\t0\t3363\n",
      "26\t3097\t0.0\t0\t3097\n",
      "27\t4\t0.0\t0\t4\n",
      "43\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121704_62_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121704_63_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121704_62_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121704_63_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121704_62_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121704_63_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121704_62_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121704_63_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.25 s (46 µs/read; 1.32 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              5,549\n",
      "  Read 1 with adapter:                   5,397 (97.3%)\n",
      "  Read 2 with adapter:                   5,015 (90.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              667 (12.0%)\n",
      "Pairs written (passing filters):         4,882 (88.0%)\n",
      "\n",
      "Total basepairs processed:     2,829,990 bp\n",
      "  Read 1:     1,692,445 bp\n",
      "  Read 2:     1,137,545 bp\n",
      "Total written (filtered):      2,276,001 bp (80.4%)\n",
      "  Read 1:     1,389,152 bp\n",
      "  Read 2:       886,849 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 5397 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t858\t0.0\t0\t858\n",
      "19\t967\t0.0\t0\t967\n",
      "20\t1009\t0.0\t0\t1009\n",
      "21\t860\t0.0\t0\t860\n",
      "22\t903\t0.0\t0\t903\n",
      "23\t797\t0.0\t0\t797\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 5015 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "20\t2\t0.0\t0\t2\n",
      "21\t964\t0.0\t0\t964\n",
      "22\t871\t0.0\t0\t871\n",
      "23\t837\t0.0\t0\t837\n",
      "24\t857\t0.0\t0\t857\n",
      "25\t780\t0.0\t0\t780\n",
      "26\t702\t0.0\t0\t702\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121705_64_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121705_65_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121705_64_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121705_65_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121705_64_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121705_65_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121705_64_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121705_65_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 3.33 s (10 µs/read; 6.21 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            344,337\n",
      "  Read 1 with adapter:                 334,766 (97.2%)\n",
      "  Read 2 with adapter:                 317,372 (92.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           35,670 (10.4%)\n",
      "Pairs written (passing filters):       308,667 (89.6%)\n",
      "\n",
      "Total basepairs processed:   175,611,870 bp\n",
      "  Read 1:   105,022,785 bp\n",
      "  Read 2:    70,589,085 bp\n",
      "Total written (filtered):    143,899,876 bp (81.9%)\n",
      "  Read 1:    87,831,910 bp\n",
      "  Read 2:    56,067,966 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 334766 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.3\t0\t1\n",
      "9\t4\t1.3\t0\t4\n",
      "10\t8\t0.3\t0\t8\n",
      "11\t7\t0.1\t0\t7\n",
      "12\t6\t0.0\t0\t6\n",
      "13\t6\t0.0\t0\t6\n",
      "14\t19\t0.0\t0\t19\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t15\t0.0\t0\t15\n",
      "17\t102\t0.0\t0\t102\n",
      "18\t52175\t0.0\t0\t52175\n",
      "19\t60493\t0.0\t0\t60493\n",
      "20\t61456\t0.0\t0\t61456\n",
      "21\t55267\t0.0\t0\t55267\n",
      "22\t56696\t0.0\t0\t56696\n",
      "23\t48458\t0.0\t0\t48458\n",
      "24\t43\t0.0\t0\t43\n",
      "26\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "63\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 317372 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t4\t0.0\t0\t4\n",
      "19\t10\t0.0\t0\t10\n",
      "20\t99\t0.0\t0\t99\n",
      "21\t59575\t0.0\t0\t59575\n",
      "22\t57160\t0.0\t0\t57160\n",
      "23\t52497\t0.0\t0\t52497\n",
      "24\t53427\t0.0\t0\t53427\n",
      "25\t48103\t0.0\t0\t48103\n",
      "26\t46434\t0.0\t0\t46434\n",
      "27\t40\t0.0\t0\t40\n",
      "28\t2\t0.0\t0\t2\n",
      "36\t1\t0.0\t0\t1\n",
      "61\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121706_66_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121706_67_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121706_66_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121706_67_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121706_66_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121706_67_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121706_66_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121706_67_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 4.17 s (10 µs/read; 6.00 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            416,322\n",
      "  Read 1 with adapter:                 405,067 (97.3%)\n",
      "  Read 2 with adapter:                 385,489 (92.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           41,066 (9.9%)\n",
      "Pairs written (passing filters):       375,256 (90.1%)\n",
      "\n",
      "Total basepairs processed:   212,324,220 bp\n",
      "  Read 1:   126,978,210 bp\n",
      "  Read 2:    85,346,010 bp\n",
      "Total written (filtered):    174,944,302 bp (82.4%)\n",
      "  Read 1:   106,779,862 bp\n",
      "  Read 2:    68,164,440 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 405067 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t6.4\t0\t4\n",
      "9\t2\t1.6\t0\t2\n",
      "10\t3\t0.4\t0\t3\n",
      "11\t5\t0.1\t0\t5\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t17\t0.0\t0\t17\n",
      "15\t12\t0.0\t0\t12\n",
      "16\t26\t0.0\t0\t26\n",
      "17\t108\t0.0\t0\t108\n",
      "18\t63733\t0.0\t0\t63733\n",
      "19\t72608\t0.0\t0\t72608\n",
      "20\t74499\t0.0\t0\t74499\n",
      "21\t66813\t0.0\t0\t66813\n",
      "22\t68675\t0.0\t0\t68675\n",
      "23\t58490\t0.0\t0\t58490\n",
      "24\t63\t0.0\t0\t63\n",
      "25\t2\t0.0\t0\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 385489 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t5\t0.0\t0\t5\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t12\t0.0\t0\t12\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t7\t0.0\t0\t7\n",
      "18\t8\t0.0\t0\t8\n",
      "19\t9\t0.0\t0\t9\n",
      "20\t138\t0.0\t0\t138\n",
      "21\t72215\t0.0\t0\t72215\n",
      "22\t69374\t0.0\t0\t69374\n",
      "23\t63845\t0.0\t0\t63845\n",
      "24\t65324\t0.0\t0\t65324\n",
      "25\t58672\t0.0\t0\t58672\n",
      "26\t55823\t0.0\t0\t55823\n",
      "27\t51\t0.0\t0\t51\n",
      "28\t2\t0.0\t0\t2\n",
      "62\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121707_68_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121707_69_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121707_68_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121707_69_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121707_68_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121707_69_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121707_68_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121707_69_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.91 s (11 µs/read; 5.35 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             81,485\n",
      "  Read 1 with adapter:                  79,303 (97.3%)\n",
      "  Read 2 with adapter:                  74,804 (91.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            8,666 (10.6%)\n",
      "Pairs written (passing filters):        72,819 (89.4%)\n",
      "\n",
      "Total basepairs processed:    41,557,350 bp\n",
      "  Read 1:    24,852,925 bp\n",
      "  Read 2:    16,704,425 bp\n",
      "Total written (filtered):     33,947,823 bp (81.7%)\n",
      "  Read 1:    20,720,493 bp\n",
      "  Read 2:    13,227,330 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 79303 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t2\t0.3\t0\t2\n",
      "10\t2\t0.1\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t34\t0.0\t0\t34\n",
      "18\t12320\t0.0\t0\t12320\n",
      "19\t14302\t0.0\t0\t14302\n",
      "20\t14387\t0.0\t0\t14387\n",
      "21\t13117\t0.0\t0\t13117\n",
      "22\t13446\t0.0\t0\t13446\n",
      "23\t11676\t0.0\t0\t11676\n",
      "24\t5\t0.0\t0\t5\n",
      "28\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 74804 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.1\t0\t2\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "19\t2\t0.0\t0\t2\n",
      "20\t25\t0.0\t0\t25\n",
      "21\t14012\t0.0\t0\t14012\n",
      "22\t13493\t0.0\t0\t13493\n",
      "23\t12382\t0.0\t0\t12382\n",
      "24\t12641\t0.0\t0\t12641\n",
      "25\t11252\t0.0\t0\t11252\n",
      "26\t10983\t0.0\t0\t10983\n",
      "27\t8\t0.0\t0\t8\n",
      "53\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121708_70_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121708_71_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121708_70_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121708_71_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121708_70_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121708_71_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121708_70_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121708_71_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.44 s (16 µs/read; 3.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             27,196\n",
      "  Read 1 with adapter:                  26,427 (97.2%)\n",
      "  Read 2 with adapter:                  24,481 (90.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            3,391 (12.5%)\n",
      "Pairs written (passing filters):        23,805 (87.5%)\n",
      "\n",
      "Total basepairs processed:    13,869,960 bp\n",
      "  Read 1:     8,294,780 bp\n",
      "  Read 2:     5,575,180 bp\n",
      "Total written (filtered):     11,097,905 bp (80.0%)\n",
      "  Read 1:     6,774,112 bp\n",
      "  Read 2:     4,323,793 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 26427 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "17\t6\t0.0\t0\t6\n",
      "18\t4153\t0.0\t0\t4153\n",
      "19\t4745\t0.0\t0\t4745\n",
      "20\t4859\t0.0\t0\t4859\n",
      "21\t4346\t0.0\t0\t4346\n",
      "22\t4473\t0.0\t0\t4473\n",
      "23\t3839\t0.0\t0\t3839\n",
      "24\t3\t0.0\t0\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 24481 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1\t0.0\t0\t1\n",
      "20\t5\t0.0\t0\t5\n",
      "21\t4531\t0.0\t0\t4531\n",
      "22\t4333\t0.0\t0\t4333\n",
      "23\t4146\t0.0\t0\t4146\n",
      "24\t4150\t0.0\t0\t4150\n",
      "25\t3714\t0.0\t0\t3714\n",
      "26\t3599\t0.0\t0\t3599\n",
      "27\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121709_72_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121709_73_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121709_72_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121709_73_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121709_72_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121709_73_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121709_72_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121709_73_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.47 s (14 µs/read; 4.23 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             32,857\n",
      "  Read 1 with adapter:                  31,953 (97.2%)\n",
      "  Read 2 with adapter:                  29,876 (90.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            3,780 (11.5%)\n",
      "Pairs written (passing filters):        29,077 (88.5%)\n",
      "\n",
      "Total basepairs processed:    16,757,070 bp\n",
      "  Read 1:    10,021,385 bp\n",
      "  Read 2:     6,735,685 bp\n",
      "Total written (filtered):     13,555,322 bp (80.9%)\n",
      "  Read 1:     8,273,880 bp\n",
      "  Read 2:     5,281,442 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 31953 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t9\t0.0\t0\t9\n",
      "18\t4981\t0.0\t0\t4981\n",
      "19\t5717\t0.0\t0\t5717\n",
      "20\t5839\t0.0\t0\t5839\n",
      "21\t5216\t0.0\t0\t5216\n",
      "22\t5501\t0.0\t0\t5501\n",
      "23\t4678\t0.0\t0\t4678\n",
      "24\t5\t0.0\t0\t5\n",
      "29\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 29876 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t12\t0.0\t0\t12\n",
      "21\t5470\t0.0\t0\t5470\n",
      "22\t5373\t0.0\t0\t5373\n",
      "23\t5010\t0.0\t0\t5010\n",
      "24\t5115\t0.0\t0\t5115\n",
      "25\t4536\t0.0\t0\t4536\n",
      "26\t4354\t0.0\t0\t4354\n",
      "27\t2\t0.0\t0\t2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121710_74_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121710_75_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121710_74_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121710_75_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121710_74_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121710_75_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121710_74_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121710_75_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.22 s (44 µs/read; 1.36 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              4,908\n",
      "  Read 1 with adapter:                   4,754 (96.9%)\n",
      "  Read 2 with adapter:                   4,257 (86.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              773 (15.7%)\n",
      "Pairs written (passing filters):         4,135 (84.3%)\n",
      "\n",
      "Total basepairs processed:     2,503,080 bp\n",
      "  Read 1:     1,496,940 bp\n",
      "  Read 2:     1,006,140 bp\n",
      "Total written (filtered):      1,927,695 bp (77.0%)\n",
      "  Read 1:     1,176,557 bp\n",
      "  Read 2:       751,138 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 4754 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t757\t0.0\t0\t757\n",
      "19\t836\t0.0\t0\t836\n",
      "20\t829\t0.0\t0\t829\n",
      "21\t772\t0.0\t0\t772\n",
      "22\t824\t0.0\t0\t824\n",
      "23\t735\t0.0\t0\t735\n",
      "24\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 4257 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "20\t2\t0.0\t0\t2\n",
      "21\t819\t0.0\t0\t819\n",
      "22\t759\t0.0\t0\t759\n",
      "23\t673\t0.0\t0\t673\n",
      "24\t752\t0.0\t0\t752\n",
      "25\t638\t0.0\t0\t638\n",
      "26\t613\t0.0\t0\t613\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121711_76_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121711_77_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121711_76_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121711_77_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121711_76_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121711_77_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121711_76_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121711_77_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.68 s (14 µs/read; 4.37 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             49,793\n",
      "  Read 1 with adapter:                  48,368 (97.1%)\n",
      "  Read 2 with adapter:                  45,264 (90.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            5,807 (11.7%)\n",
      "Pairs written (passing filters):        43,986 (88.3%)\n",
      "\n",
      "Total basepairs processed:    25,394,430 bp\n",
      "  Read 1:    15,186,865 bp\n",
      "  Read 2:    10,207,565 bp\n",
      "Total written (filtered):     20,506,528 bp (80.8%)\n",
      "  Read 1:    12,516,130 bp\n",
      "  Read 2:     7,990,398 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 48368 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.8\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t18\t0.0\t0\t18\n",
      "18\t7506\t0.0\t0\t7506\n",
      "19\t8677\t0.0\t0\t8677\n",
      "20\t8970\t0.0\t0\t8970\n",
      "21\t7956\t0.0\t0\t7956\n",
      "22\t8261\t0.0\t0\t8261\n",
      "23\t6957\t0.0\t0\t6957\n",
      "24\t10\t0.0\t0\t10\n",
      "25\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 45264 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t2\t0.0\t0\t2\n",
      "15\t1\t0.0\t0\t1\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t2\t0.0\t0\t2\n",
      "20\t15\t0.0\t0\t15\n",
      "21\t8594\t0.0\t0\t8594\n",
      "22\t8129\t0.0\t0\t8129\n",
      "23\t7417\t0.0\t0\t7417\n",
      "24\t7747\t0.0\t0\t7747\n",
      "25\t6894\t0.0\t0\t6894\n",
      "26\t6458\t0.0\t0\t6458\n",
      "27\t2\t0.0\t0\t2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121712_78_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121712_79_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121712_78_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121712_79_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121712_78_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121712_79_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121712_78_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121712_79_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.17 s (11 µs/read; 5.45 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            197,224\n",
      "  Read 1 with adapter:                 191,943 (97.3%)\n",
      "  Read 2 with adapter:                 181,981 (92.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           19,971 (10.1%)\n",
      "Pairs written (passing filters):       177,253 (89.9%)\n",
      "\n",
      "Total basepairs processed:   100,584,240 bp\n",
      "  Read 1:    60,153,320 bp\n",
      "  Read 2:    40,430,920 bp\n",
      "Total written (filtered):     82,634,027 bp (82.2%)\n",
      "  Read 1:    50,437,083 bp\n",
      "  Read 2:    32,196,944 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 191943 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t3\t0.2\t0\t3\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t8\t0.0\t0\t8\n",
      "17\t44\t0.0\t0\t44\n",
      "18\t29908\t0.0\t0\t29908\n",
      "19\t34447\t0.0\t0\t34447\n",
      "20\t35353\t0.0\t0\t35353\n",
      "21\t31785\t0.0\t0\t31785\n",
      "22\t32514\t0.0\t0\t32514\n",
      "23\t27840\t0.0\t0\t27840\n",
      "24\t22\t0.0\t0\t22\n",
      "32\t1\t0.0\t0\t1\n",
      "35\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 181981 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t9\t0.0\t0\t9\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t6\t0.0\t0\t6\n",
      "20\t60\t0.0\t0\t60\n",
      "21\t33812\t0.0\t0\t33812\n",
      "22\t32958\t0.0\t0\t32958\n",
      "23\t30086\t0.0\t0\t30086\n",
      "24\t30787\t0.0\t0\t30787\n",
      "25\t27789\t0.0\t0\t27789\n",
      "26\t26433\t0.0\t0\t26433\n",
      "27\t31\t0.0\t0\t31\n",
      "29\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "32\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121713_80_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121713_81_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121713_80_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121713_81_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121713_80_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121713_81_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121713_80_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121713_81_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.79 s (10 µs/read; 5.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            170,758\n",
      "  Read 1 with adapter:                 165,742 (97.1%)\n",
      "  Read 2 with adapter:                 156,948 (91.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           18,349 (10.7%)\n",
      "Pairs written (passing filters):       152,409 (89.3%)\n",
      "\n",
      "Total basepairs processed:    87,086,580 bp\n",
      "  Read 1:    52,081,190 bp\n",
      "  Read 2:    35,005,390 bp\n",
      "Total written (filtered):     71,052,054 bp (81.6%)\n",
      "  Read 1:    43,367,422 bp\n",
      "  Read 2:    27,684,632 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 165742 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t2.6\t0\t2\n",
      "9\t3\t0.7\t0\t3\n",
      "10\t3\t0.2\t0\t3\n",
      "11\t7\t0.0\t0\t7\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t7\t0.0\t0\t7\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t11\t0.0\t0\t11\n",
      "17\t44\t0.0\t0\t44\n",
      "18\t25727\t0.0\t0\t25727\n",
      "19\t29769\t0.0\t0\t29769\n",
      "20\t30353\t0.0\t0\t30353\n",
      "21\t27382\t0.0\t0\t27382\n",
      "22\t28371\t0.0\t0\t28371\n",
      "23\t24038\t0.0\t0\t24038\n",
      "24\t18\t0.0\t0\t18\n",
      "25\t2\t0.0\t0\t2\n",
      "28\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 156948 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t54\t0.0\t0\t54\n",
      "21\t29379\t0.0\t0\t29379\n",
      "22\t28333\t0.0\t0\t28333\n",
      "23\t25988\t0.0\t0\t25988\n",
      "24\t26572\t0.0\t0\t26572\n",
      "25\t23804\t0.0\t0\t23804\n",
      "26\t22782\t0.0\t0\t22782\n",
      "27\t21\t0.0\t0\t21\n",
      "62\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121714_82_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121714_83_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121714_82_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121714_83_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121714_82_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121714_83_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121714_82_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121714_83_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 3.09 s (10 µs/read; 6.02 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            309,657\n",
      "  Read 1 with adapter:                 301,273 (97.3%)\n",
      "  Read 2 with adapter:                 284,385 (91.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           32,799 (10.6%)\n",
      "Pairs written (passing filters):       276,858 (89.4%)\n",
      "\n",
      "Total basepairs processed:   157,925,070 bp\n",
      "  Read 1:    94,445,385 bp\n",
      "  Read 2:    63,479,685 bp\n",
      "Total written (filtered):    129,071,295 bp (81.7%)\n",
      "  Read 1:    78,780,305 bp\n",
      "  Read 2:    50,290,990 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 301273 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t4.7\t0\t2\n",
      "9\t6\t1.2\t0\t6\n",
      "10\t1\t0.3\t0\t1\n",
      "11\t4\t0.1\t0\t4\n",
      "12\t6\t0.0\t0\t6\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t14\t0.0\t0\t14\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t11\t0.0\t0\t11\n",
      "17\t103\t0.0\t0\t103\n",
      "18\t46760\t0.0\t0\t46760\n",
      "19\t54306\t0.0\t0\t54306\n",
      "20\t55444\t0.0\t0\t55444\n",
      "21\t50194\t0.0\t0\t50194\n",
      "22\t50872\t0.0\t0\t50872\n",
      "23\t43489\t0.0\t0\t43489\n",
      "24\t48\t0.0\t0\t48\n",
      "25\t1\t0.0\t0\t1\n",
      "27\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 284385 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t4\t0.0\t0\t4\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t9\t0.0\t0\t9\n",
      "20\t104\t0.0\t0\t104\n",
      "21\t53159\t0.0\t0\t53159\n",
      "22\t51530\t0.0\t0\t51530\n",
      "23\t47018\t0.0\t0\t47018\n",
      "24\t48091\t0.0\t0\t48091\n",
      "25\t43184\t0.0\t0\t43184\n",
      "26\t41236\t0.0\t0\t41236\n",
      "27\t27\t0.0\t0\t27\n",
      "29\t2\t0.0\t0\t2\n",
      "37\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121715_84_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121715_85_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121715_84_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121715_85_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121715_84_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121715_85_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121715_84_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121715_85_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 4.08 s (10 µs/read; 6.12 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            416,343\n",
      "  Read 1 with adapter:                 405,322 (97.4%)\n",
      "  Read 2 with adapter:                 382,754 (91.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           43,507 (10.4%)\n",
      "Pairs written (passing filters):       372,836 (89.6%)\n",
      "\n",
      "Total basepairs processed:   212,334,930 bp\n",
      "  Read 1:   126,984,615 bp\n",
      "  Read 2:    85,350,315 bp\n",
      "Total written (filtered):    173,816,084 bp (81.9%)\n",
      "  Read 1:   106,092,368 bp\n",
      "  Read 2:    67,723,716 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 405322 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t6.4\t0\t2\n",
      "9\t4\t1.6\t0\t4\n",
      "10\t5\t0.4\t0\t5\n",
      "11\t10\t0.1\t0\t10\n",
      "13\t7\t0.0\t0\t7\n",
      "14\t12\t0.0\t0\t12\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t22\t0.0\t0\t22\n",
      "17\t131\t0.0\t0\t131\n",
      "18\t63635\t0.0\t0\t63635\n",
      "19\t73144\t0.0\t0\t73144\n",
      "20\t74590\t0.0\t0\t74590\n",
      "21\t66828\t0.0\t0\t66828\n",
      "22\t68473\t0.0\t0\t68473\n",
      "23\t58397\t0.0\t0\t58397\n",
      "24\t48\t0.0\t0\t48\n",
      "25\t1\t0.0\t0\t1\n",
      "26\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "41\t1\t0.0\t0\t1\n",
      "43\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 382754 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t3\t0.0\t0\t3\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t6\t0.0\t0\t6\n",
      "19\t18\t0.0\t0\t18\n",
      "20\t121\t0.0\t0\t121\n",
      "21\t71493\t0.0\t0\t71493\n",
      "22\t68868\t0.0\t0\t68868\n",
      "23\t63265\t0.0\t0\t63265\n",
      "24\t65274\t0.0\t0\t65274\n",
      "25\t58104\t0.0\t0\t58104\n",
      "26\t55515\t0.0\t0\t55515\n",
      "27\t58\t0.0\t0\t58\n",
      "28\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "32\t2\t0.0\t0\t2\n",
      "45\t1\t0.0\t0\t1\n",
      "51\t1\t0.0\t0\t1\n",
      "62\t1\t0.0\t0\t1\n",
      "63\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121716_86_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121716_87_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121716_86_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121716_87_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121716_86_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121716_87_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121716_86_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121716_87_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 2.23 s (10 µs/read; 5.95 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            220,699\n",
      "  Read 1 with adapter:                 214,683 (97.3%)\n",
      "  Read 2 with adapter:                 202,229 (91.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           23,902 (10.8%)\n",
      "Pairs written (passing filters):       196,797 (89.2%)\n",
      "\n",
      "Total basepairs processed:   112,556,490 bp\n",
      "  Read 1:    67,313,195 bp\n",
      "  Read 2:    45,243,295 bp\n",
      "Total written (filtered):     91,746,847 bp (81.5%)\n",
      "  Read 1:    55,999,931 bp\n",
      "  Read 2:    35,746,916 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 214683 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.8\t0\t1\n",
      "10\t2\t0.2\t0\t2\n",
      "11\t4\t0.1\t0\t4\n",
      "12\t4\t0.0\t0\t4\n",
      "14\t11\t0.0\t0\t11\n",
      "15\t11\t0.0\t0\t11\n",
      "16\t12\t0.0\t0\t12\n",
      "17\t57\t0.0\t0\t57\n",
      "18\t33644\t0.0\t0\t33644\n",
      "19\t38542\t0.0\t0\t38542\n",
      "20\t39428\t0.0\t0\t39428\n",
      "21\t35699\t0.0\t0\t35699\n",
      "22\t36513\t0.0\t0\t36513\n",
      "23\t30727\t0.0\t0\t30727\n",
      "24\t26\t0.0\t0\t26\n",
      "25\t1\t0.0\t0\t1\n",
      "32\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 202229 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.2\t0\t2\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t4\t0.0\t0\t4\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t60\t0.0\t0\t60\n",
      "21\t37643\t0.0\t0\t37643\n",
      "22\t36409\t0.0\t0\t36409\n",
      "23\t33568\t0.0\t0\t33568\n",
      "24\t34369\t0.0\t0\t34369\n",
      "25\t30608\t0.0\t0\t30608\n",
      "26\t29509\t0.0\t0\t29509\n",
      "27\t34\t0.0\t0\t34\n",
      "32\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121717_88_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121717_89_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121717_88_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121717_89_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121717_88_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121717_89_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121717_88_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121717_89_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.23 s (11 µs/read; 5.48 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            112,481\n",
      "  Read 1 with adapter:                 109,357 (97.2%)\n",
      "  Read 2 with adapter:                 102,843 (91.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,454 (11.1%)\n",
      "Pairs written (passing filters):       100,027 (88.9%)\n",
      "\n",
      "Total basepairs processed:    57,365,310 bp\n",
      "  Read 1:    34,306,705 bp\n",
      "  Read 2:    23,058,605 bp\n",
      "Total written (filtered):     46,631,078 bp (81.3%)\n",
      "  Read 1:    28,461,891 bp\n",
      "  Read 2:    18,169,187 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 109357 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.7\t0\t1\n",
      "9\t1\t0.4\t0\t1\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t24\t0.0\t0\t24\n",
      "18\t17241\t0.0\t0\t17241\n",
      "19\t19396\t0.0\t0\t19396\n",
      "20\t19921\t0.0\t0\t19921\n",
      "21\t17857\t0.0\t0\t17857\n",
      "22\t18707\t0.0\t0\t18707\n",
      "23\t16186\t0.0\t0\t16186\n",
      "24\t7\t0.0\t0\t7\n",
      "26\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 102843 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t6\t0.0\t0\t6\n",
      "20\t45\t0.0\t0\t45\n",
      "21\t19092\t0.0\t0\t19092\n",
      "22\t18461\t0.0\t0\t18461\n",
      "23\t17158\t0.0\t0\t17158\n",
      "24\t17425\t0.0\t0\t17425\n",
      "25\t15688\t0.0\t0\t15688\n",
      "26\t14948\t0.0\t0\t14948\n",
      "27\t12\t0.0\t0\t12\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121718_90_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121718_91_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121718_90_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121718_91_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121718_90_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121718_91_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121718_90_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121718_91_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.26 s (11 µs/read; 5.36 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            112,618\n",
      "  Read 1 with adapter:                 109,503 (97.2%)\n",
      "  Read 2 with adapter:                 102,994 (91.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,401 (11.0%)\n",
      "Pairs written (passing filters):       100,217 (89.0%)\n",
      "\n",
      "Total basepairs processed:    57,435,180 bp\n",
      "  Read 1:    34,348,490 bp\n",
      "  Read 2:    23,086,690 bp\n",
      "Total written (filtered):     46,718,895 bp (81.3%)\n",
      "  Read 1:    28,515,186 bp\n",
      "  Read 2:    18,203,709 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 109503 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t35\t0.0\t0\t35\n",
      "18\t16903\t0.0\t0\t16903\n",
      "19\t19486\t0.0\t0\t19486\n",
      "20\t20143\t0.0\t0\t20143\n",
      "21\t18144\t0.0\t0\t18144\n",
      "22\t18690\t0.0\t0\t18690\n",
      "23\t16079\t0.0\t0\t16079\n",
      "24\t10\t0.0\t0\t10\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 102994 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.0\t0\t2\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t32\t0.0\t0\t32\n",
      "21\t19334\t0.0\t0\t19334\n",
      "22\t18458\t0.0\t0\t18458\n",
      "23\t17010\t0.0\t0\t17010\n",
      "24\t17502\t0.0\t0\t17502\n",
      "25\t15537\t0.0\t0\t15537\n",
      "26\t15092\t0.0\t0\t15092\n",
      "27\t10\t0.0\t0\t10\n",
      "38\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121719_92_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121719_93_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121719_92_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121719_93_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121719_92_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121719_93_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121719_92_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121719_93_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.50 s (11 µs/read; 5.50 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            137,450\n",
      "  Read 1 with adapter:                 133,616 (97.2%)\n",
      "  Read 2 with adapter:                 125,984 (91.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           14,914 (10.9%)\n",
      "Pairs written (passing filters):       122,536 (89.1%)\n",
      "\n",
      "Total basepairs processed:    70,099,500 bp\n",
      "  Read 1:    41,922,250 bp\n",
      "  Read 2:    28,177,250 bp\n",
      "Total written (filtered):     57,126,283 bp (81.5%)\n",
      "  Read 1:    34,867,786 bp\n",
      "  Read 2:    22,258,497 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 133616 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t2.1\t0\t2\n",
      "9\t1\t0.5\t0\t1\n",
      "10\t2\t0.1\t0\t2\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t11\t0.0\t0\t11\n",
      "17\t34\t0.0\t0\t34\n",
      "18\t21042\t0.0\t0\t21042\n",
      "19\t23916\t0.0\t0\t23916\n",
      "20\t24408\t0.0\t0\t24408\n",
      "21\t21855\t0.0\t0\t21855\n",
      "22\t22715\t0.0\t0\t22715\n",
      "23\t19602\t0.0\t0\t19602\n",
      "24\t17\t0.0\t0\t17\n",
      "28\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 125984 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.5\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t42\t0.0\t0\t42\n",
      "21\t23541\t0.0\t0\t23541\n",
      "22\t22951\t0.0\t0\t22951\n",
      "23\t20821\t0.0\t0\t20821\n",
      "24\t21171\t0.0\t0\t21171\n",
      "25\t19072\t0.0\t0\t19072\n",
      "26\t18364\t0.0\t0\t18364\n",
      "27\t6\t0.0\t0\t6\n",
      "29\t1\t0.0\t0\t1\n",
      "60\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121720_94_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121720_95_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121720_94_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121720_95_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121720_94_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121720_95_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121720_94_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121720_95_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 3.73 s (10 µs/read; 5.98 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            371,795\n",
      "  Read 1 with adapter:                 361,865 (97.3%)\n",
      "  Read 2 with adapter:                 340,718 (91.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           40,027 (10.8%)\n",
      "Pairs written (passing filters):       331,768 (89.2%)\n",
      "\n",
      "Total basepairs processed:   189,615,450 bp\n",
      "  Read 1:   113,397,475 bp\n",
      "  Read 2:    76,217,975 bp\n",
      "Total written (filtered):    154,669,634 bp (81.6%)\n",
      "  Read 1:    94,405,636 bp\n",
      "  Read 2:    60,263,998 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 361865 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.7\t0\t1\n",
      "9\t5\t1.4\t0\t5\n",
      "10\t5\t0.4\t0\t5\n",
      "11\t5\t0.1\t0\t5\n",
      "12\t7\t0.0\t0\t7\n",
      "13\t4\t0.0\t0\t4\n",
      "14\t16\t0.0\t0\t16\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t17\t0.0\t0\t17\n",
      "17\t104\t0.0\t0\t104\n",
      "18\t56376\t0.0\t0\t56376\n",
      "19\t65299\t0.0\t0\t65299\n",
      "20\t66080\t0.0\t0\t66080\n",
      "21\t60115\t0.0\t0\t60115\n",
      "22\t61294\t0.0\t0\t61294\n",
      "23\t52470\t0.0\t0\t52470\n",
      "24\t51\t0.0\t0\t51\n",
      "30\t1\t0.0\t0\t1\n",
      "32\t1\t0.0\t0\t1\n",
      "40\t2\t0.0\t0\t2\n",
      "49\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 340718 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t7\t0.0\t0\t7\n",
      "19\t11\t0.0\t0\t11\n",
      "20\t103\t0.0\t0\t103\n",
      "21\t63764\t0.0\t0\t63764\n",
      "22\t61383\t0.0\t0\t61383\n",
      "23\t56302\t0.0\t0\t56302\n",
      "24\t57686\t0.0\t0\t57686\n",
      "25\t51653\t0.0\t0\t51653\n",
      "26\t49749\t0.0\t0\t49749\n",
      "27\t41\t0.0\t0\t41\n",
      "39\t1\t0.0\t0\t1\n",
      "60\t1\t0.0\t0\t1\n",
      "61\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121721_96_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121721_97_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121721_96_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121721_97_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121721_96_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121721_97_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121721_96_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121721_97_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.62 s (10 µs/read; 5.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            345,448\n",
      "  Read 1 with adapter:                 336,214 (97.3%)\n",
      "  Read 2 with adapter:                 317,650 (92.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           36,167 (10.5%)\n",
      "Pairs written (passing filters):       309,281 (89.5%)\n",
      "\n",
      "Total basepairs processed:   176,178,480 bp\n",
      "  Read 1:   105,361,640 bp\n",
      "  Read 2:    70,816,840 bp\n",
      "Total written (filtered):    144,189,236 bp (81.8%)\n",
      "  Read 1:    88,006,094 bp\n",
      "  Read 2:    56,183,142 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 336214 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.3\t0\t1\n",
      "9\t3\t1.3\t0\t3\n",
      "10\t2\t0.3\t0\t2\n",
      "11\t6\t0.1\t0\t6\n",
      "12\t8\t0.0\t0\t8\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t14\t0.0\t0\t14\n",
      "15\t16\t0.0\t0\t16\n",
      "16\t16\t0.0\t0\t16\n",
      "17\t84\t0.0\t0\t84\n",
      "18\t52454\t0.0\t0\t52454\n",
      "19\t60710\t0.0\t0\t60710\n",
      "20\t61341\t0.0\t0\t61341\n",
      "21\t56018\t0.0\t0\t56018\n",
      "22\t57176\t0.0\t0\t57176\n",
      "23\t48327\t0.0\t0\t48327\n",
      "24\t29\t0.0\t0\t29\n",
      "26\t1\t0.0\t0\t1\n",
      "31\t2\t0.0\t0\t2\n",
      "33\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "51\t2\t0.0\t0\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 317650 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.3\t0\t1\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t6\t0.0\t0\t6\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t12\t0.0\t0\t12\n",
      "20\t113\t0.0\t0\t113\n",
      "21\t60040\t0.0\t0\t60040\n",
      "22\t57552\t0.0\t0\t57552\n",
      "23\t52276\t0.0\t0\t52276\n",
      "24\t53977\t0.0\t0\t53977\n",
      "25\t47798\t0.0\t0\t47798\n",
      "26\t45819\t0.0\t0\t45819\n",
      "27\t39\t0.0\t0\t39\n",
      "28\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121722_98_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121722_99_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121722_98_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121722_99_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121722_98_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121722_99_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121722_98_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121722_99_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.25 s (10 µs/read; 5.75 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            119,832\n",
      "  Read 1 with adapter:                 116,476 (97.2%)\n",
      "  Read 2 with adapter:                 110,758 (92.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,111 (10.1%)\n",
      "Pairs written (passing filters):       107,721 (89.9%)\n",
      "\n",
      "Total basepairs processed:    61,114,320 bp\n",
      "  Read 1:    36,548,760 bp\n",
      "  Read 2:    24,565,560 bp\n",
      "Total written (filtered):     50,218,813 bp (82.2%)\n",
      "  Read 1:    30,651,862 bp\n",
      "  Read 2:    19,566,951 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 116476 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t2\t0.5\t0\t2\n",
      "11\t3\t0.0\t0\t3\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t26\t0.0\t0\t26\n",
      "18\t18120\t0.0\t0\t18120\n",
      "19\t21100\t0.0\t0\t21100\n",
      "20\t21249\t0.0\t0\t21249\n",
      "21\t19183\t0.0\t0\t19183\n",
      "22\t19892\t0.0\t0\t19892\n",
      "23\t16869\t0.0\t0\t16869\n",
      "24\t12\t0.0\t0\t12\n",
      "42\t1\t0.0\t0\t1\n",
      "46\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 110758 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t4\t0.0\t0\t4\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t6\t0.0\t0\t6\n",
      "20\t42\t0.0\t0\t42\n",
      "21\t20941\t0.0\t0\t20941\n",
      "22\t19733\t0.0\t0\t19733\n",
      "23\t18154\t0.0\t0\t18154\n",
      "24\t18746\t0.0\t0\t18746\n",
      "25\t17012\t0.0\t0\t17012\n",
      "26\t16100\t0.0\t0\t16100\n",
      "27\t16\t0.0\t0\t16\n",
      "28\t1\t0.0\t0\t1\n",
      "66\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121723_100_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121723_101_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121723_100_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121723_101_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121723_100_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121723_101_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121723_100_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121723_101_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 2.58 s (10 µs/read; 5.91 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            254,695\n",
      "  Read 1 with adapter:                 247,556 (97.2%)\n",
      "  Read 2 with adapter:                 236,247 (92.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           25,012 (9.8%)\n",
      "Pairs written (passing filters):       229,683 (90.2%)\n",
      "\n",
      "Total basepairs processed:   129,894,450 bp\n",
      "  Read 1:    77,681,975 bp\n",
      "  Read 2:    52,212,475 bp\n",
      "Total written (filtered):    107,079,944 bp (82.4%)\n",
      "  Read 1:    65,356,890 bp\n",
      "  Read 2:    41,723,054 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 247556 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t3.9\t0\t1\n",
      "9\t2\t1.0\t0\t2\n",
      "10\t1\t0.2\t0\t1\n",
      "11\t6\t0.1\t0\t6\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t11\t0.0\t0\t11\n",
      "15\t11\t0.0\t0\t11\n",
      "16\t8\t0.0\t0\t8\n",
      "17\t69\t0.0\t0\t69\n",
      "18\t38539\t0.0\t0\t38539\n",
      "19\t44678\t0.0\t0\t44678\n",
      "20\t45617\t0.0\t0\t45617\n",
      "21\t40666\t0.0\t0\t40666\n",
      "22\t42090\t0.0\t0\t42090\n",
      "23\t35820\t0.0\t0\t35820\n",
      "24\t28\t0.0\t0\t28\n",
      "28\t2\t0.0\t0\t2\n",
      "29\t1\t0.0\t0\t1\n",
      "37\t2\t0.0\t0\t2\n",
      "44\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 236247 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t1.0\t0\t1\n",
      "10\t1\t0.2\t0\t1\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t4\t0.0\t0\t4\n",
      "18\t5\t0.0\t0\t5\n",
      "19\t11\t0.0\t0\t11\n",
      "20\t65\t0.0\t0\t65\n",
      "21\t44691\t0.0\t0\t44691\n",
      "22\t42849\t0.0\t0\t42849\n",
      "23\t38532\t0.0\t0\t38532\n",
      "24\t40281\t0.0\t0\t40281\n",
      "25\t35773\t0.0\t0\t35773\n",
      "26\t33982\t0.0\t0\t33982\n",
      "27\t31\t0.0\t0\t31\n",
      "29\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "49\t1\t0.0\t0\t1\n",
      "58\t2\t0.0\t0\t2\n",
      "63\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121724_102_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121724_103_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121724_102_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121724_103_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121724_102_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121724_103_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121724_102_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121724_103_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.29 s (10 µs/read; 6.09 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            334,551\n",
      "  Read 1 with adapter:                 325,355 (97.3%)\n",
      "  Read 2 with adapter:                 310,233 (92.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           32,704 (9.8%)\n",
      "Pairs written (passing filters):       301,847 (90.2%)\n",
      "\n",
      "Total basepairs processed:   170,621,010 bp\n",
      "  Read 1:   102,038,055 bp\n",
      "  Read 2:    68,582,955 bp\n",
      "Total written (filtered):    140,719,082 bp (82.5%)\n",
      "  Read 1:    85,890,636 bp\n",
      "  Read 2:    54,828,446 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 325355 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t8\t0.3\t0\t8\n",
      "11\t7\t0.1\t0\t7\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t13\t0.0\t0\t13\n",
      "15\t14\t0.0\t0\t14\n",
      "16\t7\t0.0\t0\t7\n",
      "17\t85\t0.0\t0\t85\n",
      "18\t50577\t0.0\t0\t50577\n",
      "19\t58952\t0.0\t0\t58952\n",
      "20\t59703\t0.0\t0\t59703\n",
      "21\t53628\t0.0\t0\t53628\n",
      "22\t55483\t0.0\t0\t55483\n",
      "23\t46822\t0.0\t0\t46822\n",
      "24\t39\t0.0\t0\t39\n",
      "25\t1\t0.0\t0\t1\n",
      "26\t2\t0.0\t0\t2\n",
      "27\t1\t0.0\t0\t1\n",
      "38\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "41\t1\t0.0\t0\t1\n",
      "44\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "54\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 310233 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.3\t0\t1\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t6\t0.0\t0\t6\n",
      "18\t5\t0.0\t0\t5\n",
      "19\t8\t0.0\t0\t8\n",
      "20\t99\t0.0\t0\t99\n",
      "21\t58050\t0.0\t0\t58050\n",
      "22\t55692\t0.0\t0\t55692\n",
      "23\t51370\t0.0\t0\t51370\n",
      "24\t52396\t0.0\t0\t52396\n",
      "25\t47122\t0.0\t0\t47122\n",
      "26\t45425\t0.0\t0\t45425\n",
      "27\t38\t0.0\t0\t38\n",
      "43\t1\t0.0\t0\t1\n",
      "46\t1\t0.0\t0\t1\n",
      "53\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121725_104_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121725_105_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121725_104_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121725_105_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121725_104_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121725_105_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121725_104_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121725_105_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 4.48 s (10 µs/read; 6.29 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            470,208\n",
      "  Read 1 with adapter:                 457,505 (97.3%)\n",
      "  Read 2 with adapter:                 434,779 (92.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           46,924 (10.0%)\n",
      "Pairs written (passing filters):       423,284 (90.0%)\n",
      "\n",
      "Total basepairs processed:   239,806,080 bp\n",
      "  Read 1:   143,413,440 bp\n",
      "  Read 2:    96,392,640 bp\n",
      "Total written (filtered):    197,335,315 bp (82.3%)\n",
      "  Read 1:   120,445,899 bp\n",
      "  Read 2:    76,889,416 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 457505 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t8\t7.2\t0\t8\n",
      "9\t5\t1.8\t0\t5\n",
      "10\t3\t0.4\t0\t3\n",
      "11\t8\t0.1\t0\t8\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t6\t0.0\t0\t6\n",
      "14\t21\t0.0\t0\t21\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t18\t0.0\t0\t18\n",
      "17\t122\t0.0\t0\t122\n",
      "18\t71349\t0.0\t0\t71349\n",
      "19\t82451\t0.0\t0\t82451\n",
      "20\t84064\t0.0\t0\t84064\n",
      "21\t75495\t0.0\t0\t75495\n",
      "22\t77900\t0.0\t0\t77900\n",
      "23\t65977\t0.0\t0\t65977\n",
      "24\t56\t0.0\t0\t56\n",
      "25\t1\t0.0\t0\t1\n",
      "27\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "32\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "38\t1\t0.0\t0\t1\n",
      "47\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 434779 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t1.8\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t13\t0.0\t0\t13\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t5\t0.0\t0\t5\n",
      "18\t8\t0.0\t0\t8\n",
      "19\t8\t0.0\t0\t8\n",
      "20\t135\t0.0\t0\t135\n",
      "21\t81561\t0.0\t0\t81561\n",
      "22\t78642\t0.0\t0\t78642\n",
      "23\t71863\t0.0\t0\t71863\n",
      "24\t73606\t0.0\t0\t73606\n",
      "25\t66075\t0.0\t0\t66075\n",
      "26\t62796\t0.0\t0\t62796\n",
      "27\t53\t0.0\t0\t53\n",
      "32\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "62\t1\t0.0\t0\t1\n",
      "63\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121726_106_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121726_107_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121726_106_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121726_107_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121726_106_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121726_107_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121726_106_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121726_107_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.76 s (10 µs/read; 5.72 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            168,100\n",
      "  Read 1 with adapter:                 163,558 (97.3%)\n",
      "  Read 2 with adapter:                 154,553 (91.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           17,642 (10.5%)\n",
      "Pairs written (passing filters):       150,458 (89.5%)\n",
      "\n",
      "Total basepairs processed:    85,731,000 bp\n",
      "  Read 1:    51,270,500 bp\n",
      "  Read 2:    34,460,500 bp\n",
      "Total written (filtered):     70,143,303 bp (81.8%)\n",
      "  Read 1:    42,813,447 bp\n",
      "  Read 2:    27,329,856 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 163558 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.6\t0\t1\n",
      "9\t1\t0.6\t0\t1\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "14\t6\t0.0\t0\t6\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t7\t0.0\t0\t7\n",
      "17\t34\t0.0\t0\t34\n",
      "18\t25257\t0.0\t0\t25257\n",
      "19\t29910\t0.0\t0\t29910\n",
      "20\t29975\t0.0\t0\t29975\n",
      "21\t27195\t0.0\t0\t27195\n",
      "22\t27709\t0.0\t0\t27709\n",
      "23\t23432\t0.0\t0\t23432\n",
      "24\t17\t0.0\t0\t17\n",
      "32\t1\t0.0\t0\t1\n",
      "44\t2\t0.0\t0\t2\n",
      "51\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 154553 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t4\t0.0\t0\t4\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t48\t0.0\t0\t48\n",
      "21\t28551\t0.0\t0\t28551\n",
      "22\t28058\t0.0\t0\t28058\n",
      "23\t25744\t0.0\t0\t25744\n",
      "24\t26380\t0.0\t0\t26380\n",
      "25\t23299\t0.0\t0\t23299\n",
      "26\t22433\t0.0\t0\t22433\n",
      "27\t15\t0.0\t0\t15\n",
      "32\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121727_108_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121727_109_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121727_108_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121727_109_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121727_108_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121727_109_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121727_108_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121727_109_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.07 s (11 µs/read; 5.35 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            184,058\n",
      "  Read 1 with adapter:                 179,051 (97.3%)\n",
      "  Read 2 with adapter:                 169,240 (91.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           19,329 (10.5%)\n",
      "Pairs written (passing filters):       164,729 (89.5%)\n",
      "\n",
      "Total basepairs processed:    93,869,580 bp\n",
      "  Read 1:    56,137,690 bp\n",
      "  Read 2:    37,731,890 bp\n",
      "Total written (filtered):     76,794,752 bp (81.8%)\n",
      "  Read 1:    46,873,160 bp\n",
      "  Read 2:    29,921,592 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 179051 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t2.8\t0\t3\n",
      "9\t4\t0.7\t0\t4\n",
      "11\t4\t0.0\t0\t4\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t9\t0.0\t0\t9\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t13\t0.0\t0\t13\n",
      "17\t41\t0.0\t0\t41\n",
      "18\t27921\t0.0\t0\t27921\n",
      "19\t32083\t0.0\t0\t32083\n",
      "20\t32659\t0.0\t0\t32659\n",
      "21\t29759\t0.0\t0\t29759\n",
      "22\t30582\t0.0\t0\t30582\n",
      "23\t25950\t0.0\t0\t25950\n",
      "24\t13\t0.0\t0\t13\n",
      "28\t1\t0.0\t0\t1\n",
      "33\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 169240 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.2\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t1\t0.0\t0\t1\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t59\t0.0\t0\t59\n",
      "21\t31638\t0.0\t0\t31638\n",
      "22\t30527\t0.0\t0\t30527\n",
      "23\t27622\t0.0\t0\t27622\n",
      "24\t28810\t0.0\t0\t28810\n",
      "25\t25772\t0.0\t0\t25772\n",
      "26\t24773\t0.0\t0\t24773\n",
      "27\t24\t0.0\t0\t24\n",
      "29\t1\t0.0\t0\t1\n",
      "63\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121728_110_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121728_111_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121728_110_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121728_111_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121728_110_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121728_111_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121728_110_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121728_111_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.30 s (11 µs/read; 5.58 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            120,731\n",
      "  Read 1 with adapter:                 117,343 (97.2%)\n",
      "  Read 2 with adapter:                 111,041 (92.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,746 (10.6%)\n",
      "Pairs written (passing filters):       107,985 (89.4%)\n",
      "\n",
      "Total basepairs processed:    61,572,810 bp\n",
      "  Read 1:    36,822,955 bp\n",
      "  Read 2:    24,749,855 bp\n",
      "Total written (filtered):     50,340,949 bp (81.8%)\n",
      "  Read 1:    30,726,397 bp\n",
      "  Read 2:    19,614,552 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 117343 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.1\t0\t2\n",
      "11\t5\t0.0\t0\t5\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t8\t0.0\t0\t8\n",
      "17\t33\t0.0\t0\t33\n",
      "18\t18146\t0.0\t0\t18146\n",
      "19\t21182\t0.0\t0\t21182\n",
      "20\t21424\t0.0\t0\t21424\n",
      "21\t19366\t0.0\t0\t19366\n",
      "22\t19893\t0.0\t0\t19893\n",
      "23\t17264\t0.0\t0\t17264\n",
      "24\t12\t0.0\t0\t12\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 111041 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t2\t0.0\t0\t2\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t2\t0.0\t0\t2\n",
      "20\t36\t0.0\t0\t36\n",
      "21\t20831\t0.0\t0\t20831\n",
      "22\t19718\t0.0\t0\t19718\n",
      "23\t18411\t0.0\t0\t18411\n",
      "24\t18948\t0.0\t0\t18948\n",
      "25\t16922\t0.0\t0\t16922\n",
      "26\t16148\t0.0\t0\t16148\n",
      "27\t17\t0.0\t0\t17\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121729_112_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121729_113_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121729_112_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121729_113_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121729_112_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121729_113_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121729_112_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121729_113_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.44 s (16 µs/read; 3.64 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             26,646\n",
      "  Read 1 with adapter:                  25,851 (97.0%)\n",
      "  Read 2 with adapter:                  23,760 (89.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            3,580 (13.4%)\n",
      "Pairs written (passing filters):        23,066 (86.6%)\n",
      "\n",
      "Total basepairs processed:    13,589,460 bp\n",
      "  Read 1:     8,127,030 bp\n",
      "  Read 2:     5,462,430 bp\n",
      "Total written (filtered):     10,754,283 bp (79.1%)\n",
      "  Read 1:     6,563,927 bp\n",
      "  Read 2:     4,190,356 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 25851 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "17\t11\t0.0\t0\t11\n",
      "18\t4091\t0.0\t0\t4091\n",
      "19\t4672\t0.0\t0\t4672\n",
      "20\t4659\t0.0\t0\t4659\n",
      "21\t4228\t0.0\t0\t4228\n",
      "22\t4465\t0.0\t0\t4465\n",
      "23\t3719\t0.0\t0\t3719\n",
      "24\t3\t0.0\t0\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 23760 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t1\t0.0\t0\t1\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t2\t0.0\t0\t2\n",
      "21\t4532\t0.0\t0\t4532\n",
      "22\t4335\t0.0\t0\t4335\n",
      "23\t3933\t0.0\t0\t3933\n",
      "24\t3993\t0.0\t0\t3993\n",
      "25\t3564\t0.0\t0\t3564\n",
      "26\t3397\t0.0\t0\t3397\n",
      "27\t2\t0.0\t0\t2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121730_114_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121730_115_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121730_114_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121730_115_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121730_114_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121730_115_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121730_114_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121730_115_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.29 s (11 µs/read; 5.26 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            113,245\n",
      "  Read 1 with adapter:                 110,092 (97.2%)\n",
      "  Read 2 with adapter:                 103,978 (91.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,106 (10.7%)\n",
      "Pairs written (passing filters):       101,139 (89.3%)\n",
      "\n",
      "Total basepairs processed:    57,754,950 bp\n",
      "  Read 1:    34,539,725 bp\n",
      "  Read 2:    23,215,225 bp\n",
      "Total written (filtered):     47,149,152 bp (81.6%)\n",
      "  Read 1:    28,778,393 bp\n",
      "  Read 2:    18,370,759 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 110092 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.7\t0\t1\n",
      "9\t2\t0.4\t0\t2\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t28\t0.0\t0\t28\n",
      "18\t17209\t0.0\t0\t17209\n",
      "19\t19695\t0.0\t0\t19695\n",
      "20\t20031\t0.0\t0\t20031\n",
      "21\t18433\t0.0\t0\t18433\n",
      "22\t18652\t0.0\t0\t18652\n",
      "23\t16015\t0.0\t0\t16015\n",
      "24\t11\t0.0\t0\t11\n",
      "41\t1\t0.0\t0\t1\n",
      "42\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 103978 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t28\t0.0\t0\t28\n",
      "21\t19427\t0.0\t0\t19427\n",
      "22\t18618\t0.0\t0\t18618\n",
      "23\t17055\t0.0\t0\t17055\n",
      "24\t17695\t0.0\t0\t17695\n",
      "25\t16002\t0.0\t0\t16002\n",
      "26\t15130\t0.0\t0\t15130\n",
      "27\t15\t0.0\t0\t15\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121731_116_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121731_117_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121731_116_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121731_117_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121731_116_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121731_117_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121731_116_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121731_117_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.85 s (12 µs/read; 5.05 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             71,358\n",
      "  Read 1 with adapter:                  69,352 (97.2%)\n",
      "  Read 2 with adapter:                  65,469 (91.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            7,697 (10.8%)\n",
      "Pairs written (passing filters):        63,661 (89.2%)\n",
      "\n",
      "Total basepairs processed:    36,392,580 bp\n",
      "  Read 1:    21,764,190 bp\n",
      "  Read 2:    14,628,390 bp\n",
      "Total written (filtered):     29,678,541 bp (81.6%)\n",
      "  Read 1:    18,114,837 bp\n",
      "  Read 2:    11,563,704 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 69352 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t1.1\t0\t2\n",
      "9\t1\t0.3\t0\t1\n",
      "10\t2\t0.1\t0\t2\n",
      "11\t2\t0.0\t0\t2\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t21\t0.0\t0\t21\n",
      "18\t10634\t0.0\t0\t10634\n",
      "19\t12525\t0.0\t0\t12525\n",
      "20\t12791\t0.0\t0\t12791\n",
      "21\t11476\t0.0\t0\t11476\n",
      "22\t11970\t0.0\t0\t11970\n",
      "23\t9910\t0.0\t0\t9910\n",
      "24\t7\t0.0\t0\t7\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 65469 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t2\t0.0\t0\t2\n",
      "20\t23\t0.0\t0\t23\n",
      "21\t12299\t0.0\t0\t12299\n",
      "22\t11680\t0.0\t0\t11680\n",
      "23\t10918\t0.0\t0\t10918\n",
      "24\t11132\t0.0\t0\t11132\n",
      "25\t9830\t0.0\t0\t9830\n",
      "26\t9572\t0.0\t0\t9572\n",
      "27\t6\t0.0\t0\t6\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121732_118_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121732_119_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121732_118_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121732_119_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121732_118_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121732_119_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121732_118_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121732_119_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 2.01 s (10 µs/read; 6.17 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            206,381\n",
      "  Read 1 with adapter:                 200,908 (97.3%)\n",
      "  Read 2 with adapter:                 191,308 (92.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           20,020 (9.7%)\n",
      "Pairs written (passing filters):       186,361 (90.3%)\n",
      "\n",
      "Total basepairs processed:   105,254,310 bp\n",
      "  Read 1:    62,946,205 bp\n",
      "  Read 2:    42,308,105 bp\n",
      "Total written (filtered):     86,881,900 bp (82.5%)\n",
      "  Read 1:    53,029,209 bp\n",
      "  Read 2:    33,852,691 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 200908 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t2\t0.8\t0\t2\n",
      "10\t6\t0.2\t0\t6\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t4\t0.0\t0\t4\n",
      "14\t7\t0.0\t0\t7\n",
      "15\t6\t0.0\t0\t6\n",
      "16\t13\t0.0\t0\t13\n",
      "17\t58\t0.0\t0\t58\n",
      "18\t31320\t0.0\t0\t31320\n",
      "19\t36399\t0.0\t0\t36399\n",
      "20\t36903\t0.0\t0\t36903\n",
      "21\t32934\t0.0\t0\t32934\n",
      "22\t33946\t0.0\t0\t33946\n",
      "23\t29277\t0.0\t0\t29277\n",
      "24\t29\t0.0\t0\t29\n",
      "34\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 191308 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t5\t0.0\t0\t5\n",
      "19\t11\t0.0\t0\t11\n",
      "20\t64\t0.0\t0\t64\n",
      "21\t36053\t0.0\t0\t36053\n",
      "22\t34594\t0.0\t0\t34594\n",
      "23\t31397\t0.0\t0\t31397\n",
      "24\t32441\t0.0\t0\t32441\n",
      "25\t28893\t0.0\t0\t28893\n",
      "26\t27817\t0.0\t0\t27817\n",
      "27\t21\t0.0\t0\t21\n",
      "33\t1\t0.0\t0\t1\n",
      "42\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121733_120_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121733_121_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121733_120_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121733_121_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121733_120_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121733_121_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121733_120_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121733_121_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.66 s (11 µs/read; 5.45 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            151,058\n",
      "  Read 1 with adapter:                 146,859 (97.2%)\n",
      "  Read 2 with adapter:                 139,367 (92.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           15,510 (10.3%)\n",
      "Pairs written (passing filters):       135,548 (89.7%)\n",
      "\n",
      "Total basepairs processed:    77,039,580 bp\n",
      "  Read 1:    46,072,690 bp\n",
      "  Read 2:    30,966,890 bp\n",
      "Total written (filtered):     63,193,980 bp (82.0%)\n",
      "  Read 1:    38,571,401 bp\n",
      "  Read 2:    24,622,579 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 146859 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t3\t0.6\t0\t3\n",
      "10\t2\t0.1\t0\t2\n",
      "11\t3\t0.0\t0\t3\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t39\t0.0\t0\t39\n",
      "18\t22910\t0.0\t0\t22910\n",
      "19\t26757\t0.0\t0\t26757\n",
      "20\t26867\t0.0\t0\t26867\n",
      "21\t24458\t0.0\t0\t24458\n",
      "22\t24909\t0.0\t0\t24909\n",
      "23\t20871\t0.0\t0\t20871\n",
      "24\t20\t0.0\t0\t20\n",
      "25\t1\t0.0\t0\t1\n",
      "35\t1\t0.0\t0\t1\n",
      "48\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 139367 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "15\t3\t0.0\t0\t3\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t41\t0.0\t0\t41\n",
      "21\t26308\t0.0\t0\t26308\n",
      "22\t25062\t0.0\t0\t25062\n",
      "23\t23107\t0.0\t0\t23107\n",
      "24\t23589\t0.0\t0\t23589\n",
      "25\t21010\t0.0\t0\t21010\n",
      "26\t20221\t0.0\t0\t20221\n",
      "27\t15\t0.0\t0\t15\n",
      "35\t1\t0.0\t0\t1\n",
      "60\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121734_122_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121734_123_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121734_122_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121734_123_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121734_122_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210421121734_123_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121734_122_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210421121734_123_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.80 s (13 µs/read; 4.69 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             62,412\n",
      "  Read 1 with adapter:                  60,694 (97.2%)\n",
      "  Read 2 with adapter:                  57,561 (92.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            6,383 (10.2%)\n",
      "Pairs written (passing filters):        56,029 (89.8%)\n",
      "\n",
      "Total basepairs processed:    31,830,120 bp\n",
      "  Read 1:    19,035,660 bp\n",
      "  Read 2:    12,794,460 bp\n",
      "Total written (filtered):     26,120,167 bp (82.1%)\n",
      "  Read 1:    15,943,160 bp\n",
      "  Read 2:    10,177,007 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 60694 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.2\t0\t1\n",
      "11\t3\t0.0\t0\t3\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t6\t0.0\t0\t6\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t15\t0.0\t0\t15\n",
      "18\t9391\t0.0\t0\t9391\n",
      "19\t10907\t0.0\t0\t10907\n",
      "20\t11238\t0.0\t0\t11238\n",
      "21\t10160\t0.0\t0\t10160\n",
      "22\t10334\t0.0\t0\t10334\n",
      "23\t8627\t0.0\t0\t8627\n",
      "24\t6\t0.0\t0\t6\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 57561 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t20\t0.0\t0\t20\n",
      "21\t10681\t0.0\t0\t10681\n",
      "22\t10399\t0.0\t0\t10399\n",
      "23\t9439\t0.0\t0\t9439\n",
      "24\t9883\t0.0\t0\t9883\n",
      "25\t8746\t0.0\t0\t8746\n",
      "26\t8373\t0.0\t0\t8373\n",
      "27\t9\t0.0\t0\t9\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163906_124_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163906_125_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163906_124_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163906_125_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163906_124_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163906_125_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163906_124_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163906_125_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.69 s (12 µs/read; 4.81 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             55,327\n",
      "  Read 1 with adapter:                  53,825 (97.3%)\n",
      "  Read 2 with adapter:                  50,542 (91.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            6,147 (11.1%)\n",
      "Pairs written (passing filters):        49,180 (88.9%)\n",
      "\n",
      "Total basepairs processed:    28,216,770 bp\n",
      "  Read 1:    16,874,735 bp\n",
      "  Read 2:    11,342,035 bp\n",
      "Total written (filtered):     22,927,295 bp (81.3%)\n",
      "  Read 1:    13,993,375 bp\n",
      "  Read 2:     8,933,920 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 53825 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t1\t0.0\t0\t1\n",
      "17\t22\t0.0\t0\t22\n",
      "18\t8268\t0.0\t0\t8268\n",
      "19\t9653\t0.0\t0\t9653\n",
      "20\t9875\t0.0\t0\t9875\n",
      "21\t8916\t0.0\t0\t8916\n",
      "22\t9188\t0.0\t0\t9188\n",
      "23\t7885\t0.0\t0\t7885\n",
      "24\t10\t0.0\t0\t10\n",
      "29\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 50542 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.2\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t1\t0.0\t0\t1\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t15\t0.0\t0\t15\n",
      "21\t9546\t0.0\t0\t9546\n",
      "22\t9109\t0.0\t0\t9109\n",
      "23\t8384\t0.0\t0\t8384\n",
      "24\t8557\t0.0\t0\t8557\n",
      "25\t7650\t0.0\t0\t7650\n",
      "26\t7265\t0.0\t0\t7265\n",
      "27\t5\t0.0\t0\t5\n",
      "46\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163907_126_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163907_127_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163907_126_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163907_127_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163907_126_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163907_127_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163907_126_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163907_127_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.36 s (10 µs/read; 6.19 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            243,179\n",
      "  Read 1 with adapter:                 236,854 (97.4%)\n",
      "  Read 2 with adapter:                 223,595 (91.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           25,304 (10.4%)\n",
      "Pairs written (passing filters):       217,875 (89.6%)\n",
      "\n",
      "Total basepairs processed:   124,021,290 bp\n",
      "  Read 1:    74,169,595 bp\n",
      "  Read 2:    49,851,695 bp\n",
      "Total written (filtered):    101,573,655 bp (81.9%)\n",
      "  Read 1:    61,995,458 bp\n",
      "  Read 2:    39,578,197 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 236854 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t3.7\t0\t2\n",
      "9\t2\t0.9\t0\t2\n",
      "10\t1\t0.2\t0\t1\n",
      "11\t3\t0.1\t0\t3\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t14\t0.0\t0\t14\n",
      "15\t12\t0.0\t0\t12\n",
      "16\t9\t0.0\t0\t9\n",
      "17\t66\t0.0\t0\t66\n",
      "18\t36763\t0.0\t0\t36763\n",
      "19\t42630\t0.0\t0\t42630\n",
      "20\t43467\t0.0\t0\t43467\n",
      "21\t39245\t0.0\t0\t39245\n",
      "22\t40296\t0.0\t0\t40296\n",
      "23\t34314\t0.0\t0\t34314\n",
      "24\t22\t0.0\t0\t22\n",
      "26\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "42\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 223595 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.1\t0\t2\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t7\t0.0\t0\t7\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t79\t0.0\t0\t79\n",
      "21\t41812\t0.0\t0\t41812\n",
      "22\t40819\t0.0\t0\t40819\n",
      "23\t37132\t0.0\t0\t37132\n",
      "24\t37830\t0.0\t0\t37830\n",
      "25\t33693\t0.0\t0\t33693\n",
      "26\t32185\t0.0\t0\t32185\n",
      "27\t17\t0.0\t0\t17\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163908_128_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163908_129_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163908_128_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163908_129_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163908_128_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163908_129_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163908_128_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163908_129_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.40 s (11 µs/read; 5.61 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            130,619\n",
      "  Read 1 with adapter:                 126,992 (97.2%)\n",
      "  Read 2 with adapter:                 120,039 (91.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           13,822 (10.6%)\n",
      "Pairs written (passing filters):       116,797 (89.4%)\n",
      "\n",
      "Total basepairs processed:    66,615,690 bp\n",
      "  Read 1:    39,838,795 bp\n",
      "  Read 2:    26,776,895 bp\n",
      "Total written (filtered):     54,448,150 bp (81.7%)\n",
      "  Read 1:    33,233,279 bp\n",
      "  Read 2:    21,214,871 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 126992 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.0\t0\t1\n",
      "11\t3\t0.0\t0\t3\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t11\t0.0\t0\t11\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t38\t0.0\t0\t38\n",
      "18\t19518\t0.0\t0\t19518\n",
      "19\t22757\t0.0\t0\t22757\n",
      "20\t23330\t0.0\t0\t23330\n",
      "21\t20944\t0.0\t0\t20944\n",
      "22\t21929\t0.0\t0\t21929\n",
      "23\t18437\t0.0\t0\t18437\n",
      "24\t13\t0.0\t0\t13\n",
      "25\t1\t0.0\t0\t1\n",
      "29\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "54\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 120039 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t2\t0.0\t0\t2\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t55\t0.0\t0\t55\n",
      "21\t22302\t0.0\t0\t22302\n",
      "22\t21764\t0.0\t0\t21764\n",
      "23\t19681\t0.0\t0\t19681\n",
      "24\t20134\t0.0\t0\t20134\n",
      "25\t18453\t0.0\t0\t18453\n",
      "26\t17624\t0.0\t0\t17624\n",
      "27\t12\t0.0\t0\t12\n",
      "38\t1\t0.0\t0\t1\n",
      "59\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163909_130_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163909_131_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163909_130_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163909_131_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163909_130_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163909_131_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163909_130_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163909_131_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.65 s (10 µs/read; 5.88 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            161,386\n",
      "  Read 1 with adapter:                 157,056 (97.3%)\n",
      "  Read 2 with adapter:                 148,143 (91.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           17,130 (10.6%)\n",
      "Pairs written (passing filters):       144,256 (89.4%)\n",
      "\n",
      "Total basepairs processed:    82,306,860 bp\n",
      "  Read 1:    49,222,730 bp\n",
      "  Read 2:    33,084,130 bp\n",
      "Total written (filtered):     67,251,504 bp (81.7%)\n",
      "  Read 1:    41,048,088 bp\n",
      "  Read 2:    26,203,416 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 157056 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t2.5\t0\t3\n",
      "9\t1\t0.6\t0\t1\n",
      "11\t3\t0.0\t0\t3\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t6\t0.0\t0\t6\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t10\t0.0\t0\t10\n",
      "17\t44\t0.0\t0\t44\n",
      "18\t24382\t0.0\t0\t24382\n",
      "19\t28386\t0.0\t0\t28386\n",
      "20\t28841\t0.0\t0\t28841\n",
      "21\t26127\t0.0\t0\t26127\n",
      "22\t26530\t0.0\t0\t26530\n",
      "23\t22691\t0.0\t0\t22691\n",
      "24\t20\t0.0\t0\t20\n",
      "25\t2\t0.0\t0\t2\n",
      "30\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 148143 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t50\t0.0\t0\t50\n",
      "21\t27729\t0.0\t0\t27729\n",
      "22\t26596\t0.0\t0\t26596\n",
      "23\t24468\t0.0\t0\t24468\n",
      "24\t25207\t0.0\t0\t25207\n",
      "25\t22559\t0.0\t0\t22559\n",
      "26\t21499\t0.0\t0\t21499\n",
      "27\t18\t0.0\t0\t18\n",
      "28\t1\t0.0\t0\t1\n",
      "29\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163910_132_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163910_133_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163910_132_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163910_133_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163910_132_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163910_133_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163910_132_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163910_133_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.94 s (12 µs/read; 5.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             80,755\n",
      "  Read 1 with adapter:                  78,601 (97.3%)\n",
      "  Read 2 with adapter:                  74,038 (91.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            8,658 (10.7%)\n",
      "Pairs written (passing filters):        72,097 (89.3%)\n",
      "\n",
      "Total basepairs processed:    41,185,050 bp\n",
      "  Read 1:    24,630,275 bp\n",
      "  Read 2:    16,554,775 bp\n",
      "Total written (filtered):     33,611,590 bp (81.6%)\n",
      "  Read 1:    20,515,535 bp\n",
      "  Read 2:    13,096,055 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 78601 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t22\t0.0\t0\t22\n",
      "18\t12274\t0.0\t0\t12274\n",
      "19\t14235\t0.0\t0\t14235\n",
      "20\t14471\t0.0\t0\t14471\n",
      "21\t12940\t0.0\t0\t12940\n",
      "22\t13299\t0.0\t0\t13299\n",
      "23\t11335\t0.0\t0\t11335\n",
      "24\t10\t0.0\t0\t10\n",
      "28\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 74038 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t2\t0.0\t0\t2\n",
      "20\t31\t0.0\t0\t31\n",
      "21\t14009\t0.0\t0\t14009\n",
      "22\t13191\t0.0\t0\t13191\n",
      "23\t12116\t0.0\t0\t12116\n",
      "24\t12575\t0.0\t0\t12575\n",
      "25\t11290\t0.0\t0\t11290\n",
      "26\t10807\t0.0\t0\t10807\n",
      "27\t10\t0.0\t0\t10\n",
      "36\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163911_134_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163911_135_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163911_134_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163911_135_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163911_134_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163911_135_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163911_134_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163911_135_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.90 s (9 µs/read; 6.52 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            206,001\n",
      "  Read 1 with adapter:                 200,471 (97.3%)\n",
      "  Read 2 with adapter:                 188,583 (91.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           22,340 (10.8%)\n",
      "Pairs written (passing filters):       183,661 (89.2%)\n",
      "\n",
      "Total basepairs processed:   105,060,510 bp\n",
      "  Read 1:    62,830,305 bp\n",
      "  Read 2:    42,230,205 bp\n",
      "Total written (filtered):     85,622,583 bp (81.5%)\n",
      "  Read 1:    52,260,107 bp\n",
      "  Read 2:    33,362,476 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 200471 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t3.1\t0\t2\n",
      "9\t1\t0.8\t0\t1\n",
      "10\t2\t0.2\t0\t2\n",
      "11\t2\t0.0\t0\t2\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t11\t0.0\t0\t11\n",
      "16\t9\t0.0\t0\t9\n",
      "17\t50\t0.0\t0\t50\n",
      "18\t31186\t0.0\t0\t31186\n",
      "19\t36143\t0.0\t0\t36143\n",
      "20\t36694\t0.0\t0\t36694\n",
      "21\t33127\t0.0\t0\t33127\n",
      "22\t34034\t0.0\t0\t34034\n",
      "23\t29179\t0.0\t0\t29179\n",
      "24\t22\t0.0\t0\t22\n",
      "29\t1\t0.0\t0\t1\n",
      "36\t1\t0.0\t0\t1\n",
      "38\t1\t0.0\t0\t1\n",
      "42\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 188583 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.8\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t3\t0.0\t0\t3\n",
      "15\t5\t0.0\t0\t5\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t6\t0.0\t0\t6\n",
      "20\t51\t0.0\t0\t51\n",
      "21\t35487\t0.0\t0\t35487\n",
      "22\t34213\t0.0\t0\t34213\n",
      "23\t31129\t0.0\t0\t31129\n",
      "24\t31778\t0.0\t0\t31778\n",
      "25\t28541\t0.0\t0\t28541\n",
      "26\t27339\t0.0\t0\t27339\n",
      "27\t23\t0.0\t0\t23\n",
      "29\t1\t0.0\t0\t1\n",
      "48\t1\t0.0\t0\t1\n",
      "59\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163912_136_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163912_137_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163912_136_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163912_137_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163912_136_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163912_137_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163912_136_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163912_137_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.68 s (13 µs/read; 4.68 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             53,163\n",
      "  Read 1 with adapter:                  51,676 (97.2%)\n",
      "  Read 2 with adapter:                  48,787 (91.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            5,706 (10.7%)\n",
      "Pairs written (passing filters):        47,457 (89.3%)\n",
      "\n",
      "Total basepairs processed:    27,113,130 bp\n",
      "  Read 1:    16,214,715 bp\n",
      "  Read 2:    10,898,415 bp\n",
      "Total written (filtered):     22,124,105 bp (81.6%)\n",
      "  Read 1:    13,504,555 bp\n",
      "  Read 2:     8,619,550 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 51676 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.2\t0\t1\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t13\t0.0\t0\t13\n",
      "18\t8135\t0.0\t0\t8135\n",
      "19\t9350\t0.0\t0\t9350\n",
      "20\t9494\t0.0\t0\t9494\n",
      "21\t8550\t0.0\t0\t8550\n",
      "22\t8759\t0.0\t0\t8759\n",
      "23\t7359\t0.0\t0\t7359\n",
      "24\t2\t0.0\t0\t2\n",
      "26\t1\t0.0\t0\t1\n",
      "38\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 48787 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t17\t0.0\t0\t17\n",
      "21\t8863\t0.0\t0\t8863\n",
      "22\t8801\t0.0\t0\t8801\n",
      "23\t8117\t0.0\t0\t8117\n",
      "24\t8373\t0.0\t0\t8373\n",
      "25\t7529\t0.0\t0\t7529\n",
      "26\t7077\t0.0\t0\t7077\n",
      "27\t6\t0.0\t0\t6\n",
      "36\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163913_138_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163913_139_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163913_138_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163913_139_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163913_138_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163913_139_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163913_138_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163913_139_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.48 s (11 µs/read; 5.65 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            139,176\n",
      "  Read 1 with adapter:                 135,203 (97.1%)\n",
      "  Read 2 with adapter:                 127,496 (91.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           15,223 (10.9%)\n",
      "Pairs written (passing filters):       123,953 (89.1%)\n",
      "\n",
      "Total basepairs processed:    70,979,760 bp\n",
      "  Read 1:    42,448,680 bp\n",
      "  Read 2:    28,531,080 bp\n",
      "Total written (filtered):     57,785,990 bp (81.4%)\n",
      "  Read 1:    35,270,977 bp\n",
      "  Read 2:    22,515,013 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 135203 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.5\t0\t1\n",
      "11\t3\t0.0\t0\t3\n",
      "12\t2\t0.0\t0\t2\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t35\t0.0\t0\t35\n",
      "18\t21044\t0.0\t0\t21044\n",
      "19\t24431\t0.0\t0\t24431\n",
      "20\t24740\t0.0\t0\t24740\n",
      "21\t22457\t0.0\t0\t22457\n",
      "22\t22964\t0.0\t0\t22964\n",
      "23\t19491\t0.0\t0\t19491\n",
      "24\t21\t0.0\t0\t21\n",
      "26\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 127496 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t44\t0.0\t0\t44\n",
      "21\t23678\t0.0\t0\t23678\n",
      "22\t23027\t0.0\t0\t23027\n",
      "23\t21216\t0.0\t0\t21216\n",
      "24\t21535\t0.0\t0\t21535\n",
      "25\t19316\t0.0\t0\t19316\n",
      "26\t18654\t0.0\t0\t18654\n",
      "27\t9\t0.0\t0\t9\n",
      "29\t1\t0.0\t0\t1\n",
      "49\t1\t0.0\t0\t1\n",
      "54\t1\t0.0\t0\t1\n",
      "61\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163914_140_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163914_141_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163914_140_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163914_141_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163914_140_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163914_141_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163914_140_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163914_141_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 0.64 s (14 µs/read; 4.24 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             45,103\n",
      "  Read 1 with adapter:                  43,845 (97.2%)\n",
      "  Read 2 with adapter:                  41,436 (91.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            4,788 (10.6%)\n",
      "Pairs written (passing filters):        40,315 (89.4%)\n",
      "\n",
      "Total basepairs processed:    23,002,530 bp\n",
      "  Read 1:    13,756,415 bp\n",
      "  Read 2:     9,246,115 bp\n",
      "Total written (filtered):     18,794,810 bp (81.7%)\n",
      "  Read 1:    11,472,051 bp\n",
      "  Read 2:     7,322,759 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 43845 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t8\t0.0\t0\t8\n",
      "18\t6867\t0.0\t0\t6867\n",
      "19\t7918\t0.0\t0\t7918\n",
      "20\t8141\t0.0\t0\t8141\n",
      "21\t7188\t0.0\t0\t7188\n",
      "22\t7365\t0.0\t0\t7365\n",
      "23\t6347\t0.0\t0\t6347\n",
      "24\t5\t0.0\t0\t5\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 41436 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t2\t0.0\t0\t2\n",
      "17\t3\t0.0\t0\t3\n",
      "20\t11\t0.0\t0\t11\n",
      "21\t7745\t0.0\t0\t7745\n",
      "22\t7368\t0.0\t0\t7368\n",
      "23\t6922\t0.0\t0\t6922\n",
      "24\t6962\t0.0\t0\t6962\n",
      "25\t6323\t0.0\t0\t6323\n",
      "26\t6096\t0.0\t0\t6096\n",
      "27\t4\t0.0\t0\t4\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163915_142_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163915_143_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163915_142_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163915_143_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163915_142_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163915_143_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163915_142_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163915_143_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 2.51 s (10 µs/read; 6.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            257,546\n",
      "  Read 1 with adapter:                 250,606 (97.3%)\n",
      "  Read 2 with adapter:                 237,252 (92.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           26,558 (10.3%)\n",
      "Pairs written (passing filters):       230,988 (89.7%)\n",
      "\n",
      "Total basepairs processed:   131,348,460 bp\n",
      "  Read 1:    78,551,530 bp\n",
      "  Read 2:    52,796,930 bp\n",
      "Total written (filtered):    107,685,040 bp (82.0%)\n",
      "  Read 1:    65,726,182 bp\n",
      "  Read 2:    41,958,858 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 250606 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t3.9\t0\t2\n",
      "9\t3\t1.0\t0\t3\n",
      "10\t2\t0.2\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t6\t0.0\t0\t6\n",
      "14\t9\t0.0\t0\t9\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t80\t0.0\t0\t80\n",
      "18\t38608\t0.0\t0\t38608\n",
      "19\t45103\t0.0\t0\t45103\n",
      "20\t46133\t0.0\t0\t46133\n",
      "21\t41500\t0.0\t0\t41500\n",
      "22\t42909\t0.0\t0\t42909\n",
      "23\t36201\t0.0\t0\t36201\n",
      "24\t26\t0.0\t0\t26\n",
      "25\t1\t0.0\t0\t1\n",
      "26\t1\t0.0\t0\t1\n",
      "41\t1\t0.0\t0\t1\n",
      "54\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 237252 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t6\t0.0\t0\t6\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t4\t0.0\t0\t4\n",
      "19\t8\t0.0\t0\t8\n",
      "20\t86\t0.0\t0\t86\n",
      "21\t44416\t0.0\t0\t44416\n",
      "22\t42665\t0.0\t0\t42665\n",
      "23\t39519\t0.0\t0\t39519\n",
      "24\t40423\t0.0\t0\t40423\n",
      "25\t35747\t0.0\t0\t35747\n",
      "26\t34346\t0.0\t0\t34346\n",
      "27\t21\t0.0\t0\t21\n",
      "35\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "56\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163916_144_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163916_145_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163916_144_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163916_145_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163916_144_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163916_145_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163916_144_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163916_145_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.10 s (10 µs/read; 6.09 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            314,325\n",
      "  Read 1 with adapter:                 305,481 (97.2%)\n",
      "  Read 2 with adapter:                 290,205 (92.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           32,137 (10.2%)\n",
      "Pairs written (passing filters):       282,188 (89.8%)\n",
      "\n",
      "Total basepairs processed:   160,305,750 bp\n",
      "  Read 1:    95,869,125 bp\n",
      "  Read 2:    64,436,625 bp\n",
      "Total written (filtered):    131,558,190 bp (82.1%)\n",
      "  Read 1:    80,299,014 bp\n",
      "  Read 2:    51,259,176 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 305481 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t4.8\t0\t3\n",
      "9\t5\t1.2\t0\t5\n",
      "10\t6\t0.3\t0\t6\n",
      "11\t5\t0.1\t0\t5\n",
      "12\t6\t0.0\t0\t6\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t13\t0.0\t0\t13\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t19\t0.0\t0\t19\n",
      "17\t86\t0.0\t0\t86\n",
      "18\t47777\t0.0\t0\t47777\n",
      "19\t55351\t0.0\t0\t55351\n",
      "20\t56306\t0.0\t0\t56306\n",
      "21\t50388\t0.0\t0\t50388\n",
      "22\t51687\t0.0\t0\t51687\n",
      "23\t43785\t0.0\t0\t43785\n",
      "24\t33\t0.0\t0\t33\n",
      "25\t1\t0.0\t0\t1\n",
      "27\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "38\t1\t0.0\t0\t1\n",
      "41\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 290205 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.3\t0\t1\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t5\t0.0\t0\t5\n",
      "18\t6\t0.0\t0\t6\n",
      "19\t11\t0.0\t0\t11\n",
      "20\t91\t0.0\t0\t91\n",
      "21\t54655\t0.0\t0\t54655\n",
      "22\t52202\t0.0\t0\t52202\n",
      "23\t47774\t0.0\t0\t47774\n",
      "24\t49323\t0.0\t0\t49323\n",
      "25\t44044\t0.0\t0\t44044\n",
      "26\t42034\t0.0\t0\t42034\n",
      "27\t35\t0.0\t0\t35\n",
      "29\t1\t0.0\t0\t1\n",
      "33\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "62\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163917_146_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163917_147_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163917_146_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163917_147_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163917_146_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-oyazuqe_/S210707163917_147_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163917_146_L001_R1_001.fastq.gz /tmp/qiime2-archive-7lm7gvl9/068f0c34-15b0-437a-a67e-bb0223aa7625/data/S210707163917_147_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n",
      "Finished in 1.55 s (11 µs/read; 5.53 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            142,842\n",
      "  Read 1 with adapter:                 138,968 (97.3%)\n",
      "  Read 2 with adapter:                 131,105 (91.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           15,220 (10.7%)\n",
      "Pairs written (passing filters):       127,622 (89.3%)\n",
      "\n",
      "Total basepairs processed:    72,849,420 bp\n",
      "  Read 1:    43,566,810 bp\n",
      "  Read 2:    29,282,610 bp\n",
      "Total written (filtered):     59,498,632 bp (81.7%)\n",
      "  Read 1:    36,316,219 bp\n",
      "  Read 2:    23,182,413 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 138968 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.2\t0\t1\n",
      "9\t1\t0.5\t0\t1\n",
      "10\t3\t0.1\t0\t3\n",
      "11\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t35\t0.0\t0\t35\n",
      "18\t21841\t0.0\t0\t21841\n",
      "19\t25045\t0.0\t0\t25045\n",
      "20\t25758\t0.0\t0\t25758\n",
      "21\t22904\t0.0\t0\t22904\n",
      "22\t23575\t0.0\t0\t23575\n",
      "23\t19771\t0.0\t0\t19771\n",
      "24\t17\t0.0\t0\t17\n",
      "28\t1\t0.0\t0\t1\n",
      "32\t1\t0.0\t0\t1\n",
      "38\t1\t0.0\t0\t1\n",
      "46\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 131105 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t4\t0.0\t0\t4\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t2\t0.0\t0\t2\n",
      "20\t46\t0.0\t0\t46\n",
      "21\t24694\t0.0\t0\t24694\n",
      "22\t23562\t0.0\t0\t23562\n",
      "23\t21684\t0.0\t0\t21684\n",
      "24\t22144\t0.0\t0\t22144\n",
      "25\t19786\t0.0\t0\t19786\n",
      "26\t19159\t0.0\t0\t19159\n",
      "27\t15\t0.0\t0\t15\n",
      "58\t1\t0.0\t0\t1\n"
     ]
    }
   ],
   "source": [
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')\n",
    "\n",
    "if trim and (not os.path.isfile(demux_file_trim) or replace_files):\n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = None\n",
    "    \n",
    "    if n_directions == 1:\n",
    "        res = trim_single(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front=forward_primer,\n",
    "            adapter=reverse_reverse_complement,\n",
    "            cores=threads,\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    elif n_directions == 2:\n",
    "        res = trim_paired(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front_f=forward_primer,\n",
    "            front_r=reverse_primer,\n",
    "            # adapter_f=reverse_reverse_complement,\n",
    "            # adapter_r=forward_reverse_complement,\n",
    "            cores=threads,\n",
    "            overlap=trim['overlap'],\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    else:\n",
    "        print(f'ERROR: invalid number of directions {n_directions}')\n",
    "\n",
    "    if res is not None:\n",
    "        res.save(demux_file_trim)\n",
    "        res_view = demux.visualizers.summarize(res).visualization\n",
    "        Visualization.save(res_view, filepath=demux_view_trim)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "05c4ce1f",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-23T15:39:49.017550Z",
     "start_time": "2023-08-23T15:39:49.008050Z"
    },
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "pending"
    },
    "scrolled": false,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-dd6vegka')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 19260d63-9a57-45b5-a582-8552cc8a0e4d>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res_view"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c31d6829",
   "metadata": {
    "ExecuteTime": {
     "start_time": "2023-08-23T00:38:03.024Z"
    }
   },
   "outputs": [],
   "source": [
    "demux_path = os.path.join('..', 'qiime-artifacts', 'demux-paired-trim.qza')\n",
    "!ls {demux_path}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3220b559",
   "metadata": {
    "ExecuteTime": {
     "start_time": "2023-08-23T00:38:03.993Z"
    }
   },
   "outputs": [],
   "source": [
    "filtered_demux_path = os.path.join(out_dir, 'demux-paired-trim-filtered.qza')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71a168e2",
   "metadata": {
    "ExecuteTime": {
     "start_time": "2023-08-23T00:38:04.928Z"
    }
   },
   "outputs": [],
   "source": [
    "demux_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4bb71c1c",
   "metadata": {
    "ExecuteTime": {
     "start_time": "2023-08-23T00:38:07.712Z"
    }
   },
   "outputs": [],
   "source": [
    "metadata_qa = Metadata.load(metadata_file)\n",
    "metadata_qa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f763e2af",
   "metadata": {
    "ExecuteTime": {
     "start_time": "2023-08-23T00:38:09.041Z"
    }
   },
   "outputs": [],
   "source": [
    "metadata_f = os.path.join('..', '..', '..', 'data', 'raw', 'metadata-all-ana.tsv')\n",
    "!ls {metadata_f}\n",
    "metadata_art = Metadata.load(metadata_f)\n",
    "metadata_art\n",
    "\n",
    "aux_art = Artifact.load(demux_path)\n",
    "aux_art"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0b749b1a",
   "metadata": {
    "ExecuteTime": {
     "start_time": "2023-08-23T00:38:11.744Z"
    }
   },
   "outputs": [],
   "source": [
    "filtered_demux_path = '../qiime-artifacts/demux-paired-trim-filtered.qza'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "54cbe336",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-08T22:31:25.999019Z",
     "start_time": "2023-08-08T22:31:11.227010Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "metadata_f=\"../../../data/raw/metadata-all-ana.tsv\"\n",
    "demux_path=\"../qiime-artifacts/demux-paired-trim.qza\"\n",
    "filtered_demux_path=\"../qiime-artifacts/demux-paired-trim-filtered.qza\"\n",
    "\n",
    "qiime demux filter-samples --i-demux ${demux_path} --m-metadata-file ${metadata_f} --o-filtered-demux ${filtered_demux_path}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "846f4493",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-08T22:26:06.173072Z",
     "start_time": "2023-08-08T22:26:05.890192Z"
    }
   },
   "outputs": [],
   "source": [
    "from qiime2.plugins.demux.methods import filter_samples\n",
    "\n",
    "res2 = filter_samples(demux=aux_art, metadata=metadata_art).filtered_demux"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 3191.244331,
   "end_time": "2023-08-07T21:08:41.804865",
   "environment_variables": {},
   "exception": true,
   "input_path": "nb-templates/step-prepare-data.ipynb",
   "output_path": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-all/nb-executed-steps/step-prepare-data-ana-flavia-all.ipynb",
   "parameters": {
    "base_dir": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri",
    "class_col": "group-id",
    "classifier_file": "/home/lauro/nupeb/rede-micro/models/silva-138-99-nb-classifier.qza",
    "experiment_name": "ana-flavia-all",
    "manifest_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest-all-ana.csv",
    "metadata_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata-all-ana.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "top_n": 20,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2023-08-07T20:15:30.560534",
   "version": "2.3.4"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
