{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "07c3f250",
   "metadata": {
    "papermill": {
     "duration": 0.011612,
     "end_time": "2023-08-28T14:24:34.523806",
     "exception": false,
     "start_time": "2023-08-28T14:24:34.512194",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4a3b6ccb",
   "metadata": {
    "papermill": {
     "duration": 0.009769,
     "end_time": "2023-08-28T14:24:34.543927",
     "exception": false,
     "start_time": "2023-08-28T14:24:34.534158",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5d5d500c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-10-14T14:33:37.863523Z",
     "start_time": "2022-10-14T14:33:14.145712Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-28T14:24:34.564449Z",
     "iopub.status.busy": "2023-08-28T14:24:34.564042Z",
     "iopub.status.idle": "2023-08-28T14:24:41.411119Z",
     "shell.execute_reply": "2023-08-28T14:24:41.410329Z"
    },
    "papermill": {
     "duration": 6.86094,
     "end_time": "2023-08-28T14:24:41.414450",
     "exception": false,
     "start_time": "2023-08-28T14:24:34.553510",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization, Metadata\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired, trim_single\n",
    "from qiime2.plugins.demux.methods import filter_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "60ec1c2d",
   "metadata": {
    "papermill": {
     "duration": 0.010324,
     "end_time": "2023-08-28T14:24:41.441422",
     "exception": false,
     "start_time": "2023-08-28T14:24:41.431098",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9431e74b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.594341Z",
     "start_time": "2022-06-28T18:50:41.591544Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-28T14:24:41.534684Z",
     "iopub.status.busy": "2023-08-28T14:24:41.534162Z",
     "iopub.status.idle": "2023-08-28T14:24:41.540815Z",
     "shell.execute_reply": "2023-08-28T14:24:41.539886Z"
    },
    "papermill": {
     "duration": 0.02014,
     "end_time": "2023-08-28T14:24:41.542531",
     "exception": false,
     "start_time": "2023-08-28T14:24:41.522391",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'parameter-set-01.yaml')\n",
    "experiment_name = 'exp-01'\n",
    "base_dir = os.path.join('/', 'home', 'username', 'pipeline-dir')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'manifest.csv')\n",
    "replace_files = False\n",
    "trim = None\n",
    "metadata_file = None\n",
    "threads = 1\n",
    "demux_file = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8f8dfdb5",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-28T14:24:41.567279Z",
     "iopub.status.busy": "2023-08-28T14:24:41.566850Z",
     "iopub.status.idle": "2023-08-28T14:24:41.573785Z",
     "shell.execute_reply": "2023-08-28T14:24:41.572995Z"
    },
    "papermill": {
     "duration": 0.020306,
     "end_time": "2023-08-28T14:24:41.575412",
     "exception": false,
     "start_time": "2023-08-28T14:24:41.555106",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "experiment_name = \"ana-flavia-NCxSTD-NC-trim\"\n",
    "base_dir = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri\"\n",
    "manifest_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-NCxSTD-NC.csv\"\n",
    "metadata_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-NCxSTD-NC.tsv\"\n",
    "class_col = \"group-id\"\n",
    "classifier_file = \"/home/lauro/nupeb/rede-micro/models/silva-138-99-nb-classifier.qza\"\n",
    "top_n = 20\n",
    "replace_files = False\n",
    "phred = 20\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n",
    "overlap = 12\n",
    "threads = 6\n",
    "trim = {\n",
    "    \"overlap\": 8,\n",
    "    \"forward_primer\": \"CCTACGGGRSGCAGCAG\",\n",
    "    \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\",\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7f3feace",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-18T15:40:31.070885Z",
     "start_time": "2023-08-18T15:40:30.611671Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-28T14:24:41.600767Z",
     "iopub.status.busy": "2023-08-28T14:24:41.600344Z",
     "iopub.status.idle": "2023-08-28T14:24:41.607275Z",
     "shell.execute_reply": "2023-08-28T14:24:41.606423Z"
    },
    "papermill": {
     "duration": 0.020191,
     "end_time": "2023-08-28T14:24:41.608917",
     "exception": false,
     "start_time": "2023-08-28T14:24:41.588726",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "data_dir =  os.path.abspath(os.path.join(base_dir, 'data'))\n",
    "raw_data_dir =  os.path.abspath(os.path.join(data_dir, 'raw'))\n",
    "interim_data_dir =  os.path.abspath(os.path.join(data_dir, 'interim'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fd73862f",
   "metadata": {
    "papermill": {
     "duration": 0.01129,
     "end_time": "2023-08-28T14:24:41.632192",
     "exception": false,
     "start_time": "2023-08-28T14:24:41.620902",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "9b6a53ff",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.632125Z",
     "start_time": "2022-06-28T18:50:41.619710Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-28T14:24:41.654757Z",
     "iopub.status.busy": "2023-08-28T14:24:41.654400Z",
     "iopub.status.idle": "2023-08-28T14:24:41.663218Z",
     "shell.execute_reply": "2023-08-28T14:24:41.662267Z"
    },
    "papermill": {
     "duration": 0.02212,
     "end_time": "2023-08-28T14:24:41.665048",
     "exception": false,
     "start_time": "2023-08-28T14:24:41.642928",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New qiime-artifacts folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-NCxSTD-NC-trim/qiime-artifacts\n",
      "New img folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-NCxSTD-NC-trim/imgs\n"
     ]
    }
   ],
   "source": [
    "# Create path if it not exists\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exists\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')\n",
    "\n",
    "# Define the output artifact full path\n",
    "if demux_file is None:\n",
    "    demux_file = os.path.join(out_dir, 'demux-paired.qza')\n",
    "demux_view = os.path.join(out_dir, 'demux-paired.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "74f764ce",
   "metadata": {
    "papermill": {
     "duration": 0.011375,
     "end_time": "2023-08-28T14:24:41.688956",
     "exception": false,
     "start_time": "2023-08-28T14:24:41.677581",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file of `SampleData[PairedEndSequencesWithQuality]`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "319c4de1",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-28T14:24:41.721585Z",
     "iopub.status.busy": "2023-08-28T14:24:41.721131Z",
     "iopub.status.idle": "2023-08-28T14:24:41.742817Z",
     "shell.execute_reply": "2023-08-28T14:24:41.742085Z"
    },
    "papermill": {
     "duration": 0.04489,
     "end_time": "2023-08-28T14:24:41.744882",
     "exception": false,
     "start_time": "2023-08-28T14:24:41.699992",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "manifest_df = pd.read_csv(manifest_file)\n",
    "n_directions = len(manifest_df['direction'].unique())\n",
    "if n_directions == 1:\n",
    "    d_type = 'SampleData[SequencesWithQuality]'\n",
    "    v_type = 'SingleEndFastqManifestPhred33'\n",
    "elif n_directions == 2:\n",
    "    d_type = 'SampleData[PairedEndSequencesWithQuality]'\n",
    "    v_type = 'PairedEndFastqManifestPhred33'\n",
    "else:\n",
    "    print(f'ERROR: invalid number of directions {n_directions}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1bea4734",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-28T14:24:41.772889Z",
     "iopub.status.busy": "2023-08-28T14:24:41.772452Z",
     "iopub.status.idle": "2023-08-28T14:24:41.790475Z",
     "shell.execute_reply": "2023-08-28T14:24:41.789690Z"
    },
    "papermill": {
     "duration": 0.032081,
     "end_time": "2023-08-28T14:24:41.792353",
     "exception": false,
     "start_time": "2023-08-28T14:24:41.760272",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the metadata file is not defined, use the default metadata file\n",
    "metadata_qa = Metadata.load(metadata_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1fe7cf50",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:44:58.640338Z",
     "start_time": "2022-06-28T18:50:41.633657Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-28T14:24:41.821770Z",
     "iopub.status.busy": "2023-08-28T14:24:41.821476Z",
     "iopub.status.idle": "2023-08-28T14:25:57.468839Z",
     "shell.execute_reply": "2023-08-28T14:25:57.468145Z"
    },
    "papermill": {
     "duration": 75.662825,
     "end_time": "2023-08-28T14:25:57.471871",
     "exception": false,
     "start_time": "2023-08-28T14:24:41.809046",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    interim_demux_path = os.path.join(interim_data_dir, \"demux-paired.qza\")\n",
    "    # Load Artifact from interim folder\n",
    "    if os.path.isfile(interim_demux_path):\n",
    "        artifact = Artifact.load(interim_demux_path)\n",
    "    # Create new Artifact using Manifest\n",
    "    else:\n",
    "        artifact = Artifact.import_data(d_type, manifest_file, view_type=v_type)\n",
    "        if not os.path.isfile(interim_demux_path):\n",
    "            artifact.save(interim_demux_path)\n",
    "    # Filter and Save Artifact\n",
    "    artifact = filter_samples(demux=artifact, metadata=metadata_qa).filtered_demux\n",
    "    artifact.save(demux_file)\n",
    "    # Visualization\n",
    "    if os.path.isfile(demux_view):\n",
    "        demux_view_obj = Visualization.load(demux_view)\n",
    "    else:\n",
    "        demux_view_obj = demux.visualizers.summarize(artifact).visualization\n",
    "        Visualization.save(demux_view_obj, filepath=demux_view)\n",
    "elif os.path.isfile(demux_file):\n",
    "    # Load Artifact\n",
    "    artifact = Artifact.load(demux_file)\n",
    "    artifact = filter_samples(demux=artifact, metadata=metadata_qa).filtered_demux\n",
    "    # Visualization\n",
    "    if os.path.isfile(demux_view):\n",
    "        demux_view_obj = Visualization.load(demux_view)\n",
    "    else:\n",
    "        demux_view_obj = demux.visualizers.summarize(artifact).visualization\n",
    "        Visualization.save(demux_view_obj, filepath=demux_view)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ca22824f",
   "metadata": {
    "papermill": {
     "duration": 0.010362,
     "end_time": "2023-08-28T14:25:57.500118",
     "exception": false,
     "start_time": "2023-08-28T14:25:57.489756",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "7957c00b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.609737Z",
     "start_time": "2022-06-28T19:47:56.604820Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-28T14:25:57.525849Z",
     "iopub.status.busy": "2023-08-28T14:25:57.525482Z",
     "iopub.status.idle": "2023-08-28T14:25:57.530836Z",
     "shell.execute_reply": "2023-08-28T14:25:57.529821Z"
    },
    "papermill": {
     "duration": 0.022723,
     "end_time": "2023-08-28T14:25:57.532313",
     "exception": false,
     "start_time": "2023-08-28T14:25:57.509590",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<visualization: Visualization uuid: 1d38aade-9ada-471b-8b18-18496c99c5ec>\n"
     ]
    }
   ],
   "source": [
    "print(demux_view_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "28ad0a05",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.704841Z",
     "start_time": "2022-06-28T19:47:56.611846Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-28T14:25:57.555638Z",
     "iopub.status.busy": "2023-08-28T14:25:57.555094Z",
     "iopub.status.idle": "2023-08-28T14:25:57.567119Z",
     "shell.execute_reply": "2023-08-28T14:25:57.566667Z"
    },
    "papermill": {
     "duration": 0.025351,
     "end_time": "2023-08-28T14:25:57.568575",
     "exception": false,
     "start_time": "2023-08-28T14:25:57.543224",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-j51h8qhb')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 1d38aade-9ada-471b-8b18-18496c99c5ec>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "162ee209",
   "metadata": {
    "papermill": {
     "duration": 0.010573,
     "end_time": "2023-08-28T14:25:57.591760",
     "exception": false,
     "start_time": "2023-08-28T14:25:57.581187",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6e61bdaa",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-28T14:25:57.613358Z",
     "iopub.status.busy": "2023-08-28T14:25:57.612922Z",
     "iopub.status.idle": "2023-08-28T14:27:25.388819Z",
     "shell.execute_reply": "2023-08-28T14:27:25.388018Z"
    },
    "papermill": {
     "duration": 87.790092,
     "end_time": "2023-08-28T14:27:25.392001",
     "exception": false,
     "start_time": "2023-08-28T14:25:57.601909",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121673_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121673_1_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121673_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121673_1_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121673_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121673_1_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121673_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121673_1_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.77 s (10 µs/read; 5.89 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            370,042\n",
      "  Read 1 with adapter:                 359,997 (97.3%)\n",
      "  Read 2 with adapter:                 343,009 (92.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           36,262 (9.8%)\n",
      "Pairs written (passing filters):       333,780 (90.2%)\n",
      "\n",
      "Total basepairs processed:   188,721,420 bp\n",
      "  Read 1:   112,862,810 bp\n",
      "  Read 2:    75,858,610 bp\n",
      "Total written (filtered):    155,609,062 bp (82.5%)\n",
      "  Read 1:    94,978,565 bp\n",
      "  Read 2:    60,630,497 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 359997 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t5.6\t0\t3\n",
      "9\t4\t1.4\t0\t4\n",
      "10\t3\t0.4\t0\t3\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t4\t0.0\t0\t4\n",
      "14\t12\t0.0\t0\t12\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t19\t0.0\t0\t19\n",
      "17\t109\t0.0\t0\t109\n",
      "18\t56158\t0.0\t0\t56158\n",
      "19\t64949\t0.0\t0\t64949\n",
      "20\t66248\t0.0\t0\t66248\n",
      "21\t59860\t0.0\t0\t59860\n",
      "22\t60699\t0.0\t0\t60699\n",
      "23\t51867\t0.0\t0\t51867\n",
      "24\t42\t0.0\t0\t42\n",
      "25\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 343009 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.4\t0\t1\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t14\t0.0\t0\t14\n",
      "20\t137\t0.0\t0\t137\n",
      "21\t64389\t0.0\t0\t64389\n",
      "22\t61501\t0.0\t0\t61501\n",
      "23\t56595\t0.0\t0\t56595\n",
      "24\t58584\t0.0\t0\t58584\n",
      "25\t52061\t0.0\t0\t52061\n",
      "26\t49654\t0.0\t0\t49654\n",
      "27\t42\t0.0\t0\t42\n",
      "29\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121674_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121674_3_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121674_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121674_3_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121674_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121674_3_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121674_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121674_3_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 5.08 s (9 µs/read; 6.61 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            559,807\n",
      "  Read 1 with adapter:                 544,656 (97.3%)\n",
      "  Read 2 with adapter:                 519,894 (92.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           53,686 (9.6%)\n",
      "Pairs written (passing filters):       506,121 (90.4%)\n",
      "\n",
      "Total basepairs processed:   285,501,570 bp\n",
      "  Read 1:   170,741,135 bp\n",
      "  Read 2:   114,760,435 bp\n",
      "Total written (filtered):    235,957,214 bp (82.6%)\n",
      "  Read 1:   144,019,667 bp\n",
      "  Read 2:    91,937,547 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 544656 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t8.5\t0\t5\n",
      "9\t4\t2.1\t0\t4\n",
      "10\t3\t0.5\t0\t3\n",
      "11\t7\t0.1\t0\t7\n",
      "12\t7\t0.0\t0\t7\n",
      "13\t6\t0.0\t0\t6\n",
      "14\t17\t0.0\t0\t17\n",
      "15\t16\t0.0\t0\t16\n",
      "16\t22\t0.0\t0\t22\n",
      "17\t155\t0.0\t0\t155\n",
      "18\t85410\t0.0\t0\t85410\n",
      "19\t98095\t0.0\t0\t98095\n",
      "20\t100478\t0.0\t0\t100478\n",
      "21\t90125\t0.0\t0\t90125\n",
      "22\t91840\t0.0\t0\t91840\n",
      "23\t78402\t0.0\t0\t78402\n",
      "24\t55\t0.0\t0\t55\n",
      "26\t3\t0.0\t0\t3\n",
      "29\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "35\t1\t0.0\t0\t1\n",
      "42\t1\t0.0\t0\t1\n",
      "43\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 519894 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t5\t0.0\t0\t5\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t15\t0.0\t0\t15\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t9\t0.0\t0\t9\n",
      "18\t6\t0.0\t0\t6\n",
      "19\t17\t0.0\t0\t17\n",
      "20\t180\t0.0\t0\t180\n",
      "21\t98232\t0.0\t0\t98232\n",
      "22\t93652\t0.0\t0\t93652\n",
      "23\t85816\t0.0\t0\t85816\n",
      "24\t87651\t0.0\t0\t87651\n",
      "25\t78341\t0.0\t0\t78341\n",
      "26\t75895\t0.0\t0\t75895\n",
      "27\t62\t0.0\t0\t62\n",
      "36\t1\t0.0\t0\t1\n",
      "57\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121675_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121675_5_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121675_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121675_5_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121675_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121675_5_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121675_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121675_5_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.35 s (10 µs/read; 6.19 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            345,762\n",
      "  Read 1 with adapter:                 336,819 (97.4%)\n",
      "  Read 2 with adapter:                 321,022 (92.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           32,941 (9.5%)\n",
      "Pairs written (passing filters):       312,821 (90.5%)\n",
      "\n",
      "Total basepairs processed:   176,338,620 bp\n",
      "  Read 1:   105,457,410 bp\n",
      "  Read 2:    70,881,210 bp\n",
      "Total written (filtered):    145,841,282 bp (82.7%)\n",
      "  Read 1:    89,015,404 bp\n",
      "  Read 2:    56,825,878 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 336819 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t5.3\t0\t5\n",
      "9\t3\t1.3\t0\t3\n",
      "10\t5\t0.3\t0\t5\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t5\t0.0\t0\t5\n",
      "14\t12\t0.0\t0\t12\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t11\t0.0\t0\t11\n",
      "17\t92\t0.0\t0\t92\n",
      "18\t52679\t0.0\t0\t52679\n",
      "19\t60788\t0.0\t0\t60788\n",
      "20\t62024\t0.0\t0\t62024\n",
      "21\t55828\t0.0\t0\t55828\n",
      "22\t57192\t0.0\t0\t57192\n",
      "23\t48121\t0.0\t0\t48121\n",
      "24\t34\t0.0\t0\t34\n",
      "25\t2\t0.0\t0\t2\n",
      "28\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "36\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 321022 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t3\t0.0\t0\t3\n",
      "15\t12\t0.0\t0\t12\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t11\t0.0\t0\t11\n",
      "20\t106\t0.0\t0\t106\n",
      "21\t60450\t0.0\t0\t60450\n",
      "22\t58128\t0.0\t0\t58128\n",
      "23\t53277\t0.0\t0\t53277\n",
      "24\t54587\t0.0\t0\t54587\n",
      "25\t48050\t0.0\t0\t48050\n",
      "26\t46353\t0.0\t0\t46353\n",
      "27\t34\t0.0\t0\t34\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121676_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121676_7_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121676_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121676_7_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121676_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121676_7_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121676_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121676_7_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 6.46 s (9 µs/read; 6.62 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            712,750\n",
      "  Read 1 with adapter:                 694,205 (97.4%)\n",
      "  Read 2 with adapter:                 664,033 (93.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           65,715 (9.2%)\n",
      "Pairs written (passing filters):       647,035 (90.8%)\n",
      "\n",
      "Total basepairs processed:   363,502,500 bp\n",
      "  Read 1:   217,388,750 bp\n",
      "  Read 2:   146,113,750 bp\n",
      "Total written (filtered):    301,650,677 bp (83.0%)\n",
      "  Read 1:   184,115,305 bp\n",
      "  Read 2:   117,535,372 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 694205 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t10.9\t0\t4\n",
      "9\t9\t2.7\t0\t9\n",
      "10\t5\t0.7\t0\t5\n",
      "11\t14\t0.2\t0\t14\n",
      "12\t8\t0.0\t0\t8\n",
      "13\t6\t0.0\t0\t6\n",
      "14\t30\t0.0\t0\t30\n",
      "15\t18\t0.0\t0\t18\n",
      "16\t33\t0.0\t0\t33\n",
      "17\t183\t0.0\t0\t183\n",
      "18\t108490\t0.0\t0\t108490\n",
      "19\t125122\t0.0\t0\t125122\n",
      "20\t127377\t0.0\t0\t127377\n",
      "21\t114840\t0.0\t0\t114840\n",
      "22\t117757\t0.0\t0\t117757\n",
      "23\t100233\t0.0\t0\t100233\n",
      "24\t68\t0.0\t0\t68\n",
      "25\t1\t0.0\t0\t1\n",
      "28\t2\t0.0\t0\t2\n",
      "35\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "41\t1\t0.0\t0\t1\n",
      "49\t1\t0.0\t0\t1\n",
      "57\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 664033 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.7\t0\t2\n",
      "11\t4\t0.2\t0\t4\n",
      "12\t5\t0.0\t0\t5\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t18\t0.0\t0\t18\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t4\t0.0\t0\t4\n",
      "18\t9\t0.0\t0\t9\n",
      "19\t16\t0.0\t0\t16\n",
      "20\t229\t0.0\t0\t229\n",
      "21\t124781\t0.0\t0\t124781\n",
      "22\t120299\t0.0\t0\t120299\n",
      "23\t109785\t0.0\t0\t109785\n",
      "24\t112135\t0.0\t0\t112135\n",
      "25\t100654\t0.0\t0\t100654\n",
      "26\t95988\t0.0\t0\t95988\n",
      "27\t87\t0.0\t0\t87\n",
      "28\t1\t0.0\t0\t1\n",
      "29\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "47\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "56\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121682_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121682_19_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121682_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121682_19_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121682_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121682_19_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121682_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121682_19_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.69 s (10 µs/read; 5.88 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            165,821\n",
      "  Read 1 with adapter:                 161,411 (97.3%)\n",
      "  Read 2 with adapter:                 154,014 (92.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           15,834 (9.5%)\n",
      "Pairs written (passing filters):       149,987 (90.5%)\n",
      "\n",
      "Total basepairs processed:    84,568,710 bp\n",
      "  Read 1:    50,575,405 bp\n",
      "  Read 2:    33,993,305 bp\n",
      "Total written (filtered):     69,924,570 bp (82.7%)\n",
      "  Read 1:    42,679,301 bp\n",
      "  Read 2:    27,245,269 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 161411 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t2.5\t0\t3\n",
      "9\t1\t0.6\t0\t1\n",
      "10\t3\t0.2\t0\t3\n",
      "11\t4\t0.0\t0\t4\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t46\t0.0\t0\t46\n",
      "18\t25121\t0.0\t0\t25121\n",
      "19\t28953\t0.0\t0\t28953\n",
      "20\t29908\t0.0\t0\t29908\n",
      "21\t26709\t0.0\t0\t26709\n",
      "22\t27683\t0.0\t0\t27683\n",
      "23\t22947\t0.0\t0\t22947\n",
      "24\t12\t0.0\t0\t12\n",
      "26\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "38\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 154014 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t3\t0.0\t0\t3\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t4\t0.0\t0\t4\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t54\t0.0\t0\t54\n",
      "21\t28983\t0.0\t0\t28983\n",
      "22\t27868\t0.0\t0\t27868\n",
      "23\t25302\t0.0\t0\t25302\n",
      "24\t25940\t0.0\t0\t25940\n",
      "25\t23454\t0.0\t0\t23454\n",
      "26\t22373\t0.0\t0\t22373\n",
      "27\t24\t0.0\t0\t24\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121683_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121683_21_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121683_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121683_21_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121683_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121683_21_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121683_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121683_21_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.56 s (10 µs/read; 5.74 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            149,412\n",
      "  Read 1 with adapter:                 145,228 (97.2%)\n",
      "  Read 2 with adapter:                 138,274 (92.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           14,953 (10.0%)\n",
      "Pairs written (passing filters):       134,459 (90.0%)\n",
      "\n",
      "Total basepairs processed:    76,200,120 bp\n",
      "  Read 1:    45,570,660 bp\n",
      "  Read 2:    30,629,460 bp\n",
      "Total written (filtered):     62,685,460 bp (82.3%)\n",
      "  Read 1:    38,260,442 bp\n",
      "  Read 2:    24,425,018 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 145228 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.3\t0\t1\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t10\t0.0\t0\t10\n",
      "17\t45\t0.0\t0\t45\n",
      "18\t22753\t0.0\t0\t22753\n",
      "19\t25911\t0.0\t0\t25911\n",
      "20\t26596\t0.0\t0\t26596\n",
      "21\t24288\t0.0\t0\t24288\n",
      "22\t24636\t0.0\t0\t24636\n",
      "23\t20953\t0.0\t0\t20953\n",
      "24\t17\t0.0\t0\t17\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 138274 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t1\t0.0\t0\t1\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t53\t0.0\t0\t53\n",
      "21\t25986\t0.0\t0\t25986\n",
      "22\t25058\t0.0\t0\t25058\n",
      "23\t22833\t0.0\t0\t22833\n",
      "24\t23540\t0.0\t0\t23540\n",
      "25\t20833\t0.0\t0\t20833\n",
      "26\t19937\t0.0\t0\t19937\n",
      "27\t16\t0.0\t0\t16\n",
      "33\t2\t0.0\t0\t2\n",
      "40\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121684_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121684_23_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121684_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121684_23_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121684_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121684_23_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121684_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121684_23_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.27 s (10 µs/read; 6.29 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            343,261\n",
      "  Read 1 with adapter:                 334,370 (97.4%)\n",
      "  Read 2 with adapter:                 318,364 (92.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           32,990 (9.6%)\n",
      "Pairs written (passing filters):       310,271 (90.4%)\n",
      "\n",
      "Total basepairs processed:   175,063,110 bp\n",
      "  Read 1:   104,694,605 bp\n",
      "  Read 2:    70,368,505 bp\n",
      "Total written (filtered):    144,648,355 bp (82.6%)\n",
      "  Read 1:    88,289,164 bp\n",
      "  Read 2:    56,359,191 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 334370 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.2\t0\t1\n",
      "9\t5\t1.3\t0\t5\n",
      "10\t2\t0.3\t0\t2\n",
      "11\t5\t0.1\t0\t5\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t11\t0.0\t0\t11\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t13\t0.0\t0\t13\n",
      "17\t90\t0.0\t0\t90\n",
      "18\t52368\t0.0\t0\t52368\n",
      "19\t60523\t0.0\t0\t60523\n",
      "20\t61360\t0.0\t0\t61360\n",
      "21\t54944\t0.0\t0\t54944\n",
      "22\t56828\t0.0\t0\t56828\n",
      "23\t48158\t0.0\t0\t48158\n",
      "24\t46\t0.0\t0\t46\n",
      "33\t1\t0.0\t0\t1\n",
      "35\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "44\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 318364 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t11\t0.0\t0\t11\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t5\t0.0\t0\t5\n",
      "18\t5\t0.0\t0\t5\n",
      "19\t9\t0.0\t0\t9\n",
      "20\t93\t0.0\t0\t93\n",
      "21\t59527\t0.0\t0\t59527\n",
      "22\t57389\t0.0\t0\t57389\n",
      "23\t52503\t0.0\t0\t52503\n",
      "24\t54259\t0.0\t0\t54259\n",
      "25\t47921\t0.0\t0\t47921\n",
      "26\t46600\t0.0\t0\t46600\n",
      "27\t35\t0.0\t0\t35\n",
      "35\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121694_42_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121694_43_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121694_42_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121694_43_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121694_42_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121694_43_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121694_42_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121694_43_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.72 s (12 µs/read; 4.90 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             58,992\n",
      "  Read 1 with adapter:                  57,341 (97.2%)\n",
      "  Read 2 with adapter:                  54,208 (91.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            6,281 (10.6%)\n",
      "Pairs written (passing filters):        52,711 (89.4%)\n",
      "\n",
      "Total basepairs processed:    30,085,920 bp\n",
      "  Read 1:    17,992,560 bp\n",
      "  Read 2:    12,093,360 bp\n",
      "Total written (filtered):     24,574,036 bp (81.7%)\n",
      "  Read 1:    14,999,414 bp\n",
      "  Read 2:     9,574,622 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 57341 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.2\t0\t1\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t6\t0.0\t0\t6\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t10\t0.0\t0\t10\n",
      "18\t8921\t0.0\t0\t8921\n",
      "19\t10341\t0.0\t0\t10341\n",
      "20\t10666\t0.0\t0\t10666\n",
      "21\t9376\t0.0\t0\t9376\n",
      "22\t9759\t0.0\t0\t9759\n",
      "23\t8248\t0.0\t0\t8248\n",
      "24\t6\t0.0\t0\t6\n",
      "40\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 54208 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "20\t18\t0.0\t0\t18\n",
      "21\t10166\t0.0\t0\t10166\n",
      "22\t9802\t0.0\t0\t9802\n",
      "23\t8793\t0.0\t0\t8793\n",
      "24\t9258\t0.0\t0\t9258\n",
      "25\t8219\t0.0\t0\t8219\n",
      "26\t7938\t0.0\t0\t7938\n",
      "27\t6\t0.0\t0\t6\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121695_44_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121695_45_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121695_44_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121695_45_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121695_44_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121695_45_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121695_44_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121695_45_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.28 s (10 µs/read; 6.04 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            229,809\n",
      "  Read 1 with adapter:                 223,555 (97.3%)\n",
      "  Read 2 with adapter:                 212,081 (92.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           23,391 (10.2%)\n",
      "Pairs written (passing filters):       206,418 (89.8%)\n",
      "\n",
      "Total basepairs processed:   117,202,590 bp\n",
      "  Read 1:    70,091,745 bp\n",
      "  Read 2:    47,110,845 bp\n",
      "Total written (filtered):     96,229,804 bp (82.1%)\n",
      "  Read 1:    58,736,074 bp\n",
      "  Read 2:    37,493,730 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 223555 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t3.5\t0\t2\n",
      "9\t4\t0.9\t0\t4\n",
      "10\t3\t0.2\t0\t3\n",
      "11\t6\t0.1\t0\t6\n",
      "12\t6\t0.0\t0\t6\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t14\t0.0\t0\t14\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t64\t0.0\t0\t64\n",
      "18\t34673\t0.0\t0\t34673\n",
      "19\t40049\t0.0\t0\t40049\n",
      "20\t41294\t0.0\t0\t41294\n",
      "21\t37301\t0.0\t0\t37301\n",
      "22\t38005\t0.0\t0\t38005\n",
      "23\t32087\t0.0\t0\t32087\n",
      "24\t27\t0.0\t0\t27\n",
      "26\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "35\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 212081 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t4\t0.0\t0\t4\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t63\t0.0\t0\t63\n",
      "21\t39586\t0.0\t0\t39586\n",
      "22\t38305\t0.0\t0\t38305\n",
      "23\t34643\t0.0\t0\t34643\n",
      "24\t36017\t0.0\t0\t36017\n",
      "25\t32350\t0.0\t0\t32350\n",
      "26\t31075\t0.0\t0\t31075\n",
      "27\t26\t0.0\t0\t26\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121696_46_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121696_47_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121696_46_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121696_47_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121696_46_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121696_47_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121696_46_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121696_47_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.25 s (11 µs/read; 5.52 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            114,634\n",
      "  Read 1 with adapter:                 111,413 (97.2%)\n",
      "  Read 2 with adapter:                 104,800 (91.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,724 (11.1%)\n",
      "Pairs written (passing filters):       101,910 (88.9%)\n",
      "\n",
      "Total basepairs processed:    58,463,340 bp\n",
      "  Read 1:    34,963,370 bp\n",
      "  Read 2:    23,499,970 bp\n",
      "Total written (filtered):     47,509,206 bp (81.3%)\n",
      "  Read 1:    28,997,978 bp\n",
      "  Read 2:    18,511,228 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 111413 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.7\t0\t1\n",
      "10\t4\t0.1\t0\t4\n",
      "11\t5\t0.0\t0\t5\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t5\t0.0\t0\t5\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t33\t0.0\t0\t33\n",
      "18\t17408\t0.0\t0\t17408\n",
      "19\t19914\t0.0\t0\t19914\n",
      "20\t20315\t0.0\t0\t20315\n",
      "21\t18493\t0.0\t0\t18493\n",
      "22\t18881\t0.0\t0\t18881\n",
      "23\t16324\t0.0\t0\t16324\n",
      "24\t19\t0.0\t0\t19\n",
      "28\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 104800 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.7\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t38\t0.0\t0\t38\n",
      "21\t19666\t0.0\t0\t19666\n",
      "22\t18723\t0.0\t0\t18723\n",
      "23\t17184\t0.0\t0\t17184\n",
      "24\t18017\t0.0\t0\t18017\n",
      "25\t15803\t0.0\t0\t15803\n",
      "26\t15339\t0.0\t0\t15339\n",
      "27\t17\t0.0\t0\t17\n",
      "49\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121697_48_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121697_49_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121697_48_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121697_49_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121697_48_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-8_hk7xjb/S210421121697_49_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121697_48_L001_R1_001.fastq.gz /tmp/qiime2-archive-a3dl5c10/9bc974da-6c1a-46b4-848f-a51513cc37e7/data/S210421121697_49_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.67 s (14 µs/read; 4.41 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             49,335\n",
      "  Read 1 with adapter:                  47,930 (97.2%)\n",
      "  Read 2 with adapter:                  45,553 (92.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            5,055 (10.2%)\n",
      "Pairs written (passing filters):        44,280 (89.8%)\n",
      "\n",
      "Total basepairs processed:    25,160,850 bp\n",
      "  Read 1:    15,047,175 bp\n",
      "  Read 2:    10,113,675 bp\n",
      "Total written (filtered):     20,642,581 bp (82.0%)\n",
      "  Read 1:    12,599,485 bp\n",
      "  Read 2:     8,043,096 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 47930 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.2\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t16\t0.0\t0\t16\n",
      "18\t7269\t0.0\t0\t7269\n",
      "19\t8668\t0.0\t0\t8668\n",
      "20\t8944\t0.0\t0\t8944\n",
      "21\t7771\t0.0\t0\t7771\n",
      "22\t8230\t0.0\t0\t8230\n",
      "23\t7015\t0.0\t0\t7015\n",
      "24\t7\t0.0\t0\t7\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 45553 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "20\t14\t0.0\t0\t14\n",
      "21\t8529\t0.0\t0\t8529\n",
      "22\t8218\t0.0\t0\t8218\n",
      "23\t7506\t0.0\t0\t7506\n",
      "24\t7601\t0.0\t0\t7601\n",
      "25\t6998\t0.0\t0\t6998\n",
      "26\t6678\t0.0\t0\t6678\n",
      "27\t3\t0.0\t0\t3\n",
      "32\t1\t0.0\t0\t1\n"
     ]
    }
   ],
   "source": [
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')\n",
    "\n",
    "if trim and (not os.path.isfile(demux_file_trim) or replace_files):\n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = None\n",
    "    \n",
    "    if n_directions == 1:\n",
    "        res = trim_single(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front=forward_primer,\n",
    "            adapter=reverse_reverse_complement,\n",
    "            cores=threads,\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    elif n_directions == 2:\n",
    "        res = trim_paired(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front_f=forward_primer,\n",
    "            front_r=reverse_primer,\n",
    "            # adapter_f=reverse_reverse_complement,\n",
    "            # adapter_r=forward_reverse_complement,\n",
    "            cores=threads,\n",
    "            overlap=trim['overlap'],\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    else:\n",
    "        print(f'ERROR: invalid number of directions {n_directions}')\n",
    "\n",
    "    if res is not None:\n",
    "        res.save(demux_file_trim)\n",
    "        Visualization.save(demux.visualizers.summarize(res).visualization, filepath=demux_view_trim)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 173.048409,
   "end_time": "2023-08-28T14:27:26.337200",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-templates/step-prepare-data.ipynb",
   "output_path": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-NCxSTD-NC-trim/nb-executed-steps/step-prepare-data-ana-flavia-NCxSTD-NC-trim.ipynb",
   "parameters": {
    "base_dir": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri",
    "class_col": "group-id",
    "classifier_file": "/home/lauro/nupeb/rede-micro/models/silva-138-99-nb-classifier.qza",
    "experiment_name": "ana-flavia-NCxSTD-NC-trim",
    "manifest_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-NCxSTD-NC.csv",
    "metadata_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-NCxSTD-NC.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "top_n": 20,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2023-08-28T14:24:33.288791",
   "version": "2.3.4"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}