{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "10095583",
   "metadata": {
    "papermill": {
     "duration": 0.017309,
     "end_time": "2023-08-23T18:37:39.963968",
     "exception": false,
     "start_time": "2023-08-23T18:37:39.946659",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c408f185",
   "metadata": {
    "papermill": {
     "duration": 0.038771,
     "end_time": "2023-08-23T18:37:40.090111",
     "exception": false,
     "start_time": "2023-08-23T18:37:40.051340",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3247d913",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-10-14T14:33:37.863523Z",
     "start_time": "2022-10-14T14:33:14.145712Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:37:40.138547Z",
     "iopub.status.busy": "2023-08-23T18:37:40.138044Z",
     "iopub.status.idle": "2023-08-23T18:37:46.212941Z",
     "shell.execute_reply": "2023-08-23T18:37:46.212102Z"
    },
    "papermill": {
     "duration": 6.114915,
     "end_time": "2023-08-23T18:37:46.216012",
     "exception": false,
     "start_time": "2023-08-23T18:37:40.101097",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization, Metadata\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired, trim_single\n",
    "from qiime2.plugins.demux.methods import filter_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "558c8f5c",
   "metadata": {
    "papermill": {
     "duration": 0.009587,
     "end_time": "2023-08-23T18:37:46.243146",
     "exception": false,
     "start_time": "2023-08-23T18:37:46.233559",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "18902ecf",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.594341Z",
     "start_time": "2022-06-28T18:50:41.591544Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:37:46.352499Z",
     "iopub.status.busy": "2023-08-23T18:37:46.351899Z",
     "iopub.status.idle": "2023-08-23T18:37:46.359195Z",
     "shell.execute_reply": "2023-08-23T18:37:46.358238Z"
    },
    "papermill": {
     "duration": 0.082973,
     "end_time": "2023-08-23T18:37:46.360891",
     "exception": false,
     "start_time": "2023-08-23T18:37:46.277918",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'parameter-set-01.yaml')\n",
    "experiment_name = 'exp-01'\n",
    "base_dir = os.path.join('/', 'home', 'username', 'pipeline-dir')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'manifest.csv')\n",
    "replace_files = False\n",
    "trim = None\n",
    "metadata_file = None\n",
    "threads = 1\n",
    "demux_file = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "bca9f37f",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T18:37:46.385183Z",
     "iopub.status.busy": "2023-08-23T18:37:46.384892Z",
     "iopub.status.idle": "2023-08-23T18:37:46.390401Z",
     "shell.execute_reply": "2023-08-23T18:37:46.389533Z"
    },
    "papermill": {
     "duration": 0.018173,
     "end_time": "2023-08-23T18:37:46.392039",
     "exception": false,
     "start_time": "2023-08-23T18:37:46.373866",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "experiment_name = \"ana-flavia-NCxNR-trim\"\n",
    "base_dir = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri\"\n",
    "manifest_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-NCxNR.csv\"\n",
    "metadata_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-NCxNR.tsv\"\n",
    "class_col = \"group-id\"\n",
    "classifier_file = \"/home/lauro/nupeb/rede-micro/models/silva-138-99-nb-classifier.qza\"\n",
    "top_n = 20\n",
    "replace_files = False\n",
    "phred = 20\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n",
    "overlap = 12\n",
    "threads = 6\n",
    "trim = {\n",
    "    \"overlap\": 8,\n",
    "    \"forward_primer\": \"CCTACGGGRSGCAGCAG\",\n",
    "    \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\",\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8884239a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-18T15:40:31.070885Z",
     "start_time": "2023-08-18T15:40:30.611671Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:37:46.415271Z",
     "iopub.status.busy": "2023-08-23T18:37:46.414818Z",
     "iopub.status.idle": "2023-08-23T18:37:46.421311Z",
     "shell.execute_reply": "2023-08-23T18:37:46.420345Z"
    },
    "papermill": {
     "duration": 0.020039,
     "end_time": "2023-08-23T18:37:46.423125",
     "exception": false,
     "start_time": "2023-08-23T18:37:46.403086",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "data_dir =  os.path.abspath(os.path.join(base_dir, 'data'))\n",
    "raw_data_dir =  os.path.abspath(os.path.join(data_dir, 'raw'))\n",
    "interim_data_dir =  os.path.abspath(os.path.join(data_dir, 'interim'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9b904af",
   "metadata": {
    "papermill": {
     "duration": 0.010316,
     "end_time": "2023-08-23T18:37:46.446410",
     "exception": false,
     "start_time": "2023-08-23T18:37:46.436094",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5b8dd0f5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.632125Z",
     "start_time": "2022-06-28T18:50:41.619710Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:37:46.468267Z",
     "iopub.status.busy": "2023-08-23T18:37:46.468047Z",
     "iopub.status.idle": "2023-08-23T18:37:46.473590Z",
     "shell.execute_reply": "2023-08-23T18:37:46.473054Z"
    },
    "papermill": {
     "duration": 0.018377,
     "end_time": "2023-08-23T18:37:46.475036",
     "exception": false,
     "start_time": "2023-08-23T18:37:46.456659",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New qiime-artifacts folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-NCxNR-trim/qiime-artifacts\n",
      "New img folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-NCxNR-trim/imgs\n"
     ]
    }
   ],
   "source": [
    "# Create path if it not exists\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exists\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')\n",
    "\n",
    "# Define the output artifact full path\n",
    "if demux_file is None:\n",
    "    demux_file = os.path.join(out_dir, 'demux-paired.qza')\n",
    "demux_view = os.path.join(out_dir, 'demux-paired.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "876c74c6",
   "metadata": {
    "papermill": {
     "duration": 0.010512,
     "end_time": "2023-08-23T18:37:46.496336",
     "exception": false,
     "start_time": "2023-08-23T18:37:46.485824",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file of `SampleData[PairedEndSequencesWithQuality]`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ffcbe755",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T18:37:46.525668Z",
     "iopub.status.busy": "2023-08-23T18:37:46.525459Z",
     "iopub.status.idle": "2023-08-23T18:37:46.539242Z",
     "shell.execute_reply": "2023-08-23T18:37:46.538645Z"
    },
    "papermill": {
     "duration": 0.034348,
     "end_time": "2023-08-23T18:37:46.541126",
     "exception": false,
     "start_time": "2023-08-23T18:37:46.506778",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "manifest_df = pd.read_csv(manifest_file)\n",
    "n_directions = len(manifest_df['direction'].unique())\n",
    "if n_directions == 1:\n",
    "    d_type = 'SampleData[SequencesWithQuality]'\n",
    "    v_type = 'SingleEndFastqManifestPhred33'\n",
    "elif n_directions == 2:\n",
    "    d_type = 'SampleData[PairedEndSequencesWithQuality]'\n",
    "    v_type = 'PairedEndFastqManifestPhred33'\n",
    "else:\n",
    "    print(f'ERROR: invalid number of directions {n_directions}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a218f924",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T18:37:46.565542Z",
     "iopub.status.busy": "2023-08-23T18:37:46.565186Z",
     "iopub.status.idle": "2023-08-23T18:37:46.595089Z",
     "shell.execute_reply": "2023-08-23T18:37:46.594285Z"
    },
    "papermill": {
     "duration": 0.04351,
     "end_time": "2023-08-23T18:37:46.597543",
     "exception": false,
     "start_time": "2023-08-23T18:37:46.554033",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the metadata file is not defined, use the default metadata file\n",
    "metadata_qa = Metadata.load(metadata_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "733232e5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:44:58.640338Z",
     "start_time": "2022-06-28T18:50:41.633657Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:37:46.626039Z",
     "iopub.status.busy": "2023-08-23T18:37:46.625718Z",
     "iopub.status.idle": "2023-08-23T18:39:11.571664Z",
     "shell.execute_reply": "2023-08-23T18:39:11.570783Z"
    },
    "papermill": {
     "duration": 84.959297,
     "end_time": "2023-08-23T18:39:11.573563",
     "exception": false,
     "start_time": "2023-08-23T18:37:46.614266",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    interim_demux_path = os.path.join(interim_data_dir, \"demux-paired.qza\")\n",
    "    # Load Artifact from interim folder\n",
    "    if os.path.isfile(interim_demux_path):\n",
    "        artifact = Artifact.load(interim_demux_path)\n",
    "    # Create new Artifact using Manifest\n",
    "    else:\n",
    "        artifact = Artifact.import_data(d_type, manifest_file, view_type=v_type)\n",
    "        if not os.path.isfile(interim_demux_path):\n",
    "            artifact.save(interim_demux_path)\n",
    "    # Filter and Save Artifact\n",
    "    artifact = filter_samples(demux=artifact, metadata=metadata_qa).filtered_demux\n",
    "    artifact.save(demux_file)\n",
    "    # Visualization\n",
    "    if os.path.isfile(demux_view):\n",
    "        demux_view_obj = Visualization.load(demux_view)\n",
    "    else:\n",
    "        demux_view_obj = demux.visualizers.summarize(artifact).visualization\n",
    "        Visualization.save(demux_view_obj, filepath=demux_view)\n",
    "elif os.path.isfile(demux_file):\n",
    "    # Load Artifact\n",
    "    artifact = Artifact.load(demux_file)\n",
    "    artifact = filter_samples(demux=artifact, metadata=metadata_qa).filtered_demux\n",
    "    # Visualization\n",
    "    if os.path.isfile(demux_view):\n",
    "        demux_view_obj = Visualization.load(demux_view)\n",
    "    else:\n",
    "        demux_view_obj = demux.visualizers.summarize(artifact).visualization\n",
    "        Visualization.save(demux_view_obj, filepath=demux_view)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "795ce15a",
   "metadata": {
    "papermill": {
     "duration": 0.016799,
     "end_time": "2023-08-23T18:39:11.601635",
     "exception": false,
     "start_time": "2023-08-23T18:39:11.584836",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0a03c87a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.609737Z",
     "start_time": "2022-06-28T19:47:56.604820Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:39:11.624134Z",
     "iopub.status.busy": "2023-08-23T18:39:11.623839Z",
     "iopub.status.idle": "2023-08-23T18:39:11.628052Z",
     "shell.execute_reply": "2023-08-23T18:39:11.627440Z"
    },
    "papermill": {
     "duration": 0.017395,
     "end_time": "2023-08-23T18:39:11.629632",
     "exception": false,
     "start_time": "2023-08-23T18:39:11.612237",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<visualization: Visualization uuid: e0d2ff17-7f5b-404f-a4fe-5f1ecd878f3b>\n"
     ]
    }
   ],
   "source": [
    "print(demux_view_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "526ebb45",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.704841Z",
     "start_time": "2022-06-28T19:47:56.611846Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:39:11.652983Z",
     "iopub.status.busy": "2023-08-23T18:39:11.652716Z",
     "iopub.status.idle": "2023-08-23T18:39:11.687939Z",
     "shell.execute_reply": "2023-08-23T18:39:11.687065Z"
    },
    "papermill": {
     "duration": 0.048706,
     "end_time": "2023-08-23T18:39:11.689571",
     "exception": false,
     "start_time": "2023-08-23T18:39:11.640865",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-m1t08jek')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: e0d2ff17-7f5b-404f-a4fe-5f1ecd878f3b>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "31995c03",
   "metadata": {
    "papermill": {
     "duration": 0.011794,
     "end_time": "2023-08-23T18:39:11.714561",
     "exception": false,
     "start_time": "2023-08-23T18:39:11.702767",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c010560c",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T18:39:11.741398Z",
     "iopub.status.busy": "2023-08-23T18:39:11.741129Z",
     "iopub.status.idle": "2023-08-23T18:40:41.744826Z",
     "shell.execute_reply": "2023-08-23T18:40:41.743947Z"
    },
    "papermill": {
     "duration": 90.021552,
     "end_time": "2023-08-23T18:40:41.747923",
     "exception": false,
     "start_time": "2023-08-23T18:39:11.726371",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121673_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121673_1_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121673_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121673_1_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121673_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121673_1_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121673_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121673_1_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.78 s (10 µs/read; 5.87 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            370,042\n",
      "  Read 1 with adapter:                 359,997 (97.3%)\n",
      "  Read 2 with adapter:                 343,009 (92.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           36,262 (9.8%)\n",
      "Pairs written (passing filters):       333,780 (90.2%)\n",
      "\n",
      "Total basepairs processed:   188,721,420 bp\n",
      "  Read 1:   112,862,810 bp\n",
      "  Read 2:    75,858,610 bp\n",
      "Total written (filtered):    155,609,062 bp (82.5%)\n",
      "  Read 1:    94,978,565 bp\n",
      "  Read 2:    60,630,497 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 359997 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t5.6\t0\t3\n",
      "9\t4\t1.4\t0\t4\n",
      "10\t3\t0.4\t0\t3\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t4\t0.0\t0\t4\n",
      "14\t12\t0.0\t0\t12\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t19\t0.0\t0\t19\n",
      "17\t109\t0.0\t0\t109\n",
      "18\t56158\t0.0\t0\t56158\n",
      "19\t64949\t0.0\t0\t64949\n",
      "20\t66248\t0.0\t0\t66248\n",
      "21\t59860\t0.0\t0\t59860\n",
      "22\t60699\t0.0\t0\t60699\n",
      "23\t51867\t0.0\t0\t51867\n",
      "24\t42\t0.0\t0\t42\n",
      "25\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 343009 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.4\t0\t1\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t1\t0.0\t0\t1\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t14\t0.0\t0\t14\n",
      "20\t137\t0.0\t0\t137\n",
      "21\t64389\t0.0\t0\t64389\n",
      "22\t61501\t0.0\t0\t61501\n",
      "23\t56595\t0.0\t0\t56595\n",
      "24\t58584\t0.0\t0\t58584\n",
      "25\t52061\t0.0\t0\t52061\n",
      "26\t49654\t0.0\t0\t49654\n",
      "27\t42\t0.0\t0\t42\n",
      "29\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "37\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121674_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121674_3_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121674_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121674_3_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121674_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121674_3_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121674_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121674_3_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 5.08 s (9 µs/read; 6.61 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            559,807\n",
      "  Read 1 with adapter:                 544,656 (97.3%)\n",
      "  Read 2 with adapter:                 519,894 (92.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           53,686 (9.6%)\n",
      "Pairs written (passing filters):       506,121 (90.4%)\n",
      "\n",
      "Total basepairs processed:   285,501,570 bp\n",
      "  Read 1:   170,741,135 bp\n",
      "  Read 2:   114,760,435 bp\n",
      "Total written (filtered):    235,957,214 bp (82.6%)\n",
      "  Read 1:   144,019,667 bp\n",
      "  Read 2:    91,937,547 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 544656 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t8.5\t0\t5\n",
      "9\t4\t2.1\t0\t4\n",
      "10\t3\t0.5\t0\t3\n",
      "11\t7\t0.1\t0\t7\n",
      "12\t7\t0.0\t0\t7\n",
      "13\t6\t0.0\t0\t6\n",
      "14\t17\t0.0\t0\t17\n",
      "15\t16\t0.0\t0\t16\n",
      "16\t22\t0.0\t0\t22\n",
      "17\t155\t0.0\t0\t155\n",
      "18\t85410\t0.0\t0\t85410\n",
      "19\t98095\t0.0\t0\t98095\n",
      "20\t100478\t0.0\t0\t100478\n",
      "21\t90125\t0.0\t0\t90125\n",
      "22\t91840\t0.0\t0\t91840\n",
      "23\t78402\t0.0\t0\t78402\n",
      "24\t55\t0.0\t0\t55\n",
      "26\t3\t0.0\t0\t3\n",
      "29\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "35\t1\t0.0\t0\t1\n",
      "42\t1\t0.0\t0\t1\n",
      "43\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 519894 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t5\t0.0\t0\t5\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t15\t0.0\t0\t15\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t9\t0.0\t0\t9\n",
      "18\t6\t0.0\t0\t6\n",
      "19\t17\t0.0\t0\t17\n",
      "20\t180\t0.0\t0\t180\n",
      "21\t98232\t0.0\t0\t98232\n",
      "22\t93652\t0.0\t0\t93652\n",
      "23\t85816\t0.0\t0\t85816\n",
      "24\t87651\t0.0\t0\t87651\n",
      "25\t78341\t0.0\t0\t78341\n",
      "26\t75895\t0.0\t0\t75895\n",
      "27\t62\t0.0\t0\t62\n",
      "36\t1\t0.0\t0\t1\n",
      "57\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121675_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121675_5_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121675_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121675_5_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121675_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121675_5_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121675_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121675_5_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.33 s (10 µs/read; 6.23 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            345,762\n",
      "  Read 1 with adapter:                 336,819 (97.4%)\n",
      "  Read 2 with adapter:                 321,022 (92.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           32,941 (9.5%)\n",
      "Pairs written (passing filters):       312,821 (90.5%)\n",
      "\n",
      "Total basepairs processed:   176,338,620 bp\n",
      "  Read 1:   105,457,410 bp\n",
      "  Read 2:    70,881,210 bp\n",
      "Total written (filtered):    145,841,282 bp (82.7%)\n",
      "  Read 1:    89,015,404 bp\n",
      "  Read 2:    56,825,878 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 336819 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t5.3\t0\t5\n",
      "9\t3\t1.3\t0\t3\n",
      "10\t5\t0.3\t0\t5\n",
      "11\t2\t0.1\t0\t2\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t5\t0.0\t0\t5\n",
      "14\t12\t0.0\t0\t12\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t11\t0.0\t0\t11\n",
      "17\t92\t0.0\t0\t92\n",
      "18\t52679\t0.0\t0\t52679\n",
      "19\t60788\t0.0\t0\t60788\n",
      "20\t62024\t0.0\t0\t62024\n",
      "21\t55828\t0.0\t0\t55828\n",
      "22\t57192\t0.0\t0\t57192\n",
      "23\t48121\t0.0\t0\t48121\n",
      "24\t34\t0.0\t0\t34\n",
      "25\t2\t0.0\t0\t2\n",
      "28\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "36\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 321022 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t3\t0.0\t0\t3\n",
      "15\t12\t0.0\t0\t12\n",
      "16\t4\t0.0\t0\t4\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t3\t0.0\t0\t3\n",
      "19\t11\t0.0\t0\t11\n",
      "20\t106\t0.0\t0\t106\n",
      "21\t60450\t0.0\t0\t60450\n",
      "22\t58128\t0.0\t0\t58128\n",
      "23\t53277\t0.0\t0\t53277\n",
      "24\t54587\t0.0\t0\t54587\n",
      "25\t48050\t0.0\t0\t48050\n",
      "26\t46353\t0.0\t0\t46353\n",
      "27\t34\t0.0\t0\t34\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121676_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121676_7_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121676_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121676_7_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121676_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121676_7_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121676_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121676_7_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 6.55 s (9 µs/read; 6.52 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            712,750\n",
      "  Read 1 with adapter:                 694,205 (97.4%)\n",
      "  Read 2 with adapter:                 664,033 (93.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           65,715 (9.2%)\n",
      "Pairs written (passing filters):       647,035 (90.8%)\n",
      "\n",
      "Total basepairs processed:   363,502,500 bp\n",
      "  Read 1:   217,388,750 bp\n",
      "  Read 2:   146,113,750 bp\n",
      "Total written (filtered):    301,650,677 bp (83.0%)\n",
      "  Read 1:   184,115,305 bp\n",
      "  Read 2:   117,535,372 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 694205 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t10.9\t0\t4\n",
      "9\t9\t2.7\t0\t9\n",
      "10\t5\t0.7\t0\t5\n",
      "11\t14\t0.2\t0\t14\n",
      "12\t8\t0.0\t0\t8\n",
      "13\t6\t0.0\t0\t6\n",
      "14\t30\t0.0\t0\t30\n",
      "15\t18\t0.0\t0\t18\n",
      "16\t33\t0.0\t0\t33\n",
      "17\t183\t0.0\t0\t183\n",
      "18\t108490\t0.0\t0\t108490\n",
      "19\t125122\t0.0\t0\t125122\n",
      "20\t127377\t0.0\t0\t127377\n",
      "21\t114840\t0.0\t0\t114840\n",
      "22\t117757\t0.0\t0\t117757\n",
      "23\t100233\t0.0\t0\t100233\n",
      "24\t68\t0.0\t0\t68\n",
      "25\t1\t0.0\t0\t1\n",
      "28\t2\t0.0\t0\t2\n",
      "35\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "41\t1\t0.0\t0\t1\n",
      "49\t1\t0.0\t0\t1\n",
      "57\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 664033 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.7\t0\t2\n",
      "11\t4\t0.2\t0\t4\n",
      "12\t5\t0.0\t0\t5\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t18\t0.0\t0\t18\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t4\t0.0\t0\t4\n",
      "18\t9\t0.0\t0\t9\n",
      "19\t16\t0.0\t0\t16\n",
      "20\t229\t0.0\t0\t229\n",
      "21\t124781\t0.0\t0\t124781\n",
      "22\t120299\t0.0\t0\t120299\n",
      "23\t109785\t0.0\t0\t109785\n",
      "24\t112135\t0.0\t0\t112135\n",
      "25\t100654\t0.0\t0\t100654\n",
      "26\t95988\t0.0\t0\t95988\n",
      "27\t87\t0.0\t0\t87\n",
      "28\t1\t0.0\t0\t1\n",
      "29\t1\t0.0\t0\t1\n",
      "39\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "47\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "56\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121677_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121677_9_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121677_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121677_9_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121677_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121677_9_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121677_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121677_9_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.44 s (9 µs/read; 6.38 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            365,427\n",
      "  Read 1 with adapter:                 355,998 (97.4%)\n",
      "  Read 2 with adapter:                 339,330 (92.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           34,713 (9.5%)\n",
      "Pairs written (passing filters):       330,714 (90.5%)\n",
      "\n",
      "Total basepairs processed:   186,367,770 bp\n",
      "  Read 1:   111,455,235 bp\n",
      "  Read 2:    74,912,535 bp\n",
      "Total written (filtered):    154,181,570 bp (82.7%)\n",
      "  Read 1:    94,107,785 bp\n",
      "  Read 2:    60,073,785 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 355998 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.6\t0\t1\n",
      "9\t2\t1.4\t0\t2\n",
      "10\t4\t0.3\t0\t4\n",
      "11\t4\t0.1\t0\t4\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t19\t0.0\t0\t19\n",
      "15\t9\t0.0\t0\t9\n",
      "16\t25\t0.0\t0\t25\n",
      "17\t99\t0.0\t0\t99\n",
      "18\t56045\t0.0\t0\t56045\n",
      "19\t64739\t0.0\t0\t64739\n",
      "20\t65079\t0.0\t0\t65079\n",
      "21\t58610\t0.0\t0\t58610\n",
      "22\t60073\t0.0\t0\t60073\n",
      "23\t51236\t0.0\t0\t51236\n",
      "24\t48\t0.0\t0\t48\n",
      "39\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 339330 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t1.4\t0\t1\n",
      "10\t1\t0.3\t0\t1\n",
      "12\t3\t0.0\t0\t3\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t16\t0.0\t0\t16\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t8\t0.0\t0\t8\n",
      "19\t9\t0.0\t0\t9\n",
      "20\t112\t0.0\t0\t112\n",
      "21\t63644\t0.0\t0\t63644\n",
      "22\t61206\t0.0\t0\t61206\n",
      "23\t56423\t0.0\t0\t56423\n",
      "24\t57151\t0.0\t0\t57151\n",
      "25\t51497\t0.0\t0\t51497\n",
      "26\t49209\t0.0\t0\t49209\n",
      "27\t38\t0.0\t0\t38\n",
      "28\t2\t0.0\t0\t2\n",
      "38\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "50\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121678_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121678_11_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121678_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121678_11_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121678_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121678_11_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121678_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121678_11_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.02 s (11 µs/read; 5.63 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             95,852\n",
      "  Read 1 with adapter:                  93,250 (97.3%)\n",
      "  Read 2 with adapter:                  88,539 (92.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            9,674 (10.1%)\n",
      "Pairs written (passing filters):        86,178 (89.9%)\n",
      "\n",
      "Total basepairs processed:    48,884,520 bp\n",
      "  Read 1:    29,234,860 bp\n",
      "  Read 2:    19,649,660 bp\n",
      "Total written (filtered):     40,178,039 bp (82.2%)\n",
      "  Read 1:    24,522,719 bp\n",
      "  Read 2:    15,655,320 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 93250 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t5\t0.4\t0\t5\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t5\t0.0\t0\t5\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t24\t0.0\t0\t24\n",
      "18\t14644\t0.0\t0\t14644\n",
      "19\t16889\t0.0\t0\t16889\n",
      "20\t17011\t0.0\t0\t17011\n",
      "21\t15408\t0.0\t0\t15408\n",
      "22\t16019\t0.0\t0\t16019\n",
      "23\t13225\t0.0\t0\t13225\n",
      "24\t9\t0.0\t0\t9\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 88539 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t34\t0.0\t0\t34\n",
      "21\t16889\t0.0\t0\t16889\n",
      "22\t15980\t0.0\t0\t15980\n",
      "23\t14709\t0.0\t0\t14709\n",
      "24\t14923\t0.0\t0\t14923\n",
      "25\t13235\t0.0\t0\t13235\n",
      "26\t12749\t0.0\t0\t12749\n",
      "27\t11\t0.0\t0\t11\n",
      "60\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121679_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121679_13_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121679_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121679_13_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121679_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121679_13_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121679_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121679_13_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.08 s (10 µs/read; 6.18 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            316,849\n",
      "  Read 1 with adapter:                 308,351 (97.3%)\n",
      "  Read 2 with adapter:                 292,785 (92.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           31,765 (10.0%)\n",
      "Pairs written (passing filters):       285,084 (90.0%)\n",
      "\n",
      "Total basepairs processed:   161,592,990 bp\n",
      "  Read 1:    96,638,945 bp\n",
      "  Read 2:    64,954,045 bp\n",
      "Total written (filtered):    132,905,965 bp (82.2%)\n",
      "  Read 1:    81,120,487 bp\n",
      "  Read 2:    51,785,478 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 308351 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t4.8\t0\t4\n",
      "9\t1\t1.2\t0\t1\n",
      "10\t1\t0.3\t0\t1\n",
      "11\t9\t0.1\t0\t9\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t5\t0.0\t0\t5\n",
      "14\t17\t0.0\t0\t17\n",
      "15\t20\t0.0\t0\t20\n",
      "16\t21\t0.0\t0\t21\n",
      "17\t83\t0.0\t0\t83\n",
      "18\t48009\t0.0\t0\t48009\n",
      "19\t55379\t0.0\t0\t55379\n",
      "20\t56655\t0.0\t0\t56655\n",
      "21\t51178\t0.0\t0\t51178\n",
      "22\t52436\t0.0\t0\t52436\n",
      "23\t44491\t0.0\t0\t44491\n",
      "24\t30\t0.0\t0\t30\n",
      "30\t1\t0.0\t0\t1\n",
      "31\t2\t0.0\t0\t2\n",
      "32\t1\t0.0\t0\t1\n",
      "41\t1\t0.0\t0\t1\n",
      "43\t1\t0.0\t0\t1\n",
      "60\t2\t0.0\t0\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 292785 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t3\t0.1\t0\t3\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t5\t0.0\t0\t5\n",
      "19\t16\t0.0\t0\t16\n",
      "20\t87\t0.0\t0\t87\n",
      "21\t54723\t0.0\t0\t54723\n",
      "22\t52812\t0.0\t0\t52812\n",
      "23\t48903\t0.0\t0\t48903\n",
      "24\t49680\t0.0\t0\t49680\n",
      "25\t44102\t0.0\t0\t44102\n",
      "26\t42389\t0.0\t0\t42389\n",
      "27\t46\t0.0\t0\t46\n",
      "30\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "40\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121680_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121680_15_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121680_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121680_15_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121680_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121680_15_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121680_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121680_15_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.51 s (10 µs/read; 5.93 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            248,647\n",
      "  Read 1 with adapter:                 241,815 (97.3%)\n",
      "  Read 2 with adapter:                 229,975 (92.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           24,848 (10.0%)\n",
      "Pairs written (passing filters):       223,799 (90.0%)\n",
      "\n",
      "Total basepairs processed:   126,809,970 bp\n",
      "  Read 1:    75,837,335 bp\n",
      "  Read 2:    50,972,635 bp\n",
      "Total written (filtered):    104,332,265 bp (82.3%)\n",
      "  Read 1:    63,680,774 bp\n",
      "  Read 2:    40,651,491 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 241815 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t3.8\t0\t3\n",
      "9\t1\t0.9\t0\t1\n",
      "10\t4\t0.2\t0\t4\n",
      "12\t5\t0.0\t0\t5\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t6\t0.0\t0\t6\n",
      "17\t69\t0.0\t0\t69\n",
      "18\t37474\t0.0\t0\t37474\n",
      "19\t43244\t0.0\t0\t43244\n",
      "20\t44420\t0.0\t0\t44420\n",
      "21\t40069\t0.0\t0\t40069\n",
      "22\t41569\t0.0\t0\t41569\n",
      "23\t34901\t0.0\t0\t34901\n",
      "24\t28\t0.0\t0\t28\n",
      "25\t2\t0.0\t0\t2\n",
      "27\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 229975 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t4\t0.0\t0\t4\n",
      "18\t5\t0.0\t0\t5\n",
      "19\t6\t0.0\t0\t6\n",
      "20\t74\t0.0\t0\t74\n",
      "21\t42774\t0.0\t0\t42774\n",
      "22\t41717\t0.0\t0\t41717\n",
      "23\t37843\t0.0\t0\t37843\n",
      "24\t38947\t0.0\t0\t38947\n",
      "25\t35038\t0.0\t0\t35038\n",
      "26\t33531\t0.0\t0\t33531\n",
      "27\t22\t0.0\t0\t22\n",
      "28\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121681_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121681_17_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121681_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121681_17_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121681_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-m94t6ggh/S210421121681_17_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121681_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-rf1lscat/cef1f2a8-bac6-49b1-b295-0e47d8e89e18/data/S210421121681_17_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.05 s (10 µs/read; 6.07 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            207,454\n",
      "  Read 1 with adapter:                 201,975 (97.4%)\n",
      "  Read 2 with adapter:                 192,058 (92.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           20,423 (9.8%)\n",
      "Pairs written (passing filters):       187,031 (90.2%)\n",
      "\n",
      "Total basepairs processed:   105,801,540 bp\n",
      "  Read 1:    63,273,470 bp\n",
      "  Read 2:    42,528,070 bp\n",
      "Total written (filtered):     87,193,544 bp (82.4%)\n",
      "  Read 1:    53,220,325 bp\n",
      "  Read 2:    33,973,219 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 201975 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t3.2\t0\t3\n",
      "9\t3\t0.8\t0\t3\n",
      "10\t2\t0.2\t0\t2\n",
      "11\t3\t0.0\t0\t3\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t11\t0.0\t0\t11\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t8\t0.0\t0\t8\n",
      "17\t60\t0.0\t0\t60\n",
      "18\t31587\t0.0\t0\t31587\n",
      "19\t36316\t0.0\t0\t36316\n",
      "20\t37050\t0.0\t0\t37050\n",
      "21\t33459\t0.0\t0\t33459\n",
      "22\t34304\t0.0\t0\t34304\n",
      "23\t29134\t0.0\t0\t29134\n",
      "24\t19\t0.0\t0\t19\n",
      "25\t2\t0.0\t0\t2\n",
      "31\t1\t0.0\t0\t1\n",
      "34\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 192058 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t5\t0.0\t0\t5\n",
      "19\t7\t0.0\t0\t7\n",
      "20\t54\t0.0\t0\t54\n",
      "21\t35775\t0.0\t0\t35775\n",
      "22\t34503\t0.0\t0\t34503\n",
      "23\t32154\t0.0\t0\t32154\n",
      "24\t32732\t0.0\t0\t32732\n",
      "25\t28715\t0.0\t0\t28715\n",
      "26\t28079\t0.0\t0\t28079\n",
      "27\t22\t0.0\t0\t22\n",
      "32\t1\t0.0\t0\t1\n"
     ]
    }
   ],
   "source": [
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')\n",
    "\n",
    "if trim and (not os.path.isfile(demux_file_trim) or replace_files):\n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = None\n",
    "    \n",
    "    if n_directions == 1:\n",
    "        res = trim_single(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front=forward_primer,\n",
    "            adapter=reverse_reverse_complement,\n",
    "            cores=threads,\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    elif n_directions == 2:\n",
    "        res = trim_paired(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front_f=forward_primer,\n",
    "            front_r=reverse_primer,\n",
    "            # adapter_f=reverse_reverse_complement,\n",
    "            # adapter_r=forward_reverse_complement,\n",
    "            cores=threads,\n",
    "            overlap=trim['overlap'],\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    else:\n",
    "        print(f'ERROR: invalid number of directions {n_directions}')\n",
    "\n",
    "    if res is not None:\n",
    "        res.save(demux_file_trim)\n",
    "        Visualization.save(demux.visualizers.summarize(res).visualization, filepath=demux_view_trim)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 184.430156,
   "end_time": "2023-08-23T18:40:43.090537",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-templates/step-prepare-data.ipynb",
   "output_path": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-NCxNR-trim/nb-executed-steps/step-prepare-data-ana-flavia-NCxNR-trim.ipynb",
   "parameters": {
    "base_dir": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri",
    "class_col": "group-id",
    "classifier_file": "/home/lauro/nupeb/rede-micro/models/silva-138-99-nb-classifier.qza",
    "experiment_name": "ana-flavia-NCxNR-trim",
    "manifest_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-NCxNR.csv",
    "metadata_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-NCxNR.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "top_n": 20,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2023-08-23T18:37:38.660381",
   "version": "2.3.4"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}