{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "1998ce85",
   "metadata": {
    "papermill": {
     "duration": 0.019845,
     "end_time": "2022-09-30T10:16:10.873354",
     "exception": false,
     "start_time": "2022-09-30T10:16:10.853509",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "99a8ab19",
   "metadata": {
    "papermill": {
     "duration": 0.071265,
     "end_time": "2022-09-30T10:16:10.956461",
     "exception": false,
     "start_time": "2022-09-30T10:16:10.885196",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b2c62028",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-14T17:43:20.516262Z",
     "start_time": "2022-09-14T17:43:20.510424Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-30T10:16:10.981045Z",
     "iopub.status.busy": "2022-09-30T10:16:10.980656Z",
     "iopub.status.idle": "2022-09-30T10:16:17.508568Z",
     "shell.execute_reply": "2022-09-30T10:16:17.507761Z"
    },
    "papermill": {
     "duration": 6.54399,
     "end_time": "2022-09-30T10:16:17.511893",
     "exception": false,
     "start_time": "2022-09-30T10:16:10.967903",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired\n",
    "\n",
    "from utils import *"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08516f1d",
   "metadata": {
    "papermill": {
     "duration": 0.010491,
     "end_time": "2022-09-30T10:16:17.539396",
     "exception": false,
     "start_time": "2022-09-30T10:16:17.528905",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f59cefa0",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.594341Z",
     "start_time": "2022-06-28T18:50:41.591544Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-30T10:16:17.588847Z",
     "iopub.status.busy": "2022-09-30T10:16:17.588551Z",
     "iopub.status.idle": "2022-09-30T10:16:17.594814Z",
     "shell.execute_reply": "2022-09-30T10:16:17.594013Z"
    },
    "papermill": {
     "duration": 0.019413,
     "end_time": "2022-09-30T10:16:17.596402",
     "exception": false,
     "start_time": "2022-09-30T10:16:17.576989",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'ana-flavia-superlactacao.yaml')\n",
    "experiment_name = 'jenneffer-vs-01'\n",
    "base_dir = os.path.join('/', 'home', 'lauro', 'nupeb', 'redemicro')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'not-hist-vs-manifest.csv')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'imgs'))\n",
    "replace_files = False\n",
    "trim = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5959d3ac",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-30T10:16:17.627811Z",
     "iopub.status.busy": "2022-09-30T10:16:17.627504Z",
     "iopub.status.idle": "2022-09-30T10:16:17.634310Z",
     "shell.execute_reply": "2022-09-30T10:16:17.633440Z"
    },
    "papermill": {
     "duration": 0.02706,
     "end_time": "2022-09-30T10:16:17.636012",
     "exception": false,
     "start_time": "2022-09-30T10:16:17.608952",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "experiment_name = \"ana-flavia-NCxNR-trim\"\n",
    "base_dir = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri\"\n",
    "manifest_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-NCxNR.csv\"\n",
    "metadata_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-NCxNR.tsv\"\n",
    "class_col = \"group-id\"\n",
    "classifier_file = \"/home/lauro/nupeb/dados_brutos_rede_genoma/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza\"\n",
    "replace_files = False\n",
    "phred = 20\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n",
    "overlap = 12\n",
    "threads = 6\n",
    "trim = {\n",
    "    \"overlap\": 8,\n",
    "    \"forward_primer\": \"CCTACGGGRSGCAGCAG\",\n",
    "    \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\",\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bf4cf2c9",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.610682Z",
     "start_time": "2022-06-28T18:50:41.595886Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-30T10:16:17.661224Z",
     "iopub.status.busy": "2022-09-30T10:16:17.660894Z",
     "iopub.status.idle": "2022-09-30T10:16:17.667518Z",
     "shell.execute_reply": "2022-09-30T10:16:17.666661Z"
    },
    "papermill": {
     "duration": 0.020241,
     "end_time": "2022-09-30T10:16:17.669181",
     "exception": false,
     "start_time": "2022-09-30T10:16:17.648940",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Setting hardcoded parameters for development and debbuging\n",
    "production = True\n",
    "if not production:\n",
    "    params_path = os.path.join('..', 'params', 'ana-flavia-hipotese-01')\n",
    "    with open(params_path, 'r') as stream:\n",
    "        params = yaml.safe_load(stream)\n",
    "        experiment_name = params['experiment_name']\n",
    "        base_dir = params['base_dir']\n",
    "        manifest_file = params['manifest_file']\n",
    "        replace_files = params['replace_files']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a433ca9c",
   "metadata": {
    "papermill": {
     "duration": 0.011693,
     "end_time": "2022-09-30T10:16:17.695062",
     "exception": false,
     "start_time": "2022-09-30T10:16:17.683369",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c1adb9e8",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.618098Z",
     "start_time": "2022-06-28T18:50:41.614152Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-30T10:16:17.717938Z",
     "iopub.status.busy": "2022-09-30T10:16:17.717582Z",
     "iopub.status.idle": "2022-09-30T10:16:17.722279Z",
     "shell.execute_reply": "2022-09-30T10:16:17.721397Z"
    },
    "papermill": {
     "duration": 0.018197,
     "end_time": "2022-09-30T10:16:17.723967",
     "exception": false,
     "start_time": "2022-09-30T10:16:17.705770",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# new_manifest = '/home/lauro/nupeb/redemicro/data/raw/manifest/karina-manifest.csv'\n",
    "# with open(manifest_file, 'r') as oldm, open(new_manifest, 'w') as newm:\n",
    "#     header = 'sample-id,absolute-filepath,direction\\n'\n",
    "#     newm.write(header)\n",
    "#     for line in oldm.readlines()[1:]:\n",
    "#         sid, forward, reverse = line[:-1].split(',')\n",
    "#         fline = ','.join((sid, forward, 'forward')) + '\\n'\n",
    "#         rline = ','.join((sid, reverse, 'reverse')) + '\\n'\n",
    "#         newm.write(fline)\n",
    "#         newm.write(rline)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ffc195f5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.632125Z",
     "start_time": "2022-06-28T18:50:41.619710Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-30T10:16:17.747293Z",
     "iopub.status.busy": "2022-09-30T10:16:17.746941Z",
     "iopub.status.idle": "2022-09-30T10:16:17.756561Z",
     "shell.execute_reply": "2022-09-30T10:16:17.755670Z"
    },
    "papermill": {
     "duration": 0.023768,
     "end_time": "2022-09-30T10:16:17.758539",
     "exception": false,
     "start_time": "2022-09-30T10:16:17.734771",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Define the output folder path\n",
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "\n",
    "# Create path if it not exist\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exist\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')\n",
    "\n",
    "# Define the output artifact full path\n",
    "demux_file = os.path.join(out_dir, 'demux-paired.qza')\n",
    "demux_view = os.path.join(out_dir, 'demux-paired.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60f2a5dc",
   "metadata": {
    "papermill": {
     "duration": 0.010714,
     "end_time": "2022-09-30T10:16:17.782032",
     "exception": false,
     "start_time": "2022-09-30T10:16:17.771318",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "6c27752b",
   "metadata": {
    "papermill": {
     "duration": 0.01437,
     "end_time": "2022-09-30T10:16:17.807187",
     "exception": false,
     "start_time": "2022-09-30T10:16:17.792817",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file of `SampleData[PairedEndSequencesWithQuality]`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c492bad9",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:44:58.640338Z",
     "start_time": "2022-06-28T18:50:41.633657Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-30T10:16:17.839071Z",
     "iopub.status.busy": "2022-09-30T10:16:17.838889Z",
     "iopub.status.idle": "2022-09-30T10:16:20.432601Z",
     "shell.execute_reply": "2022-09-30T10:16:20.431878Z"
    },
    "papermill": {
     "duration": 2.613162,
     "end_time": "2022-09-30T10:16:20.435798",
     "exception": false,
     "start_time": "2022-09-30T10:16:17.822636",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    \n",
    "    # Import data and create an artifact object\n",
    "    artifact = Artifact.import_data(\n",
    "        'SampleData[PairedEndSequencesWithQuality]', \n",
    "        manifest_file, \n",
    "        view_type='PairedEndFastqManifestPhred33')\n",
    "    \n",
    "    # Save the artifact object to a new qza file\n",
    "    artifact.save(demux_file)\n",
    "\n",
    "else:\n",
    "    artifact = Artifact.load(demux_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4bf7db7d",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.602855Z",
     "start_time": "2022-06-28T19:44:58.642520Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-30T10:16:20.465228Z",
     "iopub.status.busy": "2022-09-30T10:16:20.465007Z",
     "iopub.status.idle": "2022-09-30T10:16:20.533260Z",
     "shell.execute_reply": "2022-09-30T10:16:20.532740Z"
    },
    "papermill": {
     "duration": 0.082953,
     "end_time": "2022-09-30T10:16:20.536040",
     "exception": false,
     "start_time": "2022-09-30T10:16:20.453087",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "if not os.path.isfile(demux_view) or replace_files: \n",
    "    # Generate e visualization of the Artifact\n",
    "    demux_summary = demux.visualizers.summarize(artifact)\n",
    "\n",
    "    # Save a new visualization file based on the qza file\n",
    "    Visualization.save(demux_summary.visualization, filepath=demux_view)\n",
    "    \n",
    "    demux_view_obj = demux_summary.visualization\n",
    "else:\n",
    "    demux_view_obj = Visualization.load(demux_view)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a9ba063",
   "metadata": {
    "papermill": {
     "duration": 0.01136,
     "end_time": "2022-09-30T10:16:20.564732",
     "exception": false,
     "start_time": "2022-09-30T10:16:20.553372",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ab27a751",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.609737Z",
     "start_time": "2022-06-28T19:47:56.604820Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-30T10:16:20.592701Z",
     "iopub.status.busy": "2022-09-30T10:16:20.592444Z",
     "iopub.status.idle": "2022-09-30T10:16:20.598056Z",
     "shell.execute_reply": "2022-09-30T10:16:20.597071Z"
    },
    "papermill": {
     "duration": 0.023663,
     "end_time": "2022-09-30T10:16:20.599829",
     "exception": false,
     "start_time": "2022-09-30T10:16:20.576166",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<visualization: Visualization uuid: 64b9840f-fbb2-40f6-9010-834cea5a6ed5>\n"
     ]
    }
   ],
   "source": [
    "print(demux_view_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b8a8c026",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.704841Z",
     "start_time": "2022-06-28T19:47:56.611846Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-30T10:16:20.625972Z",
     "iopub.status.busy": "2022-09-30T10:16:20.625530Z",
     "iopub.status.idle": "2022-09-30T10:16:20.641205Z",
     "shell.execute_reply": "2022-09-30T10:16:20.640721Z"
    },
    "papermill": {
     "duration": 0.030393,
     "end_time": "2022-09-30T10:16:20.643117",
     "exception": false,
     "start_time": "2022-09-30T10:16:20.612724",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-3ywne2wi')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 64b9840f-fbb2-40f6-9010-834cea5a6ed5>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bda68174",
   "metadata": {
    "papermill": {
     "duration": 0.012054,
     "end_time": "2022-09-30T10:16:20.669646",
     "exception": false,
     "start_time": "2022-09-30T10:16:20.657592",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0386ea98",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-30T10:16:20.695213Z",
     "iopub.status.busy": "2022-09-30T10:16:20.694851Z",
     "iopub.status.idle": "2022-09-30T10:18:58.931966Z",
     "shell.execute_reply": "2022-09-30T10:18:58.931341Z"
    },
    "papermill": {
     "duration": 158.253203,
     "end_time": "2022-09-30T10:18:58.934840",
     "exception": false,
     "start_time": "2022-09-30T10:16:20.681637",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121673_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121673_1_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121673_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121673_1_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121673_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121673_1_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121673_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121673_1_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 5.47 s (15 µs/read; 4.06 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            370,042\n",
      "  Read 1 with adapter:                 366,460 (99.0%)\n",
      "  Read 2 with adapter:                 362,566 (98.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  10 (0.0%)\n",
      "Pairs discarded as untrimmed:           10,985 (3.0%)\n",
      "Pairs written (passing filters):       359,047 (97.0%)\n",
      "\n",
      "Total basepairs processed:   188,721,420 bp\n",
      "  Read 1:   112,862,810 bp\n",
      "  Read 2:    75,858,610 bp\n",
      "Total written (filtered):    167,383,121 bp (88.7%)\n",
      "  Read 1:   102,168,436 bp\n",
      "  Read 2:    65,214,685 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 39 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 38.5%\n",
      "  G: 61.5%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t7\t5.6\t1\t0 7\n",
      "9\t6\t1.4\t1\t0 6\n",
      "20\t1\t0.0\t3\t1\n",
      "22\t1\t0.0\t3\t1\n",
      "23\t4\t0.0\t3\t3 1\n",
      "24\t4\t0.0\t3\t4\n",
      "25\t5\t0.0\t3\t5\n",
      "26\t1\t0.0\t3\t1\n",
      "170\t1\t0.0\t3\t1\n",
      "214\t1\t0.0\t3\t1\n",
      "215\t1\t0.0\t3\t0 0 1\n",
      "219\t1\t0.0\t3\t0 0 1\n",
      "266\t1\t0.0\t3\t1\n",
      "275\t2\t0.0\t3\t2\n",
      "281\t1\t0.0\t3\t1\n",
      "282\t1\t0.0\t3\t1\n",
      "283\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 366421 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t5.6\t1\t3\n",
      "9\t6\t1.4\t1\t4 2\n",
      "10\t6\t0.4\t1\t3 3\n",
      "11\t9\t0.1\t1\t2 7\n",
      "12\t8\t0.0\t1\t4 4\n",
      "13\t7\t0.0\t1\t4 3\n",
      "14\t15\t0.0\t2\t12 1 2\n",
      "15\t14\t0.0\t2\t9 3 2\n",
      "16\t33\t0.0\t2\t19 7 7\n",
      "17\t211\t0.0\t2\t109 74 28\n",
      "18\t57200\t0.0\t2\t56152 1008 40\n",
      "19\t66208\t0.0\t2\t64944 1201 63\n",
      "20\t67345\t0.0\t2\t66242 1063 40\n",
      "21\t60893\t0.0\t2\t59858 983 52\n",
      "22\t61761\t0.0\t2\t60696 1000 65\n",
      "23\t52648\t0.0\t2\t51864 766 18\n",
      "24\t49\t0.0\t2\t42 3 4\n",
      "25\t1\t0.0\t2\t1\n",
      "28\t1\t0.0\t2\t1\n",
      "30\t1\t0.0\t2\t1\n",
      "45\t1\t0.0\t2\t1\n",
      "50\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 32 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 18.8%\n",
      "  G: 25.0%\n",
      "  T: 25.0%\n",
      "  none/other: 31.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.6\t1\t0 1\n",
      "12\t1\t0.0\t1\t0 1\n",
      "203\t9\t0.0\t2\t8 1\n",
      "204\t11\t0.0\t2\t10 1\n",
      "205\t10\t0.0\t2\t10\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 362534 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.4\t1\t1\n",
      "11\t2\t0.1\t1\t2\n",
      "12\t3\t0.0\t1\t3\n",
      "13\t1\t0.0\t1\t1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t13\t0.0\t2\t9 1 3\n",
      "16\t7\t0.0\t2\t6 0 1\n",
      "17\t7\t0.0\t2\t1 3 3\n",
      "18\t7\t0.0\t2\t3 1 3\n",
      "19\t27\t0.0\t2\t14 4 9\n",
      "20\t279\t0.0\t3\t137 82 35 25\n",
      "21\t67699\t0.0\t3\t64389 2431 589 290\n",
      "22\t64777\t0.0\t3\t61501 2388 595 293\n",
      "23\t59945\t0.0\t3\t56595 2433 617 300\n",
      "24\t61734\t0.0\t3\t58584 2300 558 292\n",
      "25\t55204\t0.0\t3\t52061 2278 544 321\n",
      "26\t52754\t0.0\t3\t49654 2268 562 270\n",
      "27\t54\t0.0\t3\t42 8 1 3\n",
      "29\t1\t0.0\t3\t1\n",
      "30\t1\t0.0\t3\t1\n",
      "32\t1\t0.0\t3\t0 0 0 1\n",
      "37\t1\t0.0\t3\t1\n",
      "39\t1\t0.0\t3\t1\n",
      "45\t1\t0.0\t3\t1\n",
      "46\t1\t0.0\t3\t0 0 0 1\n",
      "50\t1\t0.0\t3\t0 0 0 1\n",
      "58\t1\t0.0\t3\t0 0 0 1\n",
      "60\t5\t0.0\t3\t0 0 0 5\n",
      "61\t1\t0.0\t3\t0 0 1\n",
      "62\t2\t0.0\t3\t0 0 1 1\n",
      "63\t1\t0.0\t3\t0 0 1\n",
      "65\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121674_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121674_3_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121674_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121674_3_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121674_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121674_3_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121674_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121674_3_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 7.65 s (14 µs/read; 4.39 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            559,807\n",
      "  Read 1 with adapter:                 554,474 (99.0%)\n",
      "  Read 2 with adapter:                 548,879 (98.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   8 (0.0%)\n",
      "Pairs discarded as untrimmed:           16,102 (2.9%)\n",
      "Pairs written (passing filters):       543,697 (97.1%)\n",
      "\n",
      "Total basepairs processed:   285,501,570 bp\n",
      "  Read 1:   170,741,135 bp\n",
      "  Read 2:   114,760,435 bp\n",
      "Total written (filtered):    253,467,691 bp (88.8%)\n",
      "  Read 1:   154,706,569 bp\n",
      "  Read 2:    98,761,122 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 163 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 28.8%\n",
      "  G: 69.9%\n",
      "  T: 0.0%\n",
      "  none/other: 1.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t12\t8.5\t1\t0 12\n",
      "9\t28\t2.1\t1\t0 28\n",
      "10\t1\t0.5\t1\t0 1\n",
      "20\t10\t0.0\t3\t10\n",
      "21\t14\t0.0\t3\t14\n",
      "22\t6\t0.0\t3\t5 1\n",
      "23\t20\t0.0\t3\t18 1 1\n",
      "24\t14\t0.0\t3\t12 2\n",
      "25\t7\t0.0\t3\t6 1\n",
      "26\t1\t0.0\t3\t1\n",
      "29\t3\t0.0\t3\t2 1\n",
      "30\t1\t0.0\t3\t1\n",
      "33\t4\t0.0\t3\t4\n",
      "34\t2\t0.0\t3\t2\n",
      "95\t2\t0.0\t3\t2\n",
      "216\t1\t0.0\t3\t0 0 1\n",
      "254\t4\t0.0\t3\t4\n",
      "255\t5\t0.0\t3\t5\n",
      "256\t4\t0.0\t3\t4\n",
      "257\t3\t0.0\t3\t2 1\n",
      "258\t3\t0.0\t3\t3\n",
      "259\t5\t0.0\t3\t5\n",
      "277\t1\t0.0\t3\t0 1\n",
      "278\t1\t0.0\t3\t1\n",
      "279\t1\t0.0\t3\t1\n",
      "280\t3\t0.0\t3\t3\n",
      "281\t1\t0.0\t3\t0 1\n",
      "282\t2\t0.0\t3\t2\n",
      "284\t2\t0.0\t3\t2\n",
      "305\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 554311 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t8.5\t1\t5\n",
      "9\t4\t2.1\t1\t4\n",
      "10\t7\t0.5\t1\t3 4\n",
      "11\t11\t0.1\t1\t7 4\n",
      "12\t10\t0.0\t1\t7 3\n",
      "13\t8\t0.0\t1\t6 2\n",
      "14\t21\t0.0\t2\t17 3 1\n",
      "15\t35\t0.0\t2\t16 8 11\n",
      "16\t40\t0.0\t2\t22 11 7\n",
      "17\t298\t0.0\t2\t155 91 52\n",
      "18\t86945\t0.0\t2\t85395 1462 88\n",
      "19\t99905\t0.0\t2\t98077 1726 102\n",
      "20\t102223\t0.0\t2\t100454 1679 90\n",
      "21\t91692\t0.0\t2\t90110 1499 83\n",
      "22\t93369\t0.0\t2\t91816 1481 72\n",
      "23\t79657\t0.0\t2\t78382 1247 28\n",
      "24\t70\t0.0\t2\t55 4 11\n",
      "26\t4\t0.0\t2\t3 0 1\n",
      "29\t1\t0.0\t2\t1\n",
      "34\t1\t0.0\t2\t1\n",
      "35\t1\t0.0\t2\t1\n",
      "38\t1\t0.0\t2\t0 1\n",
      "42\t1\t0.0\t2\t1\n",
      "43\t1\t0.0\t2\t1\n",
      "52\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 12 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 8.3%\n",
      "  C: 8.3%\n",
      "  G: 16.7%\n",
      "  T: 16.7%\n",
      "  none/other: 50.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t8.5\t1\t0 2\n",
      "9\t1\t2.1\t1\t0 1\n",
      "11\t1\t0.1\t1\t0 1\n",
      "203\t1\t0.0\t2\t1\n",
      "204\t1\t0.0\t2\t1\n",
      "205\t6\t0.0\t2\t6\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 548867 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.5\t1\t0 1\n",
      "11\t2\t0.1\t1\t2\n",
      "12\t3\t0.0\t1\t3\n",
      "13\t5\t0.0\t1\t5\n",
      "14\t3\t0.0\t2\t1 1 1\n",
      "15\t32\t0.0\t2\t15 14 3\n",
      "16\t8\t0.0\t2\t5 1 2\n",
      "17\t15\t0.0\t2\t9 2 4\n",
      "18\t15\t0.0\t2\t6 3 6\n",
      "19\t37\t0.0\t2\t17 9 11\n",
      "20\t418\t0.0\t3\t180 119 63 56\n",
      "21\t103246\t0.0\t3\t98232 3645 930 439\n",
      "22\t98674\t0.0\t3\t93652 3603 915 504\n",
      "23\t90732\t0.0\t3\t85816 3632 863 421\n",
      "24\t92329\t0.0\t3\t87651 3345 900 433\n",
      "25\t82942\t0.0\t3\t78341 3365 816 420\n",
      "26\t80296\t0.0\t3\t75895 3250 792 359\n",
      "27\t84\t0.0\t3\t62 18 2 2\n",
      "29\t1\t0.0\t3\t0 0 1\n",
      "36\t1\t0.0\t3\t1\n",
      "54\t2\t0.0\t3\t0 0 0 2\n",
      "57\t1\t0.0\t3\t1\n",
      "59\t3\t0.0\t3\t0 0 0 3\n",
      "60\t3\t0.0\t3\t0 1 0 2\n",
      "61\t6\t0.0\t3\t0 0 4 2\n",
      "62\t3\t0.0\t3\t0 1 1 1\n",
      "63\t2\t0.0\t3\t0 0 1 1\n",
      "64\t2\t0.0\t3\t0 0 1 1\n",
      "66\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121675_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121675_5_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121675_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121675_5_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121675_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121675_5_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121675_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121675_5_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 4.95 s (14 µs/read; 4.19 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            345,762\n",
      "  Read 1 with adapter:                 342,549 (99.1%)\n",
      "  Read 2 with adapter:                 339,148 (98.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   3 (0.0%)\n",
      "Pairs discarded as untrimmed:            9,768 (2.8%)\n",
      "Pairs written (passing filters):       335,991 (97.2%)\n",
      "\n",
      "Total basepairs processed:   176,338,620 bp\n",
      "  Read 1:   105,457,410 bp\n",
      "  Read 2:    70,881,210 bp\n",
      "Total written (filtered):    156,638,059 bp (88.8%)\n",
      "  Read 1:    95,606,276 bp\n",
      "  Read 2:    61,031,783 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 67 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 17.9%\n",
      "  G: 80.6%\n",
      "  T: 1.5%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'G' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t9\t5.3\t1\t0 9\n",
      "9\t3\t1.3\t1\t0 3\n",
      "20\t2\t0.0\t3\t2\n",
      "21\t6\t0.0\t3\t6\n",
      "22\t5\t0.0\t3\t5\n",
      "23\t14\t0.0\t3\t12 0 2\n",
      "24\t2\t0.0\t3\t1 1\n",
      "25\t3\t0.0\t3\t3\n",
      "26\t1\t0.0\t3\t1\n",
      "29\t1\t0.0\t3\t1\n",
      "31\t1\t0.0\t3\t1\n",
      "32\t1\t0.0\t3\t1\n",
      "34\t1\t0.0\t3\t1\n",
      "133\t1\t0.0\t3\t1\n",
      "135\t1\t0.0\t3\t1\n",
      "215\t1\t0.0\t3\t0 0 1\n",
      "217\t1\t0.0\t3\t0 0 1\n",
      "218\t1\t0.0\t3\t1\n",
      "276\t1\t0.0\t3\t1\n",
      "277\t1\t0.0\t3\t0 1\n",
      "278\t2\t0.0\t3\t2\n",
      "279\t3\t0.0\t3\t3\n",
      "280\t2\t0.0\t3\t1 1\n",
      "281\t2\t0.0\t3\t2\n",
      "282\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 342482 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t6\t5.3\t1\t5 1\n",
      "9\t3\t1.3\t1\t3\n",
      "10\t6\t0.3\t1\t5 1\n",
      "11\t7\t0.1\t1\t2 5\n",
      "12\t7\t0.0\t1\t4 3\n",
      "13\t7\t0.0\t1\t5 2\n",
      "14\t18\t0.0\t2\t12 2 4\n",
      "15\t16\t0.0\t2\t9 2 5\n",
      "16\t28\t0.0\t2\t11 12 5\n",
      "17\t183\t0.0\t2\t92 54 37\n",
      "18\t53589\t0.0\t2\t52670 880 39\n",
      "19\t61848\t0.0\t2\t60783 1021 44\n",
      "20\t63059\t0.0\t2\t62012 986 61\n",
      "21\t56779\t0.0\t2\t55816 897 66\n",
      "22\t58075\t0.0\t2\t57184 841 50\n",
      "23\t48800\t0.0\t2\t48115 674 11\n",
      "24\t46\t0.0\t2\t34 4 8\n",
      "25\t2\t0.0\t2\t2\n",
      "28\t1\t0.0\t2\t1\n",
      "34\t1\t0.0\t2\t1\n",
      "36\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 11 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 27.3%\n",
      "  G: 36.4%\n",
      "  T: 9.1%\n",
      "  none/other: 27.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t1.3\t1\t0 1\n",
      "203\t3\t0.0\t2\t3\n",
      "204\t4\t0.0\t2\t4\n",
      "205\t3\t0.0\t2\t2 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 339137 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t5.3\t1\t0 2\n",
      "11\t1\t0.1\t1\t1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "13\t3\t0.0\t1\t3\n",
      "14\t5\t0.0\t2\t0 3 2\n",
      "15\t17\t0.0\t2\t12 3 2\n",
      "16\t10\t0.0\t2\t4 4 2\n",
      "17\t7\t0.0\t2\t2 2 3\n",
      "18\t10\t0.0\t2\t3 6 1\n",
      "19\t20\t0.0\t2\t11 6 3\n",
      "20\t251\t0.0\t3\t106 64 48 33\n",
      "21\t63498\t0.0\t3\t60450 2211 561 276\n",
      "22\t61079\t0.0\t3\t58128 2132 570 249\n",
      "23\t56280\t0.0\t3\t53277 2220 537 246\n",
      "24\t57624\t0.0\t3\t54587 2182 552 303\n",
      "25\t51013\t0.0\t3\t48050 2103 558 302\n",
      "26\t49263\t0.0\t3\t46353 2110 533 267\n",
      "27\t47\t0.0\t3\t34 7 2 4\n",
      "61\t2\t0.0\t3\t0 0 1 1\n",
      "62\t1\t0.0\t3\t0 0 1\n",
      "66\t1\t0.0\t3\t0 0 1\n",
      "67\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121676_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121676_7_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121676_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121676_7_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121676_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121676_7_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121676_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121676_7_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 9.80 s (14 µs/read; 4.36 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            712,750\n",
      "  Read 1 with adapter:                 706,137 (99.1%)\n",
      "  Read 2 with adapter:                 700,151 (98.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   8 (0.0%)\n",
      "Pairs discarded as untrimmed:           19,096 (2.7%)\n",
      "Pairs written (passing filters):       693,646 (97.3%)\n",
      "\n",
      "Total basepairs processed:   363,502,500 bp\n",
      "  Read 1:   217,388,750 bp\n",
      "  Read 2:   146,113,750 bp\n",
      "Total written (filtered):    323,379,504 bp (89.0%)\n",
      "  Read 1:   197,382,010 bp\n",
      "  Read 2:   125,997,494 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 88 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 33.0%\n",
      "  G: 67.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t13\t10.9\t1\t0 13\n",
      "9\t16\t2.7\t1\t0 16\n",
      "20\t2\t0.0\t3\t2\n",
      "21\t8\t0.0\t3\t8\n",
      "22\t15\t0.0\t3\t14 1\n",
      "23\t8\t0.0\t3\t7 1\n",
      "24\t7\t0.0\t3\t7\n",
      "25\t7\t0.0\t3\t7\n",
      "26\t5\t0.0\t3\t4 1\n",
      "82\t1\t0.0\t3\t1\n",
      "134\t2\t0.0\t3\t2\n",
      "136\t1\t0.0\t3\t1\n",
      "163\t1\t0.0\t3\t1\n",
      "281\t1\t0.0\t3\t1\n",
      "282\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 706049 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t10.9\t1\t4 1\n",
      "9\t11\t2.7\t1\t9 2\n",
      "10\t9\t0.7\t1\t5 4\n",
      "11\t20\t0.2\t1\t14 6\n",
      "12\t14\t0.0\t1\t8 6\n",
      "13\t15\t0.0\t1\t6 9\n",
      "14\t40\t0.0\t2\t30 5 5\n",
      "15\t44\t0.0\t2\t18 13 13\n",
      "16\t65\t0.0\t2\t33 20 12\n",
      "17\t372\t0.0\t2\t183 119 70\n",
      "18\t110307\t0.0\t2\t108484 1736 87\n",
      "19\t127432\t0.0\t2\t125110 2202 120\n",
      "20\t129528\t0.0\t2\t127367 2036 125\n",
      "21\t116753\t0.0\t2\t114833 1826 94\n",
      "22\t119622\t0.0\t2\t117744 1779 99\n",
      "23\t101715\t0.0\t2\t100226 1454 35\n",
      "24\t89\t0.0\t2\t68 3 18\n",
      "25\t1\t0.0\t2\t1\n",
      "28\t2\t0.0\t2\t2\n",
      "35\t1\t0.0\t2\t1\n",
      "39\t1\t0.0\t2\t1\n",
      "41\t1\t0.0\t2\t1\n",
      "49\t1\t0.0\t2\t1\n",
      "57\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 17 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 5.9%\n",
      "  C: 41.2%\n",
      "  G: 5.9%\n",
      "  T: 0.0%\n",
      "  none/other: 47.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t10.9\t1\t0 2\n",
      "9\t1\t2.7\t1\t0 1\n",
      "203\t4\t0.0\t2\t2 2\n",
      "204\t2\t0.0\t2\t2\n",
      "205\t8\t0.0\t2\t8\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 700134 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.7\t1\t2\n",
      "11\t7\t0.2\t1\t4 3\n",
      "12\t7\t0.0\t1\t5 2\n",
      "13\t4\t0.0\t1\t3 1\n",
      "14\t4\t0.0\t2\t1 3\n",
      "15\t29\t0.0\t2\t18 6 5\n",
      "16\t12\t0.0\t2\t6 2 4\n",
      "17\t9\t0.0\t2\t4 3 2\n",
      "18\t27\t0.0\t2\t9 7 11\n",
      "19\t37\t0.0\t2\t16 7 14\n",
      "20\t566\t0.0\t3\t229 158 105 74\n",
      "21\t130675\t0.0\t3\t124781 4423 1003 468\n",
      "22\t126203\t0.0\t3\t120299 4273 1093 538\n",
      "23\t115987\t0.0\t3\t109785 4610 1051 541\n",
      "24\t118220\t0.0\t3\t112135 4456 1098 531\n",
      "25\t106507\t0.0\t3\t100654 4264 1052 537\n",
      "26\t101707\t0.0\t3\t95988 4143 1086 490\n",
      "27\t111\t0.0\t3\t87 15 5 4\n",
      "28\t1\t0.0\t3\t1\n",
      "29\t1\t0.0\t3\t1\n",
      "39\t1\t0.0\t3\t1\n",
      "40\t1\t0.0\t3\t1\n",
      "44\t1\t0.0\t3\t0 1\n",
      "47\t1\t0.0\t3\t1\n",
      "50\t1\t0.0\t3\t1\n",
      "56\t1\t0.0\t3\t1\n",
      "57\t1\t0.0\t3\t0 0 0 1\n",
      "58\t1\t0.0\t3\t0 1\n",
      "59\t1\t0.0\t3\t0 0 0 1\n",
      "61\t2\t0.0\t3\t0 0 1 1\n",
      "62\t2\t0.0\t3\t0 0 0 2\n",
      "63\t2\t0.0\t3\t0 1 0 1\n",
      "64\t1\t0.0\t3\t0 0 1\n",
      "65\t2\t0.0\t3\t0 0 1 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121677_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121677_9_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121677_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121677_9_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121677_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121677_9_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121677_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121677_9_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 5.26 s (14 µs/read; 4.17 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            365,427\n",
      "  Read 1 with adapter:                 362,073 (99.1%)\n",
      "  Read 2 with adapter:                 358,323 (98.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   8 (0.0%)\n",
      "Pairs discarded as untrimmed:           10,379 (2.8%)\n",
      "Pairs written (passing filters):       355,040 (97.2%)\n",
      "\n",
      "Total basepairs processed:   186,367,770 bp\n",
      "  Read 1:   111,455,235 bp\n",
      "  Read 2:    74,912,535 bp\n",
      "Total written (filtered):    165,521,152 bp (88.8%)\n",
      "  Read 1:   101,031,128 bp\n",
      "  Read 2:    64,490,024 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 30 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 10.0%\n",
      "  G: 90.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'G' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.6\t1\t0 1\n",
      "9\t2\t1.4\t1\t0 2\n",
      "21\t5\t0.0\t3\t5\n",
      "22\t5\t0.0\t3\t5\n",
      "23\t5\t0.0\t3\t4 1\n",
      "24\t2\t0.0\t3\t2\n",
      "25\t2\t0.0\t3\t2\n",
      "30\t1\t0.0\t3\t1\n",
      "63\t1\t0.0\t3\t1\n",
      "123\t1\t0.0\t3\t1\n",
      "128\t1\t0.0\t3\t1\n",
      "137\t1\t0.0\t3\t1\n",
      "214\t2\t0.0\t3\t0 0 2\n",
      "216\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 362043 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.6\t1\t1\n",
      "9\t4\t1.4\t1\t2 2\n",
      "10\t4\t0.3\t1\t4\n",
      "11\t6\t0.1\t1\t4 2\n",
      "12\t9\t0.0\t1\t3 6\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t26\t0.0\t2\t19 3 4\n",
      "15\t14\t0.0\t2\t9 3 2\n",
      "16\t37\t0.0\t2\t25 9 3\n",
      "17\t187\t0.0\t2\t99 49 39\n",
      "18\t57029\t0.0\t2\t56041 948 40\n",
      "19\t65940\t0.0\t2\t64739 1136 65\n",
      "20\t66096\t0.0\t2\t65073 976 47\n",
      "21\t59626\t0.0\t2\t58606 954 66\n",
      "22\t61038\t0.0\t2\t60068 931 39\n",
      "23\t51967\t0.0\t2\t51230 725 12\n",
      "24\t56\t0.0\t2\t48 3 5\n",
      "39\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 16 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 6.2%\n",
      "  C: 12.5%\n",
      "  G: 25.0%\n",
      "  T: 6.2%\n",
      "  none/other: 50.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.6\t1\t0 1\n",
      "10\t1\t0.3\t1\t0 1\n",
      "113\t1\t0.0\t2\t1\n",
      "203\t1\t0.0\t2\t1\n",
      "204\t4\t0.0\t2\t3 1\n",
      "205\t8\t0.0\t2\t7 0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 358307 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t1.4\t1\t1\n",
      "10\t1\t0.3\t1\t1\n",
      "12\t3\t0.0\t1\t3\n",
      "13\t2\t0.0\t1\t2\n",
      "14\t3\t0.0\t2\t0 1 2\n",
      "15\t19\t0.0\t2\t16 1 2\n",
      "16\t5\t0.0\t2\t2 1 2\n",
      "17\t6\t0.0\t2\t3 2 1\n",
      "18\t12\t0.0\t2\t8 3 1\n",
      "19\t17\t0.0\t2\t9 4 4\n",
      "20\t287\t0.0\t3\t112 78 57 40\n",
      "21\t66800\t0.0\t3\t63644 2337 564 255\n",
      "22\t64376\t0.0\t3\t61206 2269 609 292\n",
      "23\t59606\t0.0\t3\t56423 2322 557 304\n",
      "24\t60399\t0.0\t3\t57151 2342 623 283\n",
      "25\t54547\t0.0\t3\t51497 2224 567 259\n",
      "26\t52160\t0.0\t3\t49209 2140 560 251\n",
      "27\t49\t0.0\t3\t38 5 3 3\n",
      "28\t2\t0.0\t3\t2\n",
      "38\t1\t0.0\t3\t1\n",
      "45\t1\t0.0\t3\t1\n",
      "50\t1\t0.0\t3\t1\n",
      "54\t1\t0.0\t3\t0 0 0 1\n",
      "55\t1\t0.0\t3\t0 0 0 1\n",
      "59\t2\t0.0\t3\t0 0 1 1\n",
      "60\t2\t0.0\t3\t0 0 0 2\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "64\t1\t0.0\t3\t0 0 0 1\n",
      "66\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121678_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121678_11_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121678_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121678_11_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121678_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121678_11_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121678_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121678_11_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.49 s (15 µs/read; 3.87 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             95,852\n",
      "  Read 1 with adapter:                  94,937 (99.0%)\n",
      "  Read 2 with adapter:                  93,876 (97.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   1 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,858 (3.0%)\n",
      "Pairs written (passing filters):        92,993 (97.0%)\n",
      "\n",
      "Total basepairs processed:    48,884,520 bp\n",
      "  Read 1:    29,234,860 bp\n",
      "  Read 2:    19,649,660 bp\n",
      "Total written (filtered):     43,353,503 bp (88.7%)\n",
      "  Read 1:    26,461,168 bp\n",
      "  Read 2:    16,892,335 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 54 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 1.9%\n",
      "  C: 61.1%\n",
      "  G: 37.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t16\t1.5\t1\t0 16\n",
      "9\t18\t0.4\t1\t0 18\n",
      "20\t1\t0.0\t3\t1\n",
      "21\t2\t0.0\t3\t2\n",
      "22\t2\t0.0\t3\t2\n",
      "23\t4\t0.0\t3\t4\n",
      "24\t1\t0.0\t3\t1\n",
      "25\t1\t0.0\t3\t0 0 1\n",
      "34\t1\t0.0\t3\t1\n",
      "168\t1\t0.0\t3\t1\n",
      "233\t1\t0.0\t3\t1\n",
      "236\t1\t0.0\t3\t1\n",
      "277\t1\t0.0\t3\t1\n",
      "279\t1\t0.0\t3\t1\n",
      "280\t1\t0.0\t3\t1\n",
      "282\t1\t0.0\t3\t1\n",
      "283\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 94883 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t5\t0.4\t1\t5\n",
      "10\t3\t0.1\t1\t1 2\n",
      "11\t2\t0.0\t1\t2\n",
      "12\t5\t0.0\t1\t5\n",
      "13\t2\t0.0\t1\t2\n",
      "14\t7\t0.0\t2\t1 4 2\n",
      "15\t7\t0.0\t2\t3 1 3\n",
      "16\t3\t0.0\t2\t2 0 1\n",
      "17\t51\t0.0\t2\t24 20 7\n",
      "18\t14896\t0.0\t2\t14640 249 7\n",
      "19\t17189\t0.0\t2\t16889 283 17\n",
      "20\t17313\t0.0\t2\t17006 288 19\n",
      "21\t15668\t0.0\t2\t15403 254 11\n",
      "22\t16284\t0.0\t2\t16019 248 17\n",
      "23\t13434\t0.0\t2\t13221 210 3\n",
      "24\t14\t0.0\t2\t9 2 3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 4 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 25.0%\n",
      "  G: 25.0%\n",
      "  T: 25.0%\n",
      "  none/other: 25.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t1\t0.0\t2\t1\n",
      "204\t2\t0.0\t2\t2\n",
      "205\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 93872 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.4\t1\t0 1\n",
      "11\t1\t0.0\t1\t0 1\n",
      "13\t3\t0.0\t1\t2 1\n",
      "14\t1\t0.0\t2\t0 0 1\n",
      "15\t5\t0.0\t2\t1 3 1\n",
      "16\t1\t0.0\t2\t0 1\n",
      "18\t2\t0.0\t2\t1 0 1\n",
      "19\t6\t0.0\t2\t4 1 1\n",
      "20\t78\t0.0\t3\t34 23 14 7\n",
      "21\t17770\t0.0\t3\t16889 647 153 81\n",
      "22\t16861\t0.0\t3\t15980 646 152 83\n",
      "23\t15628\t0.0\t3\t14709 686 171 62\n",
      "24\t15828\t0.0\t3\t14923 649 166 90\n",
      "25\t14073\t0.0\t3\t13235 610 144 84\n",
      "26\t13593\t0.0\t3\t12749 614 155 75\n",
      "27\t18\t0.0\t3\t11 3 3 1\n",
      "28\t1\t0.0\t3\t0 0 1\n",
      "60\t2\t0.0\t3\t1 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121679_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121679_13_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121679_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121679_13_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121679_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121679_13_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121679_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121679_13_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 4.65 s (15 µs/read; 4.09 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            316,849\n",
      "  Read 1 with adapter:                 313,871 (99.1%)\n",
      "  Read 2 with adapter:                 310,260 (97.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   3 (0.0%)\n",
      "Pairs discarded as untrimmed:            9,492 (3.0%)\n",
      "Pairs written (passing filters):       307,354 (97.0%)\n",
      "\n",
      "Total basepairs processed:   161,592,990 bp\n",
      "  Read 1:    96,638,945 bp\n",
      "  Read 2:    64,954,045 bp\n",
      "Total written (filtered):    143,285,520 bp (88.7%)\n",
      "  Read 1:    87,458,002 bp\n",
      "  Read 2:    55,827,518 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 63 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 14.3%\n",
      "  G: 84.1%\n",
      "  T: 1.6%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'G' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t6\t4.8\t1\t0 6\n",
      "9\t3\t1.2\t1\t0 3\n",
      "20\t9\t0.0\t3\t9\n",
      "21\t8\t0.0\t3\t8\n",
      "22\t3\t0.0\t3\t3\n",
      "23\t8\t0.0\t3\t6 1 1\n",
      "24\t6\t0.0\t3\t6\n",
      "25\t6\t0.0\t3\t6\n",
      "26\t7\t0.0\t3\t6 1\n",
      "28\t1\t0.0\t3\t1\n",
      "29\t1\t0.0\t3\t1\n",
      "133\t1\t0.0\t3\t1\n",
      "220\t1\t0.0\t3\t1\n",
      "239\t1\t0.0\t3\t1\n",
      "280\t1\t0.0\t3\t0 1\n",
      "281\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 313808 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t4.8\t1\t4\n",
      "9\t2\t1.2\t1\t1 1\n",
      "10\t2\t0.3\t1\t1 1\n",
      "11\t11\t0.1\t1\t9 2\n",
      "12\t6\t0.0\t1\t4 2\n",
      "13\t7\t0.0\t1\t5 2\n",
      "14\t23\t0.0\t2\t17 4 2\n",
      "15\t25\t0.0\t2\t19 1 5\n",
      "16\t29\t0.0\t2\t21 5 3\n",
      "17\t158\t0.0\t2\t83 51 24\n",
      "18\t48904\t0.0\t2\t47998 879 27\n",
      "19\t56375\t0.0\t2\t55371 950 54\n",
      "20\t57674\t0.0\t2\t56648 970 56\n",
      "21\t52026\t0.0\t2\t51171 806 49\n",
      "22\t53311\t0.0\t2\t52430 838 43\n",
      "23\t45205\t0.0\t2\t44479 714 12\n",
      "24\t38\t0.0\t2\t30 2 6\n",
      "30\t1\t0.0\t2\t1\n",
      "31\t2\t0.0\t2\t2\n",
      "32\t1\t0.0\t2\t1\n",
      "41\t1\t0.0\t2\t1\n",
      "43\t1\t0.0\t2\t1\n",
      "60\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 14 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 21.4%\n",
      "  G: 42.9%\n",
      "  T: 14.3%\n",
      "  none/other: 21.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t1\t0 1\n",
      "203\t3\t0.0\t2\t3\n",
      "204\t7\t0.0\t2\t6 1\n",
      "205\t3\t0.0\t2\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 310246 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.3\t1\t0 1\n",
      "11\t3\t0.1\t1\t3\n",
      "12\t3\t0.0\t1\t2 1\n",
      "13\t2\t0.0\t1\t2\n",
      "14\t2\t0.0\t2\t0 2\n",
      "15\t11\t0.0\t2\t7 0 4\n",
      "16\t4\t0.0\t2\t2 1 1\n",
      "17\t5\t0.0\t2\t3 2\n",
      "18\t13\t0.0\t2\t5 5 3\n",
      "19\t31\t0.0\t2\t16 9 6\n",
      "20\t223\t0.0\t3\t87 67 44 25\n",
      "21\t57713\t0.0\t3\t54723 2143 560 287\n",
      "22\t55693\t0.0\t3\t52812 2076 520 285\n",
      "23\t51934\t0.0\t3\t48903 2150 583 298\n",
      "24\t52606\t0.0\t3\t49680 2108 554 264\n",
      "25\t46805\t0.0\t3\t44102 1931 521 251\n",
      "26\t45125\t0.0\t3\t42389 2002 488 246\n",
      "27\t54\t0.0\t3\t46 4 1 3\n",
      "29\t1\t0.0\t3\t0 0 0 1\n",
      "30\t1\t0.0\t3\t1\n",
      "34\t1\t0.0\t3\t1\n",
      "40\t1\t0.0\t3\t1\n",
      "44\t1\t0.0\t3\t0 1\n",
      "48\t1\t0.0\t3\t0 0 0 1\n",
      "54\t2\t0.0\t3\t0 1 0 1\n",
      "59\t1\t0.0\t3\t0 0 0 1\n",
      "61\t4\t0.0\t3\t0 0 1 3\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "63\t3\t0.0\t3\t0 0 0 3\n",
      "64\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121680_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121680_15_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121680_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121680_15_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121680_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121680_15_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121680_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121680_15_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.58 s (14 µs/read; 4.17 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            248,647\n",
      "  Read 1 with adapter:                 246,218 (99.0%)\n",
      "  Read 2 with adapter:                 243,214 (97.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   8 (0.0%)\n",
      "Pairs discarded as untrimmed:            7,759 (3.1%)\n",
      "Pairs written (passing filters):       240,880 (96.9%)\n",
      "\n",
      "Total basepairs processed:   126,809,970 bp\n",
      "  Read 1:    75,837,335 bp\n",
      "  Read 2:    50,972,635 bp\n",
      "Total written (filtered):    112,280,927 bp (88.5%)\n",
      "  Read 1:    68,531,695 bp\n",
      "  Read 2:    43,749,232 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 113 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.5%\n",
      "  C: 42.5%\n",
      "  G: 44.2%\n",
      "  T: 9.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t21\t3.8\t1\t0 21\n",
      "9\t25\t0.9\t1\t0 25\n",
      "10\t1\t0.2\t1\t0 1\n",
      "20\t2\t0.0\t3\t1 0 1\n",
      "22\t2\t0.0\t3\t1 1\n",
      "23\t2\t0.0\t3\t1 1\n",
      "24\t2\t0.0\t3\t2\n",
      "25\t3\t0.0\t3\t2 1\n",
      "26\t1\t0.0\t3\t1\n",
      "29\t1\t0.0\t3\t1\n",
      "33\t1\t0.0\t3\t1\n",
      "128\t2\t0.0\t3\t2\n",
      "129\t2\t0.0\t3\t2\n",
      "130\t2\t0.0\t3\t2\n",
      "131\t1\t0.0\t3\t1\n",
      "132\t2\t0.0\t3\t2\n",
      "133\t2\t0.0\t3\t2\n",
      "219\t1\t0.0\t3\t1\n",
      "234\t1\t0.0\t3\t1\n",
      "236\t1\t0.0\t3\t1\n",
      "275\t1\t0.0\t3\t1\n",
      "276\t1\t0.0\t3\t1\n",
      "277\t6\t0.0\t3\t6\n",
      "278\t10\t0.0\t3\t9 1\n",
      "279\t4\t0.0\t3\t4\n",
      "280\t11\t0.0\t3\t10 1\n",
      "281\t5\t0.0\t3\t5\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 246105 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t3.8\t1\t3 1\n",
      "9\t2\t0.9\t1\t1 1\n",
      "10\t9\t0.2\t1\t4 5\n",
      "12\t7\t0.0\t1\t5 2\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t17\t0.0\t2\t8 3 6\n",
      "15\t15\t0.0\t2\t10 4 1\n",
      "16\t15\t0.0\t2\t6 7 2\n",
      "17\t135\t0.0\t2\t69 41 25\n",
      "18\t38169\t0.0\t2\t37468 677 24\n",
      "19\t44062\t0.0\t2\t43234 794 34\n",
      "20\t45176\t0.0\t2\t44407 731 38\n",
      "21\t40796\t0.0\t2\t40061 700 35\n",
      "22\t42270\t0.0\t2\t41557 683 30\n",
      "23\t35388\t0.0\t2\t34888 487 13\n",
      "24\t35\t0.0\t2\t28 3 4\n",
      "25\t2\t0.0\t2\t2\n",
      "27\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 31 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.2%\n",
      "  C: 19.4%\n",
      "  G: 35.5%\n",
      "  T: 16.1%\n",
      "  none/other: 25.8%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t3.8\t1\t0 1\n",
      "11\t1\t0.1\t1\t0 1\n",
      "203\t11\t0.0\t2\t10 1\n",
      "204\t10\t0.0\t2\t10\n",
      "205\t8\t0.0\t2\t8\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 243183 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.2\t1\t0 1\n",
      "11\t1\t0.1\t1\t1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "15\t8\t0.0\t2\t7 0 1\n",
      "16\t6\t0.0\t2\t3 1 2\n",
      "17\t7\t0.0\t2\t4 2 1\n",
      "18\t7\t0.0\t2\t5 2\n",
      "19\t13\t0.0\t2\t6 4 3\n",
      "20\t196\t0.0\t3\t74 56 36 30\n",
      "21\t44993\t0.0\t3\t42774 1641 398 180\n",
      "22\t43881\t0.0\t3\t41717 1599 382 183\n",
      "23\t40037\t0.0\t3\t37843 1569 419 206\n",
      "24\t41155\t0.0\t3\t38947 1616 394 198\n",
      "25\t37146\t0.0\t3\t35038 1520 374 214\n",
      "26\t35692\t0.0\t3\t33531 1564 412 185\n",
      "27\t31\t0.0\t3\t22 5 3 1\n",
      "28\t1\t0.0\t3\t1\n",
      "31\t1\t0.0\t3\t0 1\n",
      "34\t1\t0.0\t3\t1\n",
      "61\t3\t0.0\t3\t0 0 1 2\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121681_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121681_17_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121681_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121681_17_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121681_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-4urj2e_f/210421121681_17_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121681_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-hmux4h22/89488938-d969-489e-a3fa-6dd23632cd28/data/210421121681_17_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.95 s (14 µs/read; 4.22 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            207,454\n",
      "  Read 1 with adapter:                 205,489 (99.1%)\n",
      "  Read 2 with adapter:                 203,168 (97.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   4 (0.0%)\n",
      "Pairs discarded as untrimmed:            6,212 (3.0%)\n",
      "Pairs written (passing filters):       201,238 (97.0%)\n",
      "\n",
      "Total basepairs processed:   105,801,540 bp\n",
      "  Read 1:    63,273,470 bp\n",
      "  Read 2:    42,528,070 bp\n",
      "Total written (filtered):     93,813,343 bp (88.7%)\n",
      "  Read 1:    57,261,839 bp\n",
      "  Read 2:    36,551,504 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 53 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.8%\n",
      "  C: 47.2%\n",
      "  G: 49.1%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t12\t3.2\t1\t0 12\n",
      "9\t13\t0.8\t1\t0 13\n",
      "20\t2\t0.0\t3\t2\n",
      "21\t4\t0.0\t3\t2 1 1\n",
      "22\t4\t0.0\t3\t3 1\n",
      "23\t2\t0.0\t3\t2\n",
      "24\t4\t0.0\t3\t4\n",
      "25\t2\t0.0\t3\t1 1\n",
      "231\t1\t0.0\t3\t1\n",
      "272\t1\t0.0\t3\t1\n",
      "278\t1\t0.0\t3\t1\n",
      "280\t1\t0.0\t3\t1\n",
      "282\t1\t0.0\t3\t1\n",
      "283\t1\t0.0\t3\t1\n",
      "284\t2\t0.0\t3\t2\n",
      "286\t1\t0.0\t3\t1\n",
      "288\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 205436 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t3.2\t1\t3\n",
      "9\t5\t0.8\t1\t3 2\n",
      "10\t2\t0.2\t1\t2\n",
      "11\t5\t0.0\t1\t3 2\n",
      "12\t2\t0.0\t1\t2\n",
      "13\t4\t0.0\t1\t2 2\n",
      "14\t13\t0.0\t2\t11 0 2\n",
      "15\t10\t0.0\t2\t7 1 2\n",
      "16\t18\t0.0\t2\t8 5 5\n",
      "17\t101\t0.0\t2\t60 21 20\n",
      "18\t32140\t0.0\t2\t31585 527 28\n",
      "19\t36976\t0.0\t2\t36310 632 34\n",
      "20\t37679\t0.0\t2\t37047 592 40\n",
      "21\t34025\t0.0\t2\t33454 543 28\n",
      "22\t34817\t0.0\t2\t34298 497 22\n",
      "23\t29606\t0.0\t2\t29130 467 9\n",
      "24\t25\t0.0\t2\t19 2 4\n",
      "25\t2\t0.0\t2\t2\n",
      "31\t1\t0.0\t2\t1\n",
      "34\t1\t0.0\t2\t1\n",
      "45\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 15 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 13.3%\n",
      "  C: 20.0%\n",
      "  G: 33.3%\n",
      "  T: 6.7%\n",
      "  none/other: 26.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.2\t1\t0 1\n",
      "11\t1\t0.0\t1\t0 1\n",
      "203\t2\t0.0\t2\t2\n",
      "204\t7\t0.0\t2\t7\n",
      "205\t4\t0.0\t2\t4\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 203153 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t3.2\t1\t0 1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "15\t10\t0.0\t2\t4 4 2\n",
      "16\t4\t0.0\t2\t3 0 1\n",
      "17\t7\t0.0\t2\t2 3 2\n",
      "18\t10\t0.0\t2\t5 5\n",
      "19\t13\t0.0\t2\t7 2 4\n",
      "20\t159\t0.0\t3\t54 48 30 27\n",
      "21\t37663\t0.0\t3\t35775 1408 335 145\n",
      "22\t36312\t0.0\t3\t34503 1344 307 158\n",
      "23\t34058\t0.0\t3\t32154 1370 364 170\n",
      "24\t34584\t0.0\t3\t32732 1346 341 165\n",
      "25\t30521\t0.0\t3\t28715 1307 326 173\n",
      "26\t29766\t0.0\t3\t28079 1237 314 136\n",
      "27\t34\t0.0\t3\t22 5 6 1\n",
      "28\t1\t0.0\t3\t0 1\n",
      "32\t1\t0.0\t3\t1\n",
      "38\t1\t0.0\t3\t0 1\n",
      "60\t2\t0.0\t3\t0 0 2\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "64\t1\t0.0\t3\t0 0 0 1\n"
     ]
    }
   ],
   "source": [
    "if trim:    \n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = trim_paired(\n",
    "        demultiplexed_sequences=artifact,\n",
    "        front_f=forward_primer,\n",
    "        front_r=reverse_primer,\n",
    "        adapter_f=reverse_reverse_complement,\n",
    "        adapter_r=forward_reverse_complement,\n",
    "        cores=threads,\n",
    "        overlap=trim['overlap'],\n",
    "        indels=False,\n",
    "        match_read_wildcards=True,\n",
    "        match_adapter_wildcards=True,\n",
    "        error_rate=0.15,\n",
    "        discard_untrimmed=True,\n",
    "    ).trimmed_sequences\n",
    "\n",
    "    demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "    demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')\n",
    "    \n",
    "    demux.visualizers.summarize(res).visualization\n",
    "    res.save(demux_file_trim)\n",
    "    Visualization.save(demux.visualizers.summarize(res).visualization, filepath=demux_view_trim)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 170.646697,
   "end_time": "2022-09-30T10:18:59.878762",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-steps/step-01-prepare-data.ipynb",
   "output_path": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-NCxNR-trim/nb-executed-steps/step-01-prepare-data-ana-flavia-NCxNR-trim.ipynb",
   "parameters": {
    "base_dir": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri",
    "class_col": "group-id",
    "classifier_file": "/home/lauro/nupeb/dados_brutos_rede_genoma/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza",
    "experiment_name": "ana-flavia-NCxNR-trim",
    "manifest_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-NCxNR.csv",
    "metadata_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-NCxNR.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2022-09-30T10:16:09.232065",
   "version": "2.3.4"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}