{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "dcf69502",
   "metadata": {
    "papermill": {
     "duration": 0.019329,
     "end_time": "2022-09-29T22:15:44.470028",
     "exception": false,
     "start_time": "2022-09-29T22:15:44.450699",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ca90092c",
   "metadata": {
    "papermill": {
     "duration": 0.011424,
     "end_time": "2022-09-29T22:15:44.493482",
     "exception": false,
     "start_time": "2022-09-29T22:15:44.482058",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1625893e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-14T17:43:20.516262Z",
     "start_time": "2022-09-14T17:43:20.510424Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-29T22:15:44.517126Z",
     "iopub.status.busy": "2022-09-29T22:15:44.516836Z",
     "iopub.status.idle": "2022-09-29T22:15:50.722407Z",
     "shell.execute_reply": "2022-09-29T22:15:50.721780Z"
    },
    "papermill": {
     "duration": 6.220391,
     "end_time": "2022-09-29T22:15:50.725050",
     "exception": false,
     "start_time": "2022-09-29T22:15:44.504659",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired\n",
    "\n",
    "from utils import *"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a36ef40",
   "metadata": {
    "papermill": {
     "duration": 0.010241,
     "end_time": "2022-09-29T22:15:50.751992",
     "exception": false,
     "start_time": "2022-09-29T22:15:50.741751",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a3b26243",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.594341Z",
     "start_time": "2022-06-28T18:50:41.591544Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-29T22:15:50.805652Z",
     "iopub.status.busy": "2022-09-29T22:15:50.805414Z",
     "iopub.status.idle": "2022-09-29T22:15:50.811296Z",
     "shell.execute_reply": "2022-09-29T22:15:50.810488Z"
    },
    "papermill": {
     "duration": 0.01882,
     "end_time": "2022-09-29T22:15:50.812817",
     "exception": false,
     "start_time": "2022-09-29T22:15:50.793997",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'ana-flavia-superlactacao.yaml')\n",
    "experiment_name = 'jenneffer-vs-01'\n",
    "base_dir = os.path.join('/', 'home', 'lauro', 'nupeb', 'redemicro')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'not-hist-vs-manifest.csv')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'imgs'))\n",
    "replace_files = False\n",
    "trim = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f46f1838",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-29T22:15:50.836142Z",
     "iopub.status.busy": "2022-09-29T22:15:50.835873Z",
     "iopub.status.idle": "2022-09-29T22:15:50.842152Z",
     "shell.execute_reply": "2022-09-29T22:15:50.841327Z"
    },
    "papermill": {
     "duration": 0.020109,
     "end_time": "2022-09-29T22:15:50.843922",
     "exception": false,
     "start_time": "2022-09-29T22:15:50.823813",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "experiment_name = \"ana-flavia-HSD-NCxHSD-NR-trim\"\n",
    "base_dir = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri\"\n",
    "manifest_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-HSD-NCxHSD-NR.csv\"\n",
    "metadata_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-HSD-NCxHSD-NR.tsv\"\n",
    "class_col = \"sample-group\"\n",
    "classifier_file = \"/home/lauro/nupeb/dados_brutos_rede_genoma/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza\"\n",
    "replace_files = False\n",
    "phred = 20\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n",
    "overlap = 12\n",
    "threads = 6\n",
    "trim = {\n",
    "    \"overlap\": 8,\n",
    "    \"forward_primer\": \"CCTACGGGRSGCAGCAG\",\n",
    "    \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\",\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1b3646f7",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.610682Z",
     "start_time": "2022-06-28T18:50:41.595886Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-29T22:15:50.867771Z",
     "iopub.status.busy": "2022-09-29T22:15:50.867465Z",
     "iopub.status.idle": "2022-09-29T22:15:50.873783Z",
     "shell.execute_reply": "2022-09-29T22:15:50.872939Z"
    },
    "papermill": {
     "duration": 0.019527,
     "end_time": "2022-09-29T22:15:50.875434",
     "exception": false,
     "start_time": "2022-09-29T22:15:50.855907",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Setting hardcoded parameters for development and debbuging\n",
    "production = True\n",
    "if not production:\n",
    "    params_path = os.path.join('..', 'params', 'ana-flavia-hipotese-01')\n",
    "    with open(params_path, 'r') as stream:\n",
    "        params = yaml.safe_load(stream)\n",
    "        experiment_name = params['experiment_name']\n",
    "        base_dir = params['base_dir']\n",
    "        manifest_file = params['manifest_file']\n",
    "        replace_files = params['replace_files']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a3f7c03",
   "metadata": {
    "papermill": {
     "duration": 0.011164,
     "end_time": "2022-09-29T22:15:50.898825",
     "exception": false,
     "start_time": "2022-09-29T22:15:50.887661",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "29d9f114",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.618098Z",
     "start_time": "2022-06-28T18:50:41.614152Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-29T22:15:50.921220Z",
     "iopub.status.busy": "2022-09-29T22:15:50.920898Z",
     "iopub.status.idle": "2022-09-29T22:15:50.925390Z",
     "shell.execute_reply": "2022-09-29T22:15:50.924492Z"
    },
    "papermill": {
     "duration": 0.017741,
     "end_time": "2022-09-29T22:15:50.927051",
     "exception": false,
     "start_time": "2022-09-29T22:15:50.909310",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# new_manifest = '/home/lauro/nupeb/redemicro/data/raw/manifest/karina-manifest.csv'\n",
    "# with open(manifest_file, 'r') as oldm, open(new_manifest, 'w') as newm:\n",
    "#     header = 'sample-id,absolute-filepath,direction\\n'\n",
    "#     newm.write(header)\n",
    "#     for line in oldm.readlines()[1:]:\n",
    "#         sid, forward, reverse = line[:-1].split(',')\n",
    "#         fline = ','.join((sid, forward, 'forward')) + '\\n'\n",
    "#         rline = ','.join((sid, reverse, 'reverse')) + '\\n'\n",
    "#         newm.write(fline)\n",
    "#         newm.write(rline)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "35b27d31",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.632125Z",
     "start_time": "2022-06-28T18:50:41.619710Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-29T22:15:50.950473Z",
     "iopub.status.busy": "2022-09-29T22:15:50.950277Z",
     "iopub.status.idle": "2022-09-29T22:15:50.955212Z",
     "shell.execute_reply": "2022-09-29T22:15:50.954561Z"
    },
    "papermill": {
     "duration": 0.017896,
     "end_time": "2022-09-29T22:15:50.956831",
     "exception": false,
     "start_time": "2022-09-29T22:15:50.938935",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Define the output folder path\n",
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "\n",
    "# Create path if it not exist\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exist\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')\n",
    "\n",
    "# Define the output artifact full path\n",
    "demux_file = os.path.join(out_dir, 'demux-paired.qza')\n",
    "demux_view = os.path.join(out_dir, 'demux-paired.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4269a664",
   "metadata": {
    "papermill": {
     "duration": 0.010862,
     "end_time": "2022-09-29T22:15:50.978787",
     "exception": false,
     "start_time": "2022-09-29T22:15:50.967925",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "8efd69c5",
   "metadata": {
    "papermill": {
     "duration": 0.010477,
     "end_time": "2022-09-29T22:15:50.999839",
     "exception": false,
     "start_time": "2022-09-29T22:15:50.989362",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file of `SampleData[PairedEndSequencesWithQuality]`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "758fccea",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:44:58.640338Z",
     "start_time": "2022-06-28T18:50:41.633657Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-29T22:15:51.022108Z",
     "iopub.status.busy": "2022-09-29T22:15:51.021784Z",
     "iopub.status.idle": "2022-09-29T22:15:51.758247Z",
     "shell.execute_reply": "2022-09-29T22:15:51.757460Z"
    },
    "papermill": {
     "duration": 0.75106,
     "end_time": "2022-09-29T22:15:51.761361",
     "exception": false,
     "start_time": "2022-09-29T22:15:51.010301",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    \n",
    "    # Import data and create an artifact object\n",
    "    artifact = Artifact.import_data(\n",
    "        'SampleData[PairedEndSequencesWithQuality]', \n",
    "        manifest_file, \n",
    "        view_type='PairedEndFastqManifestPhred33')\n",
    "    \n",
    "    # Save the artifact object to a new qza file\n",
    "    artifact.save(demux_file)\n",
    "\n",
    "else:\n",
    "    artifact = Artifact.load(demux_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "aef5920e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.602855Z",
     "start_time": "2022-06-28T19:44:58.642520Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-29T22:15:51.792846Z",
     "iopub.status.busy": "2022-09-29T22:15:51.792604Z",
     "iopub.status.idle": "2022-09-29T22:15:51.814421Z",
     "shell.execute_reply": "2022-09-29T22:15:51.813820Z"
    },
    "papermill": {
     "duration": 0.035988,
     "end_time": "2022-09-29T22:15:51.816285",
     "exception": false,
     "start_time": "2022-09-29T22:15:51.780297",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "if not os.path.isfile(demux_view) or replace_files: \n",
    "    # Generate e visualization of the Artifact\n",
    "    demux_summary = demux.visualizers.summarize(artifact)\n",
    "\n",
    "    # Save a new visualization file based on the qza file\n",
    "    Visualization.save(demux_summary.visualization, filepath=demux_view)\n",
    "    \n",
    "    demux_view_obj = demux_summary.visualization\n",
    "else:\n",
    "    demux_view_obj = Visualization.load(demux_view)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f2fa0519",
   "metadata": {
    "papermill": {
     "duration": 0.012447,
     "end_time": "2022-09-29T22:15:51.842250",
     "exception": false,
     "start_time": "2022-09-29T22:15:51.829803",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "21be7839",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.609737Z",
     "start_time": "2022-06-28T19:47:56.604820Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-29T22:15:51.865595Z",
     "iopub.status.busy": "2022-09-29T22:15:51.865415Z",
     "iopub.status.idle": "2022-09-29T22:15:51.870112Z",
     "shell.execute_reply": "2022-09-29T22:15:51.869374Z"
    },
    "papermill": {
     "duration": 0.018192,
     "end_time": "2022-09-29T22:15:51.871655",
     "exception": false,
     "start_time": "2022-09-29T22:15:51.853463",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<visualization: Visualization uuid: b061a8cc-8557-4b30-bf26-e4dce3ceb743>\n"
     ]
    }
   ],
   "source": [
    "print(demux_view_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "053d9297",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.704841Z",
     "start_time": "2022-06-28T19:47:56.611846Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-29T22:15:51.896176Z",
     "iopub.status.busy": "2022-09-29T22:15:51.895940Z",
     "iopub.status.idle": "2022-09-29T22:15:51.910356Z",
     "shell.execute_reply": "2022-09-29T22:15:51.909886Z"
    },
    "papermill": {
     "duration": 0.028686,
     "end_time": "2022-09-29T22:15:51.912216",
     "exception": false,
     "start_time": "2022-09-29T22:15:51.883530",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-nc4dyld0')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: b061a8cc-8557-4b30-bf26-e4dce3ceb743>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5054b9b9",
   "metadata": {
    "papermill": {
     "duration": 0.013201,
     "end_time": "2022-09-29T22:15:51.939334",
     "exception": false,
     "start_time": "2022-09-29T22:15:51.926133",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3011e55f",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-29T22:15:51.996742Z",
     "iopub.status.busy": "2022-09-29T22:15:51.996562Z",
     "iopub.status.idle": "2022-09-29T22:17:01.361114Z",
     "shell.execute_reply": "2022-09-29T22:17:01.360387Z"
    },
    "papermill": {
     "duration": 69.396718,
     "end_time": "2022-09-29T22:17:01.364246",
     "exception": false,
     "start_time": "2022-09-29T22:15:51.967528",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121688_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121688_1_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121688_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121688_1_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121688_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121688_1_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121688_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121688_1_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.27 s (17 µs/read; 3.62 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            136,741\n",
      "  Read 1 with adapter:                 135,344 (99.0%)\n",
      "  Read 2 with adapter:                 133,661 (97.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  10 (0.0%)\n",
      "Pairs discarded as untrimmed:            4,417 (3.2%)\n",
      "Pairs written (passing filters):       132,314 (96.8%)\n",
      "\n",
      "Total basepairs processed:    69,737,910 bp\n",
      "  Read 1:    41,706,005 bp\n",
      "  Read 2:    28,031,905 bp\n",
      "Total written (filtered):     61,659,397 bp (88.4%)\n",
      "  Read 1:    37,628,045 bp\n",
      "  Read 2:    24,031,352 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 111 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 1.8%\n",
      "  C: 23.4%\n",
      "  G: 73.0%\n",
      "  T: 1.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t6\t2.1\t1\t0 6\n",
      "9\t5\t0.5\t1\t0 5\n",
      "21\t3\t0.0\t3\t2 1\n",
      "22\t2\t0.0\t3\t1 1\n",
      "24\t2\t0.0\t3\t2\n",
      "25\t1\t0.0\t3\t1\n",
      "26\t1\t0.0\t3\t1\n",
      "29\t2\t0.0\t3\t2\n",
      "30\t1\t0.0\t3\t1\n",
      "31\t2\t0.0\t3\t2\n",
      "32\t2\t0.0\t3\t2\n",
      "33\t1\t0.0\t3\t1\n",
      "34\t1\t0.0\t3\t1\n",
      "133\t1\t0.0\t3\t1\n",
      "151\t1\t0.0\t3\t0 1\n",
      "193\t1\t0.0\t3\t1\n",
      "216\t1\t0.0\t3\t1\n",
      "245\t3\t0.0\t3\t3\n",
      "246\t3\t0.0\t3\t3\n",
      "247\t3\t0.0\t3\t3\n",
      "248\t2\t0.0\t3\t1 1\n",
      "249\t1\t0.0\t3\t1\n",
      "250\t3\t0.0\t3\t2 1\n",
      "278\t1\t0.0\t3\t1\n",
      "280\t1\t0.0\t3\t0 1\n",
      "281\t2\t0.0\t3\t2\n",
      "282\t10\t0.0\t3\t9 1\n",
      "283\t14\t0.0\t3\t10 3 1\n",
      "284\t7\t0.0\t3\t7\n",
      "285\t8\t0.0\t3\t8\n",
      "286\t8\t0.0\t3\t8\n",
      "287\t10\t0.0\t3\t9 1\n",
      "288\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 135233 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.1\t1\t1\n",
      "9\t2\t0.5\t1\t2\n",
      "10\t1\t0.1\t1\t0 1\n",
      "11\t1\t0.0\t1\t0 1\n",
      "12\t5\t0.0\t1\t4 1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t14\t0.0\t2\t8 3 3\n",
      "15\t10\t0.0\t2\t5 1 4\n",
      "16\t19\t0.0\t2\t11 4 4\n",
      "17\t71\t0.0\t2\t32 27 12\n",
      "18\t20647\t0.0\t2\t20253 372 22\n",
      "19\t24041\t0.0\t2\t23576 437 28\n",
      "20\t24968\t0.0\t2\t24488 458 22\n",
      "21\t22490\t0.0\t2\t22070 392 28\n",
      "22\t23290\t0.0\t2\t22913 357 20\n",
      "23\t19653\t0.0\t2\t19351 295 7\n",
      "24\t17\t0.0\t2\t14 2 1\n",
      "45\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 26 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.8%\n",
      "  C: 11.5%\n",
      "  G: 30.8%\n",
      "  T: 15.4%\n",
      "  none/other: 38.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t2.1\t1\t0 2\n",
      "203\t2\t0.0\t2\t2\n",
      "204\t12\t0.0\t2\t12\n",
      "205\t10\t0.0\t2\t10\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 133635 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.1\t1\t0 1\n",
      "11\t1\t0.0\t1\t0 1\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t8\t0.0\t2\t7 0 1\n",
      "16\t7\t0.0\t2\t4 1 2\n",
      "17\t2\t0.0\t2\t1 0 1\n",
      "18\t5\t0.0\t2\t1 1 3\n",
      "19\t7\t0.0\t2\t4 3\n",
      "20\t93\t0.0\t3\t38 24 16 15\n",
      "21\t24828\t0.0\t3\t23579 909 223 117\n",
      "22\t23875\t0.0\t3\t22708 848 216 103\n",
      "23\t22280\t0.0\t3\t21038 929 213 100\n",
      "24\t22734\t0.0\t3\t21560 866 212 96\n",
      "25\t20325\t0.0\t3\t19189 821 199 116\n",
      "26\t19434\t0.0\t3\t18299 827 201 107\n",
      "27\t29\t0.0\t3\t23 4 0 2\n",
      "37\t1\t0.0\t3\t1\n",
      "57\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121689_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121689_3_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121689_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121689_3_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121689_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121689_3_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121689_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121689_3_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.75 s (15 µs/read; 3.96 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            181,624\n",
      "  Read 1 with adapter:                 179,835 (99.0%)\n",
      "  Read 2 with adapter:                 177,952 (98.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   7 (0.0%)\n",
      "Pairs discarded as untrimmed:            5,418 (3.0%)\n",
      "Pairs written (passing filters):       176,199 (97.0%)\n",
      "\n",
      "Total basepairs processed:    92,628,240 bp\n",
      "  Read 1:    55,395,320 bp\n",
      "  Read 2:    37,232,920 bp\n",
      "Total written (filtered):     82,133,955 bp (88.7%)\n",
      "  Read 1:    50,131,407 bp\n",
      "  Read 2:    32,002,548 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 369 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 1.6%\n",
      "  C: 13.8%\n",
      "  G: 83.7%\n",
      "  T: 0.8%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'G' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t26\t2.8\t1\t0 26\n",
      "9\t24\t0.7\t1\t0 24\n",
      "20\t3\t0.0\t3\t3\n",
      "21\t5\t0.0\t3\t5\n",
      "22\t2\t0.0\t3\t2\n",
      "23\t3\t0.0\t3\t2 1\n",
      "24\t3\t0.0\t3\t3\n",
      "25\t5\t0.0\t3\t5\n",
      "29\t37\t0.0\t3\t35 2\n",
      "30\t44\t0.0\t3\t43 1\n",
      "31\t66\t0.0\t3\t64 2\n",
      "32\t44\t0.0\t3\t41 3\n",
      "33\t42\t0.0\t3\t40 2\n",
      "34\t53\t0.0\t3\t52 1\n",
      "36\t1\t0.0\t3\t0 1\n",
      "192\t1\t0.0\t3\t1\n",
      "195\t2\t0.0\t3\t2\n",
      "196\t1\t0.0\t3\t1\n",
      "197\t2\t0.0\t3\t2\n",
      "233\t1\t0.0\t3\t1\n",
      "250\t2\t0.0\t3\t2\n",
      "253\t1\t0.0\t3\t1\n",
      "282\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 179466 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.8\t1\t1\n",
      "10\t3\t0.2\t1\t1 2\n",
      "11\t2\t0.0\t1\t2\n",
      "12\t5\t0.0\t1\t4 1\n",
      "13\t3\t0.0\t1\t2 1\n",
      "14\t11\t0.0\t2\t7 2 2\n",
      "15\t11\t0.0\t2\t3 6 2\n",
      "16\t11\t0.0\t2\t3 5 3\n",
      "17\t98\t0.0\t2\t42 35 21\n",
      "18\t27693\t0.0\t2\t27177 498 18\n",
      "19\t32014\t0.0\t2\t31417 571 26\n",
      "20\t32946\t0.0\t2\t32359 561 26\n",
      "21\t29829\t0.0\t2\t29292 506 31\n",
      "22\t30649\t0.0\t2\t30106 528 15\n",
      "23\t26157\t0.0\t2\t25767 380 10\n",
      "24\t29\t0.0\t2\t27 1 1\n",
      "25\t2\t0.0\t2\t2\n",
      "33\t1\t0.0\t2\t1\n",
      "39\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 21 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 4.8%\n",
      "  C: 19.0%\n",
      "  G: 14.3%\n",
      "  T: 28.6%\n",
      "  none/other: 33.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "179\t1\t0.0\t2\t0 0 1\n",
      "203\t4\t0.0\t2\t4\n",
      "204\t9\t0.0\t2\t9\n",
      "205\t7\t0.0\t2\t7\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 177931 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.2\t1\t1\n",
      "13\t1\t0.0\t1\t1\n",
      "14\t2\t0.0\t2\t0 2\n",
      "15\t8\t0.0\t2\t8\n",
      "16\t2\t0.0\t2\t1 1\n",
      "17\t1\t0.0\t2\t1\n",
      "18\t8\t0.0\t2\t3 4 1\n",
      "19\t8\t0.0\t2\t3 1 4\n",
      "20\t137\t0.0\t3\t56 38 22 21\n",
      "21\t33009\t0.0\t3\t31305 1276 277 151\n",
      "22\t32157\t0.0\t3\t30487 1181 344 145\n",
      "23\t29386\t0.0\t3\t27621 1320 299 146\n",
      "24\t30221\t0.0\t3\t28517 1204 328 172\n",
      "25\t27046\t0.0\t3\t25431 1158 304 153\n",
      "26\t25904\t0.0\t3\t24303 1154 306 141\n",
      "27\t27\t0.0\t3\t18 4 5\n",
      "28\t1\t0.0\t3\t0 0 1\n",
      "60\t2\t0.0\t3\t0 0 0 2\n",
      "61\t1\t0.0\t3\t0 0 1\n",
      "62\t4\t0.0\t3\t0 0 1 3\n",
      "63\t1\t0.0\t3\t0 1\n",
      "64\t2\t0.0\t3\t0 0 1 1\n",
      "65\t2\t0.0\t3\t0 0 0 2\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121690_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121690_5_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121690_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121690_5_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121690_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121690_5_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121690_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121690_5_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.09 s (16 µs/read; 3.86 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            134,399\n",
      "  Read 1 with adapter:                 133,122 (99.0%)\n",
      "  Read 2 with adapter:                 131,694 (98.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   2 (0.0%)\n",
      "Pairs discarded as untrimmed:            3,948 (2.9%)\n",
      "Pairs written (passing filters):       130,449 (97.1%)\n",
      "\n",
      "Total basepairs processed:    68,543,490 bp\n",
      "  Read 1:    40,991,695 bp\n",
      "  Read 2:    27,551,795 bp\n",
      "Total written (filtered):     60,814,970 bp (88.7%)\n",
      "  Read 1:    37,120,467 bp\n",
      "  Read 2:    23,694,503 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 78 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 46.2%\n",
      "  G: 53.8%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t23\t2.1\t1\t0 23\n",
      "9\t13\t0.5\t1\t0 13\n",
      "20\t6\t0.0\t3\t4 1 1\n",
      "21\t5\t0.0\t3\t3 1 1\n",
      "22\t6\t0.0\t3\t5 0 1\n",
      "23\t5\t0.0\t3\t5\n",
      "24\t8\t0.0\t3\t8\n",
      "25\t8\t0.0\t3\t7 1\n",
      "28\t1\t0.0\t3\t1\n",
      "31\t1\t0.0\t3\t1\n",
      "34\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 133044 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.1\t1\t0 1\n",
      "10\t2\t0.1\t1\t2\n",
      "11\t3\t0.0\t1\t2 1\n",
      "12\t3\t0.0\t1\t1 2\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t7\t0.0\t2\t5 2\n",
      "15\t2\t0.0\t2\t2\n",
      "16\t8\t0.0\t2\t6 0 2\n",
      "17\t85\t0.0\t2\t44 31 10\n",
      "18\t20575\t0.0\t2\t20211 344 20\n",
      "19\t24150\t0.0\t2\t23659 464 27\n",
      "20\t24402\t0.0\t2\t23976 398 28\n",
      "21\t22045\t0.0\t2\t21625 397 23\n",
      "22\t22706\t0.0\t2\t22309 374 23\n",
      "23\t19038\t0.0\t2\t18739 293 6\n",
      "24\t14\t0.0\t2\t10 0 4\n",
      "26\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 6 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 16.7%\n",
      "  G: 16.7%\n",
      "  T: 33.3%\n",
      "  none/other: 33.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t1\t0 1\n",
      "203\t2\t0.0\t2\t1 1\n",
      "204\t1\t0.0\t2\t1\n",
      "205\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 131688 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t5\t0.0\t2\t2 1 2\n",
      "16\t2\t0.0\t2\t1 1\n",
      "17\t6\t0.0\t2\t3 1 2\n",
      "18\t2\t0.0\t2\t2\n",
      "19\t12\t0.0\t2\t5 4 3\n",
      "20\t107\t0.0\t3\t33 38 20 16\n",
      "21\t24486\t0.0\t3\t23126 1008 235 117\n",
      "22\t23576\t0.0\t3\t22324 901 230 121\n",
      "23\t21821\t0.0\t3\t20527 954 222 118\n",
      "24\t22473\t0.0\t3\t21205 922 222 124\n",
      "25\t20054\t0.0\t3\t18881 860 208 105\n",
      "26\t19109\t0.0\t3\t17957 833 205 114\n",
      "27\t26\t0.0\t3\t22 3 0 1\n",
      "31\t1\t0.0\t3\t1\n",
      "39\t1\t0.0\t3\t1\n",
      "62\t1\t0.0\t3\t0 0 1\n",
      "63\t1\t0.0\t3\t0 0 0 1\n",
      "64\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121691_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121691_7_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121691_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121691_7_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121691_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121691_7_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121691_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121691_7_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.62 s (16 µs/read; 3.72 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            224,175\n",
      "  Read 1 with adapter:                 221,891 (99.0%)\n",
      "  Read 2 with adapter:                 219,776 (98.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   1 (0.0%)\n",
      "Pairs discarded as untrimmed:            6,634 (3.0%)\n",
      "Pairs written (passing filters):       217,540 (97.0%)\n",
      "\n",
      "Total basepairs processed:   114,329,250 bp\n",
      "  Read 1:    68,373,375 bp\n",
      "  Read 2:    45,955,875 bp\n",
      "Total written (filtered):    101,415,359 bp (88.7%)\n",
      "  Read 1:    61,901,589 bp\n",
      "  Read 2:    39,513,770 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 66 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 4.5%\n",
      "  G: 92.4%\n",
      "  T: 3.0%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'G' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t3.4\t1\t0 4\n",
      "9\t1\t0.9\t1\t0 1\n",
      "20\t5\t0.0\t3\t5\n",
      "21\t3\t0.0\t3\t2 1\n",
      "22\t8\t0.0\t3\t8\n",
      "23\t6\t0.0\t3\t6\n",
      "24\t4\t0.0\t3\t3 1\n",
      "25\t3\t0.0\t3\t2 1\n",
      "29\t2\t0.0\t3\t2\n",
      "30\t4\t0.0\t3\t4\n",
      "31\t2\t0.0\t3\t1 1\n",
      "32\t13\t0.0\t3\t13\n",
      "33\t7\t0.0\t3\t7\n",
      "34\t2\t0.0\t3\t2\n",
      "160\t1\t0.0\t3\t1\n",
      "284\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 221825 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t3.4\t1\t1\n",
      "9\t5\t0.9\t1\t4 1\n",
      "10\t3\t0.2\t1\t1 2\n",
      "11\t6\t0.1\t1\t5 1\n",
      "12\t4\t0.0\t1\t0 4\n",
      "13\t4\t0.0\t1\t2 2\n",
      "14\t10\t0.0\t2\t7 2 1\n",
      "15\t9\t0.0\t2\t5 3 1\n",
      "16\t16\t0.0\t2\t7 2 7\n",
      "17\t98\t0.0\t2\t51 30 17\n",
      "18\t34555\t0.0\t2\t33935 600 20\n",
      "19\t39994\t0.0\t2\t39193 777 24\n",
      "20\t40929\t0.0\t2\t40215 689 25\n",
      "21\t36497\t0.0\t2\t35864 599 34\n",
      "22\t37849\t0.0\t2\t37222 599 28\n",
      "23\t31809\t0.0\t2\t31328 468 13\n",
      "24\t28\t0.0\t2\t21 5 2\n",
      "27\t2\t0.0\t2\t1 1\n",
      "29\t2\t0.0\t2\t2\n",
      "31\t1\t0.0\t2\t1\n",
      "33\t1\t0.0\t2\t1\n",
      "52\t1\t0.0\t2\t1\n",
      "234\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 5 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 40.0%\n",
      "  G: 20.0%\n",
      "  T: 20.0%\n",
      "  none/other: 20.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t2\t0.0\t2\t2\n",
      "204\t2\t0.0\t2\t2\n",
      "205\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 219771 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t3\t0.0\t1\t2 1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t4\t0.0\t2\t2 1 1\n",
      "15\t7\t0.0\t2\t4 1 2\n",
      "16\t3\t0.0\t2\t2 0 1\n",
      "18\t3\t0.0\t2\t0 2 1\n",
      "19\t11\t0.0\t2\t4 3 4\n",
      "20\t170\t0.0\t3\t64 50 33 23\n",
      "21\t40830\t0.0\t3\t38764 1534 352 180\n",
      "22\t39747\t0.0\t3\t37770 1465 344 168\n",
      "23\t36360\t0.0\t3\t34391 1449 347 173\n",
      "24\t36920\t0.0\t3\t34925 1437 365 193\n",
      "25\t33615\t0.0\t3\t31714 1412 323 166\n",
      "26\t32052\t0.0\t3\t30279 1300 321 152\n",
      "27\t36\t0.0\t3\t25 7 2 2\n",
      "28\t1\t0.0\t3\t1\n",
      "30\t1\t0.0\t3\t1\n",
      "53\t1\t0.0\t3\t1\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "62\t3\t0.0\t3\t0 1 0 2\n",
      "63\t1\t0.0\t3\t0 1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121692_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121692_9_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121692_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121692_9_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121692_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121692_9_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121692_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121692_9_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.27 s (16 µs/read; 3.79 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             80,216\n",
      "  Read 1 with adapter:                  79,416 (99.0%)\n",
      "  Read 2 with adapter:                  78,121 (97.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  15 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,870 (3.6%)\n",
      "Pairs written (passing filters):        77,331 (96.4%)\n",
      "\n",
      "Total basepairs processed:    40,910,160 bp\n",
      "  Read 1:    24,465,880 bp\n",
      "  Read 2:    16,444,280 bp\n",
      "Total written (filtered):     36,048,173 bp (88.1%)\n",
      "  Read 1:    22,004,584 bp\n",
      "  Read 2:    14,043,589 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 51 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 43.1%\n",
      "  G: 54.9%\n",
      "  T: 2.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t7\t1.2\t1\t0 7\n",
      "9\t16\t0.3\t1\t0 16\n",
      "20\t5\t0.0\t3\t5\n",
      "21\t3\t0.0\t3\t3\n",
      "22\t4\t0.0\t3\t4\n",
      "23\t6\t0.0\t3\t6\n",
      "25\t5\t0.0\t3\t3 2\n",
      "30\t2\t0.0\t3\t2\n",
      "31\t1\t0.0\t3\t1\n",
      "32\t1\t0.0\t3\t1\n",
      "276\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 79365 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.2\t1\t1\n",
      "9\t1\t0.3\t1\t1\n",
      "11\t1\t0.0\t1\t0 1\n",
      "14\t4\t0.0\t2\t2 1 1\n",
      "15\t2\t0.0\t2\t0 0 2\n",
      "16\t5\t0.0\t2\t3 2\n",
      "17\t49\t0.0\t2\t23 14 12\n",
      "18\t12232\t0.0\t2\t12001 223 8\n",
      "19\t14292\t0.0\t2\t14026 243 23\n",
      "20\t14527\t0.0\t2\t14274 244 9\n",
      "21\t13220\t0.0\t2\t13012 192 16\n",
      "22\t13560\t0.0\t2\t13323 225 12\n",
      "23\t11464\t0.0\t2\t11268 190 6\n",
      "24\t6\t0.0\t2\t4 0 2\n",
      "32\t1\t0.0\t2\t0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 32 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 6.2%\n",
      "  C: 9.4%\n",
      "  G: 15.6%\n",
      "  T: 21.9%\n",
      "  none/other: 46.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.2\t1\t0 1\n",
      "203\t4\t0.0\t2\t3 1\n",
      "204\t12\t0.0\t2\t11 1\n",
      "205\t15\t0.0\t2\t15\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 78089 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t2\t0.0\t1\t2\n",
      "14\t2\t0.0\t2\t0 2\n",
      "15\t2\t0.0\t2\t2\n",
      "18\t2\t0.0\t2\t2\n",
      "19\t2\t0.0\t2\t0 0 2\n",
      "20\t61\t0.0\t3\t30 20 7 4\n",
      "21\t14486\t0.0\t3\t13728 573 112 73\n",
      "22\t13993\t0.0\t3\t13253 515 152 73\n",
      "23\t13178\t0.0\t3\t12470 521 114 73\n",
      "24\t13074\t0.0\t3\t12441 482 99 52\n",
      "25\t11830\t0.0\t3\t11187 474 99 70\n",
      "26\t11446\t0.0\t3\t10833 439 122 52\n",
      "27\t5\t0.0\t3\t3 1 1\n",
      "46\t1\t0.0\t3\t1\n",
      "52\t1\t0.0\t3\t0 0 0 1\n",
      "58\t1\t0.0\t3\t0 0 0 1\n",
      "62\t2\t0.0\t3\t0 0 1 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121693_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121693_11_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121693_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121693_11_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121693_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121693_11_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121693_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121693_11_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.42 s (15 µs/read; 4.08 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            164,608\n",
      "  Read 1 with adapter:                 163,069 (99.1%)\n",
      "  Read 2 with adapter:                 161,335 (98.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   3 (0.0%)\n",
      "Pairs discarded as untrimmed:            4,763 (2.9%)\n",
      "Pairs written (passing filters):       159,842 (97.1%)\n",
      "\n",
      "Total basepairs processed:    83,950,080 bp\n",
      "  Read 1:    50,205,440 bp\n",
      "  Read 2:    33,744,640 bp\n",
      "Total written (filtered):     74,517,810 bp (88.8%)\n",
      "  Read 1:    45,483,954 bp\n",
      "  Read 2:    29,033,856 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 101 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 82.2%\n",
      "  G: 17.8%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'C' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t38\t2.5\t1\t0 38\n",
      "9\t46\t0.6\t1\t0 46\n",
      "20\t2\t0.0\t3\t2\n",
      "21\t2\t0.0\t3\t2\n",
      "22\t6\t0.0\t3\t6\n",
      "23\t2\t0.0\t3\t2\n",
      "24\t3\t0.0\t3\t3\n",
      "25\t1\t0.0\t3\t1\n",
      "282\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 162968 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.5\t1\t1\n",
      "9\t3\t0.6\t1\t3\n",
      "10\t2\t0.2\t1\t2\n",
      "12\t3\t0.0\t1\t2 1\n",
      "14\t6\t0.0\t2\t4 1 1\n",
      "15\t13\t0.0\t2\t8 1 4\n",
      "16\t12\t0.0\t2\t5 5 2\n",
      "17\t77\t0.0\t2\t40 25 12\n",
      "18\t25475\t0.0\t2\t25001 454 20\n",
      "19\t29054\t0.0\t2\t28509 510 35\n",
      "20\t29775\t0.0\t2\t29239 501 35\n",
      "21\t27186\t0.0\t2\t26720 442 24\n",
      "22\t27872\t0.0\t2\t27381 473 18\n",
      "23\t23465\t0.0\t2\t23095 359 11\n",
      "24\t22\t0.0\t2\t19 1 2\n",
      "32\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 6 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.7%\n",
      "  C: 16.7%\n",
      "  G: 16.7%\n",
      "  T: 0.0%\n",
      "  none/other: 50.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.6\t1\t0 1\n",
      "203\t1\t0.0\t2\t1\n",
      "204\t1\t0.0\t2\t1\n",
      "205\t3\t0.0\t2\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 161329 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.2\t1\t0 1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "14\t2\t0.0\t2\t1 0 1\n",
      "15\t10\t0.0\t2\t8 0 2\n",
      "16\t4\t0.0\t2\t2 1 1\n",
      "17\t3\t0.0\t2\t2 1\n",
      "18\t2\t0.0\t2\t1 0 1\n",
      "19\t4\t0.0\t2\t1 1 2\n",
      "20\t128\t0.0\t3\t49 31 29 19\n",
      "21\t30124\t0.0\t3\t28677 1045 264 138\n",
      "22\t28918\t0.0\t3\t27458 1096 238 126\n",
      "23\t26678\t0.0\t3\t25231 1045 269 133\n",
      "24\t27385\t0.0\t3\t26023 989 236 137\n",
      "25\t24389\t0.0\t3\t22983 1008 254 144\n",
      "26\t23645\t0.0\t3\t22323 957 272 93\n",
      "27\t28\t0.0\t3\t20 3 4 1\n",
      "38\t1\t0.0\t3\t1\n",
      "59\t1\t0.0\t3\t0 0 1\n",
      "62\t3\t0.0\t3\t0 0 1 2\n",
      "65\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121698_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121698_13_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121698_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121698_13_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121698_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121698_13_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121698_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121698_13_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.43 s (30 µs/read; 2.02 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             14,319\n",
      "  Read 1 with adapter:                  14,165 (98.9%)\n",
      "  Read 2 with adapter:                  13,639 (95.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  11 (0.1%)\n",
      "Pairs discarded as untrimmed:              822 (5.7%)\n",
      "Pairs written (passing filters):        13,486 (94.2%)\n",
      "\n",
      "Total basepairs processed:     7,302,690 bp\n",
      "  Read 1:     4,367,295 bp\n",
      "  Read 2:     2,935,395 bp\n",
      "Total written (filtered):      6,284,106 bp (86.1%)\n",
      "  Read 1:     3,837,039 bp\n",
      "  Read 2:     2,447,067 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 13 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 7.7%\n",
      "  C: 38.5%\n",
      "  G: 38.5%\n",
      "  T: 7.7%\n",
      "  none/other: 7.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.2\t1\t0 1\n",
      "9\t4\t0.1\t1\t0 4\n",
      "20\t1\t0.0\t3\t1\n",
      "21\t1\t0.0\t3\t1\n",
      "23\t1\t0.0\t3\t1\n",
      "24\t1\t0.0\t3\t1\n",
      "25\t1\t0.0\t3\t1\n",
      "257\t1\t0.0\t3\t1\n",
      "282\t1\t0.0\t3\t0 1\n",
      "305\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 14152 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t1\t1\n",
      "15\t1\t0.0\t2\t1\n",
      "17\t12\t0.0\t2\t5 6 1\n",
      "18\t2212\t0.0\t2\t2169 43\n",
      "19\t2485\t0.0\t2\t2436 48 1\n",
      "20\t2558\t0.0\t2\t2493 64 1\n",
      "21\t2375\t0.0\t2\t2324 50 1\n",
      "22\t2475\t0.0\t2\t2424 49 2\n",
      "23\t2031\t0.0\t2\t2005 25 1\n",
      "24\t1\t0.0\t2\t1\n",
      "45\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 23 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 4.3%\n",
      "  C: 21.7%\n",
      "  G: 21.7%\n",
      "  T: 8.7%\n",
      "  none/other: 43.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t4\t0.0\t2\t4\n",
      "204\t9\t0.0\t2\t6 3\n",
      "205\t10\t0.0\t2\t10\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 13616 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t1\t0.0\t2\t0 1\n",
      "19\t1\t0.0\t2\t0 1\n",
      "20\t10\t0.0\t3\t3 3 1 3\n",
      "21\t2476\t0.0\t3\t2351 86 24 15\n",
      "22\t2460\t0.0\t3\t2314 101 32 13\n",
      "23\t2282\t0.0\t3\t2129 112 26 15\n",
      "24\t2295\t0.0\t3\t2175 89 20 11\n",
      "25\t2135\t0.0\t3\t2020 78 28 9\n",
      "26\t1952\t0.0\t3\t1841 87 12 12\n",
      "27\t1\t0.0\t3\t0 0 1\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "64\t2\t0.0\t3\t0 1 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121699_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121699_15_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121699_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121699_15_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121699_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121699_15_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121699_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121699_15_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.50 s (24 µs/read; 2.52 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             20,959\n",
      "  Read 1 with adapter:                  20,764 (99.1%)\n",
      "  Read 2 with adapter:                  20,210 (96.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   6 (0.0%)\n",
      "Pairs discarded as untrimmed:              930 (4.4%)\n",
      "Pairs written (passing filters):        20,023 (95.5%)\n",
      "\n",
      "Total basepairs processed:    10,689,090 bp\n",
      "  Read 1:     6,392,495 bp\n",
      "  Read 2:     4,296,595 bp\n",
      "Total written (filtered):      9,331,198 bp (87.3%)\n",
      "  Read 1:     5,696,317 bp\n",
      "  Read 2:     3,634,881 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 14 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 21.4%\n",
      "  G: 64.3%\n",
      "  T: 14.3%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.3\t1\t0 1\n",
      "9\t2\t0.1\t1\t0 2\n",
      "21\t1\t0.0\t3\t1\n",
      "22\t1\t0.0\t3\t1\n",
      "26\t1\t0.0\t3\t1\n",
      "30\t1\t0.0\t3\t1\n",
      "32\t1\t0.0\t3\t1\n",
      "33\t1\t0.0\t3\t1\n",
      "215\t1\t0.0\t3\t1\n",
      "248\t2\t0.0\t3\t2\n",
      "277\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 20750 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t3\t0.0\t2\t1 1 1\n",
      "16\t2\t0.0\t2\t1 0 1\n",
      "17\t9\t0.0\t2\t7 1 1\n",
      "18\t3158\t0.0\t2\t3084 70 4\n",
      "19\t3678\t0.0\t2\t3609 63 6\n",
      "20\t3818\t0.0\t2\t3758 57 3\n",
      "21\t3532\t0.0\t2\t3462 65 5\n",
      "22\t3549\t0.0\t2\t3485 59 5\n",
      "23\t2996\t0.0\t2\t2934 61 1\n",
      "24\t5\t0.0\t2\t5\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 16 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 12.5%\n",
      "  G: 25.0%\n",
      "  T: 25.0%\n",
      "  none/other: 37.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t3\t0.0\t2\t3\n",
      "204\t7\t0.0\t2\t6 1\n",
      "205\t6\t0.0\t2\t6\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 20194 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t2\t0.0\t2\t0 1 1\n",
      "19\t1\t0.0\t2\t1\n",
      "20\t15\t0.0\t3\t6 5 1 3\n",
      "21\t3733\t0.0\t3\t3547 126 39 21\n",
      "22\t3631\t0.0\t3\t3449 129 39 14\n",
      "23\t3299\t0.0\t3\t3103 146 28 22\n",
      "24\t3404\t0.0\t3\t3209 139 34 22\n",
      "25\t3123\t0.0\t3\t2928 142 30 23\n",
      "26\t2979\t0.0\t3\t2792 136 36 15\n",
      "27\t5\t0.0\t3\t2 1 1 1\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "66\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121700_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121700_17_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121700_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121700_17_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121700_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121700_17_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121700_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121700_17_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.52 s (26 µs/read; 2.32 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             20,194\n",
      "  Read 1 with adapter:                  19,984 (99.0%)\n",
      "  Read 2 with adapter:                  19,270 (95.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   8 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,124 (5.6%)\n",
      "Pairs written (passing filters):        19,062 (94.4%)\n",
      "\n",
      "Total basepairs processed:    10,298,940 bp\n",
      "  Read 1:     6,159,170 bp\n",
      "  Read 2:     4,139,770 bp\n",
      "Total written (filtered):      8,882,443 bp (86.2%)\n",
      "  Read 1:     5,423,098 bp\n",
      "  Read 2:     3,459,345 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 9 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 22.2%\n",
      "  C: 22.2%\n",
      "  G: 55.6%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.3\t1\t0 1\n",
      "9\t1\t0.1\t1\t0 1\n",
      "24\t1\t0.0\t3\t1\n",
      "26\t1\t0.0\t3\t1\n",
      "34\t1\t0.0\t3\t1\n",
      "277\t3\t0.0\t3\t3\n",
      "278\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 19975 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t1\t0 1\n",
      "15\t3\t0.0\t2\t3\n",
      "17\t12\t0.0\t2\t7 2 3\n",
      "18\t3069\t0.0\t2\t3002 64 3\n",
      "19\t3549\t0.0\t2\t3490 56 3\n",
      "20\t3683\t0.0\t2\t3617 61 5\n",
      "21\t3281\t0.0\t2\t3224 53 4\n",
      "22\t3480\t0.0\t2\t3413 62 5\n",
      "23\t2892\t0.0\t2\t2843 48 1\n",
      "24\t4\t0.0\t2\t4\n",
      "31\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 24 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 16.7%\n",
      "  G: 29.2%\n",
      "  T: 20.8%\n",
      "  none/other: 33.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t6\t0.0\t2\t6\n",
      "204\t10\t0.0\t2\t9 1\n",
      "205\t8\t0.0\t2\t6 2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 19246 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t2\t1\n",
      "18\t1\t0.0\t2\t0 1\n",
      "19\t2\t0.0\t2\t1 1\n",
      "20\t21\t0.0\t3\t10 6 3 2\n",
      "21\t3560\t0.0\t3\t3352 152 41 15\n",
      "22\t3422\t0.0\t3\t3231 139 32 20\n",
      "23\t3185\t0.0\t3\t2977 150 42 16\n",
      "24\t3237\t0.0\t3\t3007 169 46 15\n",
      "25\t3020\t0.0\t3\t2837 133 34 16\n",
      "26\t2791\t0.0\t3\t2625 129 20 17\n",
      "27\t4\t0.0\t3\t4\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121701_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121701_19_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121701_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121701_19_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121701_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121701_19_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121701_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121701_19_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.18 s (18 µs/read; 3.43 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             67,293\n",
      "  Read 1 with adapter:                  66,636 (99.0%)\n",
      "  Read 2 with adapter:                  65,550 (97.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  14 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,368 (3.5%)\n",
      "Pairs written (passing filters):        64,911 (96.5%)\n",
      "\n",
      "Total basepairs processed:    34,319,430 bp\n",
      "  Read 1:    20,524,365 bp\n",
      "  Read 2:    13,795,065 bp\n",
      "Total written (filtered):     30,254,663 bp (88.2%)\n",
      "  Read 1:    18,466,767 bp\n",
      "  Read 2:    11,787,896 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 44 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 2.3%\n",
      "  C: 47.7%\n",
      "  G: 50.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t10\t1.0\t1\t0 10\n",
      "9\t11\t0.3\t1\t0 11\n",
      "22\t1\t0.0\t3\t1\n",
      "30\t4\t0.0\t3\t4\n",
      "32\t1\t0.0\t3\t1\n",
      "33\t2\t0.0\t3\t2\n",
      "34\t1\t0.0\t3\t1\n",
      "275\t1\t0.0\t3\t1\n",
      "277\t3\t0.0\t3\t3\n",
      "278\t1\t0.0\t3\t1\n",
      "279\t4\t0.0\t3\t4\n",
      "280\t2\t0.0\t3\t2\n",
      "281\t3\t0.0\t3\t3\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 66592 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t1\t1\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t1\t0.0\t1\t0 1\n",
      "14\t6\t0.0\t2\t4 2\n",
      "15\t5\t0.0\t2\t2 2 1\n",
      "16\t4\t0.0\t2\t1 1 2\n",
      "17\t36\t0.0\t2\t19 10 7\n",
      "18\t10342\t0.0\t2\t10168 164 10\n",
      "19\t11793\t0.0\t2\t11595 184 14\n",
      "20\t12312\t0.0\t2\t12115 189 8\n",
      "21\t10918\t0.0\t2\t10714 191 13\n",
      "22\t11422\t0.0\t2\t11250 161 11\n",
      "23\t9740\t0.0\t2\t9600 138 2\n",
      "24\t9\t0.0\t2\t7 2\n",
      "27\t1\t0.0\t2\t1\n",
      "31\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 28 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 14.3%\n",
      "  G: 17.9%\n",
      "  T: 17.9%\n",
      "  none/other: 50.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t5\t0.0\t2\t4 1\n",
      "204\t9\t0.0\t2\t9\n",
      "205\t14\t0.0\t2\t14\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 65522 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "14\t1\t0.0\t2\t0 0 1\n",
      "15\t2\t0.0\t2\t1 0 1\n",
      "18\t1\t0.0\t2\t1\n",
      "19\t4\t0.0\t2\t2 1 1\n",
      "20\t39\t0.0\t3\t18 9 4 8\n",
      "21\t12322\t0.0\t3\t11697 449 118 58\n",
      "22\t11671\t0.0\t3\t11095 425 97 54\n",
      "23\t10860\t0.0\t3\t10267 443 101 49\n",
      "24\t10970\t0.0\t3\t10352 471 85 62\n",
      "25\t10039\t0.0\t3\t9484 406 103 46\n",
      "26\t9598\t0.0\t3\t9007 414 113 64\n",
      "27\t6\t0.0\t3\t6\n",
      "50\t1\t0.0\t3\t0 0 1\n",
      "57\t1\t0.0\t3\t0 0 0 1\n",
      "59\t1\t0.0\t3\t0 0 0 1\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "62\t1\t0.0\t3\t0 0 1\n",
      "63\t2\t0.0\t3\t0 1 1\n",
      "64\t2\t0.0\t3\t0 0 1 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121702_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121702_21_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121702_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121702_21_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121702_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121702_21_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121702_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121702_21_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.43 s (52 µs/read; 1.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              8,334\n",
      "  Read 1 with adapter:                   8,233 (98.8%)\n",
      "  Read 2 with adapter:                   7,983 (95.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   7 (0.1%)\n",
      "Pairs discarded as untrimmed:              443 (5.3%)\n",
      "Pairs written (passing filters):         7,884 (94.6%)\n",
      "\n",
      "Total basepairs processed:     4,250,340 bp\n",
      "  Read 1:     2,541,870 bp\n",
      "  Read 2:     1,708,470 bp\n",
      "Total written (filtered):      3,674,455 bp (86.5%)\n",
      "  Read 1:     2,243,538 bp\n",
      "  Read 2:     1,430,917 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 1 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 100.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "24\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 8232 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t2\t0 0 1\n",
      "16\t1\t0.0\t2\t1\n",
      "17\t3\t0.0\t2\t3\n",
      "18\t1297\t0.0\t2\t1267 30\n",
      "19\t1484\t0.0\t2\t1464 18 2\n",
      "20\t1518\t0.0\t2\t1495 20 3\n",
      "21\t1305\t0.0\t2\t1281 23 1\n",
      "22\t1420\t0.0\t2\t1390 28 2\n",
      "23\t1202\t0.0\t2\t1178 23 1\n",
      "24\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 11 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 18.2%\n",
      "  T: 18.2%\n",
      "  none/other: 63.6%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "204\t4\t0.0\t2\t4\n",
      "205\t7\t0.0\t2\t7\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 7972 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t11\t0.0\t3\t3 5 1 2\n",
      "21\t1453\t0.0\t3\t1388 47 12 6\n",
      "22\t1398\t0.0\t3\t1308 56 24 10\n",
      "23\t1322\t0.0\t3\t1248 58 12 4\n",
      "24\t1306\t0.0\t3\t1240 45 18 3\n",
      "25\t1239\t0.0\t3\t1164 57 13 5\n",
      "26\t1237\t0.0\t3\t1154 67 8 8\n",
      "27\t2\t0.0\t3\t2\n",
      "31\t1\t0.0\t3\t1\n",
      "59\t1\t0.0\t3\t1\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121707_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121707_23_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121707_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121707_23_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121707_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121707_23_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121707_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121707_23_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.53 s (19 µs/read; 3.19 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             81,485\n",
      "  Read 1 with adapter:                  80,768 (99.1%)\n",
      "  Read 2 with adapter:                  79,421 (97.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  10 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,761 (3.4%)\n",
      "Pairs written (passing filters):        78,714 (96.6%)\n",
      "\n",
      "Total basepairs processed:    41,557,350 bp\n",
      "  Read 1:    24,852,925 bp\n",
      "  Read 2:    16,704,425 bp\n",
      "Total written (filtered):     36,688,578 bp (88.3%)\n",
      "  Read 1:    22,394,693 bp\n",
      "  Read 2:    14,293,885 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 38 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 47.4%\n",
      "  C: 10.5%\n",
      "  G: 39.5%\n",
      "  T: 2.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t1.2\t1\t0 3\n",
      "9\t1\t0.3\t1\t0 1\n",
      "20\t1\t0.0\t3\t1\n",
      "21\t1\t0.0\t3\t1\n",
      "24\t1\t0.0\t3\t1\n",
      "29\t4\t0.0\t3\t2 2\n",
      "30\t3\t0.0\t3\t3\n",
      "32\t1\t0.0\t3\t1\n",
      "33\t1\t0.0\t3\t1\n",
      "34\t1\t0.0\t3\t1\n",
      "192\t5\t0.0\t3\t4 1\n",
      "193\t3\t0.0\t3\t3\n",
      "194\t1\t0.0\t3\t1\n",
      "195\t4\t0.0\t3\t4\n",
      "196\t3\t0.0\t3\t3\n",
      "197\t2\t0.0\t3\t2\n",
      "215\t1\t0.0\t3\t0 0 1\n",
      "277\t1\t0.0\t3\t0 0 1\n",
      "287\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 80730 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t2\t0.3\t1\t2\n",
      "10\t3\t0.1\t1\t2 1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "14\t6\t0.0\t2\t4 0 2\n",
      "15\t6\t0.0\t2\t3 1 2\n",
      "16\t3\t0.0\t2\t2 0 1\n",
      "17\t53\t0.0\t2\t34 11 8\n",
      "18\t12514\t0.0\t2\t12316 193 5\n",
      "19\t14571\t0.0\t2\t14298 256 17\n",
      "20\t14675\t0.0\t2\t14381 282 12\n",
      "21\t13350\t0.0\t2\t13116 221 13\n",
      "22\t13670\t0.0\t2\t13440 224 6\n",
      "23\t11864\t0.0\t2\t11665 193 6\n",
      "24\t9\t0.0\t2\t5 2 2\n",
      "28\t1\t0.0\t2\t1\n",
      "30\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 30 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.3%\n",
      "  C: 30.0%\n",
      "  G: 20.0%\n",
      "  T: 13.3%\n",
      "  none/other: 33.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t10\t0.0\t2\t10\n",
      "204\t10\t0.0\t2\t10\n",
      "205\t10\t0.0\t2\t10\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 79391 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.1\t1\t2\n",
      "11\t1\t0.0\t1\t0 1\n",
      "12\t1\t0.0\t1\t0 1\n",
      "13\t1\t0.0\t1\t0 1\n",
      "14\t2\t0.0\t2\t1 1\n",
      "15\t3\t0.0\t2\t1 1 1\n",
      "16\t1\t0.0\t2\t0 1\n",
      "17\t2\t0.0\t2\t1 0 1\n",
      "18\t1\t0.0\t2\t0 1\n",
      "19\t5\t0.0\t2\t2 0 3\n",
      "20\t60\t0.0\t3\t25 16 12 7\n",
      "21\t14780\t0.0\t3\t14012 563 143 62\n",
      "22\t14314\t0.0\t3\t13493 614 137 70\n",
      "23\t13139\t0.0\t3\t12382 556 145 56\n",
      "24\t13365\t0.0\t3\t12641 534 116 74\n",
      "25\t11962\t0.0\t3\t11252 504 134 72\n",
      "26\t11730\t0.0\t3\t10983 546 143 58\n",
      "27\t12\t0.0\t3\t8 2 1 1\n",
      "50\t1\t0.0\t3\t0 0 0 1\n",
      "53\t1\t0.0\t3\t1\n",
      "55\t1\t0.0\t3\t0 0 0 1\n",
      "61\t2\t0.0\t3\t0 0 0 2\n",
      "63\t3\t0.0\t3\t0 0 1 2\n",
      "64\t2\t0.0\t3\t0 0 2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121708_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121708_25_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121708_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121708_25_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121708_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121708_25_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121708_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121708_25_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.68 s (25 µs/read; 2.40 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             27,196\n",
      "  Read 1 with adapter:                  26,931 (99.0%)\n",
      "  Read 2 with adapter:                  26,035 (95.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  18 (0.1%)\n",
      "Pairs discarded as untrimmed:            1,407 (5.2%)\n",
      "Pairs written (passing filters):        25,771 (94.8%)\n",
      "\n",
      "Total basepairs processed:    13,869,960 bp\n",
      "  Read 1:     8,294,780 bp\n",
      "  Read 2:     5,575,180 bp\n",
      "Total written (filtered):     12,010,229 bp (86.6%)\n",
      "  Read 1:     7,333,324 bp\n",
      "  Read 2:     4,676,905 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 12 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 33.3%\n",
      "  G: 58.3%\n",
      "  T: 0.0%\n",
      "  none/other: 8.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t0.4\t1\t0 2\n",
      "9\t2\t0.1\t1\t0 2\n",
      "30\t1\t0.0\t3\t1\n",
      "31\t1\t0.0\t3\t1\n",
      "32\t2\t0.0\t3\t2\n",
      "33\t1\t0.0\t3\t1\n",
      "254\t1\t0.0\t3\t1\n",
      "257\t1\t0.0\t3\t1\n",
      "305\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 26919 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t1\t1\n",
      "14\t2\t0.0\t2\t2\n",
      "16\t1\t0.0\t2\t0 1\n",
      "17\t13\t0.0\t2\t6 6 1\n",
      "18\t4224\t0.0\t2\t4153 66 5\n",
      "19\t4838\t0.0\t2\t4744 94\n",
      "20\t4950\t0.0\t2\t4856 92 2\n",
      "21\t4449\t0.0\t2\t4345 98 6\n",
      "22\t4550\t0.0\t2\t4472 74 4\n",
      "23\t3887\t0.0\t2\t3838 48 1\n",
      "24\t4\t0.0\t2\t3 0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 38 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 5.3%\n",
      "  C: 13.2%\n",
      "  G: 28.9%\n",
      "  T: 7.9%\n",
      "  none/other: 44.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t10\t0.0\t2\t9 1\n",
      "204\t11\t0.0\t2\t11\n",
      "205\t17\t0.0\t2\t13 3 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 25997 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1\t0.0\t2\t1\n",
      "20\t14\t0.0\t3\t5 5 2 2\n",
      "21\t4795\t0.0\t3\t4531 193 46 25\n",
      "22\t4571\t0.0\t3\t4333 178 41 19\n",
      "23\t4434\t0.0\t3\t4146 200 57 31\n",
      "24\t4389\t0.0\t3\t4150 182 40 17\n",
      "25\t3946\t0.0\t3\t3714 164 39 29\n",
      "26\t3839\t0.0\t3\t3599 164 55 21\n",
      "27\t4\t0.0\t3\t1 0 3\n",
      "28\t1\t0.0\t3\t1\n",
      "54\t1\t0.0\t3\t0 0 0 1\n",
      "61\t1\t0.0\t3\t0 0 1\n",
      "64\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121709_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121709_27_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121709_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121709_27_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121709_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121709_27_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121709_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121709_27_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.74 s (23 µs/read; 2.67 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             32,857\n",
      "  Read 1 with adapter:                  32,525 (99.0%)\n",
      "  Read 2 with adapter:                  31,730 (96.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  11 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,442 (4.4%)\n",
      "Pairs written (passing filters):        31,404 (95.6%)\n",
      "\n",
      "Total basepairs processed:    16,757,070 bp\n",
      "  Read 1:    10,021,385 bp\n",
      "  Read 2:     6,735,685 bp\n",
      "Total written (filtered):     14,618,640 bp (87.2%)\n",
      "  Read 1:     8,915,974 bp\n",
      "  Read 2:     5,702,666 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 131 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 44.3%\n",
      "  C: 6.9%\n",
      "  G: 48.1%\n",
      "  T: 0.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t1\t0.0\t3\t0 1\n",
      "24\t1\t0.0\t3\t1\n",
      "25\t1\t0.0\t3\t1\n",
      "29\t1\t0.0\t3\t1\n",
      "30\t4\t0.0\t3\t4\n",
      "31\t4\t0.0\t3\t4\n",
      "32\t8\t0.0\t3\t7 1\n",
      "33\t2\t0.0\t3\t2\n",
      "34\t1\t0.0\t3\t1\n",
      "151\t2\t0.0\t3\t2\n",
      "153\t3\t0.0\t3\t3\n",
      "182\t2\t0.0\t3\t2\n",
      "183\t2\t0.0\t3\t2\n",
      "184\t1\t0.0\t3\t0 1\n",
      "187\t4\t0.0\t3\t4\n",
      "188\t1\t0.0\t3\t1\n",
      "189\t1\t0.0\t3\t1\n",
      "190\t1\t0.0\t3\t1\n",
      "191\t1\t0.0\t3\t1\n",
      "192\t9\t0.0\t3\t9\n",
      "193\t7\t0.0\t3\t7\n",
      "194\t5\t0.0\t3\t5\n",
      "195\t5\t0.0\t3\t5\n",
      "196\t8\t0.0\t3\t8\n",
      "197\t9\t0.0\t3\t9\n",
      "198\t2\t0.0\t3\t2\n",
      "215\t2\t0.0\t3\t2\n",
      "216\t2\t0.0\t3\t1 0 1\n",
      "217\t8\t0.0\t3\t8\n",
      "218\t8\t0.0\t3\t8\n",
      "219\t11\t0.0\t3\t11\n",
      "220\t5\t0.0\t3\t5\n",
      "265\t1\t0.0\t3\t0 1\n",
      "266\t1\t0.0\t3\t1\n",
      "270\t2\t0.0\t3\t2\n",
      "276\t2\t0.0\t3\t2\n",
      "278\t1\t0.0\t3\t1\n",
      "280\t1\t0.0\t3\t1\n",
      "281\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 32394 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.0\t1\t2\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t1\t0.0\t1\t1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t1\t0.0\t2\t0 1\n",
      "16\t3\t0.0\t2\t1 1 1\n",
      "17\t20\t0.0\t2\t9 6 5\n",
      "18\t5045\t0.0\t2\t4958 83 4\n",
      "19\t5807\t0.0\t2\t5693 109 5\n",
      "20\t5921\t0.0\t2\t5814 101 6\n",
      "21\t5289\t0.0\t2\t5195 92 2\n",
      "22\t5568\t0.0\t2\t5485 77 6\n",
      "23\t4729\t0.0\t2\t4661 66 2\n",
      "24\t5\t0.0\t2\t5\n",
      "29\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 19 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 15.8%\n",
      "  G: 21.1%\n",
      "  T: 5.3%\n",
      "  none/other: 57.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "92\t1\t0.0\t2\t1\n",
      "144\t1\t0.0\t2\t1\n",
      "203\t1\t0.0\t2\t1\n",
      "204\t5\t0.0\t2\t4 1\n",
      "205\t11\t0.0\t2\t10 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 31711 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t1\t1\n",
      "15\t4\t0.0\t2\t2 2\n",
      "19\t1\t0.0\t2\t1\n",
      "20\t24\t0.0\t3\t12 4 5 3\n",
      "21\t5800\t0.0\t3\t5470 241 58 31\n",
      "22\t5699\t0.0\t3\t5373 232 61 33\n",
      "23\t5351\t0.0\t3\t5010 245 58 38\n",
      "24\t5376\t0.0\t3\t5115 196 43 22\n",
      "25\t4814\t0.0\t3\t4536 206 55 17\n",
      "26\t4630\t0.0\t3\t4354 207 42 27\n",
      "27\t5\t0.0\t3\t2 2 0 1\n",
      "63\t2\t0.0\t3\t0 0 0 2\n",
      "64\t2\t0.0\t3\t0 1 0 1\n",
      "65\t1\t0.0\t3\t0 0 0 1\n",
      "66\t1\t0.0\t3\t0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121710_28_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121710_29_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121710_28_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121710_29_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121710_28_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-vwov4qgj/210421121710_29_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121710_28_L001_R1_001.fastq.gz /tmp/qiime2-archive-wak4v54g/c352367b-b277-4bd2-b830-2e674d1b4228/data/210421121710_29_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.36 s (74 µs/read; 0.81 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              4,908\n",
      "  Read 1 with adapter:                   4,853 (98.9%)\n",
      "  Read 2 with adapter:                   4,578 (93.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   3 (0.1%)\n",
      "Pairs discarded as untrimmed:              377 (7.7%)\n",
      "Pairs written (passing filters):         4,528 (92.3%)\n",
      "\n",
      "Total basepairs processed:     2,503,080 bp\n",
      "  Read 1:     1,496,940 bp\n",
      "  Read 2:     1,006,140 bp\n",
      "Total written (filtered):      2,106,079 bp (84.1%)\n",
      "  Read 1:     1,284,585 bp\n",
      "  Read 2:       821,494 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 31 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.2%\n",
      "  C: 25.8%\n",
      "  G: 58.1%\n",
      "  T: 12.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.0\t1\t0 1\n",
      "21\t1\t0.0\t3\t1\n",
      "22\t1\t0.0\t3\t1\n",
      "30\t1\t0.0\t3\t1\n",
      "31\t1\t0.0\t3\t1\n",
      "32\t5\t0.0\t3\t5\n",
      "33\t5\t0.0\t3\t5\n",
      "34\t1\t0.0\t3\t1\n",
      "132\t1\t0.0\t3\t1\n",
      "194\t1\t0.0\t3\t1\n",
      "210\t1\t0.0\t3\t0 0 1\n",
      "274\t1\t0.0\t3\t1\n",
      "279\t1\t0.0\t3\t1\n",
      "281\t6\t0.0\t3\t5 1\n",
      "283\t2\t0.0\t3\t2\n",
      "284\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 4822 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t2\t0.0\t2\t0 0 2\n",
      "17\t2\t0.0\t2\t0 2\n",
      "18\t769\t0.0\t2\t754 14 1\n",
      "19\t841\t0.0\t2\t828 11 2\n",
      "20\t839\t0.0\t2\t823 14 2\n",
      "21\t779\t0.0\t2\t766 13\n",
      "22\t839\t0.0\t2\t819 19 1\n",
      "23\t750\t0.0\t2\t734 15 1\n",
      "24\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 8 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 12.5%\n",
      "  G: 25.0%\n",
      "  T: 25.0%\n",
      "  none/other: 37.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t1\t0.0\t2\t1\n",
      "204\t4\t0.0\t2\t3 1\n",
      "205\t3\t0.0\t2\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 4570 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t2\t1\n",
      "20\t6\t0.0\t3\t2 3 0 1\n",
      "21\t864\t0.0\t3\t819 30 12 3\n",
      "22\t809\t0.0\t3\t759 31 13 6\n",
      "23\t717\t0.0\t3\t673 31 6 7\n",
      "24\t814\t0.0\t3\t752 44 12 6\n",
      "25\t686\t0.0\t3\t638 33 6 9\n",
      "26\t671\t0.0\t3\t613 38 13 7\n",
      "51\t1\t0.0\t3\t0 1\n",
      "64\t1\t0.0\t3\t0 0 0 1\n"
     ]
    }
   ],
   "source": [
    "if trim:    \n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = trim_paired(\n",
    "        demultiplexed_sequences=artifact,\n",
    "        front_f=forward_primer,\n",
    "        front_r=reverse_primer,\n",
    "        adapter_f=reverse_reverse_complement,\n",
    "        adapter_r=forward_reverse_complement,\n",
    "        cores=threads,\n",
    "        overlap=trim['overlap'],\n",
    "        indels=False,\n",
    "        match_read_wildcards=True,\n",
    "        match_adapter_wildcards=True,\n",
    "        error_rate=0.15,\n",
    "        discard_untrimmed=True,\n",
    "    ).trimmed_sequences\n",
    "\n",
    "    demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "    demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')\n",
    "    \n",
    "    demux.visualizers.summarize(res).visualization\n",
    "    res.save(demux_file_trim)\n",
    "    Visualization.save(demux.visualizers.summarize(res).visualization, filepath=demux_view_trim)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 79.310413,
   "end_time": "2022-09-29T22:17:02.411825",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-steps/step-01-prepare-data.ipynb",
   "output_path": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-HSD-NCxHSD-NR-trim/nb-executed-steps/step-01-prepare-data-ana-flavia-HSD-NCxHSD-NR-trim.ipynb",
   "parameters": {
    "base_dir": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri",
    "class_col": "sample-group",
    "classifier_file": "/home/lauro/nupeb/dados_brutos_rede_genoma/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza",
    "experiment_name": "ana-flavia-HSD-NCxHSD-NR-trim",
    "manifest_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-HSD-NCxHSD-NR.csv",
    "metadata_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-HSD-NCxHSD-NR.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2022-09-29T22:15:43.101412",
   "version": "2.3.4"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}