{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "eb75f11d",
   "metadata": {
    "papermill": {
     "duration": 0.025183,
     "end_time": "2023-08-23T18:56:50.946454",
     "exception": false,
     "start_time": "2023-08-23T18:56:50.921271",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fd9ea144",
   "metadata": {
    "papermill": {
     "duration": 0.031841,
     "end_time": "2023-08-23T18:56:51.002804",
     "exception": false,
     "start_time": "2023-08-23T18:56:50.970963",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1413bdb8",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-10-14T14:33:37.863523Z",
     "start_time": "2022-10-14T14:33:14.145712Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:56:51.058782Z",
     "iopub.status.busy": "2023-08-23T18:56:51.058333Z",
     "iopub.status.idle": "2023-08-23T18:56:57.576459Z",
     "shell.execute_reply": "2023-08-23T18:56:57.575753Z"
    },
    "papermill": {
     "duration": 6.545778,
     "end_time": "2023-08-23T18:56:57.579472",
     "exception": false,
     "start_time": "2023-08-23T18:56:51.033694",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization, Metadata\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired, trim_single\n",
    "from qiime2.plugins.demux.methods import filter_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a87a854",
   "metadata": {
    "papermill": {
     "duration": 0.015201,
     "end_time": "2023-08-23T18:56:57.610364",
     "exception": false,
     "start_time": "2023-08-23T18:56:57.595163",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "bc547d73",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.594341Z",
     "start_time": "2022-06-28T18:50:41.591544Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:56:57.630780Z",
     "iopub.status.busy": "2023-08-23T18:56:57.630432Z",
     "iopub.status.idle": "2023-08-23T18:56:57.634665Z",
     "shell.execute_reply": "2023-08-23T18:56:57.634167Z"
    },
    "papermill": {
     "duration": 0.016189,
     "end_time": "2023-08-23T18:56:57.636123",
     "exception": false,
     "start_time": "2023-08-23T18:56:57.619934",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'parameter-set-01.yaml')\n",
    "experiment_name = 'exp-01'\n",
    "base_dir = os.path.join('/', 'home', 'username', 'pipeline-dir')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'manifest.csv')\n",
    "replace_files = False\n",
    "trim = None\n",
    "metadata_file = None\n",
    "threads = 1\n",
    "demux_file = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9a99ce91",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T18:56:57.656077Z",
     "iopub.status.busy": "2023-08-23T18:56:57.655864Z",
     "iopub.status.idle": "2023-08-23T18:56:57.659705Z",
     "shell.execute_reply": "2023-08-23T18:56:57.659248Z"
    },
    "papermill": {
     "duration": 0.015306,
     "end_time": "2023-08-23T18:56:57.661079",
     "exception": false,
     "start_time": "2023-08-23T18:56:57.645773",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "experiment_name = \"ana-flavia-STD-NRxHSD-NR-trim\"\n",
    "base_dir = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri\"\n",
    "manifest_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-STD-NRxHSD-NR.csv\"\n",
    "metadata_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-STD-NRxHSD-NR.tsv\"\n",
    "class_col = \"group-id\"\n",
    "classifier_file = \"/home/lauro/nupeb/rede-micro/models/silva-138-99-nb-classifier.qza\"\n",
    "top_n = 20\n",
    "replace_files = False\n",
    "phred = 20\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n",
    "overlap = 12\n",
    "threads = 6\n",
    "trim = {\n",
    "    \"overlap\": 8,\n",
    "    \"forward_primer\": \"CCTACGGGRSGCAGCAG\",\n",
    "    \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\",\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5750065c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-18T15:40:31.070885Z",
     "start_time": "2023-08-18T15:40:30.611671Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:56:57.681055Z",
     "iopub.status.busy": "2023-08-23T18:56:57.680842Z",
     "iopub.status.idle": "2023-08-23T18:56:57.684959Z",
     "shell.execute_reply": "2023-08-23T18:56:57.684213Z"
    },
    "papermill": {
     "duration": 0.015715,
     "end_time": "2023-08-23T18:56:57.686426",
     "exception": false,
     "start_time": "2023-08-23T18:56:57.670711",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "data_dir =  os.path.abspath(os.path.join(base_dir, 'data'))\n",
    "raw_data_dir =  os.path.abspath(os.path.join(data_dir, 'raw'))\n",
    "interim_data_dir =  os.path.abspath(os.path.join(data_dir, 'interim'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b5e37006",
   "metadata": {
    "papermill": {
     "duration": 0.009813,
     "end_time": "2023-08-23T18:56:57.710465",
     "exception": false,
     "start_time": "2023-08-23T18:56:57.700652",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a67e876e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.632125Z",
     "start_time": "2022-06-28T18:50:41.619710Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:56:57.741790Z",
     "iopub.status.busy": "2023-08-23T18:56:57.741492Z",
     "iopub.status.idle": "2023-08-23T18:56:57.748380Z",
     "shell.execute_reply": "2023-08-23T18:56:57.747615Z"
    },
    "papermill": {
     "duration": 0.029682,
     "end_time": "2023-08-23T18:56:57.750063",
     "exception": false,
     "start_time": "2023-08-23T18:56:57.720381",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New qiime-artifacts folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-STD-NRxHSD-NR-trim/qiime-artifacts\n",
      "New img folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-STD-NRxHSD-NR-trim/imgs\n"
     ]
    }
   ],
   "source": [
    "# Create path if it not exists\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exists\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')\n",
    "\n",
    "# Define the output artifact full path\n",
    "if demux_file is None:\n",
    "    demux_file = os.path.join(out_dir, 'demux-paired.qza')\n",
    "demux_view = os.path.join(out_dir, 'demux-paired.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "19133f26",
   "metadata": {
    "papermill": {
     "duration": 0.010873,
     "end_time": "2023-08-23T18:56:57.772918",
     "exception": false,
     "start_time": "2023-08-23T18:56:57.762045",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file of `SampleData[PairedEndSequencesWithQuality]`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8c38e6c3",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T18:56:57.811292Z",
     "iopub.status.busy": "2023-08-23T18:56:57.810894Z",
     "iopub.status.idle": "2023-08-23T18:56:57.833753Z",
     "shell.execute_reply": "2023-08-23T18:56:57.833245Z"
    },
    "papermill": {
     "duration": 0.036408,
     "end_time": "2023-08-23T18:56:57.835676",
     "exception": false,
     "start_time": "2023-08-23T18:56:57.799268",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "manifest_df = pd.read_csv(manifest_file)\n",
    "n_directions = len(manifest_df['direction'].unique())\n",
    "if n_directions == 1:\n",
    "    d_type = 'SampleData[SequencesWithQuality]'\n",
    "    v_type = 'SingleEndFastqManifestPhred33'\n",
    "elif n_directions == 2:\n",
    "    d_type = 'SampleData[PairedEndSequencesWithQuality]'\n",
    "    v_type = 'PairedEndFastqManifestPhred33'\n",
    "else:\n",
    "    print(f'ERROR: invalid number of directions {n_directions}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3ba43bb2",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T18:56:57.864240Z",
     "iopub.status.busy": "2023-08-23T18:56:57.863879Z",
     "iopub.status.idle": "2023-08-23T18:56:57.890835Z",
     "shell.execute_reply": "2023-08-23T18:56:57.889936Z"
    },
    "papermill": {
     "duration": 0.040645,
     "end_time": "2023-08-23T18:56:57.892785",
     "exception": false,
     "start_time": "2023-08-23T18:56:57.852140",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the metadata file is not defined, use the default metadata file\n",
    "metadata_qa = Metadata.load(metadata_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "26448a58",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:44:58.640338Z",
     "start_time": "2022-06-28T18:50:41.633657Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:56:57.920421Z",
     "iopub.status.busy": "2023-08-23T18:56:57.920136Z",
     "iopub.status.idle": "2023-08-23T18:57:47.271285Z",
     "shell.execute_reply": "2023-08-23T18:57:47.270563Z"
    },
    "papermill": {
     "duration": 49.365652,
     "end_time": "2023-08-23T18:57:47.274341",
     "exception": false,
     "start_time": "2023-08-23T18:56:57.908689",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    interim_demux_path = os.path.join(interim_data_dir, \"demux-paired.qza\")\n",
    "    # Load Artifact from interim folder\n",
    "    if os.path.isfile(interim_demux_path):\n",
    "        artifact = Artifact.load(interim_demux_path)\n",
    "    # Create new Artifact using Manifest\n",
    "    else:\n",
    "        artifact = Artifact.import_data(d_type, manifest_file, view_type=v_type)\n",
    "        if not os.path.isfile(interim_demux_path):\n",
    "            artifact.save(interim_demux_path)\n",
    "    # Filter and Save Artifact\n",
    "    artifact = filter_samples(demux=artifact, metadata=metadata_qa).filtered_demux\n",
    "    artifact.save(demux_file)\n",
    "    # Visualization\n",
    "    if os.path.isfile(demux_view):\n",
    "        demux_view_obj = Visualization.load(demux_view)\n",
    "    else:\n",
    "        demux_view_obj = demux.visualizers.summarize(artifact).visualization\n",
    "        Visualization.save(demux_view_obj, filepath=demux_view)\n",
    "elif os.path.isfile(demux_file):\n",
    "    # Load Artifact\n",
    "    artifact = Artifact.load(demux_file)\n",
    "    artifact = filter_samples(demux=artifact, metadata=metadata_qa).filtered_demux\n",
    "    # Visualization\n",
    "    if os.path.isfile(demux_view):\n",
    "        demux_view_obj = Visualization.load(demux_view)\n",
    "    else:\n",
    "        demux_view_obj = demux.visualizers.summarize(artifact).visualization\n",
    "        Visualization.save(demux_view_obj, filepath=demux_view)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f4a942a",
   "metadata": {
    "papermill": {
     "duration": 0.084253,
     "end_time": "2023-08-23T18:57:47.375561",
     "exception": false,
     "start_time": "2023-08-23T18:57:47.291308",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b6fa7ee5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.609737Z",
     "start_time": "2022-06-28T19:47:56.604820Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:57:47.480316Z",
     "iopub.status.busy": "2023-08-23T18:57:47.480060Z",
     "iopub.status.idle": "2023-08-23T18:57:47.485008Z",
     "shell.execute_reply": "2023-08-23T18:57:47.484188Z"
    },
    "papermill": {
     "duration": 0.065776,
     "end_time": "2023-08-23T18:57:47.486800",
     "exception": false,
     "start_time": "2023-08-23T18:57:47.421024",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<visualization: Visualization uuid: c92ea228-809f-4760-98a6-787e0e3f05e0>\n"
     ]
    }
   ],
   "source": [
    "print(demux_view_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "63a8abf8",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.704841Z",
     "start_time": "2022-06-28T19:47:56.611846Z"
    },
    "execution": {
     "iopub.execute_input": "2023-08-23T18:57:47.512081Z",
     "iopub.status.busy": "2023-08-23T18:57:47.511787Z",
     "iopub.status.idle": "2023-08-23T18:57:47.521653Z",
     "shell.execute_reply": "2023-08-23T18:57:47.521133Z"
    },
    "papermill": {
     "duration": 0.023847,
     "end_time": "2023-08-23T18:57:47.523278",
     "exception": false,
     "start_time": "2023-08-23T18:57:47.499431",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-lnp77asq')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: c92ea228-809f-4760-98a6-787e0e3f05e0>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82ed7821",
   "metadata": {
    "papermill": {
     "duration": 0.011891,
     "end_time": "2023-08-23T18:57:47.547899",
     "exception": false,
     "start_time": "2023-08-23T18:57:47.536008",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0286e233",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-08-23T18:57:47.572129Z",
     "iopub.status.busy": "2023-08-23T18:57:47.571837Z",
     "iopub.status.idle": "2023-08-23T18:58:43.945649Z",
     "shell.execute_reply": "2023-08-23T18:58:43.944457Z"
    },
    "papermill": {
     "duration": 56.389372,
     "end_time": "2023-08-23T18:58:43.948908",
     "exception": false,
     "start_time": "2023-08-23T18:57:47.559536",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121685_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121685_25_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121685_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121685_25_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121685_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121685_25_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121685_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121685_25_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.69 s (11 µs/read; 5.67 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            254,393\n",
      "  Read 1 with adapter:                 247,382 (97.2%)\n",
      "  Read 2 with adapter:                 234,645 (92.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           26,087 (10.3%)\n",
      "Pairs written (passing filters):       228,306 (89.7%)\n",
      "\n",
      "Total basepairs processed:   129,740,430 bp\n",
      "  Read 1:    77,589,865 bp\n",
      "  Read 2:    52,150,565 bp\n",
      "Total written (filtered):    106,436,280 bp (82.0%)\n",
      "  Read 1:    64,964,669 bp\n",
      "  Read 2:    41,471,611 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 247382 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t3.9\t0\t3\n",
      "9\t3\t1.0\t0\t3\n",
      "10\t2\t0.2\t0\t2\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t6\t0.0\t0\t6\n",
      "13\t5\t0.0\t0\t5\n",
      "14\t13\t0.0\t0\t13\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t82\t0.0\t0\t82\n",
      "18\t38495\t0.0\t0\t38495\n",
      "19\t44787\t0.0\t0\t44787\n",
      "20\t45421\t0.0\t0\t45421\n",
      "21\t40692\t0.0\t0\t40692\n",
      "22\t41815\t0.0\t0\t41815\n",
      "23\t36011\t0.0\t0\t36011\n",
      "24\t29\t0.0\t0\t29\n",
      "25\t2\t0.0\t0\t2\n",
      "47\t2\t0.0\t0\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 234645 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t7\t0.0\t0\t7\n",
      "19\t5\t0.0\t0\t5\n",
      "20\t74\t0.0\t0\t74\n",
      "21\t44245\t0.0\t0\t44245\n",
      "22\t42188\t0.0\t0\t42188\n",
      "23\t38808\t0.0\t0\t38808\n",
      "24\t39553\t0.0\t0\t39553\n",
      "25\t35622\t0.0\t0\t35622\n",
      "26\t34102\t0.0\t0\t34102\n",
      "27\t20\t0.0\t0\t20\n",
      "33\t1\t0.0\t0\t1\n",
      "44\t1\t0.0\t0\t1\n",
      "64\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121686_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121686_27_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121686_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121686_27_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121686_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121686_27_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121686_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121686_27_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.18 s (11 µs/read; 5.28 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            103,905\n",
      "  Read 1 with adapter:                 101,232 (97.4%)\n",
      "  Read 2 with adapter:                  96,332 (92.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           10,027 (9.7%)\n",
      "Pairs written (passing filters):        93,878 (90.3%)\n",
      "\n",
      "Total basepairs processed:    52,991,550 bp\n",
      "  Read 1:    31,691,025 bp\n",
      "  Read 2:    21,300,525 bp\n",
      "Total written (filtered):     43,766,915 bp (82.6%)\n",
      "  Read 1:    26,714,313 bp\n",
      "  Read 2:    17,052,602 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 101232 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.6\t0\t1\n",
      "9\t2\t0.4\t0\t2\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t5\t0.0\t0\t5\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t8\t0.0\t0\t8\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t32\t0.0\t0\t32\n",
      "18\t16068\t0.0\t0\t16068\n",
      "19\t18255\t0.0\t0\t18255\n",
      "20\t18746\t0.0\t0\t18746\n",
      "21\t16620\t0.0\t0\t16620\n",
      "22\t16935\t0.0\t0\t16935\n",
      "23\t14537\t0.0\t0\t14537\n",
      "24\t12\t0.0\t0\t12\n",
      "40\t1\t0.0\t0\t1\n",
      "44\t1\t0.0\t0\t1\n",
      "51\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 96332 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.6\t0\t1\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t2\t0.0\t0\t2\n",
      "19\t6\t0.0\t0\t6\n",
      "20\t47\t0.0\t0\t47\n",
      "21\t18052\t0.0\t0\t18052\n",
      "22\t17427\t0.0\t0\t17427\n",
      "23\t15720\t0.0\t0\t15720\n",
      "24\t16399\t0.0\t0\t16399\n",
      "25\t14599\t0.0\t0\t14599\n",
      "26\t14056\t0.0\t0\t14056\n",
      "27\t15\t0.0\t0\t15\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121687_28_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121687_29_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121687_28_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121687_29_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121687_28_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121687_29_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121687_28_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121687_29_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.81 s (12 µs/read; 4.92 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             66,368\n",
      "  Read 1 with adapter:                  64,521 (97.2%)\n",
      "  Read 2 with adapter:                  61,032 (92.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            7,022 (10.6%)\n",
      "Pairs written (passing filters):        59,346 (89.4%)\n",
      "\n",
      "Total basepairs processed:    33,847,680 bp\n",
      "  Read 1:    20,242,240 bp\n",
      "  Read 2:    13,605,440 bp\n",
      "Total written (filtered):     27,666,882 bp (81.7%)\n",
      "  Read 1:    16,886,558 bp\n",
      "  Read 2:    10,780,324 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 64521 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t0\t1\n",
      "11\t2\t0.0\t0\t2\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t3\t0.0\t0\t3\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t22\t0.0\t0\t22\n",
      "18\t9869\t0.0\t0\t9869\n",
      "19\t11662\t0.0\t0\t11662\n",
      "20\t11874\t0.0\t0\t11874\n",
      "21\t10704\t0.0\t0\t10704\n",
      "22\t11032\t0.0\t0\t11032\n",
      "23\t9337\t0.0\t0\t9337\n",
      "24\t9\t0.0\t0\t9\n",
      "26\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 61032 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "19\t3\t0.0\t0\t3\n",
      "20\t16\t0.0\t0\t16\n",
      "21\t11652\t0.0\t0\t11652\n",
      "22\t10779\t0.0\t0\t10779\n",
      "23\t10214\t0.0\t0\t10214\n",
      "24\t10220\t0.0\t0\t10220\n",
      "25\t9211\t0.0\t0\t9211\n",
      "26\t8923\t0.0\t0\t8923\n",
      "27\t7\t0.0\t0\t7\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121691_36_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121691_37_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121691_36_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121691_37_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121691_36_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121691_37_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121691_36_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121691_37_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.33 s (10 µs/read; 5.77 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            224,175\n",
      "  Read 1 with adapter:                 217,928 (97.2%)\n",
      "  Read 2 with adapter:                 207,950 (92.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           21,909 (9.8%)\n",
      "Pairs written (passing filters):       202,266 (90.2%)\n",
      "\n",
      "Total basepairs processed:   114,329,250 bp\n",
      "  Read 1:    68,373,375 bp\n",
      "  Read 2:    45,955,875 bp\n",
      "Total written (filtered):     94,295,455 bp (82.5%)\n",
      "  Read 1:    57,554,676 bp\n",
      "  Read 2:    36,740,779 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 217928 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t3.4\t0\t1\n",
      "9\t4\t0.9\t0\t4\n",
      "10\t1\t0.2\t0\t1\n",
      "11\t5\t0.1\t0\t5\n",
      "13\t2\t0.0\t0\t2\n",
      "14\t7\t0.0\t0\t7\n",
      "15\t5\t0.0\t0\t5\n",
      "16\t7\t0.0\t0\t7\n",
      "17\t51\t0.0\t0\t51\n",
      "18\t33939\t0.0\t0\t33939\n",
      "19\t39206\t0.0\t0\t39206\n",
      "20\t40233\t0.0\t0\t40233\n",
      "21\t35873\t0.0\t0\t35873\n",
      "22\t37230\t0.0\t0\t37230\n",
      "23\t31336\t0.0\t0\t31336\n",
      "24\t21\t0.0\t0\t21\n",
      "27\t1\t0.0\t0\t1\n",
      "29\t2\t0.0\t0\t2\n",
      "31\t1\t0.0\t0\t1\n",
      "33\t1\t0.0\t0\t1\n",
      "52\t1\t0.0\t0\t1\n",
      "234\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 207950 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t2\t0.0\t0\t2\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "15\t4\t0.0\t0\t4\n",
      "16\t2\t0.0\t0\t2\n",
      "19\t4\t0.0\t0\t4\n",
      "20\t64\t0.0\t0\t64\n",
      "21\t38764\t0.0\t0\t38764\n",
      "22\t37770\t0.0\t0\t37770\n",
      "23\t34391\t0.0\t0\t34391\n",
      "24\t34925\t0.0\t0\t34925\n",
      "25\t31714\t0.0\t0\t31714\n",
      "26\t30279\t0.0\t0\t30279\n",
      "27\t25\t0.0\t0\t25\n",
      "28\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "53\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121692_38_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121692_39_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121692_38_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121692_39_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121692_38_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121692_39_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121692_38_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121692_39_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.92 s (11 µs/read; 5.22 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             80,216\n",
      "  Read 1 with adapter:                  77,965 (97.2%)\n",
      "  Read 2 with adapter:                  73,953 (92.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            8,302 (10.3%)\n",
      "Pairs written (passing filters):        71,914 (89.7%)\n",
      "\n",
      "Total basepairs processed:    40,910,160 bp\n",
      "  Read 1:    24,465,880 bp\n",
      "  Read 2:    16,444,280 bp\n",
      "Total written (filtered):     33,525,312 bp (81.9%)\n",
      "  Read 1:    20,462,757 bp\n",
      "  Read 2:    13,062,555 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 77965 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.2\t0\t1\n",
      "9\t1\t0.3\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "16\t3\t0.0\t0\t3\n",
      "17\t23\t0.0\t0\t23\n",
      "18\t12006\t0.0\t0\t12006\n",
      "19\t14026\t0.0\t0\t14026\n",
      "20\t14281\t0.0\t0\t14281\n",
      "21\t13017\t0.0\t0\t13017\n",
      "22\t13328\t0.0\t0\t13328\n",
      "23\t11273\t0.0\t0\t11273\n",
      "24\t4\t0.0\t0\t4\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 73953 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t2\t0.0\t0\t2\n",
      "15\t2\t0.0\t0\t2\n",
      "18\t2\t0.0\t0\t2\n",
      "20\t30\t0.0\t0\t30\n",
      "21\t13728\t0.0\t0\t13728\n",
      "22\t13253\t0.0\t0\t13253\n",
      "23\t12470\t0.0\t0\t12470\n",
      "24\t12441\t0.0\t0\t12441\n",
      "25\t11187\t0.0\t0\t11187\n",
      "26\t10833\t0.0\t0\t10833\n",
      "27\t3\t0.0\t0\t3\n",
      "46\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121693_40_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121693_41_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121693_40_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121693_41_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121693_40_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121693_41_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121693_40_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121693_41_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.68 s (10 µs/read; 5.88 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            164,608\n",
      "  Read 1 with adapter:                 160,048 (97.2%)\n",
      "  Read 2 with adapter:                 152,781 (92.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           15,982 (9.7%)\n",
      "Pairs written (passing filters):       148,626 (90.3%)\n",
      "\n",
      "Total basepairs processed:    83,950,080 bp\n",
      "  Read 1:    50,205,440 bp\n",
      "  Read 2:    33,744,640 bp\n",
      "Total written (filtered):     69,288,385 bp (82.5%)\n",
      "  Read 1:    42,290,904 bp\n",
      "  Read 2:    26,997,481 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 160048 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.5\t0\t1\n",
      "9\t3\t0.6\t0\t3\n",
      "10\t2\t0.2\t0\t2\n",
      "12\t2\t0.0\t0\t2\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t5\t0.0\t0\t5\n",
      "17\t40\t0.0\t0\t40\n",
      "18\t25002\t0.0\t0\t25002\n",
      "19\t28512\t0.0\t0\t28512\n",
      "20\t29241\t0.0\t0\t29241\n",
      "21\t26726\t0.0\t0\t26726\n",
      "22\t27383\t0.0\t0\t27383\n",
      "23\t23098\t0.0\t0\t23098\n",
      "24\t19\t0.0\t0\t19\n",
      "32\t2\t0.0\t0\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 152781 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t8\t0.0\t0\t8\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t2\t0.0\t0\t2\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t49\t0.0\t0\t49\n",
      "21\t28677\t0.0\t0\t28677\n",
      "22\t27458\t0.0\t0\t27458\n",
      "23\t25231\t0.0\t0\t25231\n",
      "24\t26023\t0.0\t0\t26023\n",
      "25\t22983\t0.0\t0\t22983\n",
      "26\t22323\t0.0\t0\t22323\n",
      "27\t20\t0.0\t0\t20\n",
      "38\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121703_60_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121703_61_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121703_60_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121703_61_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121703_60_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121703_61_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121703_60_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121703_61_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.41 s (17 µs/read; 3.53 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             24,189\n",
      "  Read 1 with adapter:                  23,503 (97.2%)\n",
      "  Read 2 with adapter:                  21,685 (89.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            3,102 (12.8%)\n",
      "Pairs written (passing filters):        21,087 (87.2%)\n",
      "\n",
      "Total basepairs processed:    12,336,390 bp\n",
      "  Read 1:     7,377,645 bp\n",
      "  Read 2:     4,958,745 bp\n",
      "Total written (filtered):      9,830,774 bp (79.7%)\n",
      "  Read 1:     6,000,254 bp\n",
      "  Read 2:     3,830,520 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 23503 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t4\t0.0\t0\t4\n",
      "18\t3615\t0.0\t0\t3615\n",
      "19\t4126\t0.0\t0\t4126\n",
      "20\t4339\t0.0\t0\t4339\n",
      "21\t3919\t0.0\t0\t3919\n",
      "22\t4058\t0.0\t0\t4058\n",
      "23\t3438\t0.0\t0\t3438\n",
      "24\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 21685 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t10\t0.0\t0\t10\n",
      "21\t4101\t0.0\t0\t4101\n",
      "22\t3941\t0.0\t0\t3941\n",
      "23\t3556\t0.0\t0\t3556\n",
      "24\t3609\t0.0\t0\t3609\n",
      "25\t3363\t0.0\t0\t3363\n",
      "26\t3097\t0.0\t0\t3097\n",
      "27\t4\t0.0\t0\t4\n",
      "43\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121704_62_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121704_63_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121704_62_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121704_63_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121704_62_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121704_63_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121704_62_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121704_63_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.24 s (43 µs/read; 1.39 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              5,549\n",
      "  Read 1 with adapter:                   5,397 (97.3%)\n",
      "  Read 2 with adapter:                   5,015 (90.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              667 (12.0%)\n",
      "Pairs written (passing filters):         4,882 (88.0%)\n",
      "\n",
      "Total basepairs processed:     2,829,990 bp\n",
      "  Read 1:     1,692,445 bp\n",
      "  Read 2:     1,137,545 bp\n",
      "Total written (filtered):      2,276,001 bp (80.4%)\n",
      "  Read 1:     1,389,152 bp\n",
      "  Read 2:       886,849 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 5397 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t858\t0.0\t0\t858\n",
      "19\t967\t0.0\t0\t967\n",
      "20\t1009\t0.0\t0\t1009\n",
      "21\t860\t0.0\t0\t860\n",
      "22\t903\t0.0\t0\t903\n",
      "23\t797\t0.0\t0\t797\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 5015 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "20\t2\t0.0\t0\t2\n",
      "21\t964\t0.0\t0\t964\n",
      "22\t871\t0.0\t0\t871\n",
      "23\t837\t0.0\t0\t837\n",
      "24\t857\t0.0\t0\t857\n",
      "25\t780\t0.0\t0\t780\n",
      "26\t702\t0.0\t0\t702\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121705_64_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121705_65_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121705_64_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121705_65_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121705_64_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121705_65_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121705_64_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121705_65_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.46 s (10 µs/read; 5.97 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            344,337\n",
      "  Read 1 with adapter:                 334,766 (97.2%)\n",
      "  Read 2 with adapter:                 317,372 (92.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           35,670 (10.4%)\n",
      "Pairs written (passing filters):       308,667 (89.6%)\n",
      "\n",
      "Total basepairs processed:   175,611,870 bp\n",
      "  Read 1:   105,022,785 bp\n",
      "  Read 2:    70,589,085 bp\n",
      "Total written (filtered):    143,899,876 bp (81.9%)\n",
      "  Read 1:    87,831,910 bp\n",
      "  Read 2:    56,067,966 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 334766 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.3\t0\t1\n",
      "9\t4\t1.3\t0\t4\n",
      "10\t8\t0.3\t0\t8\n",
      "11\t7\t0.1\t0\t7\n",
      "12\t6\t0.0\t0\t6\n",
      "13\t6\t0.0\t0\t6\n",
      "14\t19\t0.0\t0\t19\n",
      "15\t7\t0.0\t0\t7\n",
      "16\t15\t0.0\t0\t15\n",
      "17\t102\t0.0\t0\t102\n",
      "18\t52175\t0.0\t0\t52175\n",
      "19\t60493\t0.0\t0\t60493\n",
      "20\t61456\t0.0\t0\t61456\n",
      "21\t55267\t0.0\t0\t55267\n",
      "22\t56696\t0.0\t0\t56696\n",
      "23\t48458\t0.0\t0\t48458\n",
      "24\t43\t0.0\t0\t43\n",
      "26\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "63\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 317372 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t10\t0.0\t0\t10\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t4\t0.0\t0\t4\n",
      "19\t10\t0.0\t0\t10\n",
      "20\t99\t0.0\t0\t99\n",
      "21\t59575\t0.0\t0\t59575\n",
      "22\t57160\t0.0\t0\t57160\n",
      "23\t52497\t0.0\t0\t52497\n",
      "24\t53427\t0.0\t0\t53427\n",
      "25\t48103\t0.0\t0\t48103\n",
      "26\t46434\t0.0\t0\t46434\n",
      "27\t40\t0.0\t0\t40\n",
      "28\t2\t0.0\t0\t2\n",
      "36\t1\t0.0\t0\t1\n",
      "61\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121706_66_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121706_67_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121706_66_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121706_67_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121706_66_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121706_67_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121706_66_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121706_67_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 4.08 s (10 µs/read; 6.12 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            416,322\n",
      "  Read 1 with adapter:                 405,067 (97.3%)\n",
      "  Read 2 with adapter:                 385,489 (92.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:           41,066 (9.9%)\n",
      "Pairs written (passing filters):       375,256 (90.1%)\n",
      "\n",
      "Total basepairs processed:   212,324,220 bp\n",
      "  Read 1:   126,978,210 bp\n",
      "  Read 2:    85,346,010 bp\n",
      "Total written (filtered):    174,944,302 bp (82.4%)\n",
      "  Read 1:   106,779,862 bp\n",
      "  Read 2:    68,164,440 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 405067 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t6.4\t0\t4\n",
      "9\t2\t1.6\t0\t2\n",
      "10\t3\t0.4\t0\t3\n",
      "11\t5\t0.1\t0\t5\n",
      "12\t4\t0.0\t0\t4\n",
      "13\t3\t0.0\t0\t3\n",
      "14\t17\t0.0\t0\t17\n",
      "15\t12\t0.0\t0\t12\n",
      "16\t26\t0.0\t0\t26\n",
      "17\t108\t0.0\t0\t108\n",
      "18\t63733\t0.0\t0\t63733\n",
      "19\t72608\t0.0\t0\t72608\n",
      "20\t74499\t0.0\t0\t74499\n",
      "21\t66813\t0.0\t0\t66813\n",
      "22\t68675\t0.0\t0\t68675\n",
      "23\t58490\t0.0\t0\t58490\n",
      "24\t63\t0.0\t0\t63\n",
      "25\t2\t0.0\t0\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 385489 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t0\t1\n",
      "12\t5\t0.0\t0\t5\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t12\t0.0\t0\t12\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t7\t0.0\t0\t7\n",
      "18\t8\t0.0\t0\t8\n",
      "19\t9\t0.0\t0\t9\n",
      "20\t138\t0.0\t0\t138\n",
      "21\t72215\t0.0\t0\t72215\n",
      "22\t69374\t0.0\t0\t69374\n",
      "23\t63845\t0.0\t0\t63845\n",
      "24\t65324\t0.0\t0\t65324\n",
      "25\t58672\t0.0\t0\t58672\n",
      "26\t55823\t0.0\t0\t55823\n",
      "27\t51\t0.0\t0\t51\n",
      "28\t2\t0.0\t0\t2\n",
      "62\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121707_68_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121707_69_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121707_68_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121707_69_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121707_68_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121707_69_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121707_68_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121707_69_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.91 s (11 µs/read; 5.37 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             81,485\n",
      "  Read 1 with adapter:                  79,303 (97.3%)\n",
      "  Read 2 with adapter:                  74,804 (91.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            8,666 (10.6%)\n",
      "Pairs written (passing filters):        72,819 (89.4%)\n",
      "\n",
      "Total basepairs processed:    41,557,350 bp\n",
      "  Read 1:    24,852,925 bp\n",
      "  Read 2:    16,704,425 bp\n",
      "Total written (filtered):     33,947,823 bp (81.7%)\n",
      "  Read 1:    20,720,493 bp\n",
      "  Read 2:    13,227,330 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 79303 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t2\t0.3\t0\t2\n",
      "10\t2\t0.1\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t34\t0.0\t0\t34\n",
      "18\t12320\t0.0\t0\t12320\n",
      "19\t14302\t0.0\t0\t14302\n",
      "20\t14387\t0.0\t0\t14387\n",
      "21\t13117\t0.0\t0\t13117\n",
      "22\t13446\t0.0\t0\t13446\n",
      "23\t11676\t0.0\t0\t11676\n",
      "24\t5\t0.0\t0\t5\n",
      "28\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 74804 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.1\t0\t2\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "19\t2\t0.0\t0\t2\n",
      "20\t25\t0.0\t0\t25\n",
      "21\t14012\t0.0\t0\t14012\n",
      "22\t13493\t0.0\t0\t13493\n",
      "23\t12382\t0.0\t0\t12382\n",
      "24\t12641\t0.0\t0\t12641\n",
      "25\t11252\t0.0\t0\t11252\n",
      "26\t10983\t0.0\t0\t10983\n",
      "27\t8\t0.0\t0\t8\n",
      "53\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121708_70_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121708_71_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121708_70_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121708_71_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121708_70_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121708_71_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121708_70_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121708_71_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.45 s (17 µs/read; 3.62 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             27,196\n",
      "  Read 1 with adapter:                  26,427 (97.2%)\n",
      "  Read 2 with adapter:                  24,481 (90.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            3,391 (12.5%)\n",
      "Pairs written (passing filters):        23,805 (87.5%)\n",
      "\n",
      "Total basepairs processed:    13,869,960 bp\n",
      "  Read 1:     8,294,780 bp\n",
      "  Read 2:     5,575,180 bp\n",
      "Total written (filtered):     11,097,905 bp (80.0%)\n",
      "  Read 1:     6,774,112 bp\n",
      "  Read 2:     4,323,793 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 26427 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "17\t6\t0.0\t0\t6\n",
      "18\t4153\t0.0\t0\t4153\n",
      "19\t4745\t0.0\t0\t4745\n",
      "20\t4859\t0.0\t0\t4859\n",
      "21\t4346\t0.0\t0\t4346\n",
      "22\t4473\t0.0\t0\t4473\n",
      "23\t3839\t0.0\t0\t3839\n",
      "24\t3\t0.0\t0\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 24481 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1\t0.0\t0\t1\n",
      "20\t5\t0.0\t0\t5\n",
      "21\t4531\t0.0\t0\t4531\n",
      "22\t4333\t0.0\t0\t4333\n",
      "23\t4146\t0.0\t0\t4146\n",
      "24\t4150\t0.0\t0\t4150\n",
      "25\t3714\t0.0\t0\t3714\n",
      "26\t3599\t0.0\t0\t3599\n",
      "27\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121709_72_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121709_73_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121709_72_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121709_73_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121709_72_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121709_73_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121709_72_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121709_73_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.48 s (14 µs/read; 4.14 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             32,857\n",
      "  Read 1 with adapter:                  31,953 (97.2%)\n",
      "  Read 2 with adapter:                  29,876 (90.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            3,780 (11.5%)\n",
      "Pairs written (passing filters):        29,077 (88.5%)\n",
      "\n",
      "Total basepairs processed:    16,757,070 bp\n",
      "  Read 1:    10,021,385 bp\n",
      "  Read 2:     6,735,685 bp\n",
      "Total written (filtered):     13,555,322 bp (80.9%)\n",
      "  Read 1:     8,273,880 bp\n",
      "  Read 2:     5,281,442 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 31953 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t9\t0.0\t0\t9\n",
      "18\t4981\t0.0\t0\t4981\n",
      "19\t5717\t0.0\t0\t5717\n",
      "20\t5839\t0.0\t0\t5839\n",
      "21\t5216\t0.0\t0\t5216\n",
      "22\t5501\t0.0\t0\t5501\n",
      "23\t4678\t0.0\t0\t4678\n",
      "24\t5\t0.0\t0\t5\n",
      "29\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 29876 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t12\t0.0\t0\t12\n",
      "21\t5470\t0.0\t0\t5470\n",
      "22\t5373\t0.0\t0\t5373\n",
      "23\t5010\t0.0\t0\t5010\n",
      "24\t5115\t0.0\t0\t5115\n",
      "25\t4536\t0.0\t0\t4536\n",
      "26\t4354\t0.0\t0\t4354\n",
      "27\t2\t0.0\t0\t2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121710_74_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121710_75_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121710_74_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121710_75_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121710_74_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-kqpkqmxi/S210421121710_75_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121710_74_L001_R1_001.fastq.gz /tmp/qiime2-archive-s1x7mq7p/baf0bbb7-e484-4372-8cd7-d68fd293544f/data/S210421121710_75_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.22 s (46 µs/read; 1.31 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              4,908\n",
      "  Read 1 with adapter:                   4,754 (96.9%)\n",
      "  Read 2 with adapter:                   4,257 (86.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              773 (15.7%)\n",
      "Pairs written (passing filters):         4,135 (84.3%)\n",
      "\n",
      "Total basepairs processed:     2,503,080 bp\n",
      "  Read 1:     1,496,940 bp\n",
      "  Read 2:     1,006,140 bp\n",
      "Total written (filtered):      1,927,695 bp (77.0%)\n",
      "  Read 1:     1,176,557 bp\n",
      "  Read 2:       751,138 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 4754 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t757\t0.0\t0\t757\n",
      "19\t836\t0.0\t0\t836\n",
      "20\t829\t0.0\t0\t829\n",
      "21\t772\t0.0\t0\t772\n",
      "22\t824\t0.0\t0\t824\n",
      "23\t735\t0.0\t0\t735\n",
      "24\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 4257 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "20\t2\t0.0\t0\t2\n",
      "21\t819\t0.0\t0\t819\n",
      "22\t759\t0.0\t0\t759\n",
      "23\t673\t0.0\t0\t673\n",
      "24\t752\t0.0\t0\t752\n",
      "25\t638\t0.0\t0\t638\n",
      "26\t613\t0.0\t0\t613\n"
     ]
    }
   ],
   "source": [
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')\n",
    "\n",
    "if trim and (not os.path.isfile(demux_file_trim) or replace_files):\n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = None\n",
    "    \n",
    "    if n_directions == 1:\n",
    "        res = trim_single(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front=forward_primer,\n",
    "            adapter=reverse_reverse_complement,\n",
    "            cores=threads,\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    elif n_directions == 2:\n",
    "        res = trim_paired(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front_f=forward_primer,\n",
    "            front_r=reverse_primer,\n",
    "            # adapter_f=reverse_reverse_complement,\n",
    "            # adapter_r=forward_reverse_complement,\n",
    "            cores=threads,\n",
    "            overlap=trim['overlap'],\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    else:\n",
    "        print(f'ERROR: invalid number of directions {n_directions}')\n",
    "\n",
    "    if res is not None:\n",
    "        res.save(demux_file_trim)\n",
    "        Visualization.save(demux.visualizers.summarize(res).visualization, filepath=demux_view_trim)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 115.258711,
   "end_time": "2023-08-23T18:58:44.892272",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-templates/step-prepare-data.ipynb",
   "output_path": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-STD-NRxHSD-NR-trim/nb-executed-steps/step-prepare-data-ana-flavia-STD-NRxHSD-NR-trim.ipynb",
   "parameters": {
    "base_dir": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri",
    "class_col": "group-id",
    "classifier_file": "/home/lauro/nupeb/rede-micro/models/silva-138-99-nb-classifier.qza",
    "experiment_name": "ana-flavia-STD-NRxHSD-NR-trim",
    "manifest_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-STD-NRxHSD-NR.csv",
    "metadata_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-STD-NRxHSD-NR.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "top_n": 20,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2023-08-23T18:56:49.633561",
   "version": "2.3.4"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}