{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "a466d8b3",
   "metadata": {
    "papermill": {
     "duration": 0.018003,
     "end_time": "2022-10-24T19:54:22.826389",
     "exception": false,
     "start_time": "2022-10-24T19:54:22.808386",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a5cad82",
   "metadata": {
    "papermill": {
     "duration": 0.037221,
     "end_time": "2022-10-24T19:54:22.875727",
     "exception": false,
     "start_time": "2022-10-24T19:54:22.838506",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "670d3d58",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-10-14T14:33:37.863523Z",
     "start_time": "2022-10-14T14:33:14.145712Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T19:54:22.900060Z",
     "iopub.status.busy": "2022-10-24T19:54:22.899633Z",
     "iopub.status.idle": "2022-10-24T19:54:29.637436Z",
     "shell.execute_reply": "2022-10-24T19:54:29.636916Z"
    },
    "papermill": {
     "duration": 6.753559,
     "end_time": "2022-10-24T19:54:29.640607",
     "exception": false,
     "start_time": "2022-10-24T19:54:22.887048",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1045d135",
   "metadata": {
    "papermill": {
     "duration": 0.010308,
     "end_time": "2022-10-24T19:54:29.669417",
     "exception": false,
     "start_time": "2022-10-24T19:54:29.659109",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "899a65bf",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.594341Z",
     "start_time": "2022-06-28T18:50:41.591544Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T19:54:29.725133Z",
     "iopub.status.busy": "2022-10-24T19:54:29.724844Z",
     "iopub.status.idle": "2022-10-24T19:54:29.731739Z",
     "shell.execute_reply": "2022-10-24T19:54:29.730763Z"
    },
    "papermill": {
     "duration": 0.020478,
     "end_time": "2022-10-24T19:54:29.733435",
     "exception": false,
     "start_time": "2022-10-24T19:54:29.712957",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'ana-flavia-superlactacao.yaml')\n",
    "experiment_name = 'jenneffer-vs-01'\n",
    "base_dir = os.path.join('/', 'home', 'lauro', 'nupeb', 'redemicro')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'not-hist-vs-manifest.csv')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'imgs'))\n",
    "replace_files = False\n",
    "trim = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9ce582ac",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-10-24T19:54:29.757832Z",
     "iopub.status.busy": "2022-10-24T19:54:29.757500Z",
     "iopub.status.idle": "2022-10-24T19:54:29.764839Z",
     "shell.execute_reply": "2022-10-24T19:54:29.763811Z"
    },
    "papermill": {
     "duration": 0.021623,
     "end_time": "2022-10-24T19:54:29.766816",
     "exception": false,
     "start_time": "2022-10-24T19:54:29.745193",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "experiment_name = \"ana-flavia-STD-NRxHSD-NR-trim\"\n",
    "base_dir = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri\"\n",
    "manifest_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-STD-NRxHSD-NR.csv\"\n",
    "metadata_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-STD-NRxHSD-NR.tsv\"\n",
    "class_col = \"group-id\"\n",
    "classifier_file = (\n",
    "    \"/home/lauro/nupeb/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza\"\n",
    ")\n",
    "replace_files = False\n",
    "phred = 20\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n",
    "overlap = 12\n",
    "threads = 6\n",
    "trim = {\n",
    "    \"overlap\": 8,\n",
    "    \"forward_primer\": \"CCTACGGGRSGCAGCAG\",\n",
    "    \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\",\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "97bcf393",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.610682Z",
     "start_time": "2022-06-28T18:50:41.595886Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T19:54:29.790361Z",
     "iopub.status.busy": "2022-10-24T19:54:29.789969Z",
     "iopub.status.idle": "2022-10-24T19:54:29.797188Z",
     "shell.execute_reply": "2022-10-24T19:54:29.796083Z"
    },
    "papermill": {
     "duration": 0.021042,
     "end_time": "2022-10-24T19:54:29.799188",
     "exception": false,
     "start_time": "2022-10-24T19:54:29.778146",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Setting hardcoded parameters for development and debbuging\n",
    "production = True\n",
    "if not production:\n",
    "    params_path = os.path.join('..', 'params', 'ana-flavia-hipotese-01')\n",
    "    with open(params_path, 'r') as stream:\n",
    "        params = yaml.safe_load(stream)\n",
    "        experiment_name = params['experiment_name']\n",
    "        base_dir = params['base_dir']\n",
    "        manifest_file = params['manifest_file']\n",
    "        replace_files = params['replace_files']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b2fe36d7",
   "metadata": {
    "papermill": {
     "duration": 0.010603,
     "end_time": "2022-10-24T19:54:29.822349",
     "exception": false,
     "start_time": "2022-10-24T19:54:29.811746",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d691bcb7",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.618098Z",
     "start_time": "2022-06-28T18:50:41.614152Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T19:54:29.844951Z",
     "iopub.status.busy": "2022-10-24T19:54:29.844595Z",
     "iopub.status.idle": "2022-10-24T19:54:29.849659Z",
     "shell.execute_reply": "2022-10-24T19:54:29.848557Z"
    },
    "papermill": {
     "duration": 0.018536,
     "end_time": "2022-10-24T19:54:29.851443",
     "exception": false,
     "start_time": "2022-10-24T19:54:29.832907",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# new_manifest = '/home/lauro/nupeb/redemicro/data/raw/manifest/karina-manifest.csv'\n",
    "# with open(manifest_file, 'r') as oldm, open(new_manifest, 'w') as newm:\n",
    "#     header = 'sample-id,absolute-filepath,direction\\n'\n",
    "#     newm.write(header)\n",
    "#     for line in oldm.readlines()[1:]:\n",
    "#         sid, forward, reverse = line[:-1].split(',')\n",
    "#         fline = ','.join((sid, forward, 'forward')) + '\\n'\n",
    "#         rline = ','.join((sid, reverse, 'reverse')) + '\\n'\n",
    "#         newm.write(fline)\n",
    "#         newm.write(rline)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a832990c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.632125Z",
     "start_time": "2022-06-28T18:50:41.619710Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T19:54:29.875022Z",
     "iopub.status.busy": "2022-10-24T19:54:29.874668Z",
     "iopub.status.idle": "2022-10-24T19:54:29.886803Z",
     "shell.execute_reply": "2022-10-24T19:54:29.885692Z"
    },
    "papermill": {
     "duration": 0.026722,
     "end_time": "2022-10-24T19:54:29.889061",
     "exception": false,
     "start_time": "2022-10-24T19:54:29.862339",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New qiime-artifacts folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-STD-NRxHSD-NR-trim/qiime-artifacts\n",
      "New img folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-STD-NRxHSD-NR-trim/imgs\n"
     ]
    }
   ],
   "source": [
    "# Define the output folder path\n",
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "\n",
    "# Create path if it not exist\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exist\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')\n",
    "\n",
    "# Define the output artifact full path\n",
    "demux_file = os.path.join(out_dir, 'demux-paired.qza')\n",
    "demux_view = os.path.join(out_dir, 'demux-paired.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eda9e19b",
   "metadata": {
    "papermill": {
     "duration": 0.01163,
     "end_time": "2022-10-24T19:54:29.913878",
     "exception": false,
     "start_time": "2022-10-24T19:54:29.902248",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file of `SampleData[PairedEndSequencesWithQuality]`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c7f2ec18",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-10-24T19:54:29.937965Z",
     "iopub.status.busy": "2022-10-24T19:54:29.937519Z",
     "iopub.status.idle": "2022-10-24T19:54:29.970882Z",
     "shell.execute_reply": "2022-10-24T19:54:29.970236Z"
    },
    "papermill": {
     "duration": 0.047729,
     "end_time": "2022-10-24T19:54:29.972832",
     "exception": false,
     "start_time": "2022-10-24T19:54:29.925103",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "manifest_df = pd.read_csv(manifest_file)\n",
    "n_directions = len(manifest_df['direction'].unique())\n",
    "if n_directions == 1:\n",
    "    d_type = 'SampleData[SequencesWithQuality]'\n",
    "    v_type = 'SingleEndFastqManifestPhred33'\n",
    "elif n_directions == 2:\n",
    "    d_type = 'SampleData[PairedEndSequencesWithQuality]'\n",
    "    v_type = 'PairedEndFastqManifestPhred33'\n",
    "else:\n",
    "    print(f'ERROR: invalid number of directions {n_directions}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4e85b1d3",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:44:58.640338Z",
     "start_time": "2022-06-28T18:50:41.633657Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T19:54:30.002095Z",
     "iopub.status.busy": "2022-10-24T19:54:30.001871Z",
     "iopub.status.idle": "2022-10-24T20:02:06.617027Z",
     "shell.execute_reply": "2022-10-24T20:02:06.616264Z"
    },
    "papermill": {
     "duration": 456.63057,
     "end_time": "2022-10-24T20:02:06.620119",
     "exception": false,
     "start_time": "2022-10-24T19:54:29.989549",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    \n",
    "    # Import data and create an artifact object\n",
    "    artifact = Artifact.import_data(\n",
    "        d_type, \n",
    "        manifest_file, \n",
    "        view_type=v_type)\n",
    "    \n",
    "    # Save the artifact object to a new qza file\n",
    "    artifact.save(demux_file)\n",
    "\n",
    "else:\n",
    "    artifact = Artifact.load(demux_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2eb4712e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.602855Z",
     "start_time": "2022-06-28T19:44:58.642520Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T20:02:06.653371Z",
     "iopub.status.busy": "2022-10-24T20:02:06.653140Z",
     "iopub.status.idle": "2022-10-24T20:02:35.289381Z",
     "shell.execute_reply": "2022-10-24T20:02:35.288525Z"
    },
    "papermill": {
     "duration": 28.654912,
     "end_time": "2022-10-24T20:02:35.292536",
     "exception": false,
     "start_time": "2022-10-24T20:02:06.637624",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "if not os.path.isfile(demux_view) or replace_files: \n",
    "    # Generate e visualization of the Artifact\n",
    "    demux_summary = demux.visualizers.summarize(artifact)\n",
    "\n",
    "    # Save a new visualization file based on the qza file\n",
    "    Visualization.save(demux_summary.visualization, filepath=demux_view)\n",
    "    \n",
    "    demux_view_obj = demux_summary.visualization\n",
    "else:\n",
    "    demux_view_obj = Visualization.load(demux_view)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e14175c2",
   "metadata": {
    "papermill": {
     "duration": 0.011704,
     "end_time": "2022-10-24T20:02:35.324357",
     "exception": false,
     "start_time": "2022-10-24T20:02:35.312653",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ddd8a2f9",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.609737Z",
     "start_time": "2022-06-28T19:47:56.604820Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T20:02:35.364309Z",
     "iopub.status.busy": "2022-10-24T20:02:35.364068Z",
     "iopub.status.idle": "2022-10-24T20:02:35.369010Z",
     "shell.execute_reply": "2022-10-24T20:02:35.368194Z"
    },
    "papermill": {
     "duration": 0.019398,
     "end_time": "2022-10-24T20:02:35.370678",
     "exception": false,
     "start_time": "2022-10-24T20:02:35.351280",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<visualization: Visualization uuid: 6fa595a3-aad6-413f-886f-d8fe05a36646>\n"
     ]
    }
   ],
   "source": [
    "print(demux_view_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "4825ed20",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.704841Z",
     "start_time": "2022-06-28T19:47:56.611846Z"
    },
    "execution": {
     "iopub.execute_input": "2022-10-24T20:02:35.396803Z",
     "iopub.status.busy": "2022-10-24T20:02:35.396535Z",
     "iopub.status.idle": "2022-10-24T20:02:35.456294Z",
     "shell.execute_reply": "2022-10-24T20:02:35.455688Z"
    },
    "papermill": {
     "duration": 0.076808,
     "end_time": "2022-10-24T20:02:35.460220",
     "exception": false,
     "start_time": "2022-10-24T20:02:35.383412",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-f9e6mc94')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 6fa595a3-aad6-413f-886f-d8fe05a36646>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ef07bca",
   "metadata": {
    "papermill": {
     "duration": 0.022505,
     "end_time": "2022-10-24T20:02:35.501594",
     "exception": false,
     "start_time": "2022-10-24T20:02:35.479089",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e9b74dd8",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-10-24T20:02:35.541835Z",
     "iopub.status.busy": "2022-10-24T20:02:35.541548Z",
     "iopub.status.idle": "2022-10-24T20:04:11.687376Z",
     "shell.execute_reply": "2022-10-24T20:04:11.686780Z"
    },
    "papermill": {
     "duration": 96.174963,
     "end_time": "2022-10-24T20:04:11.689421",
     "exception": false,
     "start_time": "2022-10-24T20:02:35.514458",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121685_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121685_1_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121685_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121685_1_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121685_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121685_1_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121685_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121685_1_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.87 s (15 µs/read; 3.94 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            254,393\n",
      "  Read 1 with adapter:                 251,998 (99.1%)\n",
      "  Read 2 with adapter:                 248,860 (97.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  10 (0.0%)\n",
      "Pairs discarded as untrimmed:            7,871 (3.1%)\n",
      "Pairs written (passing filters):       246,512 (96.9%)\n",
      "\n",
      "Total basepairs processed:   129,740,430 bp\n",
      "  Read 1:    77,589,865 bp\n",
      "  Read 2:    52,150,565 bp\n",
      "Total written (filtered):    114,920,232 bp (88.6%)\n",
      "  Read 1:    70,145,471 bp\n",
      "  Read 2:    44,774,761 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 128 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 34.4%\n",
      "  G: 60.9%\n",
      "  T: 4.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t28\t3.9\t1\t0 28\n",
      "9\t18\t1.0\t1\t0 18\n",
      "10\t1\t0.2\t1\t0 1\n",
      "20\t9\t0.0\t3\t9\n",
      "21\t10\t0.0\t3\t8 2\n",
      "22\t14\t0.0\t3\t12 1 1\n",
      "23\t14\t0.0\t3\t13 1\n",
      "24\t10\t0.0\t3\t8 2\n",
      "25\t13\t0.0\t3\t12 0 1\n",
      "26\t2\t0.0\t3\t1 1\n",
      "30\t1\t0.0\t3\t1\n",
      "218\t1\t0.0\t3\t1\n",
      "234\t1\t0.0\t3\t1\n",
      "236\t1\t0.0\t3\t1\n",
      "237\t1\t0.0\t3\t0 1\n",
      "253\t1\t0.0\t3\t1\n",
      "279\t1\t0.0\t3\t0 1\n",
      "302\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 251870 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t3.9\t1\t3 1\n",
      "9\t5\t1.0\t1\t3 2\n",
      "10\t3\t0.2\t1\t2 1\n",
      "11\t4\t0.1\t1\t1 3\n",
      "12\t7\t0.0\t1\t6 1\n",
      "13\t7\t0.0\t1\t5 2\n",
      "14\t20\t0.0\t2\t13 2 5\n",
      "15\t11\t0.0\t2\t8 2 1\n",
      "16\t18\t0.0\t2\t5 7 6\n",
      "17\t152\t0.0\t2\t82 43 27\n",
      "18\t39230\t0.0\t2\t38482 722 26\n",
      "19\t45580\t0.0\t2\t44776 764 40\n",
      "20\t46235\t0.0\t2\t45407 794 34\n",
      "21\t41414\t0.0\t2\t40682 694 38\n",
      "22\t42544\t0.0\t2\t41795 719 30\n",
      "23\t36597\t0.0\t2\t36001 585 11\n",
      "24\t35\t0.0\t2\t29 2 4\n",
      "25\t2\t0.0\t2\t2\n",
      "47\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 25 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 8.0%\n",
      "  C: 8.0%\n",
      "  G: 36.0%\n",
      "  T: 8.0%\n",
      "  none/other: 40.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t4\t0.0\t2\t4\n",
      "204\t11\t0.0\t2\t11\n",
      "205\t10\t0.0\t2\t10\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 248835 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t3.9\t1\t0 1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "13\t3\t0.0\t1\t3\n",
      "14\t4\t0.0\t2\t1 1 2\n",
      "15\t12\t0.0\t2\t10 1 1\n",
      "16\t4\t0.0\t2\t1 3\n",
      "17\t6\t0.0\t2\t2 2 2\n",
      "18\t9\t0.0\t2\t7 1 1\n",
      "19\t12\t0.0\t2\t5 5 2\n",
      "20\t185\t0.0\t3\t74 55 37 19\n",
      "21\t46670\t0.0\t3\t44245 1778 427 220\n",
      "22\t44425\t0.0\t3\t42188 1648 399 190\n",
      "23\t41247\t0.0\t3\t38808 1831 381 227\n",
      "24\t41891\t0.0\t3\t39553 1702 433 203\n",
      "25\t37932\t0.0\t3\t35622 1661 436 213\n",
      "26\t36391\t0.0\t3\t34102 1664 425 200\n",
      "27\t34\t0.0\t3\t20 9 4 1\n",
      "33\t1\t0.0\t3\t1\n",
      "44\t1\t0.0\t3\t1\n",
      "57\t1\t0.0\t3\t0 0 0 1\n",
      "60\t1\t0.0\t3\t0 0 1\n",
      "61\t2\t0.0\t3\t0 0 0 2\n",
      "64\t1\t0.0\t3\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121686_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121686_3_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121686_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121686_3_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121686_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121686_3_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121686_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121686_3_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.76 s (17 µs/read; 3.54 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            103,905\n",
      "  Read 1 with adapter:                 102,946 (99.1%)\n",
      "  Read 2 with adapter:                 101,888 (98.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,954 (2.8%)\n",
      "Pairs written (passing filters):       100,951 (97.2%)\n",
      "\n",
      "Total basepairs processed:    52,991,550 bp\n",
      "  Read 1:    31,691,025 bp\n",
      "  Read 2:    21,300,525 bp\n",
      "Total written (filtered):     47,063,687 bp (88.8%)\n",
      "  Read 1:    28,727,135 bp\n",
      "  Read 2:    18,336,552 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 42 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 4.8%\n",
      "  C: 14.3%\n",
      "  G: 78.6%\n",
      "  T: 2.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t1.6\t1\t0 3\n",
      "9\t4\t0.4\t1\t0 4\n",
      "21\t4\t0.0\t3\t3 1\n",
      "22\t1\t0.0\t3\t1\n",
      "23\t6\t0.0\t3\t6\n",
      "24\t2\t0.0\t3\t2\n",
      "25\t2\t0.0\t3\t2\n",
      "29\t2\t0.0\t3\t2\n",
      "30\t6\t0.0\t3\t6\n",
      "31\t5\t0.0\t3\t5\n",
      "32\t1\t0.0\t3\t1\n",
      "33\t1\t0.0\t3\t1\n",
      "34\t1\t0.0\t3\t1\n",
      "78\t1\t0.0\t3\t1\n",
      "85\t1\t0.0\t3\t1\n",
      "166\t1\t0.0\t3\t0 0 1\n",
      "234\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 102904 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.6\t1\t1\n",
      "9\t2\t0.4\t1\t2\n",
      "10\t1\t0.1\t1\t1\n",
      "11\t6\t0.0\t1\t5 1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "14\t9\t0.0\t2\t8 1\n",
      "15\t5\t0.0\t2\t4 1\n",
      "16\t5\t0.0\t2\t2 0 3\n",
      "17\t56\t0.0\t2\t32 18 6\n",
      "18\t16355\t0.0\t2\t16066 273 16\n",
      "19\t18584\t0.0\t2\t18252 317 15\n",
      "20\t19044\t0.0\t2\t18741 292 11\n",
      "21\t16875\t0.0\t2\t16611 250 14\n",
      "22\t17209\t0.0\t2\t16927 266 16\n",
      "23\t14733\t0.0\t2\t14532 197 4\n",
      "24\t14\t0.0\t2\t12 1 1\n",
      "40\t1\t0.0\t2\t1\n",
      "44\t1\t0.0\t2\t1\n",
      "51\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 3 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 100.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "204\t3\t0.0\t2\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 101885 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t6\t1.6\t1\t1 5\n",
      "11\t2\t0.0\t1\t1 1\n",
      "12\t2\t0.0\t1\t2\n",
      "13\t1\t0.0\t1\t1\n",
      "14\t2\t0.0\t2\t0 0 2\n",
      "15\t3\t0.0\t2\t1 2\n",
      "16\t3\t0.0\t2\t1 2\n",
      "17\t3\t0.0\t2\t2 1\n",
      "18\t4\t0.0\t2\t2 1 1\n",
      "19\t11\t0.0\t2\t6 4 1\n",
      "20\t93\t0.0\t3\t47 23 17 6\n",
      "21\t19021\t0.0\t3\t18052 727 151 91\n",
      "22\t18364\t0.0\t3\t17427 671 172 94\n",
      "23\t16633\t0.0\t3\t15720 660 173 80\n",
      "24\t17334\t0.0\t3\t16399 664 178 93\n",
      "25\t15483\t0.0\t3\t14599 640 167 77\n",
      "26\t14896\t0.0\t3\t14056 607 158 75\n",
      "27\t20\t0.0\t3\t15 4 0 1\n",
      "60\t1\t0.0\t3\t0 0 0 1\n",
      "61\t1\t0.0\t3\t0 1\n",
      "63\t2\t0.0\t3\t0 0 1 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121687_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121687_5_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121687_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121687_5_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121687_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121687_5_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121687_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121687_5_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.17 s (18 µs/read; 3.39 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             66,368\n",
      "  Read 1 with adapter:                  65,760 (99.1%)\n",
      "  Read 2 with adapter:                  64,834 (97.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   2 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,121 (3.2%)\n",
      "Pairs written (passing filters):        64,245 (96.8%)\n",
      "\n",
      "Total basepairs processed:    33,847,680 bp\n",
      "  Read 1:    20,242,240 bp\n",
      "  Read 2:    13,605,440 bp\n",
      "Total written (filtered):     29,949,537 bp (88.5%)\n",
      "  Read 1:    18,280,890 bp\n",
      "  Read 2:    11,668,647 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 85 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 63.5%\n",
      "  G: 36.5%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t25\t1.0\t1\t1 24\n",
      "9\t29\t0.3\t1\t0 29\n",
      "21\t2\t0.0\t3\t2\n",
      "22\t4\t0.0\t3\t3 1\n",
      "23\t4\t0.0\t3\t3 1\n",
      "24\t3\t0.0\t3\t3\n",
      "25\t5\t0.0\t3\t5\n",
      "29\t2\t0.0\t3\t2\n",
      "31\t1\t0.0\t3\t1\n",
      "32\t2\t0.0\t3\t2\n",
      "33\t2\t0.0\t3\t2\n",
      "34\t4\t0.0\t3\t3 1\n",
      "278\t1\t0.0\t3\t1\n",
      "284\t1\t0.0\t3\t0 1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 65675 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t1\t1\n",
      "11\t4\t0.0\t1\t2 2\n",
      "13\t3\t0.0\t1\t2 1\n",
      "14\t5\t0.0\t2\t3 0 2\n",
      "15\t3\t0.0\t2\t2 0 1\n",
      "16\t3\t0.0\t2\t1 1 1\n",
      "17\t36\t0.0\t2\t22 8 6\n",
      "18\t10073\t0.0\t2\t9860 201 12\n",
      "19\t11875\t0.0\t2\t11657 206 12\n",
      "20\t12063\t0.0\t2\t11869 186 8\n",
      "21\t10903\t0.0\t2\t10700 196 7\n",
      "22\t11233\t0.0\t2\t11028 191 14\n",
      "23\t9462\t0.0\t2\t9334 123 5\n",
      "24\t10\t0.0\t2\t9 1\n",
      "26\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 9 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 33.3%\n",
      "  G: 33.3%\n",
      "  T: 11.1%\n",
      "  none/other: 22.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.3\t1\t0 1\n",
      "203\t3\t0.0\t2\t3\n",
      "204\t3\t0.0\t2\t3\n",
      "205\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 64825 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.0\t1\t0 1\n",
      "12\t1\t0.0\t1\t1\n",
      "14\t1\t0.0\t2\t0 1\n",
      "15\t5\t0.0\t2\t4 1\n",
      "16\t1\t0.0\t2\t1\n",
      "17\t4\t0.0\t2\t1 2 1\n",
      "19\t5\t0.0\t2\t3 0 2\n",
      "20\t36\t0.0\t3\t16 12 3 5\n",
      "21\t12261\t0.0\t3\t11652 454 107 48\n",
      "22\t11400\t0.0\t3\t10779 446 117 58\n",
      "23\t10858\t0.0\t3\t10214 496 100 48\n",
      "24\t10877\t0.0\t3\t10220 473 117 67\n",
      "25\t9802\t0.0\t3\t9211 432 102 57\n",
      "26\t9560\t0.0\t3\t8923 475 116 46\n",
      "27\t11\t0.0\t3\t7 2 1 1\n",
      "28\t1\t0.0\t3\t0 0 1\n",
      "53\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121691_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121691_7_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121691_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121691_7_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121691_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121691_7_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121691_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121691_7_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.48 s (16 µs/read; 3.86 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            224,175\n",
      "  Read 1 with adapter:                 221,891 (99.0%)\n",
      "  Read 2 with adapter:                 219,776 (98.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   1 (0.0%)\n",
      "Pairs discarded as untrimmed:            6,634 (3.0%)\n",
      "Pairs written (passing filters):       217,540 (97.0%)\n",
      "\n",
      "Total basepairs processed:   114,329,250 bp\n",
      "  Read 1:    68,373,375 bp\n",
      "  Read 2:    45,955,875 bp\n",
      "Total written (filtered):    101,415,359 bp (88.7%)\n",
      "  Read 1:    61,901,589 bp\n",
      "  Read 2:    39,513,770 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 66 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 4.5%\n",
      "  G: 92.4%\n",
      "  T: 3.0%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'G' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t3.4\t1\t0 4\n",
      "9\t1\t0.9\t1\t0 1\n",
      "20\t5\t0.0\t3\t5\n",
      "21\t3\t0.0\t3\t2 1\n",
      "22\t8\t0.0\t3\t8\n",
      "23\t6\t0.0\t3\t6\n",
      "24\t4\t0.0\t3\t3 1\n",
      "25\t3\t0.0\t3\t2 1\n",
      "29\t2\t0.0\t3\t2\n",
      "30\t4\t0.0\t3\t4\n",
      "31\t2\t0.0\t3\t1 1\n",
      "32\t13\t0.0\t3\t13\n",
      "33\t7\t0.0\t3\t7\n",
      "34\t2\t0.0\t3\t2\n",
      "160\t1\t0.0\t3\t1\n",
      "284\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 221825 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t3.4\t1\t1\n",
      "9\t5\t0.9\t1\t4 1\n",
      "10\t3\t0.2\t1\t1 2\n",
      "11\t6\t0.1\t1\t5 1\n",
      "12\t4\t0.0\t1\t0 4\n",
      "13\t4\t0.0\t1\t2 2\n",
      "14\t10\t0.0\t2\t7 2 1\n",
      "15\t9\t0.0\t2\t5 3 1\n",
      "16\t16\t0.0\t2\t7 2 7\n",
      "17\t98\t0.0\t2\t51 30 17\n",
      "18\t34555\t0.0\t2\t33935 600 20\n",
      "19\t39994\t0.0\t2\t39193 777 24\n",
      "20\t40929\t0.0\t2\t40215 689 25\n",
      "21\t36497\t0.0\t2\t35864 599 34\n",
      "22\t37849\t0.0\t2\t37222 599 28\n",
      "23\t31809\t0.0\t2\t31328 468 13\n",
      "24\t28\t0.0\t2\t21 5 2\n",
      "27\t2\t0.0\t2\t1 1\n",
      "29\t2\t0.0\t2\t2\n",
      "31\t1\t0.0\t2\t1\n",
      "33\t1\t0.0\t2\t1\n",
      "52\t1\t0.0\t2\t1\n",
      "234\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 5 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 40.0%\n",
      "  G: 20.0%\n",
      "  T: 20.0%\n",
      "  none/other: 20.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t2\t0.0\t2\t2\n",
      "204\t2\t0.0\t2\t2\n",
      "205\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 219771 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t3\t0.0\t1\t2 1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t4\t0.0\t2\t2 1 1\n",
      "15\t7\t0.0\t2\t4 1 2\n",
      "16\t3\t0.0\t2\t2 0 1\n",
      "18\t3\t0.0\t2\t0 2 1\n",
      "19\t11\t0.0\t2\t4 3 4\n",
      "20\t170\t0.0\t3\t64 50 33 23\n",
      "21\t40830\t0.0\t3\t38764 1534 352 180\n",
      "22\t39747\t0.0\t3\t37770 1465 344 168\n",
      "23\t36360\t0.0\t3\t34391 1449 347 173\n",
      "24\t36920\t0.0\t3\t34925 1437 365 193\n",
      "25\t33615\t0.0\t3\t31714 1412 323 166\n",
      "26\t32052\t0.0\t3\t30279 1300 321 152\n",
      "27\t36\t0.0\t3\t25 7 2 2\n",
      "28\t1\t0.0\t3\t1\n",
      "30\t1\t0.0\t3\t1\n",
      "53\t1\t0.0\t3\t1\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "62\t3\t0.0\t3\t0 1 0 2\n",
      "63\t1\t0.0\t3\t0 1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121692_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121692_9_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121692_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121692_9_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121692_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121692_9_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121692_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121692_9_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.32 s (16 µs/read; 3.65 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             80,216\n",
      "  Read 1 with adapter:                  79,416 (99.0%)\n",
      "  Read 2 with adapter:                  78,121 (97.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  15 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,870 (3.6%)\n",
      "Pairs written (passing filters):        77,331 (96.4%)\n",
      "\n",
      "Total basepairs processed:    40,910,160 bp\n",
      "  Read 1:    24,465,880 bp\n",
      "  Read 2:    16,444,280 bp\n",
      "Total written (filtered):     36,048,173 bp (88.1%)\n",
      "  Read 1:    22,004,584 bp\n",
      "  Read 2:    14,043,589 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 51 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 43.1%\n",
      "  G: 54.9%\n",
      "  T: 2.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t7\t1.2\t1\t0 7\n",
      "9\t16\t0.3\t1\t0 16\n",
      "20\t5\t0.0\t3\t5\n",
      "21\t3\t0.0\t3\t3\n",
      "22\t4\t0.0\t3\t4\n",
      "23\t6\t0.0\t3\t6\n",
      "25\t5\t0.0\t3\t3 2\n",
      "30\t2\t0.0\t3\t2\n",
      "31\t1\t0.0\t3\t1\n",
      "32\t1\t0.0\t3\t1\n",
      "276\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 79365 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.2\t1\t1\n",
      "9\t1\t0.3\t1\t1\n",
      "11\t1\t0.0\t1\t0 1\n",
      "14\t4\t0.0\t2\t2 1 1\n",
      "15\t2\t0.0\t2\t0 0 2\n",
      "16\t5\t0.0\t2\t3 2\n",
      "17\t49\t0.0\t2\t23 14 12\n",
      "18\t12232\t0.0\t2\t12001 223 8\n",
      "19\t14292\t0.0\t2\t14026 243 23\n",
      "20\t14527\t0.0\t2\t14274 244 9\n",
      "21\t13220\t0.0\t2\t13012 192 16\n",
      "22\t13560\t0.0\t2\t13323 225 12\n",
      "23\t11464\t0.0\t2\t11268 190 6\n",
      "24\t6\t0.0\t2\t4 0 2\n",
      "32\t1\t0.0\t2\t0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 32 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 6.2%\n",
      "  C: 9.4%\n",
      "  G: 15.6%\n",
      "  T: 21.9%\n",
      "  none/other: 46.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.2\t1\t0 1\n",
      "203\t4\t0.0\t2\t3 1\n",
      "204\t12\t0.0\t2\t11 1\n",
      "205\t15\t0.0\t2\t15\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 78089 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t2\t0.0\t1\t2\n",
      "14\t2\t0.0\t2\t0 2\n",
      "15\t2\t0.0\t2\t2\n",
      "18\t2\t0.0\t2\t2\n",
      "19\t2\t0.0\t2\t0 0 2\n",
      "20\t61\t0.0\t3\t30 20 7 4\n",
      "21\t14486\t0.0\t3\t13728 573 112 73\n",
      "22\t13993\t0.0\t3\t13253 515 152 73\n",
      "23\t13178\t0.0\t3\t12470 521 114 73\n",
      "24\t13074\t0.0\t3\t12441 482 99 52\n",
      "25\t11830\t0.0\t3\t11187 474 99 70\n",
      "26\t11446\t0.0\t3\t10833 439 122 52\n",
      "27\t5\t0.0\t3\t3 1 1\n",
      "46\t1\t0.0\t3\t1\n",
      "52\t1\t0.0\t3\t0 0 0 1\n",
      "58\t1\t0.0\t3\t0 0 0 1\n",
      "62\t2\t0.0\t3\t0 0 1 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121693_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121693_11_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121693_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121693_11_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121693_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121693_11_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121693_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121693_11_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.51 s (15 µs/read; 3.93 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            164,608\n",
      "  Read 1 with adapter:                 163,069 (99.1%)\n",
      "  Read 2 with adapter:                 161,335 (98.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   3 (0.0%)\n",
      "Pairs discarded as untrimmed:            4,763 (2.9%)\n",
      "Pairs written (passing filters):       159,842 (97.1%)\n",
      "\n",
      "Total basepairs processed:    83,950,080 bp\n",
      "  Read 1:    50,205,440 bp\n",
      "  Read 2:    33,744,640 bp\n",
      "Total written (filtered):     74,517,810 bp (88.8%)\n",
      "  Read 1:    45,483,954 bp\n",
      "  Read 2:    29,033,856 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 101 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 82.2%\n",
      "  G: 17.8%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'C' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t38\t2.5\t1\t0 38\n",
      "9\t46\t0.6\t1\t0 46\n",
      "20\t2\t0.0\t3\t2\n",
      "21\t2\t0.0\t3\t2\n",
      "22\t6\t0.0\t3\t6\n",
      "23\t2\t0.0\t3\t2\n",
      "24\t3\t0.0\t3\t3\n",
      "25\t1\t0.0\t3\t1\n",
      "282\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 162968 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.5\t1\t1\n",
      "9\t3\t0.6\t1\t3\n",
      "10\t2\t0.2\t1\t2\n",
      "12\t3\t0.0\t1\t2 1\n",
      "14\t6\t0.0\t2\t4 1 1\n",
      "15\t13\t0.0\t2\t8 1 4\n",
      "16\t12\t0.0\t2\t5 5 2\n",
      "17\t77\t0.0\t2\t40 25 12\n",
      "18\t25475\t0.0\t2\t25001 454 20\n",
      "19\t29054\t0.0\t2\t28509 510 35\n",
      "20\t29775\t0.0\t2\t29239 501 35\n",
      "21\t27186\t0.0\t2\t26720 442 24\n",
      "22\t27872\t0.0\t2\t27381 473 18\n",
      "23\t23465\t0.0\t2\t23095 359 11\n",
      "24\t22\t0.0\t2\t19 1 2\n",
      "32\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 6 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 16.7%\n",
      "  C: 16.7%\n",
      "  G: 16.7%\n",
      "  T: 0.0%\n",
      "  none/other: 50.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.6\t1\t0 1\n",
      "203\t1\t0.0\t2\t1\n",
      "204\t1\t0.0\t2\t1\n",
      "205\t3\t0.0\t2\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 161329 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.2\t1\t0 1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "14\t2\t0.0\t2\t1 0 1\n",
      "15\t10\t0.0\t2\t8 0 2\n",
      "16\t4\t0.0\t2\t2 1 1\n",
      "17\t3\t0.0\t2\t2 1\n",
      "18\t2\t0.0\t2\t1 0 1\n",
      "19\t4\t0.0\t2\t1 1 2\n",
      "20\t128\t0.0\t3\t49 31 29 19\n",
      "21\t30124\t0.0\t3\t28677 1045 264 138\n",
      "22\t28918\t0.0\t3\t27458 1096 238 126\n",
      "23\t26678\t0.0\t3\t25231 1045 269 133\n",
      "24\t27385\t0.0\t3\t26023 989 236 137\n",
      "25\t24389\t0.0\t3\t22983 1008 254 144\n",
      "26\t23645\t0.0\t3\t22323 957 272 93\n",
      "27\t28\t0.0\t3\t20 3 4 1\n",
      "38\t1\t0.0\t3\t1\n",
      "59\t1\t0.0\t3\t0 0 1\n",
      "62\t3\t0.0\t3\t0 0 1 2\n",
      "65\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121703_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121703_13_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121703_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121703_13_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121703_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121703_13_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121703_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121703_13_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.57 s (24 µs/read; 2.55 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             24,189\n",
      "  Read 1 with adapter:                  23,955 (99.0%)\n",
      "  Read 2 with adapter:                  23,066 (95.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  14 (0.1%)\n",
      "Pairs discarded as untrimmed:            1,339 (5.5%)\n",
      "Pairs written (passing filters):        22,836 (94.4%)\n",
      "\n",
      "Total basepairs processed:    12,336,390 bp\n",
      "  Read 1:     7,377,645 bp\n",
      "  Read 2:     4,958,745 bp\n",
      "Total written (filtered):     10,633,459 bp (86.2%)\n",
      "  Read 1:     6,488,861 bp\n",
      "  Read 2:     4,144,598 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 56 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 44.6%\n",
      "  G: 51.8%\n",
      "  T: 3.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t0.4\t1\t0 5\n",
      "9\t4\t0.1\t1\t0 4\n",
      "21\t1\t0.0\t3\t1\n",
      "23\t2\t0.0\t3\t2\n",
      "24\t3\t0.0\t3\t3\n",
      "29\t1\t0.0\t3\t1\n",
      "32\t2\t0.0\t3\t2\n",
      "33\t1\t0.0\t3\t1\n",
      "245\t2\t0.0\t3\t2\n",
      "246\t3\t0.0\t3\t1 2\n",
      "247\t3\t0.0\t3\t3\n",
      "248\t5\t0.0\t3\t5\n",
      "249\t3\t0.0\t3\t2 0 1\n",
      "250\t1\t0.0\t3\t1\n",
      "282\t4\t0.0\t3\t4\n",
      "283\t2\t0.0\t3\t2\n",
      "284\t4\t0.0\t3\t4\n",
      "285\t5\t0.0\t3\t5\n",
      "286\t2\t0.0\t3\t2\n",
      "287\t3\t0.0\t3\t3\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 23899 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.0\t1\t1\n",
      "15\t1\t0.0\t2\t0 1\n",
      "16\t2\t0.0\t2\t1 1\n",
      "17\t7\t0.0\t2\t4 1 2\n",
      "18\t3680\t0.0\t2\t3610 63 7\n",
      "19\t4202\t0.0\t2\t4120 78 4\n",
      "20\t4407\t0.0\t2\t4323 81 3\n",
      "21\t3980\t0.0\t2\t3912 61 7\n",
      "22\t4140\t0.0\t2\t4052 85 3\n",
      "23\t3476\t0.0\t2\t3432 44\n",
      "24\t2\t0.0\t2\t1 0 1\n",
      "28\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 33 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.0%\n",
      "  C: 9.1%\n",
      "  G: 27.3%\n",
      "  T: 18.2%\n",
      "  none/other: 42.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t5\t0.0\t2\t4 1\n",
      "204\t14\t0.0\t2\t14\n",
      "205\t14\t0.0\t2\t13 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 23033 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t1\t0.0\t1\t0 1\n",
      "17\t1\t0.0\t2\t0 0 1\n",
      "18\t1\t0.0\t2\t1\n",
      "19\t1\t0.0\t2\t1\n",
      "20\t20\t0.0\t3\t10 4 3 3\n",
      "21\t4309\t0.0\t3\t4101 147 39 22\n",
      "22\t4174\t0.0\t3\t3941 164 44 25\n",
      "23\t3791\t0.0\t3\t3556 170 43 22\n",
      "24\t3843\t0.0\t3\t3609 158 47 29\n",
      "25\t3575\t0.0\t3\t3363 147 40 25\n",
      "26\t3304\t0.0\t3\t3097 150 33 24\n",
      "27\t8\t0.0\t3\t4 3 1\n",
      "43\t1\t0.0\t3\t1\n",
      "61\t2\t0.0\t3\t0 0 0 2\n",
      "62\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121704_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121704_15_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121704_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121704_15_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121704_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121704_15_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121704_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121704_15_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.41 s (74 µs/read; 0.81 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              5,549\n",
      "  Read 1 with adapter:                   5,499 (99.1%)\n",
      "  Read 2 with adapter:                   5,356 (96.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              240 (4.3%)\n",
      "Pairs written (passing filters):         5,309 (95.7%)\n",
      "\n",
      "Total basepairs processed:     2,829,990 bp\n",
      "  Read 1:     1,692,445 bp\n",
      "  Read 2:     1,137,545 bp\n",
      "Total written (filtered):      2,474,084 bp (87.4%)\n",
      "  Read 1:     1,510,232 bp\n",
      "  Read 2:       963,852 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 4 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 100.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.1\t1\t0 1\n",
      "9\t1\t0.0\t1\t0 1\n",
      "279\t1\t0.0\t3\t1\n",
      "283\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 5495 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t4\t0.0\t2\t3 1\n",
      "18\t880\t0.0\t2\t858 22\n",
      "19\t985\t0.0\t2\t966 19\n",
      "20\t1025\t0.0\t2\t1009 13 3\n",
      "21\t879\t0.0\t2\t860 19\n",
      "22\t914\t0.0\t2\t903 10 1\n",
      "23\t808\t0.0\t2\t796 10 2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 3 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 66.7%\n",
      "  G: 0.0%\n",
      "  T: 33.3%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t3\t0.0\t2\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 5353 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t2\t1\n",
      "18\t2\t0.0\t2\t1 0 1\n",
      "20\t4\t0.0\t3\t2 1 1\n",
      "21\t1027\t0.0\t3\t964 42 13 8\n",
      "22\t925\t0.0\t3\t871 35 10 9\n",
      "23\t888\t0.0\t3\t837 37 8 6\n",
      "24\t903\t0.0\t3\t857 33 5 8\n",
      "25\t850\t0.0\t3\t780 36 21 13\n",
      "26\t752\t0.0\t3\t702 28 15 7\n",
      "62\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121705_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121705_17_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121705_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121705_17_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121705_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121705_17_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121705_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121705_17_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 4.98 s (14 µs/read; 4.15 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            344,337\n",
      "  Read 1 with adapter:                 340,922 (99.0%)\n",
      "  Read 2 with adapter:                 337,158 (97.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  12 (0.0%)\n",
      "Pairs discarded as untrimmed:           10,517 (3.1%)\n",
      "Pairs written (passing filters):       333,808 (96.9%)\n",
      "\n",
      "Total basepairs processed:   175,611,870 bp\n",
      "  Read 1:   105,022,785 bp\n",
      "  Read 2:    70,589,085 bp\n",
      "Total written (filtered):    155,616,970 bp (88.6%)\n",
      "  Read 1:    94,986,971 bp\n",
      "  Read 2:    60,629,999 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 43 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 32.6%\n",
      "  G: 67.4%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t7\t5.3\t1\t0 7\n",
      "9\t7\t1.3\t1\t0 7\n",
      "20\t2\t0.0\t3\t2\n",
      "21\t7\t0.0\t3\t6 1\n",
      "22\t7\t0.0\t3\t6 0 1\n",
      "23\t1\t0.0\t3\t1\n",
      "24\t2\t0.0\t3\t2\n",
      "25\t4\t0.0\t3\t4\n",
      "26\t1\t0.0\t3\t1\n",
      "166\t1\t0.0\t3\t1\n",
      "169\t2\t0.0\t3\t2\n",
      "236\t1\t0.0\t3\t1\n",
      "237\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 340879 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.3\t1\t1\n",
      "9\t4\t1.3\t1\t4\n",
      "10\t9\t0.3\t1\t8 1\n",
      "11\t10\t0.1\t1\t7 3\n",
      "12\t6\t0.0\t1\t6\n",
      "13\t10\t0.0\t1\t6 4\n",
      "14\t30\t0.0\t2\t19 7 4\n",
      "15\t16\t0.0\t2\t7 4 5\n",
      "16\t24\t0.0\t2\t15 6 3\n",
      "17\t199\t0.0\t2\t102 57 40\n",
      "18\t53126\t0.0\t2\t52172 916 38\n",
      "19\t61651\t0.0\t2\t60490 1088 73\n",
      "20\t62544\t0.0\t2\t61451 1030 63\n",
      "21\t56263\t0.0\t2\t55262 938 63\n",
      "22\t57693\t0.0\t2\t56689 936 68\n",
      "23\t49232\t0.0\t2\t48452 763 17\n",
      "24\t57\t0.0\t2\t43 5 9\n",
      "26\t1\t0.0\t2\t1\n",
      "34\t1\t0.0\t2\t0 1\n",
      "45\t1\t0.0\t2\t1\n",
      "63\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 26 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.8%\n",
      "  C: 15.4%\n",
      "  G: 23.1%\n",
      "  T: 11.5%\n",
      "  none/other: 46.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.3\t1\t0 1\n",
      "203\t7\t0.0\t2\t7\n",
      "204\t6\t0.0\t2\t6\n",
      "205\t12\t0.0\t2\t12\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 337132 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.1\t1\t1 1\n",
      "13\t1\t0.0\t1\t0 1\n",
      "14\t6\t0.0\t2\t4 2\n",
      "15\t14\t0.0\t2\t10 3 1\n",
      "16\t2\t0.0\t2\t1 0 1\n",
      "17\t7\t0.0\t2\t3 2 2\n",
      "18\t9\t0.0\t2\t4 1 4\n",
      "19\t20\t0.0\t2\t10 4 6\n",
      "20\t252\t0.0\t3\t99 68 45 40\n",
      "21\t62962\t0.0\t3\t59575 2455 647 285\n",
      "22\t60384\t0.0\t3\t57160 2384 549 291\n",
      "23\t55799\t0.0\t3\t52497 2376 601 325\n",
      "24\t56809\t0.0\t3\t53427 2402 666 314\n",
      "25\t51187\t0.0\t3\t48103 2189 587 308\n",
      "26\t49591\t0.0\t3\t46434 2273 594 290\n",
      "27\t52\t0.0\t3\t40 8 4\n",
      "28\t2\t0.0\t3\t2\n",
      "32\t1\t0.0\t3\t0 0 1\n",
      "33\t1\t0.0\t3\t0 0 1\n",
      "35\t1\t0.0\t3\t0 0 1\n",
      "36\t1\t0.0\t3\t1\n",
      "52\t1\t0.0\t3\t0 0 0 1\n",
      "56\t1\t0.0\t3\t0 0 1\n",
      "57\t2\t0.0\t3\t0 0 0 2\n",
      "58\t1\t0.0\t3\t0 0 0 1\n",
      "59\t2\t0.0\t3\t0 0 2\n",
      "60\t1\t0.0\t3\t0 0 0 1\n",
      "61\t5\t0.0\t3\t1 0 0 4\n",
      "62\t4\t0.0\t3\t0 0 3 1\n",
      "63\t4\t0.0\t3\t0 1 2 1\n",
      "64\t3\t0.0\t3\t0 0 0 3\n",
      "65\t2\t0.0\t3\t0 0 1 1\n",
      "66\t2\t0.0\t3\t0 0 0 2\n",
      "67\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121706_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121706_19_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121706_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121706_19_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121706_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121706_19_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121706_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121706_19_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 6.05 s (15 µs/read; 4.13 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            416,322\n",
      "  Read 1 with adapter:                 412,332 (99.0%)\n",
      "  Read 2 with adapter:                 407,730 (97.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  21 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,483 (3.0%)\n",
      "Pairs written (passing filters):       403,818 (97.0%)\n",
      "\n",
      "Total basepairs processed:   212,324,220 bp\n",
      "  Read 1:   126,978,210 bp\n",
      "  Read 2:    85,346,010 bp\n",
      "Total written (filtered):    188,257,614 bp (88.7%)\n",
      "  Read 1:   114,910,144 bp\n",
      "  Read 2:    73,347,470 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 75 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 40.0%\n",
      "  G: 54.7%\n",
      "  T: 5.3%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t19\t6.4\t1\t0 19\n",
      "9\t15\t1.6\t1\t0 15\n",
      "18\t1\t0.0\t2\t1\n",
      "20\t3\t0.0\t3\t3\n",
      "21\t7\t0.0\t3\t5 1 1\n",
      "22\t5\t0.0\t3\t5\n",
      "23\t7\t0.0\t3\t6 1\n",
      "24\t9\t0.0\t3\t7 2\n",
      "25\t3\t0.0\t3\t3\n",
      "26\t4\t0.0\t3\t2 0 2\n",
      "84\t1\t0.0\t3\t1\n",
      "282\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 412257 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t6.4\t1\t4 1\n",
      "9\t3\t1.6\t1\t2 1\n",
      "10\t5\t0.4\t1\t3 2\n",
      "11\t8\t0.1\t1\t5 3\n",
      "12\t7\t0.0\t1\t4 3\n",
      "13\t8\t0.0\t1\t3 5\n",
      "14\t27\t0.0\t2\t17 7 3\n",
      "15\t21\t0.0\t2\t12 1 8\n",
      "16\t35\t0.0\t2\t26 7 2\n",
      "17\t183\t0.0\t2\t108 52 23\n",
      "18\t64944\t0.0\t2\t63727 1168 49\n",
      "19\t73895\t0.0\t2\t72605 1226 64\n",
      "20\t75810\t0.0\t2\t74487 1261 62\n",
      "21\t67970\t0.0\t2\t66807 1096 67\n",
      "22\t69866\t0.0\t2\t68669 1144 53\n",
      "23\t59387\t0.0\t2\t58482 880 25\n",
      "24\t81\t0.0\t2\t63 6 12\n",
      "25\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 37 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 16.2%\n",
      "  G: 18.9%\n",
      "  T: 8.1%\n",
      "  none/other: 56.8%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.4\t1\t0 1\n",
      "61\t1\t0.0\t2\t0 0 1\n",
      "203\t7\t0.0\t2\t6 1\n",
      "204\t7\t0.0\t2\t7\n",
      "205\t21\t0.0\t2\t20 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 407693 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t6.4\t1\t0 1\n",
      "11\t1\t0.1\t1\t1\n",
      "12\t7\t0.0\t1\t5 2\n",
      "13\t3\t0.0\t1\t0 3\n",
      "14\t3\t0.0\t2\t1 1 1\n",
      "15\t22\t0.0\t2\t12 3 7\n",
      "16\t6\t0.0\t2\t1 1 4\n",
      "17\t11\t0.0\t2\t7 2 2\n",
      "18\t15\t0.0\t2\t8 5 2\n",
      "19\t17\t0.0\t2\t9 5 3\n",
      "20\t328\t0.0\t3\t138 84 57 49\n",
      "21\t75920\t0.0\t3\t72215 2693 651 361\n",
      "22\t73065\t0.0\t3\t69374 2679 661 351\n",
      "23\t67778\t0.0\t3\t63845 2911 692 330\n",
      "24\t69099\t0.0\t3\t65324 2732 710 333\n",
      "25\t62151\t0.0\t3\t58672 2553 622 304\n",
      "26\t59158\t0.0\t3\t55823 2417 622 296\n",
      "27\t63\t0.0\t3\t51 5 4 3\n",
      "28\t2\t0.0\t3\t2\n",
      "30\t1\t0.0\t3\t0 0 1\n",
      "36\t1\t0.0\t3\t0 1\n",
      "55\t1\t0.0\t3\t0 0 1\n",
      "57\t1\t0.0\t3\t0 0 0 1\n",
      "58\t4\t0.0\t3\t0 2 0 2\n",
      "59\t2\t0.0\t3\t0 0 0 2\n",
      "60\t3\t0.0\t3\t0 0 1 2\n",
      "61\t6\t0.0\t3\t0 0 4 2\n",
      "62\t10\t0.0\t3\t1 0 3 6\n",
      "63\t2\t0.0\t3\t0 1 0 1\n",
      "64\t7\t0.0\t3\t0 0 3 4\n",
      "65\t2\t0.0\t3\t0 0 1 1\n",
      "66\t3\t0.0\t3\t0 0 1 2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121707_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121707_21_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121707_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121707_21_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121707_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121707_21_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121707_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121707_21_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.37 s (17 µs/read; 3.58 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             81,485\n",
      "  Read 1 with adapter:                  80,768 (99.1%)\n",
      "  Read 2 with adapter:                  79,421 (97.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  10 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,761 (3.4%)\n",
      "Pairs written (passing filters):        78,714 (96.6%)\n",
      "\n",
      "Total basepairs processed:    41,557,350 bp\n",
      "  Read 1:    24,852,925 bp\n",
      "  Read 2:    16,704,425 bp\n",
      "Total written (filtered):     36,688,578 bp (88.3%)\n",
      "  Read 1:    22,394,693 bp\n",
      "  Read 2:    14,293,885 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 38 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 47.4%\n",
      "  C: 10.5%\n",
      "  G: 39.5%\n",
      "  T: 2.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t1.2\t1\t0 3\n",
      "9\t1\t0.3\t1\t0 1\n",
      "20\t1\t0.0\t3\t1\n",
      "21\t1\t0.0\t3\t1\n",
      "24\t1\t0.0\t3\t1\n",
      "29\t4\t0.0\t3\t2 2\n",
      "30\t3\t0.0\t3\t3\n",
      "32\t1\t0.0\t3\t1\n",
      "33\t1\t0.0\t3\t1\n",
      "34\t1\t0.0\t3\t1\n",
      "192\t5\t0.0\t3\t4 1\n",
      "193\t3\t0.0\t3\t3\n",
      "194\t1\t0.0\t3\t1\n",
      "195\t4\t0.0\t3\t4\n",
      "196\t3\t0.0\t3\t3\n",
      "197\t2\t0.0\t3\t2\n",
      "215\t1\t0.0\t3\t0 0 1\n",
      "277\t1\t0.0\t3\t0 0 1\n",
      "287\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 80730 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t2\t0.3\t1\t2\n",
      "10\t3\t0.1\t1\t2 1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "14\t6\t0.0\t2\t4 0 2\n",
      "15\t6\t0.0\t2\t3 1 2\n",
      "16\t3\t0.0\t2\t2 0 1\n",
      "17\t53\t0.0\t2\t34 11 8\n",
      "18\t12514\t0.0\t2\t12316 193 5\n",
      "19\t14571\t0.0\t2\t14298 256 17\n",
      "20\t14675\t0.0\t2\t14381 282 12\n",
      "21\t13350\t0.0\t2\t13116 221 13\n",
      "22\t13670\t0.0\t2\t13440 224 6\n",
      "23\t11864\t0.0\t2\t11665 193 6\n",
      "24\t9\t0.0\t2\t5 2 2\n",
      "28\t1\t0.0\t2\t1\n",
      "30\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 30 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.3%\n",
      "  C: 30.0%\n",
      "  G: 20.0%\n",
      "  T: 13.3%\n",
      "  none/other: 33.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t10\t0.0\t2\t10\n",
      "204\t10\t0.0\t2\t10\n",
      "205\t10\t0.0\t2\t10\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 79391 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.1\t1\t2\n",
      "11\t1\t0.0\t1\t0 1\n",
      "12\t1\t0.0\t1\t0 1\n",
      "13\t1\t0.0\t1\t0 1\n",
      "14\t2\t0.0\t2\t1 1\n",
      "15\t3\t0.0\t2\t1 1 1\n",
      "16\t1\t0.0\t2\t0 1\n",
      "17\t2\t0.0\t2\t1 0 1\n",
      "18\t1\t0.0\t2\t0 1\n",
      "19\t5\t0.0\t2\t2 0 3\n",
      "20\t60\t0.0\t3\t25 16 12 7\n",
      "21\t14780\t0.0\t3\t14012 563 143 62\n",
      "22\t14314\t0.0\t3\t13493 614 137 70\n",
      "23\t13139\t0.0\t3\t12382 556 145 56\n",
      "24\t13365\t0.0\t3\t12641 534 116 74\n",
      "25\t11962\t0.0\t3\t11252 504 134 72\n",
      "26\t11730\t0.0\t3\t10983 546 143 58\n",
      "27\t12\t0.0\t3\t8 2 1 1\n",
      "50\t1\t0.0\t3\t0 0 0 1\n",
      "53\t1\t0.0\t3\t1\n",
      "55\t1\t0.0\t3\t0 0 0 1\n",
      "61\t2\t0.0\t3\t0 0 0 2\n",
      "63\t3\t0.0\t3\t0 0 1 2\n",
      "64\t2\t0.0\t3\t0 0 2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121708_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121708_23_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121708_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121708_23_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121708_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121708_23_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121708_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121708_23_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.64 s (24 µs/read; 2.55 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             27,196\n",
      "  Read 1 with adapter:                  26,931 (99.0%)\n",
      "  Read 2 with adapter:                  26,035 (95.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  18 (0.1%)\n",
      "Pairs discarded as untrimmed:            1,407 (5.2%)\n",
      "Pairs written (passing filters):        25,771 (94.8%)\n",
      "\n",
      "Total basepairs processed:    13,869,960 bp\n",
      "  Read 1:     8,294,780 bp\n",
      "  Read 2:     5,575,180 bp\n",
      "Total written (filtered):     12,010,229 bp (86.6%)\n",
      "  Read 1:     7,333,324 bp\n",
      "  Read 2:     4,676,905 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 12 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 33.3%\n",
      "  G: 58.3%\n",
      "  T: 0.0%\n",
      "  none/other: 8.3%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t0.4\t1\t0 2\n",
      "9\t2\t0.1\t1\t0 2\n",
      "30\t1\t0.0\t3\t1\n",
      "31\t1\t0.0\t3\t1\n",
      "32\t2\t0.0\t3\t2\n",
      "33\t1\t0.0\t3\t1\n",
      "254\t1\t0.0\t3\t1\n",
      "257\t1\t0.0\t3\t1\n",
      "305\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 26919 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t1\t1\n",
      "14\t2\t0.0\t2\t2\n",
      "16\t1\t0.0\t2\t0 1\n",
      "17\t13\t0.0\t2\t6 6 1\n",
      "18\t4224\t0.0\t2\t4153 66 5\n",
      "19\t4838\t0.0\t2\t4744 94\n",
      "20\t4950\t0.0\t2\t4856 92 2\n",
      "21\t4449\t0.0\t2\t4345 98 6\n",
      "22\t4550\t0.0\t2\t4472 74 4\n",
      "23\t3887\t0.0\t2\t3838 48 1\n",
      "24\t4\t0.0\t2\t3 0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 38 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 5.3%\n",
      "  C: 13.2%\n",
      "  G: 28.9%\n",
      "  T: 7.9%\n",
      "  none/other: 44.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t10\t0.0\t2\t9 1\n",
      "204\t11\t0.0\t2\t11\n",
      "205\t17\t0.0\t2\t13 3 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 25997 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1\t0.0\t2\t1\n",
      "20\t14\t0.0\t3\t5 5 2 2\n",
      "21\t4795\t0.0\t3\t4531 193 46 25\n",
      "22\t4571\t0.0\t3\t4333 178 41 19\n",
      "23\t4434\t0.0\t3\t4146 200 57 31\n",
      "24\t4389\t0.0\t3\t4150 182 40 17\n",
      "25\t3946\t0.0\t3\t3714 164 39 29\n",
      "26\t3839\t0.0\t3\t3599 164 55 21\n",
      "27\t4\t0.0\t3\t1 0 3\n",
      "28\t1\t0.0\t3\t1\n",
      "54\t1\t0.0\t3\t0 0 0 1\n",
      "61\t1\t0.0\t3\t0 0 1\n",
      "64\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121709_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121709_25_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121709_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121709_25_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121709_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121709_25_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121709_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121709_25_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.72 s (22 µs/read; 2.75 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             32,857\n",
      "  Read 1 with adapter:                  32,525 (99.0%)\n",
      "  Read 2 with adapter:                  31,730 (96.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  11 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,442 (4.4%)\n",
      "Pairs written (passing filters):        31,404 (95.6%)\n",
      "\n",
      "Total basepairs processed:    16,757,070 bp\n",
      "  Read 1:    10,021,385 bp\n",
      "  Read 2:     6,735,685 bp\n",
      "Total written (filtered):     14,618,640 bp (87.2%)\n",
      "  Read 1:     8,915,974 bp\n",
      "  Read 2:     5,702,666 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 131 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 44.3%\n",
      "  C: 6.9%\n",
      "  G: 48.1%\n",
      "  T: 0.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "21\t1\t0.0\t3\t0 1\n",
      "24\t1\t0.0\t3\t1\n",
      "25\t1\t0.0\t3\t1\n",
      "29\t1\t0.0\t3\t1\n",
      "30\t4\t0.0\t3\t4\n",
      "31\t4\t0.0\t3\t4\n",
      "32\t8\t0.0\t3\t7 1\n",
      "33\t2\t0.0\t3\t2\n",
      "34\t1\t0.0\t3\t1\n",
      "151\t2\t0.0\t3\t2\n",
      "153\t3\t0.0\t3\t3\n",
      "182\t2\t0.0\t3\t2\n",
      "183\t2\t0.0\t3\t2\n",
      "184\t1\t0.0\t3\t0 1\n",
      "187\t4\t0.0\t3\t4\n",
      "188\t1\t0.0\t3\t1\n",
      "189\t1\t0.0\t3\t1\n",
      "190\t1\t0.0\t3\t1\n",
      "191\t1\t0.0\t3\t1\n",
      "192\t9\t0.0\t3\t9\n",
      "193\t7\t0.0\t3\t7\n",
      "194\t5\t0.0\t3\t5\n",
      "195\t5\t0.0\t3\t5\n",
      "196\t8\t0.0\t3\t8\n",
      "197\t9\t0.0\t3\t9\n",
      "198\t2\t0.0\t3\t2\n",
      "215\t2\t0.0\t3\t2\n",
      "216\t2\t0.0\t3\t1 0 1\n",
      "217\t8\t0.0\t3\t8\n",
      "218\t8\t0.0\t3\t8\n",
      "219\t11\t0.0\t3\t11\n",
      "220\t5\t0.0\t3\t5\n",
      "265\t1\t0.0\t3\t0 1\n",
      "266\t1\t0.0\t3\t1\n",
      "270\t2\t0.0\t3\t2\n",
      "276\t2\t0.0\t3\t2\n",
      "278\t1\t0.0\t3\t1\n",
      "280\t1\t0.0\t3\t1\n",
      "281\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 32394 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.0\t1\t2\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t1\t0.0\t1\t1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t1\t0.0\t2\t0 1\n",
      "16\t3\t0.0\t2\t1 1 1\n",
      "17\t20\t0.0\t2\t9 6 5\n",
      "18\t5045\t0.0\t2\t4958 83 4\n",
      "19\t5807\t0.0\t2\t5693 109 5\n",
      "20\t5921\t0.0\t2\t5814 101 6\n",
      "21\t5289\t0.0\t2\t5195 92 2\n",
      "22\t5568\t0.0\t2\t5485 77 6\n",
      "23\t4729\t0.0\t2\t4661 66 2\n",
      "24\t5\t0.0\t2\t5\n",
      "29\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 19 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 15.8%\n",
      "  G: 21.1%\n",
      "  T: 5.3%\n",
      "  none/other: 57.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "92\t1\t0.0\t2\t1\n",
      "144\t1\t0.0\t2\t1\n",
      "203\t1\t0.0\t2\t1\n",
      "204\t5\t0.0\t2\t4 1\n",
      "205\t11\t0.0\t2\t10 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 31711 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t1\t1\n",
      "15\t4\t0.0\t2\t2 2\n",
      "19\t1\t0.0\t2\t1\n",
      "20\t24\t0.0\t3\t12 4 5 3\n",
      "21\t5800\t0.0\t3\t5470 241 58 31\n",
      "22\t5699\t0.0\t3\t5373 232 61 33\n",
      "23\t5351\t0.0\t3\t5010 245 58 38\n",
      "24\t5376\t0.0\t3\t5115 196 43 22\n",
      "25\t4814\t0.0\t3\t4536 206 55 17\n",
      "26\t4630\t0.0\t3\t4354 207 42 27\n",
      "27\t5\t0.0\t3\t2 2 0 1\n",
      "63\t2\t0.0\t3\t0 0 0 2\n",
      "64\t2\t0.0\t3\t0 1 0 1\n",
      "65\t1\t0.0\t3\t0 0 0 1\n",
      "66\t1\t0.0\t3\t0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121710_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121710_27_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121710_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121710_27_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121710_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-z5kdrf1y/210421121710_27_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121710_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-xpuveehl/399d6386-b73f-448c-a5a2-dc63a94a4515/data/210421121710_27_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.32 s (65 µs/read; 0.92 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              4,908\n",
      "  Read 1 with adapter:                   4,853 (98.9%)\n",
      "  Read 2 with adapter:                   4,578 (93.3%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   3 (0.1%)\n",
      "Pairs discarded as untrimmed:              377 (7.7%)\n",
      "Pairs written (passing filters):         4,528 (92.3%)\n",
      "\n",
      "Total basepairs processed:     2,503,080 bp\n",
      "  Read 1:     1,496,940 bp\n",
      "  Read 2:     1,006,140 bp\n",
      "Total written (filtered):      2,106,079 bp (84.1%)\n",
      "  Read 1:     1,284,585 bp\n",
      "  Read 2:       821,494 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 31 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.2%\n",
      "  C: 25.8%\n",
      "  G: 58.1%\n",
      "  T: 12.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.0\t1\t0 1\n",
      "21\t1\t0.0\t3\t1\n",
      "22\t1\t0.0\t3\t1\n",
      "30\t1\t0.0\t3\t1\n",
      "31\t1\t0.0\t3\t1\n",
      "32\t5\t0.0\t3\t5\n",
      "33\t5\t0.0\t3\t5\n",
      "34\t1\t0.0\t3\t1\n",
      "132\t1\t0.0\t3\t1\n",
      "194\t1\t0.0\t3\t1\n",
      "210\t1\t0.0\t3\t0 0 1\n",
      "274\t1\t0.0\t3\t1\n",
      "279\t1\t0.0\t3\t1\n",
      "281\t6\t0.0\t3\t5 1\n",
      "283\t2\t0.0\t3\t2\n",
      "284\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 4822 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t2\t0.0\t2\t0 0 2\n",
      "17\t2\t0.0\t2\t0 2\n",
      "18\t769\t0.0\t2\t754 14 1\n",
      "19\t841\t0.0\t2\t828 11 2\n",
      "20\t839\t0.0\t2\t823 14 2\n",
      "21\t779\t0.0\t2\t766 13\n",
      "22\t839\t0.0\t2\t819 19 1\n",
      "23\t750\t0.0\t2\t734 15 1\n",
      "24\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 8 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 12.5%\n",
      "  G: 25.0%\n",
      "  T: 25.0%\n",
      "  none/other: 37.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t1\t0.0\t2\t1\n",
      "204\t4\t0.0\t2\t3 1\n",
      "205\t3\t0.0\t2\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 4570 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t2\t1\n",
      "20\t6\t0.0\t3\t2 3 0 1\n",
      "21\t864\t0.0\t3\t819 30 12 3\n",
      "22\t809\t0.0\t3\t759 31 13 6\n",
      "23\t717\t0.0\t3\t673 31 6 7\n",
      "24\t814\t0.0\t3\t752 44 12 6\n",
      "25\t686\t0.0\t3\t638 33 6 9\n",
      "26\t671\t0.0\t3\t613 38 13 7\n",
      "51\t1\t0.0\t3\t0 1\n",
      "64\t1\t0.0\t3\t0 0 0 1\n"
     ]
    }
   ],
   "source": [
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')\n",
    "\n",
    "if trim and (not os.path.isfile(demux_file_trim) or replace_files):\n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = trim_paired(\n",
    "        demultiplexed_sequences=artifact,\n",
    "        front_f=forward_primer,\n",
    "        front_r=reverse_primer,\n",
    "        adapter_f=reverse_reverse_complement,\n",
    "        adapter_r=forward_reverse_complement,\n",
    "        cores=threads,\n",
    "        overlap=trim['overlap'],\n",
    "        indels=False,\n",
    "        match_read_wildcards=True,\n",
    "        match_adapter_wildcards=True,\n",
    "        error_rate=0.15,\n",
    "        discard_untrimmed=True,\n",
    "    ).trimmed_sequences\n",
    "\n",
    "    demux.visualizers.summarize(res).visualization\n",
    "    res.save(demux_file_trim)\n",
    "    Visualization.save(demux.visualizers.summarize(res).visualization, filepath=demux_view_trim)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 591.128253,
   "end_time": "2022-10-24T20:04:12.633524",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-templates/step-prepare-data.ipynb",
   "output_path": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-STD-NRxHSD-NR-trim/nb-executed-steps/step-prepare-data-ana-flavia-STD-NRxHSD-NR-trim.ipynb",
   "parameters": {
    "base_dir": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri",
    "class_col": "group-id",
    "classifier_file": "/home/lauro/nupeb/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza",
    "experiment_name": "ana-flavia-STD-NRxHSD-NR-trim",
    "manifest_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-STD-NRxHSD-NR.csv",
    "metadata_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-STD-NRxHSD-NR.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2022-10-24T19:54:21.505271",
   "version": "2.3.4"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}