{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "23e025ca",
   "metadata": {
    "papermill": {
     "duration": 0.018459,
     "end_time": "2022-09-15T17:16:33.971018",
     "exception": false,
     "start_time": "2022-09-15T17:16:33.952559",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1434d405",
   "metadata": {
    "papermill": {
     "duration": 0.027826,
     "end_time": "2022-09-15T17:16:34.088681",
     "exception": false,
     "start_time": "2022-09-15T17:16:34.060855",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6c033073",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-14T17:43:20.516262Z",
     "start_time": "2022-09-14T17:43:20.510424Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-15T17:16:34.113437Z",
     "iopub.status.busy": "2022-09-15T17:16:34.112982Z",
     "iopub.status.idle": "2022-09-15T17:16:40.776248Z",
     "shell.execute_reply": "2022-09-15T17:16:40.775368Z"
    },
    "papermill": {
     "duration": 6.679381,
     "end_time": "2022-09-15T17:16:40.779529",
     "exception": false,
     "start_time": "2022-09-15T17:16:34.100148",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired\n",
    "\n",
    "from utils import *"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e97b8e5",
   "metadata": {
    "papermill": {
     "duration": 0.016868,
     "end_time": "2022-09-15T17:16:40.812514",
     "exception": false,
     "start_time": "2022-09-15T17:16:40.795646",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "928d1e4f",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.594341Z",
     "start_time": "2022-06-28T18:50:41.591544Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-15T17:16:40.850901Z",
     "iopub.status.busy": "2022-09-15T17:16:40.850610Z",
     "iopub.status.idle": "2022-09-15T17:16:40.855935Z",
     "shell.execute_reply": "2022-09-15T17:16:40.855188Z"
    },
    "papermill": {
     "duration": 0.036042,
     "end_time": "2022-09-15T17:16:40.857356",
     "exception": false,
     "start_time": "2022-09-15T17:16:40.821314",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'ana-flavia-superlactacao.yaml')\n",
    "experiment_name = 'jenneffer-vs-01'\n",
    "base_dir = os.path.join('/', 'home', 'lauro', 'nupeb', 'redemicro')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'not-hist-vs-manifest.csv')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'imgs'))\n",
    "replace_files = False\n",
    "trim = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0e973e1b",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-15T17:16:40.879536Z",
     "iopub.status.busy": "2022-09-15T17:16:40.879341Z",
     "iopub.status.idle": "2022-09-15T17:16:40.883132Z",
     "shell.execute_reply": "2022-09-15T17:16:40.882580Z"
    },
    "papermill": {
     "duration": 0.018673,
     "end_time": "2022-09-15T17:16:40.886479",
     "exception": false,
     "start_time": "2022-09-15T17:16:40.867806",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "experiment_name = \"ana-flavia-STD-NCxSTD-NR-trim\"\n",
    "base_dir = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri\"\n",
    "manifest_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-STD-NCxSTD-NR.csv\"\n",
    "metadata_file = \"/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-STD-NCxSTD-NR.tsv\"\n",
    "class_col = \"sample-group\"\n",
    "classifier_file = \"/home/lauro/nupeb/dados_brutos_rede_genoma/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza\"\n",
    "replace_files = False\n",
    "phred = 20\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n",
    "overlap = 12\n",
    "threads = 6\n",
    "trim = {\n",
    "    \"overlap\": 8,\n",
    "    \"forward_primer\": \"CCTACGGGRSGCAGCAG\",\n",
    "    \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\",\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b37263c2",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.610682Z",
     "start_time": "2022-06-28T18:50:41.595886Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-15T17:16:40.911591Z",
     "iopub.status.busy": "2022-09-15T17:16:40.911388Z",
     "iopub.status.idle": "2022-09-15T17:16:40.916112Z",
     "shell.execute_reply": "2022-09-15T17:16:40.915348Z"
    },
    "papermill": {
     "duration": 0.021354,
     "end_time": "2022-09-15T17:16:40.917475",
     "exception": false,
     "start_time": "2022-09-15T17:16:40.896121",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Setting hardcoded parameters for development and debbuging\n",
    "production = True\n",
    "if not production:\n",
    "    params_path = os.path.join('..', 'params', 'ana-flavia-hipotese-01')\n",
    "    with open(params_path, 'r') as stream:\n",
    "        params = yaml.safe_load(stream)\n",
    "        experiment_name = params['experiment_name']\n",
    "        base_dir = params['base_dir']\n",
    "        manifest_file = params['manifest_file']\n",
    "        replace_files = params['replace_files']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0c3a6da",
   "metadata": {
    "papermill": {
     "duration": 0.010277,
     "end_time": "2022-09-15T17:16:40.937900",
     "exception": false,
     "start_time": "2022-09-15T17:16:40.927623",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d8fee342",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.618098Z",
     "start_time": "2022-06-28T18:50:41.614152Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-15T17:16:40.964777Z",
     "iopub.status.busy": "2022-09-15T17:16:40.964514Z",
     "iopub.status.idle": "2022-09-15T17:16:40.967930Z",
     "shell.execute_reply": "2022-09-15T17:16:40.967278Z"
    },
    "papermill": {
     "duration": 0.014914,
     "end_time": "2022-09-15T17:16:40.969261",
     "exception": false,
     "start_time": "2022-09-15T17:16:40.954347",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# new_manifest = '/home/lauro/nupeb/redemicro/data/raw/manifest/karina-manifest.csv'\n",
    "# with open(manifest_file, 'r') as oldm, open(new_manifest, 'w') as newm:\n",
    "#     header = 'sample-id,absolute-filepath,direction\\n'\n",
    "#     newm.write(header)\n",
    "#     for line in oldm.readlines()[1:]:\n",
    "#         sid, forward, reverse = line[:-1].split(',')\n",
    "#         fline = ','.join((sid, forward, 'forward')) + '\\n'\n",
    "#         rline = ','.join((sid, reverse, 'reverse')) + '\\n'\n",
    "#         newm.write(fline)\n",
    "#         newm.write(rline)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d55757f0",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.632125Z",
     "start_time": "2022-06-28T18:50:41.619710Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-15T17:16:40.989347Z",
     "iopub.status.busy": "2022-09-15T17:16:40.989087Z",
     "iopub.status.idle": "2022-09-15T17:16:40.996508Z",
     "shell.execute_reply": "2022-09-15T17:16:40.995837Z"
    },
    "papermill": {
     "duration": 0.019027,
     "end_time": "2022-09-15T17:16:40.997860",
     "exception": false,
     "start_time": "2022-09-15T17:16:40.978833",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New qiime-artifacts folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-STD-NCxSTD-NR-trim/qiime-artifacts\n",
      "New img folder path created: /home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-STD-NCxSTD-NR-trim/imgs\n"
     ]
    }
   ],
   "source": [
    "# Define the output folder path\n",
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "\n",
    "# Create path if it not exist\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exist\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')\n",
    "\n",
    "# Define the output artifact full path\n",
    "demux_file = os.path.join(out_dir, 'demux-paired.qza')\n",
    "demux_view = os.path.join(out_dir, 'demux-paired.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3cfe5d71",
   "metadata": {
    "papermill": {
     "duration": 0.010342,
     "end_time": "2022-09-15T17:16:41.019664",
     "exception": false,
     "start_time": "2022-09-15T17:16:41.009322",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "7a729a2e",
   "metadata": {
    "papermill": {
     "duration": 0.009478,
     "end_time": "2022-09-15T17:16:41.038629",
     "exception": false,
     "start_time": "2022-09-15T17:16:41.029151",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file of `SampleData[PairedEndSequencesWithQuality]`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a3a8700b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:44:58.640338Z",
     "start_time": "2022-06-28T18:50:41.633657Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-15T17:16:41.064091Z",
     "iopub.status.busy": "2022-09-15T17:16:41.063755Z",
     "iopub.status.idle": "2022-09-15T17:26:16.070408Z",
     "shell.execute_reply": "2022-09-15T17:26:16.069629Z"
    },
    "papermill": {
     "duration": 575.025432,
     "end_time": "2022-09-15T17:26:16.073481",
     "exception": false,
     "start_time": "2022-09-15T17:16:41.048049",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    \n",
    "    # Import data and create an artifact object\n",
    "    artifact = Artifact.import_data(\n",
    "        'SampleData[PairedEndSequencesWithQuality]', \n",
    "        manifest_file, \n",
    "        view_type='PairedEndFastqManifestPhred33')\n",
    "    \n",
    "    # Save the artifact object to a new qza file\n",
    "    artifact.save(demux_file)\n",
    "\n",
    "else:\n",
    "    artifact = Artifact.load(demux_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ffd1b02e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.602855Z",
     "start_time": "2022-06-28T19:44:58.642520Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-15T17:26:16.195470Z",
     "iopub.status.busy": "2022-09-15T17:26:16.195296Z",
     "iopub.status.idle": "2022-09-15T17:26:51.195481Z",
     "shell.execute_reply": "2022-09-15T17:26:51.194892Z"
    },
    "papermill": {
     "duration": 35.106457,
     "end_time": "2022-09-15T17:26:51.198482",
     "exception": false,
     "start_time": "2022-09-15T17:26:16.092025",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "if not os.path.isfile(demux_view) or replace_files: \n",
    "    # Generate e visualization of the Artifact\n",
    "    demux_summary = demux.visualizers.summarize(artifact)\n",
    "\n",
    "    # Save a new visualization file based on the qza file\n",
    "    Visualization.save(demux_summary.visualization, filepath=demux_view)\n",
    "    \n",
    "    demux_view_obj = demux_summary.visualization\n",
    "else:\n",
    "    demux_view_obj = Visualization.load(demux_view)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "63aaacff",
   "metadata": {
    "papermill": {
     "duration": 0.081826,
     "end_time": "2022-09-15T17:26:51.298465",
     "exception": false,
     "start_time": "2022-09-15T17:26:51.216639",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2cb826d8",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.609737Z",
     "start_time": "2022-06-28T19:47:56.604820Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-15T17:26:51.365366Z",
     "iopub.status.busy": "2022-09-15T17:26:51.365120Z",
     "iopub.status.idle": "2022-09-15T17:26:51.370145Z",
     "shell.execute_reply": "2022-09-15T17:26:51.369459Z"
    },
    "papermill": {
     "duration": 0.020879,
     "end_time": "2022-09-15T17:26:51.372901",
     "exception": false,
     "start_time": "2022-09-15T17:26:51.352022",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<visualization: Visualization uuid: cc615010-9536-4402-b973-f7a739541f4d>\n"
     ]
    }
   ],
   "source": [
    "print(demux_view_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "5df1dafb",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.704841Z",
     "start_time": "2022-06-28T19:47:56.611846Z"
    },
    "execution": {
     "iopub.execute_input": "2022-09-15T17:26:51.399044Z",
     "iopub.status.busy": "2022-09-15T17:26:51.398841Z",
     "iopub.status.idle": "2022-09-15T17:26:51.407923Z",
     "shell.execute_reply": "2022-09-15T17:26:51.407465Z"
    },
    "papermill": {
     "duration": 0.023194,
     "end_time": "2022-09-15T17:26:51.409354",
     "exception": false,
     "start_time": "2022-09-15T17:26:51.386160",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2-archive-cdmv2gdy')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: cc615010-9536-4402-b973-f7a739541f4d>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "547b4999",
   "metadata": {
    "papermill": {
     "duration": 0.013862,
     "end_time": "2022-09-15T17:26:51.438337",
     "exception": false,
     "start_time": "2022-09-15T17:26:51.424475",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "32f34286",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2022-09-15T17:26:51.464375Z",
     "iopub.status.busy": "2022-09-15T17:26:51.464207Z",
     "iopub.status.idle": "2022-09-15T17:28:51.711162Z",
     "shell.execute_reply": "2022-09-15T17:28:51.710621Z"
    },
    "papermill": {
     "duration": 120.263312,
     "end_time": "2022-09-15T17:28:51.714194",
     "exception": false,
     "start_time": "2022-09-15T17:26:51.450882",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121682_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121682_1_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121682_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121682_1_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121682_0_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121682_1_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121682_0_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121682_1_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.72 s (16 µs/read; 3.65 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            165,821\n",
      "  Read 1 with adapter:                 164,234 (99.0%)\n",
      "  Read 2 with adapter:                 162,538 (98.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   2 (0.0%)\n",
      "Pairs discarded as untrimmed:            4,837 (2.9%)\n",
      "Pairs written (passing filters):       160,982 (97.1%)\n",
      "\n",
      "Total basepairs processed:    84,568,710 bp\n",
      "  Read 1:    50,575,405 bp\n",
      "  Read 2:    33,993,305 bp\n",
      "Total written (filtered):     75,046,538 bp (88.7%)\n",
      "  Read 1:    45,805,681 bp\n",
      "  Read 2:    29,240,857 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 38 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 18.4%\n",
      "  G: 73.7%\n",
      "  T: 7.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t2.5\t1\t0 2\n",
      "9\t3\t0.6\t1\t0 3\n",
      "20\t1\t0.0\t3\t1\n",
      "21\t3\t0.0\t3\t2 1\n",
      "22\t3\t0.0\t3\t3\n",
      "23\t3\t0.0\t3\t2 1\n",
      "24\t8\t0.0\t3\t7 1\n",
      "25\t1\t0.0\t3\t1\n",
      "29\t1\t0.0\t3\t1\n",
      "177\t1\t0.0\t3\t1\n",
      "255\t1\t0.0\t3\t0 1\n",
      "258\t1\t0.0\t3\t1\n",
      "265\t1\t0.0\t3\t1\n",
      "266\t1\t0.0\t3\t1\n",
      "268\t1\t0.0\t3\t0 0 1\n",
      "277\t2\t0.0\t3\t2\n",
      "278\t2\t0.0\t3\t2\n",
      "280\t1\t0.0\t3\t0 1\n",
      "281\t1\t0.0\t3\t1\n",
      "284\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 164196 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t2.5\t1\t3\n",
      "9\t1\t0.6\t1\t1\n",
      "10\t6\t0.2\t1\t3 3\n",
      "11\t7\t0.0\t1\t4 3\n",
      "12\t2\t0.0\t1\t1 1\n",
      "13\t3\t0.0\t1\t3\n",
      "14\t13\t0.0\t2\t8 2 3\n",
      "15\t5\t0.0\t2\t1 2 2\n",
      "16\t11\t0.0\t2\t4 4 3\n",
      "17\t85\t0.0\t2\t46 27 12\n",
      "18\t25564\t0.0\t2\t25119 428 17\n",
      "19\t29498\t0.0\t2\t28946 517 35\n",
      "20\t30404\t0.0\t2\t29904 482 18\n",
      "21\t27173\t0.0\t2\t26703 442 28\n",
      "22\t28128\t0.0\t2\t27677 430 21\n",
      "23\t23272\t0.0\t2\t22942 325 5\n",
      "24\t17\t0.0\t2\t12 1 4\n",
      "26\t1\t0.0\t2\t1\n",
      "31\t1\t0.0\t2\t1\n",
      "37\t1\t0.0\t2\t1\n",
      "38\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 8 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 37.5%\n",
      "  G: 25.0%\n",
      "  T: 12.5%\n",
      "  none/other: 25.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.5\t1\t0 1\n",
      "203\t2\t0.0\t2\t2\n",
      "204\t3\t0.0\t2\t3\n",
      "205\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 162530 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t2.5\t1\t0 1\n",
      "11\t2\t0.0\t1\t1 1\n",
      "12\t1\t0.0\t1\t1\n",
      "14\t2\t0.0\t2\t1 1\n",
      "15\t10\t0.0\t2\t3 3 4\n",
      "16\t5\t0.0\t2\t0 4 1\n",
      "17\t4\t0.0\t2\t1 2 1\n",
      "18\t7\t0.0\t2\t4 1 2\n",
      "19\t9\t0.0\t2\t5 3 1\n",
      "20\t121\t0.0\t3\t54 32 18 17\n",
      "21\t30398\t0.0\t3\t28983 1051 239 125\n",
      "22\t29264\t0.0\t3\t27868 1039 219 138\n",
      "23\t26760\t0.0\t3\t25302 1067 262 129\n",
      "24\t27456\t0.0\t3\t25940 1130 248 138\n",
      "25\t24814\t0.0\t3\t23454 977 250 133\n",
      "26\t23641\t0.0\t3\t22373 932 245 91\n",
      "27\t28\t0.0\t3\t24 2 0 2\n",
      "36\t1\t0.0\t3\t0 0 1\n",
      "52\t1\t0.0\t3\t0 0 0 1\n",
      "54\t1\t0.0\t3\t0 0 0 1\n",
      "59\t1\t0.0\t3\t0 0 0 1\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "62\t2\t0.0\t3\t0 0 0 2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121683_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121683_3_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121683_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121683_3_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121683_2_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121683_3_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121683_2_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121683_3_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 2.32 s (16 µs/read; 3.87 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            149,412\n",
      "  Read 1 with adapter:                 147,947 (99.0%)\n",
      "  Read 2 with adapter:                 146,155 (97.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   9 (0.0%)\n",
      "Pairs discarded as untrimmed:            4,686 (3.1%)\n",
      "Pairs written (passing filters):       144,717 (96.9%)\n",
      "\n",
      "Total basepairs processed:    76,200,120 bp\n",
      "  Read 1:    45,570,660 bp\n",
      "  Read 2:    30,629,460 bp\n",
      "Total written (filtered):     67,462,124 bp (88.5%)\n",
      "  Read 1:    41,176,861 bp\n",
      "  Read 2:    26,285,263 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 105 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 27.6%\n",
      "  G: 71.4%\n",
      "  T: 1.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t8\t2.3\t1\t0 8\n",
      "9\t16\t0.6\t1\t0 16\n",
      "20\t5\t0.0\t3\t5\n",
      "21\t5\t0.0\t3\t5\n",
      "22\t4\t0.0\t3\t4\n",
      "23\t4\t0.0\t3\t4\n",
      "24\t3\t0.0\t3\t3\n",
      "25\t3\t0.0\t3\t3\n",
      "26\t1\t0.0\t3\t1\n",
      "29\t4\t0.0\t3\t4\n",
      "30\t11\t0.0\t3\t10 1\n",
      "31\t7\t0.0\t3\t7\n",
      "32\t12\t0.0\t3\t11 1\n",
      "33\t3\t0.0\t3\t3\n",
      "34\t6\t0.0\t3\t6\n",
      "153\t1\t0.0\t3\t1\n",
      "257\t1\t0.0\t3\t1\n",
      "258\t3\t0.0\t3\t3\n",
      "260\t2\t0.0\t3\t2\n",
      "261\t1\t0.0\t3\t1\n",
      "280\t1\t0.0\t3\t1\n",
      "281\t1\t0.0\t3\t1\n",
      "282\t2\t0.0\t3\t2\n",
      "284\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 147842 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t2.3\t1\t1 1\n",
      "10\t1\t0.1\t1\t1\n",
      "11\t4\t0.0\t1\t2 2\n",
      "12\t4\t0.0\t1\t2 2\n",
      "13\t5\t0.0\t1\t3 2\n",
      "14\t14\t0.0\t2\t8 5 1\n",
      "15\t7\t0.0\t2\t2 1 4\n",
      "16\t13\t0.0\t2\t10 2 1\n",
      "17\t75\t0.0\t2\t45 21 9\n",
      "18\t23193\t0.0\t2\t22745 428 20\n",
      "19\t26424\t0.0\t2\t25902 503 19\n",
      "20\t27036\t0.0\t2\t26577 441 18\n",
      "21\t24700\t0.0\t2\t24277 404 19\n",
      "22\t25051\t0.0\t2\t24615 412 24\n",
      "23\t21290\t0.0\t2\t20941 343 6\n",
      "24\t22\t0.0\t2\t17 3 2\n",
      "37\t1\t0.0\t2\t0 0 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 24 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 12.5%\n",
      "  C: 16.7%\n",
      "  G: 20.8%\n",
      "  T: 12.5%\n",
      "  none/other: 37.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t9\t0.0\t2\t7 2\n",
      "204\t6\t0.0\t2\t6\n",
      "205\t9\t0.0\t2\t9\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 146131 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t2\t0.0\t1\t2\n",
      "14\t2\t0.0\t2\t1 0 1\n",
      "15\t3\t0.0\t2\t3\n",
      "16\t2\t0.0\t2\t2\n",
      "17\t1\t0.0\t2\t1\n",
      "18\t5\t0.0\t2\t0 1 4\n",
      "19\t11\t0.0\t2\t5 5 1\n",
      "20\t116\t0.0\t3\t53 33 14 16\n",
      "21\t27337\t0.0\t3\t25986 1000 239 112\n",
      "22\t26355\t0.0\t3\t25058 941 213 143\n",
      "23\t24213\t0.0\t3\t22833 1030 227 123\n",
      "24\t24844\t0.0\t3\t23540 980 222 102\n",
      "25\t22091\t0.0\t3\t20833 932 228 98\n",
      "26\t21122\t0.0\t3\t19937 843 226 116\n",
      "27\t22\t0.0\t3\t16 2 2 2\n",
      "33\t2\t0.0\t3\t2\n",
      "40\t1\t0.0\t3\t1\n",
      "45\t1\t0.0\t3\t1\n",
      "61\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121684_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121684_5_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121684_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121684_5_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121684_4_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121684_5_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121684_4_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121684_5_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 5.02 s (15 µs/read; 4.10 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            343,261\n",
      "  Read 1 with adapter:                 340,079 (99.1%)\n",
      "  Read 2 with adapter:                 336,731 (98.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   1 (0.0%)\n",
      "Pairs discarded as untrimmed:            9,645 (2.8%)\n",
      "Pairs written (passing filters):       333,615 (97.2%)\n",
      "\n",
      "Total basepairs processed:   175,063,110 bp\n",
      "  Read 1:   104,694,605 bp\n",
      "  Read 2:    70,368,505 bp\n",
      "Total written (filtered):    155,517,885 bp (88.8%)\n",
      "  Read 1:    94,921,854 bp\n",
      "  Read 2:    60,596,031 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 162 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.6%\n",
      "  C: 56.2%\n",
      "  G: 22.8%\n",
      "  T: 20.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t38\t5.2\t1\t0 38\n",
      "9\t41\t1.3\t1\t0 41\n",
      "20\t3\t0.0\t3\t2 1\n",
      "21\t7\t0.0\t3\t7\n",
      "22\t4\t0.0\t3\t3 0 1\n",
      "23\t6\t0.0\t3\t5 1\n",
      "24\t4\t0.0\t3\t4\n",
      "25\t7\t0.0\t3\t7\n",
      "26\t1\t0.0\t3\t1\n",
      "29\t1\t0.0\t3\t1\n",
      "136\t1\t0.0\t3\t1\n",
      "218\t2\t0.0\t3\t0 0 2\n",
      "256\t1\t0.0\t3\t1\n",
      "277\t8\t0.0\t3\t7 1\n",
      "278\t5\t0.0\t3\t5\n",
      "279\t5\t0.0\t3\t5\n",
      "280\t8\t0.0\t3\t7 1\n",
      "281\t11\t0.0\t3\t11\n",
      "282\t6\t0.0\t3\t6\n",
      "283\t3\t0.0\t3\t2 1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 339917 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.2\t1\t1\n",
      "9\t6\t1.3\t1\t5 1\n",
      "10\t4\t0.3\t1\t2 2\n",
      "11\t6\t0.1\t1\t5 1\n",
      "12\t4\t0.0\t1\t2 2\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t17\t0.0\t2\t11 1 5\n",
      "15\t20\t0.0\t2\t9 6 5\n",
      "16\t34\t0.0\t2\t13 12 9\n",
      "17\t167\t0.0\t2\t90 51 26\n",
      "18\t53282\t0.0\t2\t52362 885 35\n",
      "19\t61544\t0.0\t2\t60504 999 41\n",
      "20\t62323\t0.0\t2\t61346 940 37\n",
      "21\t55889\t0.0\t2\t54937 902 50\n",
      "22\t57711\t0.0\t2\t56814 848 49\n",
      "23\t48851\t0.0\t2\t48142 690 19\n",
      "24\t52\t0.0\t2\t46 4 2\n",
      "33\t1\t0.0\t2\t1\n",
      "35\t1\t0.0\t2\t1\n",
      "40\t1\t0.0\t2\t1\n",
      "44\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 14 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 7.1%\n",
      "  C: 7.1%\n",
      "  G: 50.0%\n",
      "  T: 28.6%\n",
      "  none/other: 7.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t1.3\t1\t0 1\n",
      "10\t1\t0.3\t1\t0 1\n",
      "180\t1\t0.0\t2\t1\n",
      "203\t6\t0.0\t2\t3 2 1\n",
      "204\t4\t0.0\t2\t4\n",
      "205\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 336717 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t5.2\t1\t0 2\n",
      "10\t1\t0.3\t1\t0 1\n",
      "11\t1\t0.1\t1\t1\n",
      "12\t2\t0.0\t1\t2\n",
      "13\t2\t0.0\t1\t0 2\n",
      "14\t8\t0.0\t2\t2 5 1\n",
      "15\t20\t0.0\t2\t11 7 2\n",
      "16\t6\t0.0\t2\t1 3 2\n",
      "17\t12\t0.0\t2\t5 1 6\n",
      "18\t7\t0.0\t2\t5 0 2\n",
      "19\t22\t0.0\t2\t9 7 6\n",
      "20\t266\t0.0\t3\t93 92 49 32\n",
      "21\t62624\t0.0\t3\t59527 2272 574 251\n",
      "22\t60388\t0.0\t3\t57389 2182 559 258\n",
      "23\t55640\t0.0\t3\t52503 2291 570 276\n",
      "24\t57281\t0.0\t3\t54259 2205 530 287\n",
      "25\t50943\t0.0\t3\t47921 2110 603 309\n",
      "26\t49434\t0.0\t3\t46600 2071 525 238\n",
      "27\t51\t0.0\t3\t35 7 3 6\n",
      "35\t1\t0.0\t3\t1\n",
      "61\t4\t0.0\t3\t0 1 3\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "64\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121685_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121685_7_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121685_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121685_7_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121685_6_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121685_7_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121685_6_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121685_7_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.85 s (15 µs/read; 3.97 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            254,393\n",
      "  Read 1 with adapter:                 251,998 (99.1%)\n",
      "  Read 2 with adapter:                 248,860 (97.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  10 (0.0%)\n",
      "Pairs discarded as untrimmed:            7,871 (3.1%)\n",
      "Pairs written (passing filters):       246,512 (96.9%)\n",
      "\n",
      "Total basepairs processed:   129,740,430 bp\n",
      "  Read 1:    77,589,865 bp\n",
      "  Read 2:    52,150,565 bp\n",
      "Total written (filtered):    114,920,232 bp (88.6%)\n",
      "  Read 1:    70,145,471 bp\n",
      "  Read 2:    44,774,761 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 128 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 34.4%\n",
      "  G: 60.9%\n",
      "  T: 4.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t28\t3.9\t1\t0 28\n",
      "9\t18\t1.0\t1\t0 18\n",
      "10\t1\t0.2\t1\t0 1\n",
      "20\t9\t0.0\t3\t9\n",
      "21\t10\t0.0\t3\t8 2\n",
      "22\t14\t0.0\t3\t12 1 1\n",
      "23\t14\t0.0\t3\t13 1\n",
      "24\t10\t0.0\t3\t8 2\n",
      "25\t13\t0.0\t3\t12 0 1\n",
      "26\t2\t0.0\t3\t1 1\n",
      "30\t1\t0.0\t3\t1\n",
      "218\t1\t0.0\t3\t1\n",
      "234\t1\t0.0\t3\t1\n",
      "236\t1\t0.0\t3\t1\n",
      "237\t1\t0.0\t3\t0 1\n",
      "253\t1\t0.0\t3\t1\n",
      "279\t1\t0.0\t3\t0 1\n",
      "302\t2\t0.0\t3\t2\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 251870 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t3.9\t1\t3 1\n",
      "9\t5\t1.0\t1\t3 2\n",
      "10\t3\t0.2\t1\t2 1\n",
      "11\t4\t0.1\t1\t1 3\n",
      "12\t7\t0.0\t1\t6 1\n",
      "13\t7\t0.0\t1\t5 2\n",
      "14\t20\t0.0\t2\t13 2 5\n",
      "15\t11\t0.0\t2\t8 2 1\n",
      "16\t18\t0.0\t2\t5 7 6\n",
      "17\t152\t0.0\t2\t82 43 27\n",
      "18\t39230\t0.0\t2\t38482 722 26\n",
      "19\t45580\t0.0\t2\t44776 764 40\n",
      "20\t46235\t0.0\t2\t45407 794 34\n",
      "21\t41414\t0.0\t2\t40682 694 38\n",
      "22\t42544\t0.0\t2\t41795 719 30\n",
      "23\t36597\t0.0\t2\t36001 585 11\n",
      "24\t35\t0.0\t2\t29 2 4\n",
      "25\t2\t0.0\t2\t2\n",
      "47\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 25 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 8.0%\n",
      "  C: 8.0%\n",
      "  G: 36.0%\n",
      "  T: 8.0%\n",
      "  none/other: 40.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t4\t0.0\t2\t4\n",
      "204\t11\t0.0\t2\t11\n",
      "205\t10\t0.0\t2\t10\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 248835 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t3.9\t1\t0 1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "13\t3\t0.0\t1\t3\n",
      "14\t4\t0.0\t2\t1 1 2\n",
      "15\t12\t0.0\t2\t10 1 1\n",
      "16\t4\t0.0\t2\t1 3\n",
      "17\t6\t0.0\t2\t2 2 2\n",
      "18\t9\t0.0\t2\t7 1 1\n",
      "19\t12\t0.0\t2\t5 5 2\n",
      "20\t185\t0.0\t3\t74 55 37 19\n",
      "21\t46670\t0.0\t3\t44245 1778 427 220\n",
      "22\t44425\t0.0\t3\t42188 1648 399 190\n",
      "23\t41247\t0.0\t3\t38808 1831 381 227\n",
      "24\t41891\t0.0\t3\t39553 1702 433 203\n",
      "25\t37932\t0.0\t3\t35622 1661 436 213\n",
      "26\t36391\t0.0\t3\t34102 1664 425 200\n",
      "27\t34\t0.0\t3\t20 9 4 1\n",
      "33\t1\t0.0\t3\t1\n",
      "44\t1\t0.0\t3\t1\n",
      "57\t1\t0.0\t3\t0 0 0 1\n",
      "60\t1\t0.0\t3\t0 0 1\n",
      "61\t2\t0.0\t3\t0 0 0 2\n",
      "64\t1\t0.0\t3\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121686_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121686_9_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121686_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121686_9_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121686_8_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121686_9_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121686_8_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121686_9_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.67 s (16 µs/read; 3.74 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            103,905\n",
      "  Read 1 with adapter:                 102,946 (99.1%)\n",
      "  Read 2 with adapter:                 101,888 (98.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,954 (2.8%)\n",
      "Pairs written (passing filters):       100,951 (97.2%)\n",
      "\n",
      "Total basepairs processed:    52,991,550 bp\n",
      "  Read 1:    31,691,025 bp\n",
      "  Read 2:    21,300,525 bp\n",
      "Total written (filtered):     47,063,687 bp (88.8%)\n",
      "  Read 1:    28,727,135 bp\n",
      "  Read 2:    18,336,552 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 42 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 4.8%\n",
      "  C: 14.3%\n",
      "  G: 78.6%\n",
      "  T: 2.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t3\t1.6\t1\t0 3\n",
      "9\t4\t0.4\t1\t0 4\n",
      "21\t4\t0.0\t3\t3 1\n",
      "22\t1\t0.0\t3\t1\n",
      "23\t6\t0.0\t3\t6\n",
      "24\t2\t0.0\t3\t2\n",
      "25\t2\t0.0\t3\t2\n",
      "29\t2\t0.0\t3\t2\n",
      "30\t6\t0.0\t3\t6\n",
      "31\t5\t0.0\t3\t5\n",
      "32\t1\t0.0\t3\t1\n",
      "33\t1\t0.0\t3\t1\n",
      "34\t1\t0.0\t3\t1\n",
      "78\t1\t0.0\t3\t1\n",
      "85\t1\t0.0\t3\t1\n",
      "166\t1\t0.0\t3\t0 0 1\n",
      "234\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 102904 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.6\t1\t1\n",
      "9\t2\t0.4\t1\t2\n",
      "10\t1\t0.1\t1\t1\n",
      "11\t6\t0.0\t1\t5 1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "14\t9\t0.0\t2\t8 1\n",
      "15\t5\t0.0\t2\t4 1\n",
      "16\t5\t0.0\t2\t2 0 3\n",
      "17\t56\t0.0\t2\t32 18 6\n",
      "18\t16355\t0.0\t2\t16066 273 16\n",
      "19\t18584\t0.0\t2\t18252 317 15\n",
      "20\t19044\t0.0\t2\t18741 292 11\n",
      "21\t16875\t0.0\t2\t16611 250 14\n",
      "22\t17209\t0.0\t2\t16927 266 16\n",
      "23\t14733\t0.0\t2\t14532 197 4\n",
      "24\t14\t0.0\t2\t12 1 1\n",
      "40\t1\t0.0\t2\t1\n",
      "44\t1\t0.0\t2\t1\n",
      "51\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 3 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 100.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "204\t3\t0.0\t2\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 101885 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t6\t1.6\t1\t1 5\n",
      "11\t2\t0.0\t1\t1 1\n",
      "12\t2\t0.0\t1\t2\n",
      "13\t1\t0.0\t1\t1\n",
      "14\t2\t0.0\t2\t0 0 2\n",
      "15\t3\t0.0\t2\t1 2\n",
      "16\t3\t0.0\t2\t1 2\n",
      "17\t3\t0.0\t2\t2 1\n",
      "18\t4\t0.0\t2\t2 1 1\n",
      "19\t11\t0.0\t2\t6 4 1\n",
      "20\t93\t0.0\t3\t47 23 17 6\n",
      "21\t19021\t0.0\t3\t18052 727 151 91\n",
      "22\t18364\t0.0\t3\t17427 671 172 94\n",
      "23\t16633\t0.0\t3\t15720 660 173 80\n",
      "24\t17334\t0.0\t3\t16399 664 178 93\n",
      "25\t15483\t0.0\t3\t14599 640 167 77\n",
      "26\t14896\t0.0\t3\t14056 607 158 75\n",
      "27\t20\t0.0\t3\t15 4 0 1\n",
      "60\t1\t0.0\t3\t0 0 0 1\n",
      "61\t1\t0.0\t3\t0 1\n",
      "63\t2\t0.0\t3\t0 0 1 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121687_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121687_11_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121687_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121687_11_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121687_10_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121687_11_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121687_10_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121687_11_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.16 s (17 µs/read; 3.44 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             66,368\n",
      "  Read 1 with adapter:                  65,760 (99.1%)\n",
      "  Read 2 with adapter:                  64,834 (97.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   2 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,121 (3.2%)\n",
      "Pairs written (passing filters):        64,245 (96.8%)\n",
      "\n",
      "Total basepairs processed:    33,847,680 bp\n",
      "  Read 1:    20,242,240 bp\n",
      "  Read 2:    13,605,440 bp\n",
      "Total written (filtered):     29,949,537 bp (88.5%)\n",
      "  Read 1:    18,280,890 bp\n",
      "  Read 2:    11,668,647 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 85 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 63.5%\n",
      "  G: 36.5%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t25\t1.0\t1\t1 24\n",
      "9\t29\t0.3\t1\t0 29\n",
      "21\t2\t0.0\t3\t2\n",
      "22\t4\t0.0\t3\t3 1\n",
      "23\t4\t0.0\t3\t3 1\n",
      "24\t3\t0.0\t3\t3\n",
      "25\t5\t0.0\t3\t5\n",
      "29\t2\t0.0\t3\t2\n",
      "31\t1\t0.0\t3\t1\n",
      "32\t2\t0.0\t3\t2\n",
      "33\t2\t0.0\t3\t2\n",
      "34\t4\t0.0\t3\t3 1\n",
      "278\t1\t0.0\t3\t1\n",
      "284\t1\t0.0\t3\t0 1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 65675 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.1\t1\t1\n",
      "11\t4\t0.0\t1\t2 2\n",
      "13\t3\t0.0\t1\t2 1\n",
      "14\t5\t0.0\t2\t3 0 2\n",
      "15\t3\t0.0\t2\t2 0 1\n",
      "16\t3\t0.0\t2\t1 1 1\n",
      "17\t36\t0.0\t2\t22 8 6\n",
      "18\t10073\t0.0\t2\t9860 201 12\n",
      "19\t11875\t0.0\t2\t11657 206 12\n",
      "20\t12063\t0.0\t2\t11869 186 8\n",
      "21\t10903\t0.0\t2\t10700 196 7\n",
      "22\t11233\t0.0\t2\t11028 191 14\n",
      "23\t9462\t0.0\t2\t9334 123 5\n",
      "24\t10\t0.0\t2\t9 1\n",
      "26\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 9 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 33.3%\n",
      "  G: 33.3%\n",
      "  T: 11.1%\n",
      "  none/other: 22.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.3\t1\t0 1\n",
      "203\t3\t0.0\t2\t3\n",
      "204\t3\t0.0\t2\t3\n",
      "205\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 64825 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.0\t1\t0 1\n",
      "12\t1\t0.0\t1\t1\n",
      "14\t1\t0.0\t2\t0 1\n",
      "15\t5\t0.0\t2\t4 1\n",
      "16\t1\t0.0\t2\t1\n",
      "17\t4\t0.0\t2\t1 2 1\n",
      "19\t5\t0.0\t2\t3 0 2\n",
      "20\t36\t0.0\t3\t16 12 3 5\n",
      "21\t12261\t0.0\t3\t11652 454 107 48\n",
      "22\t11400\t0.0\t3\t10779 446 117 58\n",
      "23\t10858\t0.0\t3\t10214 496 100 48\n",
      "24\t10877\t0.0\t3\t10220 473 117 67\n",
      "25\t9802\t0.0\t3\t9211 432 102 57\n",
      "26\t9560\t0.0\t3\t8923 475 116 46\n",
      "27\t11\t0.0\t3\t7 2 1 1\n",
      "28\t1\t0.0\t3\t0 0 1\n",
      "53\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121694_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121694_13_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121694_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121694_13_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121694_12_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121694_13_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121694_12_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121694_13_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.08 s (18 µs/read; 3.29 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             58,992\n",
      "  Read 1 with adapter:                  58,444 (99.1%)\n",
      "  Read 2 with adapter:                  57,446 (97.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  12 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,078 (3.5%)\n",
      "Pairs written (passing filters):        56,902 (96.5%)\n",
      "\n",
      "Total basepairs processed:    30,085,920 bp\n",
      "  Read 1:    17,992,560 bp\n",
      "  Read 2:    12,093,360 bp\n",
      "Total written (filtered):     26,521,205 bp (88.2%)\n",
      "  Read 1:    16,187,827 bp\n",
      "  Read 2:    10,333,378 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 88 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 2.3%\n",
      "  C: 35.2%\n",
      "  G: 59.1%\n",
      "  T: 3.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t6\t0.9\t1\t0 6\n",
      "9\t11\t0.2\t1\t0 11\n",
      "20\t1\t0.0\t3\t1\n",
      "21\t1\t0.0\t3\t1\n",
      "22\t1\t0.0\t3\t1\n",
      "24\t1\t0.0\t3\t1\n",
      "29\t4\t0.0\t3\t4\n",
      "30\t9\t0.0\t3\t8 1\n",
      "31\t8\t0.0\t3\t8\n",
      "32\t7\t0.0\t3\t6 1\n",
      "33\t13\t0.0\t3\t12 0 1\n",
      "34\t7\t0.0\t3\t7\n",
      "110\t1\t0.0\t3\t1\n",
      "246\t1\t0.0\t3\t1\n",
      "252\t4\t0.0\t3\t4\n",
      "253\t2\t0.0\t3\t2\n",
      "254\t2\t0.0\t3\t2\n",
      "255\t2\t0.0\t3\t2\n",
      "256\t5\t0.0\t3\t5\n",
      "257\t1\t0.0\t3\t1\n",
      "283\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 58356 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.2\t1\t1\n",
      "10\t1\t0.1\t1\t0 1\n",
      "11\t1\t0.0\t1\t1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t8\t0.0\t2\t6 1 1\n",
      "16\t4\t0.0\t2\t3 1\n",
      "17\t23\t0.0\t2\t10 9 4\n",
      "18\t9079\t0.0\t2\t8912 158 9\n",
      "19\t10538\t0.0\t2\t10322 207 9\n",
      "20\t10856\t0.0\t2\t10657 191 8\n",
      "21\t9542\t0.0\t2\t9365 166 11\n",
      "22\t9915\t0.0\t2\t9746 162 7\n",
      "23\t8378\t0.0\t2\t8239 136 3\n",
      "24\t6\t0.0\t2\t6\n",
      "40\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 28 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.6%\n",
      "  C: 14.3%\n",
      "  G: 17.9%\n",
      "  T: 21.4%\n",
      "  none/other: 42.9%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.9\t1\t0 1\n",
      "11\t1\t0.0\t1\t0 1\n",
      "150\t1\t0.0\t2\t0 1\n",
      "203\t4\t0.0\t2\t4\n",
      "204\t9\t0.0\t2\t9\n",
      "205\t12\t0.0\t2\t12\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 57418 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t1\t1\n",
      "14\t1\t0.0\t2\t1\n",
      "15\t2\t0.0\t2\t2\n",
      "16\t1\t0.0\t2\t1\n",
      "17\t2\t0.0\t2\t1 0 1\n",
      "18\t3\t0.0\t2\t2 1\n",
      "19\t5\t0.0\t2\t0 2 3\n",
      "20\t38\t0.0\t3\t18 6 7 7\n",
      "21\t10758\t0.0\t3\t10166 438 100 54\n",
      "22\t10350\t0.0\t3\t9802 401 99 48\n",
      "23\t9350\t0.0\t3\t8793 400 114 43\n",
      "24\t9766\t0.0\t3\t9258 373 80 55\n",
      "25\t8706\t0.0\t3\t8219 343 90 54\n",
      "26\t8422\t0.0\t3\t7938 358 82 44\n",
      "27\t10\t0.0\t3\t6 2 0 2\n",
      "58\t1\t0.0\t3\t0 0 0 1\n",
      "61\t2\t0.0\t3\t0 0 0 2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121695_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121695_15_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121695_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121695_15_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121695_14_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121695_15_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121695_14_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121695_15_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 3.62 s (16 µs/read; 3.81 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            229,809\n",
      "  Read 1 with adapter:                 227,628 (99.1%)\n",
      "  Read 2 with adapter:                 224,970 (97.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  10 (0.0%)\n",
      "Pairs discarded as untrimmed:            6,969 (3.0%)\n",
      "Pairs written (passing filters):       222,830 (97.0%)\n",
      "\n",
      "Total basepairs processed:   117,202,590 bp\n",
      "  Read 1:    70,091,745 bp\n",
      "  Read 2:    47,110,845 bp\n",
      "Total written (filtered):    103,871,094 bp (88.6%)\n",
      "  Read 1:    63,399,872 bp\n",
      "  Read 2:    40,471,222 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 372 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.2%\n",
      "  C: 12.1%\n",
      "  G: 82.0%\n",
      "  T: 2.7%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by 'G' extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 5' end.\n",
      "    Ignore this warning when trimming primers.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t21\t3.5\t1\t0 21\n",
      "9\t23\t0.9\t1\t0 23\n",
      "10\t1\t0.2\t1\t0 1\n",
      "20\t2\t0.0\t3\t2\n",
      "21\t3\t0.0\t3\t3\n",
      "22\t4\t0.0\t3\t4\n",
      "23\t6\t0.0\t3\t6\n",
      "24\t3\t0.0\t3\t3\n",
      "25\t2\t0.0\t3\t2\n",
      "29\t38\t0.0\t3\t35 3\n",
      "30\t52\t0.0\t3\t49 1 2\n",
      "31\t52\t0.0\t3\t51 1\n",
      "32\t52\t0.0\t3\t51 1\n",
      "33\t43\t0.0\t3\t40 3\n",
      "34\t42\t0.0\t3\t42\n",
      "129\t2\t0.0\t3\t2\n",
      "187\t1\t0.0\t3\t1\n",
      "192\t1\t0.0\t3\t1\n",
      "193\t1\t0.0\t3\t1\n",
      "195\t3\t0.0\t3\t2 1\n",
      "196\t1\t0.0\t3\t1\n",
      "197\t2\t0.0\t3\t2\n",
      "209\t1\t0.0\t3\t1\n",
      "215\t1\t0.0\t3\t1\n",
      "217\t2\t0.0\t3\t1 1\n",
      "218\t1\t0.0\t3\t1\n",
      "219\t3\t0.0\t3\t2 1\n",
      "239\t1\t0.0\t3\t1\n",
      "243\t2\t0.0\t3\t1 1\n",
      "247\t1\t0.0\t3\t1\n",
      "248\t2\t0.0\t3\t2\n",
      "249\t1\t0.0\t3\t1\n",
      "277\t1\t0.0\t3\t1\n",
      "281\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 227256 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t2\t3.5\t1\t2\n",
      "9\t4\t0.9\t1\t4\n",
      "10\t4\t0.2\t1\t3 1\n",
      "11\t6\t0.1\t1\t6\n",
      "12\t8\t0.0\t1\t6 2\n",
      "13\t3\t0.0\t1\t2 1\n",
      "14\t19\t0.0\t2\t14 3 2\n",
      "15\t17\t0.0\t2\t8 5 4\n",
      "16\t10\t0.0\t2\t6 1 3\n",
      "17\t119\t0.0\t2\t64 43 12\n",
      "18\t35246\t0.0\t2\t34628 590 28\n",
      "19\t40769\t0.0\t2\t39998 740 31\n",
      "20\t41965\t0.0\t2\t41231 693 41\n",
      "21\t37951\t0.0\t2\t37242 666 43\n",
      "22\t38582\t0.0\t2\t37947 606 29\n",
      "23\t32515\t0.0\t2\t32044 462 9\n",
      "24\t31\t0.0\t2\t27 1 3\n",
      "26\t1\t0.0\t2\t1\n",
      "27\t1\t0.0\t2\t0 1\n",
      "31\t1\t0.0\t2\t1\n",
      "34\t1\t0.0\t2\t1\n",
      "35\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 26 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 11.5%\n",
      "  C: 7.7%\n",
      "  G: 19.2%\n",
      "  T: 23.1%\n",
      "  none/other: 38.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.9\t1\t0 1\n",
      "10\t1\t0.2\t1\t0 1\n",
      "86\t1\t0.0\t2\t1\n",
      "203\t6\t0.0\t2\t6\n",
      "204\t7\t0.0\t2\t7\n",
      "205\t10\t0.0\t2\t9 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 224944 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.1\t1\t0 1\n",
      "12\t1\t0.0\t1\t0 1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t2\t0.0\t2\t0 1 1\n",
      "15\t5\t0.0\t2\t5\n",
      "16\t7\t0.0\t2\t1 4 2\n",
      "17\t6\t0.0\t2\t4 2\n",
      "18\t4\t0.0\t2\t0 3 1\n",
      "19\t14\t0.0\t2\t5 6 3\n",
      "20\t160\t0.0\t3\t63 42 25 30\n",
      "21\t41811\t0.0\t3\t39586 1635 404 186\n",
      "22\t40375\t0.0\t3\t38305 1501 372 197\n",
      "23\t36809\t0.0\t3\t34643 1598 382 186\n",
      "24\t38209\t0.0\t3\t36017 1623 373 196\n",
      "25\t34398\t0.0\t3\t32350 1484 373 191\n",
      "26\t33094\t0.0\t3\t31075 1494 350 175\n",
      "27\t33\t0.0\t3\t26 5 1 1\n",
      "54\t1\t0.0\t3\t0 0 0 1\n",
      "58\t1\t0.0\t3\t0 0 0 1\n",
      "59\t2\t0.0\t3\t0 0 1 1\n",
      "60\t2\t0.0\t3\t0 0 0 2\n",
      "61\t4\t0.0\t3\t0 1 1 2\n",
      "63\t1\t0.0\t3\t0 0 1\n",
      "64\t1\t0.0\t3\t0 0 0 1\n",
      "65\t1\t0.0\t3\t0 1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121696_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121696_17_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121696_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121696_17_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121696_16_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121696_17_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121696_16_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121696_17_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 1.84 s (16 µs/read; 3.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            114,634\n",
      "  Read 1 with adapter:                 113,440 (99.0%)\n",
      "  Read 2 with adapter:                 111,771 (97.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  14 (0.0%)\n",
      "Pairs discarded as untrimmed:            4,005 (3.5%)\n",
      "Pairs written (passing filters):       110,615 (96.5%)\n",
      "\n",
      "Total basepairs processed:    58,463,340 bp\n",
      "  Read 1:    34,963,370 bp\n",
      "  Read 2:    23,499,970 bp\n",
      "Total written (filtered):     51,562,559 bp (88.2%)\n",
      "  Read 1:    31,472,632 bp\n",
      "  Read 2:    20,089,927 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 47 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 34.0%\n",
      "  G: 63.8%\n",
      "  T: 2.1%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t7\t1.7\t1\t0 7\n",
      "21\t1\t0.0\t3\t1\n",
      "22\t4\t0.0\t3\t3 0 1\n",
      "23\t2\t0.0\t3\t2\n",
      "24\t6\t0.0\t3\t6\n",
      "25\t2\t0.0\t3\t2\n",
      "26\t4\t0.0\t3\t4\n",
      "30\t2\t0.0\t3\t2\n",
      "31\t2\t0.0\t3\t2\n",
      "32\t3\t0.0\t3\t3\n",
      "34\t3\t0.0\t3\t3\n",
      "244\t1\t0.0\t3\t1\n",
      "279\t1\t0.0\t3\t1\n",
      "280\t2\t0.0\t3\t2\n",
      "281\t2\t0.0\t3\t2\n",
      "282\t2\t0.0\t3\t2\n",
      "283\t2\t0.0\t3\t2\n",
      "284\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 113393 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.7\t1\t1\n",
      "9\t1\t0.4\t1\t0 1\n",
      "10\t4\t0.1\t1\t4\n",
      "11\t6\t0.0\t1\t5 1\n",
      "12\t2\t0.0\t1\t1 1\n",
      "13\t2\t0.0\t1\t1 1\n",
      "14\t6\t0.0\t2\t5 0 1\n",
      "15\t10\t0.0\t2\t5 1 4\n",
      "16\t5\t0.0\t2\t3 1 1\n",
      "17\t60\t0.0\t2\t33 19 8\n",
      "18\t17704\t0.0\t2\t17399 295 10\n",
      "19\t20264\t0.0\t2\t19910 336 18\n",
      "20\t20686\t0.0\t2\t20306 361 19\n",
      "21\t18830\t0.0\t2\t18485 323 22\n",
      "22\t19209\t0.0\t2\t18874 322 13\n",
      "23\t16580\t0.0\t2\t16322 254 4\n",
      "24\t22\t0.0\t2\t19 1 2\n",
      "28\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 25 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 4.0%\n",
      "  C: 16.0%\n",
      "  G: 0.0%\n",
      "  T: 24.0%\n",
      "  none/other: 56.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t1.7\t1\t0 1\n",
      "203\t6\t0.0\t2\t6\n",
      "204\t4\t0.0\t2\t4\n",
      "205\t14\t0.0\t2\t14\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 111746 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t4\t1.7\t1\t1 3\n",
      "13\t1\t0.0\t1\t1\n",
      "15\t4\t0.0\t2\t2 2\n",
      "16\t3\t0.0\t2\t2 0 1\n",
      "17\t4\t0.0\t2\t1 2 1\n",
      "18\t6\t0.0\t2\t2 2 2\n",
      "19\t8\t0.0\t2\t3 2 3\n",
      "20\t93\t0.0\t3\t38 28 17 10\n",
      "21\t20831\t0.0\t3\t19666 863 188 114\n",
      "22\t19961\t0.0\t3\t18723 875 244 119\n",
      "23\t18302\t0.0\t3\t17184 798 210 110\n",
      "24\t19164\t0.0\t3\t18017 823 198 126\n",
      "25\t16910\t0.0\t3\t15803 792 202 113\n",
      "26\t16422\t0.0\t3\t15339 774 200 109\n",
      "27\t22\t0.0\t3\t17 3 1 1\n",
      "49\t1\t0.0\t3\t1\n",
      "55\t1\t0.0\t3\t0 0 0 1\n",
      "59\t1\t0.0\t3\t0 0 1\n",
      "60\t2\t0.0\t3\t0 2\n",
      "61\t3\t0.0\t3\t0 0 1 2\n",
      "62\t1\t0.0\t3\t0 0 0 1\n",
      "63\t1\t0.0\t3\t0 0 0 1\n",
      "64\t1\t0.0\t3\t0 0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121697_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121697_19_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121697_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121697_19_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121697_18_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121697_19_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121697_18_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121697_19_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.92 s (19 µs/read; 3.20 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             49,335\n",
      "  Read 1 with adapter:                  48,895 (99.1%)\n",
      "  Read 2 with adapter:                  48,247 (97.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   5 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,514 (3.1%)\n",
      "Pairs written (passing filters):        47,816 (96.9%)\n",
      "\n",
      "Total basepairs processed:    25,160,850 bp\n",
      "  Read 1:    15,047,175 bp\n",
      "  Read 2:    10,113,675 bp\n",
      "Total written (filtered):     22,289,755 bp (88.6%)\n",
      "  Read 1:    13,605,875 bp\n",
      "  Read 2:     8,683,880 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 31 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.2%\n",
      "  C: 74.2%\n",
      "  G: 19.4%\n",
      "  T: 3.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t14\t0.8\t1\t0 14\n",
      "9\t10\t0.2\t1\t0 10\n",
      "20\t4\t0.0\t3\t3 0 1\n",
      "21\t1\t0.0\t3\t1\n",
      "25\t1\t0.0\t3\t1\n",
      "262\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 48864 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t1\t0.2\t1\t1\n",
      "10\t1\t0.0\t1\t0 1\n",
      "11\t2\t0.0\t1\t2\n",
      "12\t1\t0.0\t1\t1\n",
      "14\t2\t0.0\t2\t1 0 1\n",
      "15\t5\t0.0\t2\t2 1 2\n",
      "16\t7\t0.0\t2\t3 3 1\n",
      "17\t31\t0.0\t2\t16 10 5\n",
      "18\t7412\t0.0\t2\t7268 138 6\n",
      "19\t8842\t0.0\t2\t8668 166 8\n",
      "20\t9116\t0.0\t2\t8944 161 11\n",
      "21\t7915\t0.0\t2\t7771 135 9\n",
      "22\t8401\t0.0\t2\t8228 165 8\n",
      "23\t7119\t0.0\t2\t7011 108\n",
      "24\t9\t0.0\t2\t7 2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 11 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 9.1%\n",
      "  G: 18.2%\n",
      "  T: 27.3%\n",
      "  none/other: 45.5%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t5\t0.0\t2\t4 1\n",
      "204\t1\t0.0\t2\t1\n",
      "205\t5\t0.0\t2\t4 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 48236 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t1\t1\n",
      "15\t2\t0.0\t2\t1 0 1\n",
      "16\t1\t0.0\t2\t1\n",
      "17\t1\t0.0\t2\t1\n",
      "18\t2\t0.0\t2\t1 1\n",
      "19\t1\t0.0\t2\t0 1\n",
      "20\t34\t0.0\t3\t14 11 6 3\n",
      "21\t8963\t0.0\t3\t8529 330 67 37\n",
      "22\t8658\t0.0\t3\t8218 313 77 50\n",
      "23\t7982\t0.0\t3\t7506 350 89 37\n",
      "24\t8054\t0.0\t3\t7601 339 67 47\n",
      "25\t7425\t0.0\t3\t6998 304 89 34\n",
      "26\t7102\t0.0\t3\t6678 304 77 43\n",
      "27\t3\t0.0\t3\t3\n",
      "32\t1\t0.0\t3\t1\n",
      "57\t1\t0.0\t3\t0 0 0 1\n",
      "58\t1\t0.0\t3\t0 0 0 1\n",
      "63\t2\t0.0\t3\t0 0 2\n",
      "64\t2\t0.0\t3\t0 0 0 2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121703_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121703_21_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121703_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121703_21_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121703_20_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121703_21_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121703_20_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121703_21_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.62 s (25 µs/read; 2.36 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             24,189\n",
      "  Read 1 with adapter:                  23,955 (99.0%)\n",
      "  Read 2 with adapter:                  23,066 (95.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  14 (0.1%)\n",
      "Pairs discarded as untrimmed:            1,339 (5.5%)\n",
      "Pairs written (passing filters):        22,836 (94.4%)\n",
      "\n",
      "Total basepairs processed:    12,336,390 bp\n",
      "  Read 1:     7,377,645 bp\n",
      "  Read 2:     4,958,745 bp\n",
      "Total written (filtered):     10,633,459 bp (86.2%)\n",
      "  Read 1:     6,488,861 bp\n",
      "  Read 2:     4,144,598 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 56 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 44.6%\n",
      "  G: 51.8%\n",
      "  T: 3.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t0.4\t1\t0 5\n",
      "9\t4\t0.1\t1\t0 4\n",
      "21\t1\t0.0\t3\t1\n",
      "23\t2\t0.0\t3\t2\n",
      "24\t3\t0.0\t3\t3\n",
      "29\t1\t0.0\t3\t1\n",
      "32\t2\t0.0\t3\t2\n",
      "33\t1\t0.0\t3\t1\n",
      "245\t2\t0.0\t3\t2\n",
      "246\t3\t0.0\t3\t1 2\n",
      "247\t3\t0.0\t3\t3\n",
      "248\t5\t0.0\t3\t5\n",
      "249\t3\t0.0\t3\t2 0 1\n",
      "250\t1\t0.0\t3\t1\n",
      "282\t4\t0.0\t3\t4\n",
      "283\t2\t0.0\t3\t2\n",
      "284\t4\t0.0\t3\t4\n",
      "285\t5\t0.0\t3\t5\n",
      "286\t2\t0.0\t3\t2\n",
      "287\t3\t0.0\t3\t3\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 23899 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.0\t1\t1\n",
      "15\t1\t0.0\t2\t0 1\n",
      "16\t2\t0.0\t2\t1 1\n",
      "17\t7\t0.0\t2\t4 1 2\n",
      "18\t3680\t0.0\t2\t3610 63 7\n",
      "19\t4202\t0.0\t2\t4120 78 4\n",
      "20\t4407\t0.0\t2\t4323 81 3\n",
      "21\t3980\t0.0\t2\t3912 61 7\n",
      "22\t4140\t0.0\t2\t4052 85 3\n",
      "23\t3476\t0.0\t2\t3432 44\n",
      "24\t2\t0.0\t2\t1 0 1\n",
      "28\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 33 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.0%\n",
      "  C: 9.1%\n",
      "  G: 27.3%\n",
      "  T: 18.2%\n",
      "  none/other: 42.4%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t5\t0.0\t2\t4 1\n",
      "204\t14\t0.0\t2\t14\n",
      "205\t14\t0.0\t2\t13 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 23033 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "12\t1\t0.0\t1\t1\n",
      "13\t1\t0.0\t1\t0 1\n",
      "17\t1\t0.0\t2\t0 0 1\n",
      "18\t1\t0.0\t2\t1\n",
      "19\t1\t0.0\t2\t1\n",
      "20\t20\t0.0\t3\t10 4 3 3\n",
      "21\t4309\t0.0\t3\t4101 147 39 22\n",
      "22\t4174\t0.0\t3\t3941 164 44 25\n",
      "23\t3791\t0.0\t3\t3556 170 43 22\n",
      "24\t3843\t0.0\t3\t3609 158 47 29\n",
      "25\t3575\t0.0\t3\t3363 147 40 25\n",
      "26\t3304\t0.0\t3\t3097 150 33 24\n",
      "27\t8\t0.0\t3\t4 3 1\n",
      "43\t1\t0.0\t3\t1\n",
      "61\t2\t0.0\t3\t0 0 0 2\n",
      "62\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121704_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121704_23_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121704_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121704_23_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121704_22_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121704_23_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121704_22_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121704_23_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.37 s (67 µs/read; 0.89 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              5,549\n",
      "  Read 1 with adapter:                   5,499 (99.1%)\n",
      "  Read 2 with adapter:                   5,356 (96.5%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              240 (4.3%)\n",
      "Pairs written (passing filters):         5,309 (95.7%)\n",
      "\n",
      "Total basepairs processed:     2,829,990 bp\n",
      "  Read 1:     1,692,445 bp\n",
      "  Read 2:     1,137,545 bp\n",
      "Total written (filtered):      2,474,084 bp (87.4%)\n",
      "  Read 1:     1,510,232 bp\n",
      "  Read 2:       963,852 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 4 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 100.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t0.1\t1\t0 1\n",
      "9\t1\t0.0\t1\t0 1\n",
      "279\t1\t0.0\t3\t1\n",
      "283\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 5495 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "17\t4\t0.0\t2\t3 1\n",
      "18\t880\t0.0\t2\t858 22\n",
      "19\t985\t0.0\t2\t966 19\n",
      "20\t1025\t0.0\t2\t1009 13 3\n",
      "21\t879\t0.0\t2\t860 19\n",
      "22\t914\t0.0\t2\t903 10 1\n",
      "23\t808\t0.0\t2\t796 10 2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 3 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 66.7%\n",
      "  G: 0.0%\n",
      "  T: 33.3%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "203\t3\t0.0\t2\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 5353 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t2\t1\n",
      "18\t2\t0.0\t2\t1 0 1\n",
      "20\t4\t0.0\t3\t2 1 1\n",
      "21\t1027\t0.0\t3\t964 42 13 8\n",
      "22\t925\t0.0\t3\t871 35 10 9\n",
      "23\t888\t0.0\t3\t837 37 8 6\n",
      "24\t903\t0.0\t3\t857 33 5 8\n",
      "25\t850\t0.0\t3\t780 36 21 13\n",
      "26\t752\t0.0\t3\t702 28 15 7\n",
      "62\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121705_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121705_25_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121705_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121705_25_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121705_24_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121705_25_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121705_24_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121705_25_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 5.16 s (15 µs/read; 4.00 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            344,337\n",
      "  Read 1 with adapter:                 340,922 (99.0%)\n",
      "  Read 2 with adapter:                 337,158 (97.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  12 (0.0%)\n",
      "Pairs discarded as untrimmed:           10,517 (3.1%)\n",
      "Pairs written (passing filters):       333,808 (96.9%)\n",
      "\n",
      "Total basepairs processed:   175,611,870 bp\n",
      "  Read 1:   105,022,785 bp\n",
      "  Read 2:    70,589,085 bp\n",
      "Total written (filtered):    155,616,970 bp (88.6%)\n",
      "  Read 1:    94,986,971 bp\n",
      "  Read 2:    60,629,999 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 43 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 32.6%\n",
      "  G: 67.4%\n",
      "  T: 0.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t7\t5.3\t1\t0 7\n",
      "9\t7\t1.3\t1\t0 7\n",
      "20\t2\t0.0\t3\t2\n",
      "21\t7\t0.0\t3\t6 1\n",
      "22\t7\t0.0\t3\t6 0 1\n",
      "23\t1\t0.0\t3\t1\n",
      "24\t2\t0.0\t3\t2\n",
      "25\t4\t0.0\t3\t4\n",
      "26\t1\t0.0\t3\t1\n",
      "166\t1\t0.0\t3\t1\n",
      "169\t2\t0.0\t3\t2\n",
      "236\t1\t0.0\t3\t1\n",
      "237\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 340879 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.3\t1\t1\n",
      "9\t4\t1.3\t1\t4\n",
      "10\t9\t0.3\t1\t8 1\n",
      "11\t10\t0.1\t1\t7 3\n",
      "12\t6\t0.0\t1\t6\n",
      "13\t10\t0.0\t1\t6 4\n",
      "14\t30\t0.0\t2\t19 7 4\n",
      "15\t16\t0.0\t2\t7 4 5\n",
      "16\t24\t0.0\t2\t15 6 3\n",
      "17\t199\t0.0\t2\t102 57 40\n",
      "18\t53126\t0.0\t2\t52172 916 38\n",
      "19\t61651\t0.0\t2\t60490 1088 73\n",
      "20\t62544\t0.0\t2\t61451 1030 63\n",
      "21\t56263\t0.0\t2\t55262 938 63\n",
      "22\t57693\t0.0\t2\t56689 936 68\n",
      "23\t49232\t0.0\t2\t48452 763 17\n",
      "24\t57\t0.0\t2\t43 5 9\n",
      "26\t1\t0.0\t2\t1\n",
      "34\t1\t0.0\t2\t0 1\n",
      "45\t1\t0.0\t2\t1\n",
      "63\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 26 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 3.8%\n",
      "  C: 15.4%\n",
      "  G: 23.1%\n",
      "  T: 11.5%\n",
      "  none/other: 46.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t5.3\t1\t0 1\n",
      "203\t7\t0.0\t2\t7\n",
      "204\t6\t0.0\t2\t6\n",
      "205\t12\t0.0\t2\t12\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 337132 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.1\t1\t1 1\n",
      "13\t1\t0.0\t1\t0 1\n",
      "14\t6\t0.0\t2\t4 2\n",
      "15\t14\t0.0\t2\t10 3 1\n",
      "16\t2\t0.0\t2\t1 0 1\n",
      "17\t7\t0.0\t2\t3 2 2\n",
      "18\t9\t0.0\t2\t4 1 4\n",
      "19\t20\t0.0\t2\t10 4 6\n",
      "20\t252\t0.0\t3\t99 68 45 40\n",
      "21\t62962\t0.0\t3\t59575 2455 647 285\n",
      "22\t60384\t0.0\t3\t57160 2384 549 291\n",
      "23\t55799\t0.0\t3\t52497 2376 601 325\n",
      "24\t56809\t0.0\t3\t53427 2402 666 314\n",
      "25\t51187\t0.0\t3\t48103 2189 587 308\n",
      "26\t49591\t0.0\t3\t46434 2273 594 290\n",
      "27\t52\t0.0\t3\t40 8 4\n",
      "28\t2\t0.0\t3\t2\n",
      "32\t1\t0.0\t3\t0 0 1\n",
      "33\t1\t0.0\t3\t0 0 1\n",
      "35\t1\t0.0\t3\t0 0 1\n",
      "36\t1\t0.0\t3\t1\n",
      "52\t1\t0.0\t3\t0 0 0 1\n",
      "56\t1\t0.0\t3\t0 0 1\n",
      "57\t2\t0.0\t3\t0 0 0 2\n",
      "58\t1\t0.0\t3\t0 0 0 1\n",
      "59\t2\t0.0\t3\t0 0 2\n",
      "60\t1\t0.0\t3\t0 0 0 1\n",
      "61\t5\t0.0\t3\t1 0 0 4\n",
      "62\t4\t0.0\t3\t0 0 3 1\n",
      "63\t4\t0.0\t3\t0 1 2 1\n",
      "64\t3\t0.0\t3\t0 0 0 3\n",
      "65\t2\t0.0\t3\t0 0 1 1\n",
      "66\t2\t0.0\t3\t0 0 0 2\n",
      "67\t1\t0.0\t3\t0 0 1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121706_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121706_27_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121706_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121706_27_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 3.5 with Python 3.8.12\n",
      "Command line parameters: --cores 6 --error-rate 0.15 --times 1 --overlap 8 --minimum-length 1 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121706_26_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-ivu_boz8/210421121706_27_L001_R2_001.fastq.gz --adapter ATTAGAWACCCBDGTAGTCC --front CCTACGGGRSGCAGCAG -A CTGCTGCSYCCCGTAGG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121706_26_L001_R1_001.fastq.gz /tmp/qiime2-archive-slftvfbd/b6afce83-cc33-4219-ac0a-c83a609026ce/data/210421121706_27_L001_R2_001.fastq.gz\n",
      "Processing reads on 6 cores in paired-end mode ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 6.15 s (15 µs/read; 4.06 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:            416,322\n",
      "  Read 1 with adapter:                 412,332 (99.0%)\n",
      "  Read 2 with adapter:                 407,730 (97.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                  21 (0.0%)\n",
      "Pairs discarded as untrimmed:           12,483 (3.0%)\n",
      "Pairs written (passing filters):       403,818 (97.0%)\n",
      "\n",
      "Total basepairs processed:   212,324,220 bp\n",
      "  Read 1:   126,978,210 bp\n",
      "  Read 2:    85,346,010 bp\n",
      "Total written (filtered):    188,257,614 bp (88.7%)\n",
      "  Read 1:   114,910,144 bp\n",
      "  Read 2:    73,347,470 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: ATTAGAWACCCBDGTAGTCC; Type: regular 3'; Length: 20; Trimmed: 75 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 40.0%\n",
      "  G: 54.7%\n",
      "  T: 5.3%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t19\t6.4\t1\t0 19\n",
      "9\t15\t1.6\t1\t0 15\n",
      "18\t1\t0.0\t2\t1\n",
      "20\t3\t0.0\t3\t3\n",
      "21\t7\t0.0\t3\t5 1 1\n",
      "22\t5\t0.0\t3\t5\n",
      "23\t7\t0.0\t3\t6 1\n",
      "24\t9\t0.0\t3\t7 2\n",
      "25\t3\t0.0\t3\t3\n",
      "26\t4\t0.0\t3\t2 0 2\n",
      "84\t1\t0.0\t3\t1\n",
      "282\t1\t0.0\t3\t1\n",
      "\n",
      "\n",
      "=== First read: Adapter 2 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 412257 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t5\t6.4\t1\t4 1\n",
      "9\t3\t1.6\t1\t2 1\n",
      "10\t5\t0.4\t1\t3 2\n",
      "11\t8\t0.1\t1\t5 3\n",
      "12\t7\t0.0\t1\t4 3\n",
      "13\t8\t0.0\t1\t3 5\n",
      "14\t27\t0.0\t2\t17 7 3\n",
      "15\t21\t0.0\t2\t12 1 8\n",
      "16\t35\t0.0\t2\t26 7 2\n",
      "17\t183\t0.0\t2\t108 52 23\n",
      "18\t64944\t0.0\t2\t63727 1168 49\n",
      "19\t73895\t0.0\t2\t72605 1226 64\n",
      "20\t75810\t0.0\t2\t74487 1261 62\n",
      "21\t67970\t0.0\t2\t66807 1096 67\n",
      "22\t69866\t0.0\t2\t68669 1144 53\n",
      "23\t59387\t0.0\t2\t58482 880 25\n",
      "24\t81\t0.0\t2\t63 6 12\n",
      "25\t2\t0.0\t2\t2\n",
      "\n",
      "\n",
      "=== Second read: Adapter 3 ===\n",
      "\n",
      "Sequence: CTGCTGCSYCCCGTAGG; Type: regular 3'; Length: 17; Trimmed: 37 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-17 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 16.2%\n",
      "  G: 18.9%\n",
      "  T: 8.1%\n",
      "  none/other: 56.8%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t1\t0.4\t1\t0 1\n",
      "61\t1\t0.0\t2\t0 0 1\n",
      "203\t7\t0.0\t2\t6 1\n",
      "204\t7\t0.0\t2\t7\n",
      "205\t21\t0.0\t2\t20 1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 4 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 407693 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-5 bp: 0; 6-12 bp: 1; 13-19 bp: 2; 20 bp: 3\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "8\t1\t6.4\t1\t0 1\n",
      "11\t1\t0.1\t1\t1\n",
      "12\t7\t0.0\t1\t5 2\n",
      "13\t3\t0.0\t1\t0 3\n",
      "14\t3\t0.0\t2\t1 1 1\n",
      "15\t22\t0.0\t2\t12 3 7\n",
      "16\t6\t0.0\t2\t1 1 4\n",
      "17\t11\t0.0\t2\t7 2 2\n",
      "18\t15\t0.0\t2\t8 5 2\n",
      "19\t17\t0.0\t2\t9 5 3\n",
      "20\t328\t0.0\t3\t138 84 57 49\n",
      "21\t75920\t0.0\t3\t72215 2693 651 361\n",
      "22\t73065\t0.0\t3\t69374 2679 661 351\n",
      "23\t67778\t0.0\t3\t63845 2911 692 330\n",
      "24\t69099\t0.0\t3\t65324 2732 710 333\n",
      "25\t62151\t0.0\t3\t58672 2553 622 304\n",
      "26\t59158\t0.0\t3\t55823 2417 622 296\n",
      "27\t63\t0.0\t3\t51 5 4 3\n",
      "28\t2\t0.0\t3\t2\n",
      "30\t1\t0.0\t3\t0 0 1\n",
      "36\t1\t0.0\t3\t0 1\n",
      "55\t1\t0.0\t3\t0 0 1\n",
      "57\t1\t0.0\t3\t0 0 0 1\n",
      "58\t4\t0.0\t3\t0 2 0 2\n",
      "59\t2\t0.0\t3\t0 0 0 2\n",
      "60\t3\t0.0\t3\t0 0 1 2\n",
      "61\t6\t0.0\t3\t0 0 4 2\n",
      "62\t10\t0.0\t3\t1 0 3 6\n",
      "63\t2\t0.0\t3\t0 1 0 1\n",
      "64\t7\t0.0\t3\t0 0 3 4\n",
      "65\t2\t0.0\t3\t0 0 1 1\n",
      "66\t3\t0.0\t3\t0 0 1 2\n"
     ]
    }
   ],
   "source": [
    "if trim:    \n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = trim_paired(\n",
    "        demultiplexed_sequences=artifact,\n",
    "        front_f=forward_primer,\n",
    "        front_r=reverse_primer,\n",
    "        adapter_f=reverse_reverse_complement,\n",
    "        adapter_r=forward_reverse_complement,\n",
    "        cores=threads,\n",
    "        overlap=trim['overlap'],\n",
    "        indels=False,\n",
    "        match_read_wildcards=True,\n",
    "        match_adapter_wildcards=True,\n",
    "        error_rate=0.15,\n",
    "        discard_untrimmed=True,\n",
    "    ).trimmed_sequences\n",
    "\n",
    "    demux_file_trim = os.path.join(out_dir, 'demux-paired-trim.qza')\n",
    "    demux_view_trim = os.path.join(out_dir, 'demux-paired-trim.qzv')\n",
    "    \n",
    "    demux.visualizers.summarize(res).visualization\n",
    "    res.save(demux_file_trim)\n",
    "    Visualization.save(demux.visualizers.summarize(res).visualization, filepath=demux_view_trim)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 740.208481,
   "end_time": "2022-09-15T17:28:52.663459",
   "environment_variables": {},
   "exception": null,
   "input_path": "./step-01-prepare-data.ipynb",
   "output_path": "../executed/step-01-prepare-data-STD-NCxSTD-NR.ipynb",
   "parameters": {
    "base_dir": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri",
    "class_col": "sample-group",
    "classifier_file": "/home/lauro/nupeb/dados_brutos_rede_genoma/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza",
    "experiment_name": "ana-flavia-STD-NCxSTD-NR-trim",
    "manifest_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-STD-NCxSTD-NR.csv",
    "metadata_file": "/home/lauro/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-STD-NCxSTD-NR.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2022-09-15T17:16:32.454978",
   "version": "2.3.4"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}