{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "15bcfa06",
   "metadata": {},
   "source": [
    " \n",
    "# Alpha and Beta Diversity\n",
    "\n",
    "Different higher-level measures are often used to describe the microbiome in a sample. These do not provide information on changes in the abundance of specific taxa but allow us to access a broader change or difference in the composition of microorganisms. Alpha and beta diversity are examples of such measures.\n",
    "\n",
    "Different measures exist to estimate diversity within a single sample, jointly called alpha diversity. The different measures reflect the richness (number) or distribution (evenness) of a microbial sample or aim to reflect a combination of both properties.\n",
    "\n",
    "Rarefaction curves are often used when calculating alpha diversity indices because increasing numbers of sequenced taxa allow increasingly accurate estimates of total population diversity. Rarefaction curves can therefore be used to estimate the full sample richness, as compared to the observed sample richness.\n",
    "\n",
    "While alpha diversity is a measure of microbiome diversity applicable to a single sample, beta diversity is a measure of the similarity or dissimilarity of two communities. As for alpha diversity, many indices exist, each reflecting different aspects of community heterogeneity. Key differences relate to how the indices value variation in rare species if they consider presence/absence only or incorporate abundance, and how they interpret shared absence. Bray-Curtis dissimilarity is a popular measure that considers both size (overall abundance per sample) and shape (abundance of each taxon) of the communities (Bray, 1957). Beta diversity is an essential measure for many popular statistical methods in ecology, such as ordination-based methods, and is widely used for studying the association between environmental variables and microbial composition.\n",
    "\n",
    "In summary, alpha diversity measures can be seen as a summary statistic of a single population (within-sample diversity), while beta diversity measures are estimates of similarity or dissimilarity between populations (between samples).\n",
    "\n",
    "**Source**: (https://biomcare.com/info/key-terms-in-microbiome-projects/)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e7f2330",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-22T19:36:33.828188Z",
     "start_time": "2022-09-22T19:36:33.817323Z"
    }
   },
   "source": [
    "### STEP : Diversity Analysis\n",
    "\n",
    "Using QIIME2 to create diversity analisys graphs and calculations.\n",
    "\n",
    "- [QIIME2 Workflow Overview](https://docs.qiime2.org/2022.8/tutorials/overview/)\n",
    "\n",
    "\n",
    "#### Methods\n",
    "- [diversity](https://docs.qiime2.org/2022.8/plugins/available/diversity/)\n",
    "- [diversity alpha](https://docs.qiime2.org/2022.8/plugins/available/diversity/alpha/)\n",
    "- [diversity alpha_phylogenetic](https://docs.qiime2.org/2022.8/plugins/available/diversity/alpha-phylogenetic/)\n",
    "- [diversity beta](https://docs.qiime2.org/2022.8/plugins/available/diversity/beta/)\n",
    "- [diversity core_metrics](https://docs.qiime2.org/2022.8/plugins/available/diversity/core-metrics/)\n",
    "- [diversity alpha_group_significance](https://docs.qiime2.org/2022.8/plugins/available/diversity/alpha-group-significance/)\n",
    "- [diversity beta_group_significance](https://docs.qiime2.org/2022.8/plugins/available/diversity/beta-group-significance/)\n",
    "- [feature_table core_features](https://docs.qiime2.org/2022.8/plugins/available/feature-table/core-features/)\n",
    "- [feature_table summarize](https://docs.qiime2.org/2022.8/plugins/available/feature-table/summarize/)\n",
    "- [taxa filter-table](https://docs.qiime2.org/2022.8/plugins/available/taxa/filter-table/)\n",
    "- [taxa collapse](https://docs.qiime2.org/2022.8/plugins/available/taxa/collapse/)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "843debbf",
   "metadata": {},
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "965e28e5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:14.786765Z",
     "start_time": "2022-09-29T20:54:08.286473Z"
    }
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import pandas as pd\n",
    "from qiime2 import Artifact\n",
    "from qiime2 import Visualization\n",
    "from qiime2 import Metadata\n",
    "\n",
    "from qiime2.plugins.phylogeny.pipelines import align_to_tree_mafft_fasttree\n",
    "\n",
    "from qiime2.plugins.diversity.pipelines import alpha\n",
    "from qiime2.plugins.diversity.pipelines import beta\n",
    "from qiime2.plugins.diversity.pipelines import core_metrics\n",
    "from qiime2.plugins.diversity.pipelines import alpha_phylogenetic\n",
    "\n",
    "from qiime2.plugins.diversity.visualizers import alpha_group_significance\n",
    "from qiime2.plugins.diversity.visualizers import beta_group_significance\n",
    "from qiime2.plugins.diversity.visualizers import alpha_correlation\n",
    "from qiime2.plugins.diversity.visualizers import beta_rarefaction\n",
    "\n",
    "from qiime2.plugins.taxa.methods import filter_table\n",
    "from qiime2.plugins.taxa.methods import collapse\n",
    "\n",
    "from qiime2.plugins.feature_table.visualizers import tabulate_seqs\n",
    "from qiime2.plugins.feature_table.visualizers import summarize\n",
    "from qiime2.plugins.feature_table.visualizers import core_features\n",
    "\n",
    "from qiime2.plugins.alignment.methods import mafft\n",
    "\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "from utils import *\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "530ceb60",
   "metadata": {},
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "18d874ea",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:14.793270Z",
     "start_time": "2022-09-29T20:54:14.789367Z"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "metadata_file = '/home/lauro/nupeb/rede-micro/redemicro-miliane-nutri/data/raw/metadata/miliane-metadata-CxAC.tsv'\n",
    "base_dir = os.path.join('/', 'home', 'lauro', 'nupeb', 'rede-micro', 'redemicro-miliane-nutri')\n",
    "experiment_name = 'miliane-CxAC-trim'\n",
    "class_col = 'group-id'\n",
    "replace_files = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f38617d9",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:14.804575Z",
     "start_time": "2022-09-29T20:54:14.794927Z"
    }
   },
   "outputs": [],
   "source": [
    "experiment_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name))\n",
    "img_folder = os.path.abspath(os.path.join(experiment_folder, 'imgs'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea9b90fb",
   "metadata": {},
   "source": [
    "### Defining names, paths and flags"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "45727eef",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:14.816686Z",
     "start_time": "2022-09-29T20:54:14.806238Z"
    }
   },
   "outputs": [],
   "source": [
    "# QIIME2 Artifacts folder\n",
    "qiime_folder = os.path.join(experiment_folder, 'qiime-artifacts')\n",
    "\n",
    "# Input - DADA2 Artifacts\n",
    "dada2_tabs_path = os.path.join(qiime_folder, 'dada2-tabs.qza')\n",
    "dada2_reps_path = os.path.join(qiime_folder, 'dada2-reps.qza')\n",
    "dada2_stat_path = os.path.join(qiime_folder, 'dada2-stat.qza')\n",
    "\n",
    "# Input - Taxonaomic Artifacts\n",
    "taxonomy_path = os.path.join(qiime_folder, 'metatax.qza')\n",
    "\n",
    "# Output -Diversity Artifacts\n",
    "alpha_diversity_path = os.path.join(qiime_folder, 'alpha-diversity.qza')\n",
    "alpha_diversity_view_path = os.path.join(qiime_folder, 'alpha-diversity.qzv')\n",
    "beta_diversity_path = os.path.join(qiime_folder, 'beta-diversity.qza')\n",
    "beta_diversity_view_path = os.path.join(qiime_folder, 'beta-diversity.qzv')\n",
    "\n",
    "dada2_tabs_view = os.path.join(qiime_folder, 'dada2-tabs.qzv')\n",
    "dada2_reps_view = os.path.join(qiime_folder, 'dada2-reps.qzv')\n",
    "dada2_core_view = os.path.join(qiime_folder, 'dada2-core.qzv')\n",
    "\n",
    "# Flag - Load or create files\n",
    "need_tax = not (os.path.isfile(alpha_diversity_path)) or replace_files\n",
    "need_view = not (os.path.isfile(alpha_diversity_view_path)) or replace_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "601ae263",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:14.829117Z",
     "start_time": "2022-09-29T20:54:14.818272Z"
    }
   },
   "outputs": [],
   "source": [
    "def filter_and_collapse(tab, seqs, tax, meta, lvl, exclude=True, exclude_list='uncultured,unidentified,metagenome'):\n",
    "    from qiime2.plugins.taxa.methods import collapse\n",
    "    from qiime2.plugins.taxa.methods import filter_table\n",
    "    from qiime2.plugins.feature_table.methods import filter_seqs\n",
    "    from qiime2.plugins.feature_table.visualizers import summarize\n",
    "    \n",
    "    to_include = ('d', 'p', 'c', 'o', 'f', 'g', 's')[lvl-1]\n",
    "    to_include += '__'\n",
    "    to_exclude = exclude_list if exclude else None\n",
    "    \n",
    "    filtered_tabs = filter_table(\n",
    "        table=tab, \n",
    "        taxonomy=tax,\n",
    "        include=to_include,\n",
    "        exclude=to_exclude,\n",
    "        mode='contains').filtered_table\n",
    "    \n",
    "    filtered_seqs = filter_seqs(\n",
    "        data = seqs,\n",
    "        table = filtered_tabs,\n",
    "    ).filtered_data\n",
    "    \n",
    "    collapsed_table = collapse(table=filtered_tabs, taxonomy=tax, level=lvl).collapsed_table\n",
    "    collapsed_table_view = summarize(table=collapsed_table, sample_metadata=meta).visualization\n",
    "    \n",
    "    return collapsed_table, collapsed_table_view, filtered_seqs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0999a1b2",
   "metadata": {},
   "source": [
    "## Step execution\n",
    "\n",
    "### Load input files\n",
    "\n",
    "This Step import the QIIME2 `FeatureTable[Frequency]` Artifact and the `Metadata` file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8f299d1a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:15.559837Z",
     "start_time": "2022-09-29T20:54:14.830405Z"
    }
   },
   "outputs": [],
   "source": [
    "#Load Metadata\n",
    "metadata_qa = Metadata.load(metadata_file)\n",
    "\n",
    "#Load FeatureTable[Frequency]\n",
    "tabs = Artifact.load(dada2_tabs_path)\n",
    "tabs_df = tabs.view(Metadata).to_dataframe().T\n",
    "\n",
    "# FeatureData[Sequence]\n",
    "reps = Artifact.load(dada2_reps_path)\n",
    "\n",
    "# FeatureData[Taxonomy]\n",
    "tax = Artifact.load(taxonomy_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c8dfe530",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:15.564968Z",
     "start_time": "2022-09-29T20:54:15.561875Z"
    }
   },
   "outputs": [],
   "source": [
    "# lvl = 7\n",
    "# exclude = True\n",
    "# tabs, collapsed_table_view, reps = filter_and_collapse(\n",
    "#                     tabs, reps, tax, metadata_qa, \n",
    "#                     lvl=lvl,\n",
    "#                     exclude=exclude, \n",
    "#                     exclude_list='uncultured,unidentified,metagenome')\n",
    "# collapsed_table_view"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8b66a6e5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:20.434382Z",
     "start_time": "2022-09-29T20:54:15.567722Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/lauro/nupeb/rede-micro/redemicro-miliane-nutri/experiments/miliane-CxAC-trim/qiime-artifacts/dada2-core.qzv'"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAY0AAAEKCAYAAADuEgmxAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/YYfK9AAAACXBIWXMAAAsTAAALEwEAmpwYAAAcAUlEQVR4nO3df5xddX3n8debyYTMQCCDZBCTIESjIbFCdWS12jS1KoiVHyoYq9ugbIk+aKnb1QW2D6PiuqLWPtzWUpOlFtQqZluBqT9AGjeOKIoTjcrgULODJVOyDuoIgcmPmcln/zjfgZuQmfnOMGfuucn7+Xjcxz3n3HPufd87yf3c7/me8z2KCMzMzHIcVe8AZmbWOFw0zMwsm4uGmZllc9EwM7NsLhpmZpbNRcPMzLKVWjQk/UzSjyVtk9Sdlp0g6Q5JP033bTXrXy1pu6T7JJ1dZjYzM5u62Whp/G5EnBkRHWn+KmBzRCwDNqd5JK0A1gArgXOA6yQ1zUI+MzPLVI/dU+cDN6bpG4ELapbfFBF7I+J+YDtw1uzHMzOz8cwp+fkD+JqkADZExEbgpIjYCRAROyW1p3UXAd+p2bY/LTuApMuAywCa5s574bEnPZP9Ecw56iiWLjymzPdiZnZY2Lp16y8iYuF0ti27aLw0Ih5MheEOSb0TrKtDLHvSGCep8GwEWHDK8vjtd19PRPDw7mG+eeXLZya1mdlhTNK/TXfbUndPRcSD6X4AuJlid9PPJZ0MkO4H0ur9wJKazRcDD+a8zu7hURa3tc5UbDMzG0dpRUPSMZLmj00DrwLuATqBtWm1tcCtaboTWCPpaEmnAcuAuyd7naF9IwyPButWLZ3pt2BmZgcpc/fUScDNksZe53MRcZuk7wGbJF0KPABcBBARPZI2AfcCI8DlETE60QuM7g/a589j3aqlrF7ePtGqZmY2A9TIQ6N3dHREd3d3vWOYmTUUSVtrToOYEp8RbmZm2Vw0zMwsm4uGmZllc9EwM7NsLhpmZpbNRcPMzLK5aJiZWTYXDTMzy+aiYWZm2Vw0zMwsm4uGmZllc9EwM7NsLhpmZpbNRcPMzLK5aJiZWTYXDTMzy+aiYWZm2Vw0zMwsm4uGmZllc9EwM7NsLhpmZpbNRcPMzLK5aJiZWTYXDTMzy+aiYWZm2Vw0zMwsm4uGmZllc9EwM7NsLhpmZpbNRcPMzLK5aJiZWTYXDTMzy+aiYWZm2Vw0zMwsm4uGmZllK71oSGqS9ANJX0rzJ0i6Q9JP031bzbpXS9ou6T5JZ5edzczMpmY2Whp/CvykZv4qYHNELAM2p3kkrQDWACuBc4DrJDXNQj4zM8tUatGQtBh4DXB9zeLzgRvT9I3ABTXLb4qIvRFxP7AdOKvMfGZmNjVltzQ+DvxXYH/NspMiYidAum9PyxcBO2rW60/LDiDpMkndkrofeuihUkKbmdmhlVY0JP0+MBARW3M3OcSyeNKCiI0R0RERHQsXLnxKGc3MbGrmlPjcLwXOk3QuMA84TtJngZ9LOjkidko6GRhI6/cDS2q2Xww8WGI+MzObotJaGhFxdUQsjohTKTq4vx4RbwE6gbVptbXArWm6E1gj6WhJpwHLgLvLymdmZlNXZktjPNcCmyRdCjwAXAQQET2SNgH3AiPA5RExWod8ZmY2DkU8qdugYXR0dER3d3e9Y5iZNRRJWyOiYzrb+oxwMzPL5qJhZmbZXDTMzCybi4aZmWVz0TAzs2wuGmZmls1Fw8zMsrlomJlZNhcNMzPL5qJhZmbZXDTMzCybi4aZmWVz0TAzs2wuGmZmls1Fw8zMsrlomJlZNhcNMzPL5qJhZmbZXDTMzCybi4aZmWVz0TAzs2wuGmZmls1Fw8zMsrlomJlZtjn1DnA42dI7wIauPnYMDrGkrZV1q5ayenl7vWOZmc0YtzRmyJbeAdZ39jCwaw8LWpoZ2LWH9Z09bOkdqHc0M7MZM2nRkPSZnGVHug1dfTQ3ida5c5CK++YmsaGrr97RzMxmTE5LY2XtjKQm4IXlxGlcOwaHaGluOmBZS3MT/YNDdUpkZjbzxi0akq6WtAt4vqRH0m0XMADcOmsJG8SStlZ2D48esGz38CiL21rrlMjMbOaNWzQi4kMRMR/4aEQcl27zI+JpEXH1LGZsCOtWLWV4NBjaN0JEcT88GqxbtbTe0czMZsykR09FxNWSFgHPrF0/IrrKDNZoVi9v5xqKvo3+wSEW++gpMzsMTVo0JF0LrAHuBcb2vwTgonGQ1cvbXSTM7LCWc57GhcBzI2Jv2WHMzKzaco6e6gOayw5iZmbVl9PSGAK2SdoMPN7aiIgrJtpI0jyKXVhHp9f5x4h4r6QTgC8ApwI/Ay6OiMG0zdXApRS7wa6IiNun+obMzKw8OUWjM92mai/w8oh4VFIzcKekrwKvAzZHxLWSrgKuAq6UtIKi72Ql8AzgXyQ9JyJGx3sBMzObXTlHT904nSeOiAAeTbPN6RbA+cDqtPxGYAtwZVp+U+o7uV/SduAs4K7pvL6Zmc28nKOn7qf4sj9AREx6AkI6e3wr8GzgbyLiu5JOioid6Tl2Sho73GgR8J2azfvTsoOf8zLgMoBTTjllsghmZjaDcnZPddRMzwMuAk7IefK0a+lMSQuAmyU9b4LVdainOMRzbgQ2AnR0dDzpcTMzK8+kR09FxC9rbv8eER8HXj6VF4mIX1PshjoH+LmkkwHS/dgwsP3AkprNFgMPTuV1zMysXDmj3L6g5tYh6e3A/IztFqYWBpJagFcAvRSd6mvTamt5YhyrTmCNpKMlnQYsA+6e6hsyM7Py5Oye+ljN9AjpMNmM7U4Gbkz9GkcBmyLiS5LuAjZJuhR4gGJ3FxHRI2kTxZnnI8DlPnLKzKxaVBzk1Jg6Ojqiu7u73jHMzBqKpK0R0TH5mk+Ws3vqeEl/Kak73T4m6fjpvJiZmTW2nGFEPgXsotgldTHwCPD3ZYYyM7NqyunTeFZEvL5m/v2StpWUx8zMKiynpbFb0svGZiS9FNhdXiQzM6uqnJbGOyiOghrrxxgELiktkZmZVVbO2FPbgDMkHZfmHyk7lJmZVVPO0VP/Q9KCiHgkIh6R1Cbpv89GODMzq5acPo1Xp2FAAEjXvji3tERmZlZZOUWjSdLRYzNpSJCjJ1jfzMwOUzkd4Z8FNkv6e4pRZ99GcR0MMzM7wuR0hH9E0o8oBhwU8AFfhtXM7MiU09IgIm4Dbis5i5mZVVxOn4aZmRngomFmZlOQVTQktUh6btlhzMys2nJO7nstsI3UpyHpTEmdJecyM7MKymlpvA84C/g1PD6syKllBTIzs+rKKRojEfFw6UnMzKzycg65vUfSH1CcGb4MuAL4drmxzMysinJaGn8CrAT2Ap8DHgbeWWImMzOrqAlbGpKagM6IeAXw57MTyWbSlt4BNnT1sWNwiCVtraxbtZTVy9vrHcvMGtSELY2IGAWGai7AZA1kS+8A6zt7GNi1hwUtzQzs2sP6zh629A7UO5qZNaicPo09wI8l3QE8NrYwIq4oLZXNiA1dfTQ3ida5xZ+5de4chvaNsKGrz60NM5uWnKLx5XSzBrNjcIgFLc0HLGtpbqJ/cKhOicys0eWMcuth0BvUkrZWBnbtebylAbB7eJTFba11TGVmjSznjPD7JfUdfJuNcPbUrFu1lOHRYGjfCBHF/fBosG7V0npHM7MGlbN7qqNmeh5wEXBCOXFsJq1e3s41FH0b/YNDLPbRU2b2FCkipr6RdGdEvKyEPFPS0dER3d3d9Y5hZtZQJG2NiI7J13yySVsakl5QM3sURctj/nRezMzMGlvO7qmP1UyPAPcDF5cTx8zMqiynaFwaEQd0fEs6raQ8ZmZWYTljT/1j5jIzMzvMjdvSkLScYqDC4yW9ruah4yiOojIzsyPMRLunngv8PrAAeG3N8l3AH5WYyczMKmrcohERtwK3SnpJRNw1i5nMzKyicjrCfyDpcopdVY/vloqIt020kaQlwKeBpwP7gY0R8T8lnQB8geKSsT8DLo6IwbTN1cClwChwRUTcPtU3ZGZm5cnpCP8MxRf/2cA3gMUUu6gmMwL8l4g4HXgxcLmkFcBVwOaIWAZsTvOkx9ZQFKdzgOvS9TzMzKwicorGsyPiPcBjafDC1wC/MdlGEbEzIr6fpncBPwEWAecDY4Mg3ghckKbPB26KiL0RcT+wHThrCu/FzMxKllM0htP9ryU9DzieYtdSNkmnAr8JfBc4KSJ2QlFYgLGBkBYBO2o260/LDn6uyyR1S+p+6KGHphLDzMyeopyisVFSG/AeoBO4F/hI7gtIOhb4J+CdEfHIRKseYtmTBsaKiI0R0RERHQsXLsyNYWZmMyDnehrXp8lvAFMaU1tSM0XB+IeI+GJa/HNJJ0fETkknA2PXHu0HltRsvhh4cCqvZ2Zm5cq5nsZJkv5O0lfT/ApJl2ZsJ+DvgJ9ExF/WPNQJrE3Ta4Fba5avkXR0GqZkGXB3/lsxM7Oy5eyeugG4HXhGmv9X4J0Z270U+I/AyyVtS7dzgWuBV0r6KfDKNE9E9ACbKHZ/3QZcHhGj+W/FzMzKlnOexokRsSmdQ0FEjEia9Ms8Iu7k0P0UAL83zjYfBD6YkcnMzOogp6XxmKSnkTqlJb0YeLjUVGZmVkk5LY0/o+hveJakbwELgTeUmsrMzCppolFuT4mIByLi+5J+h2IAQwH3RcTweNuZmdnha6LdU7fUTH8hInoi4h4XDDOzI9dEu6dqO7GndH6G2US29A6woauPHYNDLGlrZd2qpaxe3j75hmZWdxO1NGKcabNp29I7wPrOHgZ27WFBSzMDu/awvrOHLb0Dk29sZnU3UdE4Q9IjknYBz0/Tj0jaJWmi4UDMxrWhq4/mJtE6dw5Scd/cJDZ09U2+sZnV3UQXYfKw5DbjdgwOsaCl+YBlLc1N9A8O1SmRmU1FznkaZjNmSVsru4cPPDd09/Aoi9ta65TIzKbCRcNm1bpVSxkeDYb2jRBR3A+PButW+VgLs0bgomGzavXydq45byXt8+fx8O5h2ufP45rzVvroKbMGkXNGuNmMWr283UXCrEG5pWFmZtlcNMzMLJuLhpmZZXPRMDOzbC4aZmaWzUXDzMyyuWiYmVk2Fw0zM8vmomFmZtlcNMzMLJuLhpmZZXPRMDOzbC4aZmaWzUXDzMyyeWh0M2BL7wAbuvrYMTjEkrZW1q1a6uHbzQ7BLQ074m3pHWB9Zw8Du/awoKWZgV17WN/Zw5begXpHM6scFw074m3o6qO5SbTOnYNU3Dc3iQ1dffWOZlY5Lhp2xNsxOERLc9MBy1qam+gfHKpTIrPqctGwI96StlZ2D48esGz38CiL21rrlMisulw07Ii3btVShkeDoX0jRBT3w6PBulVL6x3NrHJcNOyIt3p5O9ect5L2+fN4ePcw7fPncc15K330lNkh+JBbM4rC4SJhNrnSWhqSPiVpQNI9NctOkHSHpJ+m+7aax66WtF3SfZLOLiuXmZlNX5m7p24Azjlo2VXA5ohYBmxO80haAawBVqZtrpPUhJmZVUppRSMiuoBfHbT4fODGNH0jcEHN8psiYm9E3A9sB84qK5uZmU3PbHeEnxQROwHS/dhO5EXAjpr1+tOyJ5F0maRuSd0PPfRQqWHNzOxAVTl6SodYFodaMSI2RkRHRHQsXLiw5FhmZlZrtovGzyWdDJDuxwb36QeW1Ky3GHhwlrOZmdkkZrtodAJr0/Ra4Naa5WskHS3pNGAZcPcsZzMzs0mUdp6GpM8Dq4ETJfUD7wWuBTZJuhR4ALgIICJ6JG0C7gVGgMsjYvSQT2xmZnVTWtGIiDeN89DvjbP+B4EPlpXHzMyeuqp0hJuZWQNw0TAzs2wuGmZmls1Fw8zMsrlomJlZNhcNMzPL5qJhZmbZfBEms4ra0jvAhq4+dgwOsaStlXWrlvpCUVZ3bmmYVdCW3gHWd/YwsGsPC1qaGdi1h/WdPWzpHZh8Y7MSuWiYVdCGrj6am0Tr3DlIxX1zk9jQ1VfvaHaEc9Ewq6Adg0O0NB948cqW5ib6B4fqlMis4KJhVkFL2lrZPXzgmJ27h0dZ3NZap0RmBRcNswpat2opw6PB0L4RIor74dFg3aql9Y5mRzgXDbMKWr28nWvOW0n7/Hk8vHuY9vnzuOa8lT56yurOh9yaVdTq5e0uElY5bmmYmVk2Fw0zM8vm3VNmNiU+U/3I5paGmWXzmermomFm2XymurlomFk2n6lu7tMws2xL2loZ2LWH1rlPfHVU4Ux197PMHrc0zCxbFc9Udz/L7HLRMLNsVTxT3f0ss8u7p8xsSqp2pvqOwSEWtDQfsKwK/SyH6y4ztzTMrKFVcUTgw3mXmYuGmTW0KvazVHWX2ZbeAd608Ts0Lzz1N6b7HC4aZtbQqtjPUsVDk2tbP8T+kek+j/s0zKzhVa2fpYqHJte2fp4KtzTMzGZYFXeZHar1Mx0uGmZmM6yKu8wOdcDAdHj3lJlZCaq2y2zdqqWs7+xhaN+0uzMAtzTMzI4Ita0fdNS0GwyKiJnMNas6Ojqiu7u73jHMzBqKpK0R0TGdbd3SMDOzbJUrGpLOkXSfpO2Srqp3HjMze0KlioakJuBvgFcDK4A3SVpR31RmZjamUkUDOAvYHhF9EbEPuAk4v86ZzMwsqdoht4uAHTXz/cB/qF1B0mXAZWl2r6R7ZinbVJwI/KLeIQ7iTHmcKV8VczlTnudOd8OqFQ0dYtkBh3dFxEZgI4Ck7ukeAVCmKuZypjzOlK+KuZwpj6RpH3Zatd1T/cCSmvnFwIN1ymJmZgepWtH4HrBM0mmS5gJrgM46ZzIzs6RSu6ciYkTSHwO3A03ApyKiZ4JNNs5OsimrYi5nyuNM+aqYy5nyTDtTQ58RbmZms6tqu6fMzKzCXDTMzCxbQxSNyYYWkbRc0l2S9kp6V0UyvVnSj9Lt25LOqECm81OebZK6Jb2s7Ew5uWrWe5GkUUlvqHcmSaslPZw+q22S1tc7U02ubZJ6JH2j3pkkvbvmM7on/f1OqHOm4yX9s6Qfps/prWXmmUKuNkk3p/+Dd0t6Xsl5PiVpYLxz2VT4q5T3R5JekPXEEVHpG0WH+P8FlgJzgR8CKw5apx14EfBB4F0VyfRbQFuafjXw3QpkOpYn+rGeD/RW4bOqWe/rwFeAN9Q7E7Aa+FLZn88UMy0A7gVOSfPt9c500PqvBb5e70zAfwM+nKYXAr8C5lYg10eB96bp5cDmkjOtAl4A3DPO4+cCX6U4P+7Fud9RjdDSmHRokYgYiIjvAcMVyvTtiBhMs9+hOOek3pkejfSvBTiGg06crFeu5E+AfwIGKpRpNuVk+gPgixHxABT/7iuQqdabgM9XIFMA8yWJ4ofSr4CnduWhmcm1AtgMEBG9wKmSTiorUER0Ubz38ZwPfDoK3wEWSDp5sudthKJxqKFFFtUpy5ipZrqUoqKXKSuTpAsl9QJfBt5WcqasXJIWARcCn5yFPFmZkpekXRxflbSyApmeA7RJ2iJpq6Q/rEAmACS1AudQFP56Z/oEcDrFicE/Bv40IvZXINcPgdcBSDoLeCbl/5icyLS+WxuhaEw6tEgdZGeS9LsURePKUhNlZoqImyNiOXAB8IGSM0Fero8DV0bEU7+AcZ6cTN8HnhkRZwB/DdxSgUxzgBcCrwHOBt4j6Tl1zjTmtcC3ImKiX7YzISfT2cA24BnAmcAnJB1XbqysXNdSFP1tFC3rH1B+C2gi0/purdTJfeOo4tAiWZkkPR+4Hnh1RPyyCpnGRESXpGdJOjEiyhxMLSdXB3BTsTeBE4FzJY1ExC31yhQRj9RMf0XSdSV/VjmfUz/wi4h4DHhMUhdwBvCvdcw0Zg3l75qCvExvBa5Nu2K3S7qfog/h7nrmSv+m3gpFJzRwf7rVy/S+W8vsiJmhzpw5QB9wGk90MK0cZ933MTsd4ZNmAk4BtgO/VZXPCXg2T3SEvwD497H5Kvz90vo3UH5HeM5n9fSaz+os4IEyP6vMTKdT7BOfA7QC9wDPq/ffDjieYt/5MWX+3abwOf0t8L40fVL6d35iBXItIHXIA39E0Z9Q9ud1KuN3hL+GAzvC7855zsq3NGKcoUUkvT09/klJTwe6geOA/ZLeSXHkwiPjPW/ZmYD1wNOA69Iv6JEocaTLzEyvB/5Q0jCwG3hjpH89dc41qzIzvQF4h6QRis9qTZmfVU6miPiJpNuAHwH7gesjorRLA0zhb3ch8LUoWkClysz0AeAGST+m+EK8MsptTefmOh34tKRRiqPgLi0zk6TPUxwFeKKkfuC9QHNNnq9QHEG1HRgitYImfd6SvzPMzOww0ggd4WZmVhEuGmZmls1Fw8zMsrlomJlZNhcNMzPL5qJhlZVGTN1Wczv1KT7fmZLOrZk/b7zRZGeKpCsk/UTSP5T5Ohk5Hq3n69vhw4fcWmVJejQijh3nMVH8+80eU0jSJUBHRPzxDEXMec1eihEB6nnm74SfpdlUuKVhDUPSqelX+3UUY0MtkfS3Kq4N0iPp/TXrvkjFdUx+mK5dcDxwDfDG1Gp5o6RLJH0irf9MSZvTdQU2SzolLb8hXXPg25L6NM61PiT9mYrrSdyTTi5F0icphsrulPSfD1p/Zcq1Lb3msrT8ljQYYY+ky2rWf1TSh9Nj/yLprDRwYZ+k89I6l0i6VdJtKq7r8N5xsr5b0vfS674/LTtG0pfT53WPpDdO649kh7+yT2P3zbfp3oBRioHntgE3UwyJsB94cc06J6T7JmALxXVC5lIM6fCi9NhxFMM8XAJ8ombbx+eBfwbWpum3Abek6RuA/03xA2sFxfDXB+d8IcVoqsdQDMXdA/xmeuxnHGIIC4pBEN+cpucCLQe9nxaKYUKeluaDosVC+iy+RnF27xnAtpr3s5NiJIKx7TvSY4+m+1cBGynOlD4K+BLFdRdeD/yvmnzH1/vv71s1b5UfRsSOaLsj4syxmdSn8W9RjP0/5uL0i3wOcDLFF3sAO6O4xgqRhpNJw7mM5yWkYauBzwAfqXnslih2g92rQ1//4GXAzZGG0ZD0ReC3KUYxHc9dwJ9LWkxxjYyfpuVXSLowTS8BlgG/BPYBt6XlPwb2RsRwGirj1JrnvSPS4Jgpx8sohtgZ86p0G8t2bHqNbwJ/IenDFBef+uYE2e0I5qJhjebx8Y0knQa8i6JFMSjpBmAexa/op9pZV7v93prpQ1WeCavRIZ884nOSvksxaNztkv4TRSvqFcBLImJI0haK9wMwHBFjmfaPZYqI/ZJq/x8f/L4PnhfwoYjY8KQ3Ib2QYiyiD0n6WkRcM9X3ZYc/92lYIzuOoog8nFoAr07Le4FnSHoRgKT56Yt1FzB/nOf6NsXw3gBvBu6cQo4u4AJJrZKOoRjAb8Jf6pKWAn0R8VdAJ8VuteOBwVQwllOMPDpVr5R0gqQWimumfOugx28H3ibp2JRjkaR2Sc8AhiLis8BfUIyCbPYkbmlYw4qIH0r6AUUfQh/pCzIi9qWO3L9OX567KX7B/x/gKhUXwfnQQU93BfApSe8GHiJzxM/0et9PrZyx6zVcHxET7ZoCeCPwFhUjDv8/ik76x4C3S/oRcB/FZYKn6k6K3WvPBj4XEbW7poiIr0k6Hbgr7a57FHhLWv+jkvZTXDb5HdN4bTsC+JBbs8NEPQ4ptiOPd0+ZmVk2tzTMzCybWxpmZpbNRcPMzLK5aJiZWTYXDTMzy+aiYWZm2f4/q5CQ2bfI+p8AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "tabs_view = summarize(table=tabs, sample_metadata=metadata_qa).visualization\n",
    "tabs_view.save(dada2_tabs_view)\n",
    "reps_view = tabulate_seqs(data=reps).visualization\n",
    "reps_view.save(dada2_reps_view)\n",
    "core_view = core_features(table=tabs, min_fraction=.1, steps=10).visualization\n",
    "core_view.save(dada2_core_view)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b2bf36b8",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:20.449299Z",
     "start_time": "2022-09-29T20:54:20.435814Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['6f060e2c-ee21-425e-9345-522910e77455/data/core-features-1.000.tsv']\n",
      "                          Feature ID      2%       9%      25%     50%  \\\n",
      "0   c688e3b1ada46ed57f5b7e7e0d56664f  400.74   567.83  3168.50  9079.5   \n",
      "1   fca5b1ccd94b107a3ddf3e99feaafb6f  812.32  1278.94  2834.25  5627.0   \n",
      "2   26fb6a43293a39fa4463bc35a69457b2  420.58   643.11  2381.00  3918.0   \n",
      "3   55b8f708d3561852ca20801e513b08a7  412.98   612.41   900.75  3566.5   \n",
      "4   ff6880f38ac8c1569873d503a3d53c2a  901.16  1384.72  1836.25  3381.0   \n",
      "5   bc798de4a9acd3ff7ba51c244523be14  361.44   401.48  1101.00  2853.0   \n",
      "6   a1bc173c37d38b7a95f6f464d62fa86e  348.68   729.06  1012.50  1532.5   \n",
      "7   60ad690613589cad95c200968df41bb5  227.00   304.00   454.25  1313.0   \n",
      "8   33be3de1f8ec58c58a1d87db108d913c  453.74   736.33   924.25  1263.0   \n",
      "9   bfbcddb34034c8e1327bc8e4a3817f73  445.62   731.29   847.75  1012.0   \n",
      "10  94da0800673da2590ace91fc6e62c80a   91.86   101.87   377.50   809.5   \n",
      "11  b6b804323b11530a7bb8c25d42de7c58  341.68   529.56   569.50   774.0   \n",
      "12  06376bd58e034461292535056d63710f   73.60   135.20   426.25   648.5   \n",
      "13  72e62b56bb188827c70b57b9b1abdc71  122.02   192.09   304.50   473.0   \n",
      "14  d54ba858dcfc687c0433becca93d891c   58.64    67.88   211.25   468.5   \n",
      "15  f91de93a2a6c016c964ba9583e72190c  121.56   158.52   288.00   366.0   \n",
      "16  191bca7ba3d98899bc0037f6a2b82f1b   77.88    80.96   160.00   238.5   \n",
      "17  b803764a3bf4c19193100cbe10778b1b   47.80    78.60   115.00   237.0   \n",
      "18  3aa324222f8ba93b3e7263588f586d1c   39.36    68.62   104.25   192.0   \n",
      "19  d87bf25a6ca7762903d692d567a10280   14.54    19.93    56.25   120.0   \n",
      "\n",
      "         75%       91%       98%  \n",
      "0   16105.25  26750.65  29726.70  \n",
      "1   12796.75  19986.71  21350.38  \n",
      "2    7309.50  10340.15  11968.70  \n",
      "3    7336.00  10128.06  13828.68  \n",
      "4    4071.50   6905.00   7059.00  \n",
      "5    4038.00   9131.38  10238.64  \n",
      "6    2903.00   5647.49   6840.22  \n",
      "7    2946.50   6544.14   8479.92  \n",
      "8    1858.25   3611.53   3652.34  \n",
      "9    2206.00   3507.63   4326.14  \n",
      "10   1114.50   1575.23   2439.94  \n",
      "11   1511.75   2805.77   3096.06  \n",
      "12   1110.00   3597.02   3752.56  \n",
      "13    670.00   1269.68   1476.04  \n",
      "14    580.25    736.29    912.62  \n",
      "15    636.25   1047.23   2219.94  \n",
      "16    486.25    585.74    950.72  \n",
      "17    747.50   1667.63   1947.14  \n",
      "18    406.25    520.61    567.58  \n",
      "19    196.25    250.53    291.34  \n"
     ]
    }
   ],
   "source": [
    "import zipfile\n",
    "from io import StringIO\n",
    "import pandas as pd\n",
    "with zipfile.ZipFile(dada2_core_view, 'r') as zip_ref:\n",
    "    names = [x for x in zip_ref.namelist() if x.endswith('1.000.tsv')]\n",
    "    print(names)\n",
    "    with zip_ref.open(names[0]) as myfile:\n",
    "        txt = str(myfile.read(), 'utf-8')\n",
    "        txt = StringIO(txt)\n",
    "        df = pd.read_csv(txt, sep='\\t')\n",
    "        print(df)   "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9f43843a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-22T22:48:07.523952Z",
     "start_time": "2022-09-22T22:48:07.517565Z"
    }
   },
   "source": [
    "## Alpha diversity analysis\n",
    "\n",
    "#### Reference\n",
    "- [The Use and Types of Alpha-Diversity Metrics in Microbial NGS](https://www.cd-genomics.com/microbioseq/the-use-and-types-of-alpha-diversity-metrics-in-microbial-ngs.html)\n",
    "- [Alpha diversity metrics](http://scikit-bio.org/docs/0.2.0/generated/skbio.diversity.alpha.html)\n",
    "\n",
    "#### Methods\n",
    "- [diversity alpha](https://docs.qiime2.org/2022.8/plugins/available/diversity/alpha/): Computes a user-specified alpha diversity metric for all samples in a\n",
    "feature table.\n",
    "- [diversity alpha_phylogenetic](https://docs.qiime2.org/2022.8/plugins/available/diversity/alpha-phylogenetic/): Computes a user-specified phylogenetic alpha diversity metric for all\n",
    "samples in a feature table.\n",
    "- [diversity alpha_correlation](https://docs.qiime2.org/2022.8/plugins/available/diversity/alpha-correlation/): Determine whether numeric sample metadata columns are correlated with alpha diversity.\n",
    "- [diversity alpha_group_significance](https://docs.qiime2.org/2022.8/plugins/available/diversity/alpha-group-significance/): Visually and statistically compare groups of alpha diversity values."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "606b9c75",
   "metadata": {},
   "source": [
    "### Compute Alpha Diversity vectors\n",
    "- [diversity alpha](https://docs.qiime2.org/2022.8/plugins/available/diversity/alpha/): Computes a user-specified alpha diversity metric for all samples in a feature table.\n",
    "- [Alpha diversity metrics](http://scikit-bio.org/docs/0.2.0/generated/skbio.diversity.alpha.html)\n",
    " - Choices: ('ace', 'berger_parker_d', 'brillouin_d', 'chao1', 'chao1_ci', 'dominance', 'doubles', 'enspie', 'esty_ci', 'fisher_alpha', 'gini_index', 'goods_coverage', 'heip_e', 'kempton_taylor_q', 'lladser_pe', 'margalef', 'mcintosh_d', 'mcintosh_e', 'menhinick', 'michaelis_menten_fit', 'observed_features', 'osd', 'pielou_e', 'robbins', 'shannon', 'simpson', 'simpson_e', 'singles', 'strong')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "556e5264",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:20.790166Z",
     "start_time": "2022-09-29T20:54:20.451134Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating alpha diversty: chao1\n",
      "Calculating alpha diversty: simpson\n",
      "Calculating alpha diversty: shannon\n",
      "Calculating alpha diversty: ace\n",
      "Calculating alpha diversty: observed_features\n"
     ]
    }
   ],
   "source": [
    "metrics = ('chao1', 'simpson', 'shannon', 'ace', 'observed_features')\n",
    "alpha_diversities = dict()\n",
    "for metric in metrics:\n",
    "    print(f\"Calculating alpha diversty: {metric}\")\n",
    "    alpha_diversity = alpha(table=tabs, metric=metric).alpha_diversity\n",
    "    alpha_diversities[metric] = alpha_diversity"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "184bcb7b",
   "metadata": {},
   "source": [
    "### Create Phylogenetic inference\n",
    "\n",
    "- [alignment align_to_tree_mafft_fasttree](https://docs.qiime2.org/2022.8/plugins/available/phylogeny/align-to-tree-mafft-fasttree/): Build a phylogenetic tree using fasttree and mafft alignment\n",
    "\n",
    "This pipeline will start by creating a sequence alignment using MAFFT,\n",
    "after which any alignment columns that are phylogenetically uninformative\n",
    "or ambiguously aligned will be removed (masked). The resulting masked\n",
    "alignment will be used to infer a phylogenetic tree and then subsequently\n",
    "rooted at its midpoint. Output files from each step of the pipeline will be\n",
    "saved. This includes both the unmasked and masked MAFFT alignment from\n",
    "q2-alignment methods, and both the rooted and unrooted phylogenies from\n",
    "q2-phylogeny methods.\n",
    "\n",
    "\n",
    "Returns\n",
    "- alignment : FeatureData[AlignedSequence] : The aligned sequences.\n",
    "- masked_alignment : FeatureData[AlignedSequence] : The masked alignment.\n",
    "- tree : Phylogeny[Unrooted] : The unrooted phylogenetic tree.\n",
    "- rooted_tree : Phylogeny[Rooted] : The rooted phylogenetic tree."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "9c8cfe47",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:35.139531Z",
     "start_time": "2022-09-29T20:54:20.792130Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: mafft --preservecase --inputorder --thread 6 /tmp/qiime2-archive-v04ml_kj/2f80a2b8-d473-4e42-82e6-a2f605b0e0d9/data/dna-sequences.fasta\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "inputfile = orig\n",
      "918 x 430 - 271 d\n",
      "nthread = 6\n",
      "nthreadpair = 6\n",
      "nthreadtb = 6\n",
      "ppenalty_ex = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "  901 / 918 (thread    1)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "  910 / 918\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP   701 / 917 (thread    4)\n",
      "Reallocating..done. *alloclen = 1868\n",
      "STEP   901 / 917 (thread    5)\n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "  900 / 918 (thread    5)\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "  910 / 918\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP   801 / 917 (thread    5)\n",
      "Reallocating..done. *alloclen = 1875\n",
      "STEP   901 / 917 (thread    5) h\n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.490\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "6 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The command being run is below. This command cannot be manually re-run as it will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: FastTreeMP -quote -nt /tmp/qiime2-archive-yiu3jme4/eb5454fb-90bd-4dee-9989-8e6cf8994526/data/aligned-dna-sequences.fasta\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "FastTree Version 2.1.10 Double precision (No SSE3), OpenMP (6 threads)\n",
      "Alignment: /tmp/qiime2-archive-yiu3jme4/eb5454fb-90bd-4dee-9989-8e6cf8994526/data/aligned-dna-sequences.fasta\n",
      "Nucleotide distances: Jukes-Cantor Joins: balanced Support: SH-like 1000\n",
      "Search: Normal +NNI +SPR (2 rounds range 10) +ML-NNI opt-each=1\n",
      "TopHits: 1.00*sqrtN close=default refresh=0.80\n",
      "ML Model: Jukes-Cantor, CAT approximation with 20 rate categories\n",
      "      0.11 seconds: Joined    100 of    870\n",
      "      0.23 seconds: Joined    300 of    870\n",
      "      0.37 seconds: Joined    500 of    870\n",
      "      0.49 seconds: Joined    700 of    870\n",
      "Initial topology in 0.60 seconds\n",
      "Refining topology: 39 rounds ME-NNIs, 2 rounds ME-SPRs, 20 rounds ML-NNIs\n",
      "      0.60 seconds: ME NNI round 1 of 39, 1 of 871 splits\n",
      "      0.70 seconds: ME NNI round 4 of 39, 401 of 871 splits, 12 changes (max delta 0.007)\n",
      "      0.83 seconds: SPR round   1 of   2, 201 of 1744 nodes\n",
      "      0.96 seconds: SPR round   1 of   2, 401 of 1744 nodes\n",
      "      1.11 seconds: SPR round   1 of   2, 601 of 1744 nodes\n",
      "      1.26 seconds: SPR round   1 of   2, 801 of 1744 nodes\n",
      "      1.40 seconds: SPR round   1 of   2, 1001 of 1744 nodes\n",
      "      1.52 seconds: SPR round   1 of   2, 1201 of 1744 nodes\n",
      "      1.64 seconds: SPR round   1 of   2, 1401 of 1744 nodes\n",
      "      1.78 seconds: SPR round   1 of   2, 1601 of 1744 nodes\n",
      "      1.88 seconds: ME NNI round 14 of 39, 501 of 871 splits, 12 changes (max delta 0.020)\n",
      "      1.99 seconds: SPR round   2 of   2, 101 of 1744 nodes\n",
      "      2.11 seconds: SPR round   2 of   2, 301 of 1744 nodes\n",
      "      2.22 seconds: SPR round   2 of   2, 501 of 1744 nodes\n",
      "      2.34 seconds: SPR round   2 of   2, 701 of 1744 nodes\n",
      "      2.45 seconds: SPR round   2 of   2, 901 of 1744 nodes\n",
      "      2.58 seconds: SPR round   2 of   2, 1101 of 1744 nodes\n",
      "      2.68 seconds: SPR round   2 of   2, 1301 of 1744 nodes\n",
      "      2.82 seconds: SPR round   2 of   2, 1601 of 1744 nodes\n",
      "      2.93 seconds: ME NNI round 27 of 39, 301 of 871 splits, 0 changes\n",
      "Total branch-length 19.047 after 2.98 sec\n",
      "      3.03 seconds: ML Lengths 201 of 871 splits\n",
      "      3.14 seconds: ML Lengths 601 of 871 splits\n",
      "      3.31 seconds: ML NNI round 1 of 20, 101 of 871 splits, 18 changes (max delta 5.135)\n",
      "      3.46 seconds: ML NNI round 1 of 20, 301 of 871 splits, 57 changes (max delta 7.588)\n",
      "      3.61 seconds: ML NNI round 1 of 20, 501 of 871 splits, 83 changes (max delta 8.409)\n",
      "      3.75 seconds: ML NNI round 1 of 20, 701 of 871 splits, 112 changes (max delta 9.826)\n",
      "ML-NNI round 1: LogLk = -49085.220 NNIs 145 max delta 9.83 Time 3.88\n",
      "      3.88 seconds: Site likelihoods with rate category 1 of 20\n",
      "      3.98 seconds: Site likelihoods with rate category 12 of 20\n",
      "Switched to using 20 rate categories (CAT approximation)\n",
      "Rate categories were divided by 1.099 so that average rate = 1.0\n",
      "CAT-based log-likelihoods may not be comparable across runs\n",
      "Use -gamma for approximate but comparable Gamma(20) log-likelihoods\n",
      "      4.15 seconds: ML NNI round 2 of 20, 101 of 871 splits, 8 changes (max delta 4.492)\n",
      "      4.30 seconds: ML NNI round 2 of 20, 301 of 871 splits, 23 changes (max delta 4.492)\n",
      "      4.46 seconds: ML NNI round 2 of 20, 501 of 871 splits, 45 changes (max delta 8.038)\n",
      "      4.63 seconds: ML NNI round 2 of 20, 701 of 871 splits, 70 changes (max delta 8.038)\n",
      "ML-NNI round 2: LogLk = -40962.999 NNIs 82 max delta 8.04 Time 4.78\n",
      "      4.77 seconds: ML NNI round 3 of 20, 1 of 871 splits\n",
      "      4.92 seconds: ML NNI round 3 of 20, 201 of 871 splits, 5 changes (max delta 0.126)\n",
      "      5.07 seconds: ML NNI round 3 of 20, 401 of 871 splits, 20 changes (max delta 5.434)\n",
      "ML-NNI round 3: LogLk = -40928.563 NNIs 29 max delta 5.43 Time 5.23\n",
      "      5.23 seconds: ML NNI round 4 of 20, 1 of 871 splits\n",
      "      5.37 seconds: ML NNI round 4 of 20, 201 of 871 splits, 13 changes (max delta 3.681)\n",
      "ML-NNI round 4: LogLk = -40910.586 NNIs 16 max delta 3.68 Time 5.50\n",
      "      5.50 seconds: ML NNI round 5 of 20, 1 of 871 splits\n",
      "ML-NNI round 5: LogLk = -40902.540 NNIs 6 max delta 1.76 Time 5.66\n",
      "      5.66 seconds: ML NNI round 6 of 20, 1 of 871 splits\n",
      "ML-NNI round 6: LogLk = -40902.393 NNIs 0 max delta 0.00 Time 5.76\n",
      "Turning off heuristics for final round of ML NNIs (converged)\n",
      "      5.82 seconds: ML NNI round 7 of 20, 101 of 871 splits, 1 changes (max delta 0.000)\n",
      "      5.98 seconds: ML NNI round 7 of 20, 301 of 871 splits, 4 changes (max delta 0.467)\n",
      "      6.12 seconds: ML NNI round 7 of 20, 501 of 871 splits, 4 changes (max delta 0.467)\n",
      "      6.28 seconds: ML NNI round 7 of 20, 701 of 871 splits, 5 changes (max delta 0.467)\n",
      "ML-NNI round 7: LogLk = -40899.041 NNIs 5 max delta 0.47 Time 6.41 (final)\n",
      "      6.41 seconds: ML Lengths 1 of 871 splits\n",
      "      6.52 seconds: ML Lengths 501 of 871 splits\n",
      "Optimize all lengths: LogLk = -40898.803 Time 6.62\n",
      "      6.74 seconds: ML split tests for    100 of    870 internal splits\n",
      "      6.87 seconds: ML split tests for    200 of    870 internal splits\n",
      "      7.00 seconds: ML split tests for    300 of    870 internal splits\n",
      "      7.13 seconds: ML split tests for    400 of    870 internal splits\n",
      "      7.26 seconds: ML split tests for    500 of    870 internal splits\n",
      "      7.40 seconds: ML split tests for    600 of    870 internal splits\n",
      "      7.53 seconds: ML split tests for    700 of    870 internal splits\n",
      "      7.66 seconds: ML split tests for    800 of    870 internal splits\n",
      "Total time: 7.75 seconds Unique: 873/918 Bad splits: 2/870 Worst delta-LogLk 1.503\n"
     ]
    }
   ],
   "source": [
    "mafft_alignment, mafft_masked_alignment, mafft_tree, mafft_rooted_tree = align_to_tree_mafft_fasttree(\n",
    "    sequences=reps, n_threads=6, )"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0768d598",
   "metadata": {},
   "source": [
    "### Compute Alpha Diversity (Phylogeny)\n",
    "- [diversity alpha_phylogenetic](https://docs.qiime2.org/2022.8/plugins/available/diversity/alpha-phylogenetic/): Computes a user-specified phylogenetic alpha diversity metric for all samples in a feature table.\n",
    "- Metrics: Choices ('faith_pd')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "0383aa19",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:35.310009Z",
     "start_time": "2022-09-29T20:54:35.141333Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating alpha diversity: faith_pd\n"
     ]
    }
   ],
   "source": [
    "metrics = ('faith_pd', )\n",
    "alpha_diversities_phylogenetic = dict()\n",
    "for metric in metrics:\n",
    "    print(f\"Calculating alpha diversity: {metric}\")\n",
    "    alpha_diversity = alpha_phylogenetic(table=tabs, phylogeny=mafft_rooted_tree, metric=metric).alpha_diversity\n",
    "    alpha_diversities_phylogenetic[metric] = alpha_diversity"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c7752cfd",
   "metadata": {},
   "source": [
    "### Alpha diversity correlation\n",
    "\n",
    "This method only process `numeric` columns.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2814f613",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:35.317618Z",
     "start_time": "2022-09-29T20:54:35.312094Z"
    }
   },
   "outputs": [],
   "source": [
    "methods = ('spearman', 'pearson')\n",
    "numerics_cols = metadata_qa.filter_columns(column_type='numeric')\n",
    "if numerics_cols.column_count > 0:\n",
    "    for metric, alpha_values in alpha_diversities.items():\n",
    "        for method in methods:\n",
    "            corr_view = alpha_correlation(alpha_diversity=alpha_values, metadata=numerics_cols, \n",
    "                                      method=method, intersect_ids=True).visualization\n",
    "            view_path = os.path.join(qiime_folder, f'alpha-correlation-{metric}-{method}.qzv')\n",
    "            corr_view.save(view_path)\n",
    "            corr_view"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ea83c8a",
   "metadata": {},
   "source": [
    "## Alpha diversity comparisons\n",
    "\n",
    "Visually and statistically compare groups of alpha diversity values.\n",
    "\n",
    "[diversity alpha_group_significance](https://docs.qiime2.org/2022.8/plugins/available/diversity/alpha-group-significance/)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "276d1c18",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:36.089290Z",
     "start_time": "2022-09-29T20:54:35.318963Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing alpha_group_significance: chao1\n",
      "Processing alpha_group_significance: simpson\n",
      "Processing alpha_group_significance: shannon\n",
      "Processing alpha_group_significance: ace\n",
      "Processing alpha_group_significance: observed_features\n"
     ]
    }
   ],
   "source": [
    "for metric, alpha_values in alpha_diversities.items():\n",
    "    print(f\"Processing alpha_group_significance: {metric}\")\n",
    "    significance_view = alpha_group_significance(alpha_diversity=alpha_values, metadata=metadata_qa).visualization\n",
    "    view_path = os.path.join(qiime_folder, f'alpha-group-significance-{metric}.qzv')\n",
    "    significance_view.save(view_path)\n",
    "    significance_view"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4349d357",
   "metadata": {},
   "source": [
    "## Beta diversity analysis\n",
    "\n",
    "#### Reference\n",
    "- [diversity beta](https://docs.qiime2.org/2022.8/plugins/available/diversity/beta/): Computes a user-specified beta diversity metric for all pairs of samples in a feature table.\n",
    "- [Beta diversity metrics](http://scikit-bio.org/docs/0.2.0/generated/skbio.diversity.beta.html)\n",
    "\n",
    "- Metric Choices('aitchison', 'braycurtis', 'canberra', 'canberra_adkins', 'chebyshev', 'cityblock', 'correlation', 'cosine', 'dice', 'euclidean', 'hamming', 'jaccard', 'jensenshannon', 'kulsinski', 'matching', 'minkowski', 'rogerstanimoto', 'russellrao', 'seuclidean', 'sokalmichener', 'sokalsneath', 'sqeuclidean', 'yule')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "f6f358dd",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:54:36.335038Z",
     "start_time": "2022-09-29T20:54:36.091288Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating beta diversity: dice\n",
      "Calculating beta diversity: braycurtis\n",
      "Calculating beta diversity: chebyshev\n",
      "Calculating beta diversity: jaccard\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/lauro/anaconda3/envs/qiime2-2022.2/lib/python3.8/site-packages/sklearn/metrics/pairwise.py:1761: DataConversionWarning: Data was converted to boolean for metric dice\n",
      "  warnings.warn(msg, DataConversionWarning)\n",
      "/home/lauro/anaconda3/envs/qiime2-2022.2/lib/python3.8/site-packages/sklearn/metrics/pairwise.py:1761: DataConversionWarning: Data was converted to boolean for metric jaccard\n",
      "  warnings.warn(msg, DataConversionWarning)\n"
     ]
    }
   ],
   "source": [
    "metrics = ('dice', 'braycurtis', 'chebyshev', 'jaccard')\n",
    "beta_diversities = dict()\n",
    "for metric in metrics:\n",
    "    print(f\"Calculating beta diversity: {metric}\")\n",
    "    beta_diversity = beta(table=tabs, metric=metric, n_jobs=6, pseudocount=1).distance_matrix\n",
    "    beta_diversities[metric] = beta_diversity"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "200a2fa7",
   "metadata": {},
   "source": [
    "### Beta group significance\n",
    "\n",
    "- [diversity beta_group_significance](https://docs.qiime2.org/2022.8/plugins/available/diversity/beta-group-significance/): Determine whether groups of samples are significantly different from one another using a permutation-based statistical test.\n",
    "- Marti J Anderson. A new method for non-parametric multivariate analysis of variance. Austral ecology, 26(1):32â€“46, 2001. doi:https://doi.org/10.1111/j.1442-9993.2001.01070.pp.x."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "b8dd42b6",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-29T20:55:08.313860Z",
     "start_time": "2022-09-29T20:54:36.336993Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating beta group significance with method permanova and metric dice\n",
      "Calculating beta group significance with method permanova and metric braycurtis\n",
      "Calculating beta group significance with method permanova and metric chebyshev\n",
      "Calculating beta group significance with method permanova and metric jaccard\n",
      "Calculating beta group significance with method anosim and metric dice\n",
      "Calculating beta group significance with method anosim and metric braycurtis\n",
      "Calculating beta group significance with method anosim and metric chebyshev\n",
      "Calculating beta group significance with method anosim and metric jaccard\n",
      "Calculating beta group significance with method permdisp and metric dice\n",
      "Calculating beta group significance with method permdisp and metric braycurtis\n",
      "Calculating beta group significance with method permdisp and metric chebyshev\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/lauro/anaconda3/envs/qiime2-2022.2/lib/python3.8/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:143: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -95033973.72889368 and the largest is 1022480902.9013429.\n",
      "  warn(\n",
      "/home/lauro/anaconda3/envs/qiime2-2022.2/lib/python3.8/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:143: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -95033973.72889373 and the largest is 1022480902.9013419.\n",
      "  warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating beta group significance with method permdisp and metric jaccard\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "methods = ('permanova', 'anosim', 'permdisp')\n",
    "for method in methods:\n",
    "    for metric, beta_diversity in beta_diversities.items():\n",
    "        print(f'Calculating beta group significance with method {method} and metric {metric}')\n",
    "        beta_view = beta_group_significance(distance_matrix=beta_diversity, \n",
    "                                            metadata=metadata_qa.get_column('group-id'), \n",
    "                                            pairwise=True, method=method).visualization\n",
    "        view_name = os.path.join(qiime_folder, f'beta-group-significance-{metric}-{method}.qzv')\n",
    "        beta_view.save(view_name)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0de85cf",
   "metadata": {},
   "source": [
    "### Beta group Rarefaction\n",
    "\n",
    "- [diversity beta_rarefaction](https://docs.qiime2.org/2022.8/plugins/available/diversity/beta-rarefaction/): Repeatedly rarefy a feature table to compare beta diversity results within a given rarefaction depth.  For a given beta diversity metric, this visualizer will provide: an Emperor jackknifed PCoA plot, samples clustered by UPGMA or neighbor joining with support calculation, and a heatmap showing the correlation between rarefaction trials of that beta diversity metric."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a8e37df4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
